<!-- MarkdownTOC -->

- [RNA-Seq Tumor and Margin](#rna-seq-tumor-and-margin)
- [Import & fastqC](#import--fastqc)
- [Transcript quantification with kallisto](#transcript-quantification-with-kallisto)
    - [Expression at gene level](#expression-at-gene-level)
- [Expression vs Methylation](#expression-vs-methylation)
- [Map reads to genome/transcriptome \(tophat\)](#map-reads-to-genometranscriptome-tophat)
- [Quantitate transcripts with cufflinks](#quantitate-transcripts-with-cufflinks)

<!-- /MarkdownTOC -->


## RNA-Seq Tumor and Margin

Downloading from box, login with

Name: Dario

pwd: t...a

email: dario.beraldi@cruk.cam.ac.uk

These Files will be removed August 2016.

```
cd /nas/sblab_data1/berald01/repository/fastq/
R
library(data.table)
txt<- fread('find ./ -name "*.fastq.gz"', header= FALSE)
names(txt)<- c('fullpath')
txt$filename<- basename(txt$fullpath)

txt$bname<- sub('_S._L00._R._00..fastq.gz', '', txt$filename)
txt[, sampleName := sub('.*-', '', bname)]
xkey<- data.table(sampleName= c("F2", "F3", "M1", "M3"), library_id= c('ear046_F2', 'ear047_F3', 'ear048_M1', 'ear049_M3'))
txt<- merge(txt, xkey, by= 'sampleName')
txt[, newname := paste(library_id, filename, sep= '.')]
file.rename(txt$fullpath, txt$newname)
## Copy to lustre ...
```

## Import & fastqC

```
cd /lustre/sblab/berald01/repository/fastq/

for fq in ear04[6-9]*.fastq.gz
do
    bsub -J fastqc-${fq} "
    fastqfile_to_sblab.py -f $fq -p 'Illumina HiSeq' -s '' &&
    fastqc_md5.py -i $fq -f ' --noextract' &&
        ## NOTE suffix .fastq.gz
    fastqc_to_pgtable.py -i ${fq%.fastq.gz}_fastqc.zip &&
    rm ${fq%.fastq.gz}_fastqc.zip" 
done
```

## Transcript quantification with kallisto

Roughly mammalian cells have ~200,00 (10^5-10^6) mRNAs per cell see http://book.bionumbers.org/how-many-mrnas-are-in-a-cell/ See also
[From single-cell to cell-pool transcriptomes: Stochasticity in gene expression and RNA splicing](http://genome.cshlp.org/content/24/3/496) quoting
50,000-300,000.

```
cd /lustre/sblab/berald01/projects/20150501_methylation_brain/20160303_rnaseq/

## Do we really need to bootstrap?

samples=`find /lustre/sblab/berald01/repository/fastq/ear04[6-9]*_R1_001.fastq.gz -exec basename {} \; | sort | cut -d '.' -f 1 | uniq`
for x in $samples
do
    fq1=`find /lustre/sblab/berald01/repository/fastq/ -name ${x}*_R1_001.fastq.gz | sort`
    fq1=`echo $fq1`
    fq2=${fq1//_R1_/_R2_}
    echo "kallisto quant --bias -i /lustre/sblab/berald01/reference_data/kallisto/Homo_sapiens.GRCh37.cdna.all.idx -o ${x}.tx <(zcat $fq1) <(zcat $fq2)" > $x.tmp.sh
    bsub -oo $x.log -R "rusage[mem=8192]" "bash $x.tmp.sh"
done

## Rename and transfer
for x in ear049_M3 ear047_F3 ear048_M1 ear046_F2
do
    mv ${x}.tx/abundance.tsv ${x}.tx_quant.tsv
done
mvsync *.tx_quant.tsv sblab-srv001:/nas/sblab_data1/berald01/projects/20150501_methylation_brain/20160303_rnaseq/
```

Transcripts in fasta for hg19/GRCh37 from ftp://ftp.ensembl.org/pub/grch37/release-83/fasta/homo_sapiens/cdna/Homo_sapiens.GRCh37.cdna.all.fa.gz

Annotation `mart_export.GRCh37.txt` to link ensembl transcripts to gene name from http://grch37.ensembl.org/biomart/martview/ with the appropropriate Attributes.
Header line edited to replace whitespace with underscore and parenthesis remove.

```
cd /nas/sblab_data1/group_folders/berald01/projects/20150501_methylation_brain/20160303_rnaseq/
R
library(data.table)
library(ggplot2)
tx<- fread('tableCat.py -H -i ear04*.tx_quant.tsv -r "\\..*" -id "library_id"')
ann<- fread('zcat mart_export.GRCh37.txt.gz')
tx[, Ensembl_Transcript_ID := sub('\\..*', '', target_id)]
tx<- merge(tx, ann[], by= 'Ensembl_Transcript_ID')

## Keep only regular chroms and rename to UCSC format. 
## Note this will make the sum(TPM) < 1e6
tx<- tx[Chromosome_Name %in% c(1:22, 'MT', 'X', 'Y')]
tx[, chrom := ifelse(Chromosome_Name %in% c(1:22, 'X', 'Y'), paste0('chr', Chromosome_Name), Chromosome_Name)]
tx[ , chrom := ifelse(Chromosome_Name == 'MT', 'chrM', chrom)]
stopifnot(unique(sort(tx$chrom)) == c("chr1", "chr10", "chr11", "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr2", "chr20", "chr21", "chr22", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chrM", "chrX", "chrY"))
tx[, Strand := ifelse(Strand == 1, '+', '-')]

## Note Change to bed coords!!
tx[, Transcript_Start_bp := Transcript_Start_bp-1]
tx[, Gene_Start_bp := Gene_Start_bp-1] 

## Table logFC
## ===========
txLfc<- dcast.data.table(data= tx[library_id %in% c("ear047_F3", "ear049_M3")], Ensembl_Transcript_ID ~ library_id, value.var= 'tpm')
## Note using unique(..., by= NULL)!!!
txLfc<- merge(txLfc, 
            unique(tx[, list(Ensembl_Transcript_ID, chrom, Transcript_Start_bp, Transcript_End_bp, Associated_Gene_Name, Strand)], by= NULL), 
        by= 'Ensembl_Transcript_ID')

txLfc[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
txLfc$tpmLog2Avg<- log2(rowMeans(txLfc[, list(ear047_F3, ear049_M3)]) + 0.01)
txLfc[, promStart := ifelse(Strand == '+', Transcript_Start_bp-1000, Transcript_End_bp)]
txLfc[, promEnd := promStart + 1000]
write.table(file= 'tx_quant_lfc.bed',
    txLfc[, list(chrom, promStart, promEnd, Ensembl_Transcript_ID, Associated_Gene_Name, Strand, Transcript_Start_bp, Transcript_End_bp, tpmLog2FC, tpmLog2Avg, ear047_F3, ear049_M3)][order(chrom, Transcript_Start_bp, Transcript_End_bp)],
    row.names= FALSE, quote= FALSE, sep= '\t')

## Table all quants
txout<- tx[, list(chrom, Transcript_Start_bp, Transcript_End_bp, Ensembl_Transcript_ID, Associated_Gene_Name, Strand, tpm, est_counts, library_id)][order(library_id, chrom, Transcript_Start_bp, Transcript_End_bp)]
write.table(file= 'tx_quant.bed', txout, row.names= FALSE, quote= FALSE, sep= '\t')

gx<- tx[, list(tpm= sum(tpm), est_counts= sum(est_counts)), by= list(Associated_Gene_Name, library_id)]
cmb<- combn(unique(gx$library_id), 2)
pdf('gene_tpm_pairs.pdf', w= 24/2.54, h= 16/2.54, pointsize= 10)
par(mfrow= c(2, 3), las= 1,  mgp= c(1.5, 0.5, 0), mar= c(3, 3, 2, 0.1))
for(i in 1:ncol(cmb)){
    dat<- dcast.data.table(data= gx[library_id %in% cmb[, i]], Associated_Gene_Name ~ library_id, value.var= 'tpm')
    dat[, Associated_Gene_Name := NULL]
    avgExpr<- apply(dat, 1, function(x) {mean(log2(x + 0.01))})
    logFC<- apply(dat, 1, function(x) { log2((x[1]+0.01) / (x[2]+0.01)) })
    smoothScatter(x= avgExpr, y= logFC, xlab= 'Mean log2(TPM)', ylab= 'Log2FC', main= paste(names(dat), collapse= '/'), nrpoints= 1000)
    lines(loess.smooth(x= avgExpr, y= logFC, span= 0.05), col= 'grey', lwd= 2)
    grid()
    abline(h= 0, col= 'red', lty= 'dotted')
}
dev.off()
system('rsync --remove-source-files gene_tpm_pairs.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20160303_rnaseq/')

gg<- ggplot(gx, aes(x= tpm+0.01)) +
    stat_ecdf(geom = "step") +
    geom_vline(xintercept= 1, colour= 'red', linetype= 'dotted') +
    facet_wrap(~library_id) +
    scale_x_continuous(trans= 'log2') +
    xlab('log2(TPM + 0.01)') + 
    ylab('% number of genes') +
    ggtitle('Cumulative distribution of TPM')
ggsave('gene_tpm_cdf.pdf', w= 22/2.54, h= 14/2.54)
system('rsync --remove-source-files gene_tpm_cdf.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20160303_rnaseq/')

## Number of genes with large-ish logFC
dat<- dcast.data.table(data= gx[library_id %in% c('ear047_F3', 'ear049_M3')], Associated_Gene_Name ~ library_id, value.var= 'tpm')
dat$avgLogExpr<- apply(dat[, list(ear047_F3, ear049_M3)], 1, function(x) {mean(log2(x + 0.01))})

## LogFC is [Tumor - Margin] so +ve means tumor > margin
dat$logFC<- apply(dat[, list(ear047_F3, ear049_M3)], 1, function(x) { log2((x[1]+0.01) / (x[2]+0.01)) })

smry<- ifelse(dat$logFC > 2, 'T > M',
    ifelse(dat$logFC < -2, 'T < M', 'T == M'))
table(smry)
# smry
# T < M T == M  T > M 
#  5602  26679   2539 
```

### Expression at gene level

```
cd /nas/sblab_data1/group_folders/berald01/projects/20150501_methylation_brain/20160303_rnaseq/
R
library(data.table)
tx<- fread('tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmLog2Avg<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmLog2Avg)][order(chrom, promStart, promEnd)]
```

## Expression vs Methylation

Table of methylation in promoters generated by Sergio

```
cd /nas/sblab_data1/group_folders/berald01/projects/20150501_methylation_brain/20160303_rnaseq/
R
library(data.table)

## Prepare expression at gene level
## 
tx<- fread('tx_quant_lfc.bed')

gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmLog2Avg<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmLog2Avg)][order(chrom, promStart, promEnd)]
setnames(gx, 'Associated_Gene_Name', 'gene_name')

## Table of promoter methylation
mc<- fread('/nas/sblab_data1/group_folders/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/tables/M8.T3.promoter.txt')
mc[, cancerGene := ifelse(glioblastoma == 'y', 'gbm', ifelse(cancer == 'y', 'cancer', 'na'))]
mc[, cancer := NULL]
mc[, glioblastoma := NULL]

## Combine expression and methylation
gx<- merge(gx, mc, by= 'gene_name')
gxDat<- rbindlist(list(
  gx[cancerGene == 'gbm', list(gene_name, tpmLog2= log2(ear047_F3 + 0.01), pct_mc= pct_5mC.T3, pct_hmc= pct_5hmC.T3, tissue= 'Tumor')],
  gx[cancerGene == 'gbm', list(gene_name, tpmLog2= log2(ear049_M3 + 0.01), pct_mc= pct_5mC.M8, pct_hmc= pct_5hmC.M8, tissue= 'Margin')]),
)

## Expression vs 5mC
gg<- ggplot(data= gxDat, aes(x= tpmLog2, y= pct_mc, colour= tissue, label= gene_name)) +
    geom_text()
    
## Expression vs 5hmC
gg<- ggplot(data= gxDat, aes(x= tpmLog2, y= pct_hmc, colour= tissue, label= gene_name)) +
    geom_text()

```


```
cd /nas/sblab_data1/berald01/projects/20150501_methylation_brain/20160303_rnaseq/
R
library(data.table)
library(ggplot2)
library(gridExtra)
winsor2<- function (x, multiple=3){
    ## From http://www.r-bloggers.com/winsorization/
   if(length(multiple) != 1 || multiple <= 0) {
      stop("bad value for 'multiple'")
   }
   med <- median(x, na.rm= TRUE)
   y <- x - med
   sc <- mad(y, center=0) * multiple
   y[ y > sc ] <- sc
   y[ y < -sc ] <- -sc
   y + med
}

mc<- fread('/nas/sblab_data1/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/tables/M8.T3.promoter.txt')
mc[, cancerGene := ifelse(glioblastoma == 'y', 'gbm', ifelse(cancer == 'y', 'cancer', 'na'))]
mc[, cancer := NULL]
mc[, glioblastoma := NULL]
tx<- fread('tx_quant_lfc.bed')
setnames(tx, 'Associated_Gene_Name', 'gene_name')
tx<- merge(tx, mc, by= 'gene_name')
tx[, Transcript_Start_bp := NULL]
tx[, Transcript_End_bp := NULL]

tx[, TvsMexpr := ifelse(tpmLog2Avg > 1 & tpmLog2FC > 2, 'up', ifelse(tpmLog2Avg > 1 & tpmLog2FC < -2, 'down', 'na'))]

## Expression of cancer/glioblastoma/other genes
g1<- ggplot(data= tx[tpmLog2Avg > 1, ], aes(x= TvsMexpr, y= winsor2(tpmLog2FC, 4))) +
    geom_boxplot(varwidth= TRUE, size= 0.2, outlier.size = 0.2) +
    ylab('log2(Tum/Mar)') + ggtitle('Expression (TPM)') + xlab('') +
    geom_hline(yintercept= 0, col= 'red', linetype= 'dotted')
g2<- ggplot(data= tx[tpmLog2Avg > 1, ], aes(x= TvsMexpr, y= winsor2(pct_5mC.T3 - pct_5mC.M8))) +
    geom_boxplot(varwidth= TRUE, size= 0.2, outlier.size = 0.2) +
    ylab('Tum - Mar') + ggtitle('5mC') + xlab('') +
    geom_hline(yintercept= 0, col= 'red', linetype= 'dotted')
g3<- ggplot(data= tx[tpmLog2Avg > 1, ], aes(x= TvsMexpr, y= winsor2(pct_5hmC.T3 - pct_5hmC.M8))) +
    geom_boxplot(varwidth= TRUE, size= 0.2, outlier.size = 0.2) +
    ylab('Tum - Mar') + ggtitle('5hmC') + xlab('') +
    geom_hline(yintercept= 0, col= 'red', linetype= 'dotted')
gg<- arrangeGrob(g1, g2, g3, nrow= 1)
ggsave('boxplot_expr_vs_met.pdf', gg, w= 22/2.54, h= 8/2.54)
system('rsync --remove-source-files boxplot_expr_vs_met.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20160303_rnaseq/')

## Expression vs promoter methylation
tx[, tpm.M3 := ifelse(log2(ear049_M3+0.01) < 0.1, 'M_low', ifelse(log2(ear049_M3+0.01) < 2, 'M_mid', 'M_zHigh'))]
tx[, tpm.F3 := ifelse(log2(ear047_F3+0.01) < 0.1, 'T_low', ifelse(log2(ear047_F3+0.01) < 2, 'T_mid', 'T_zHigh'))]  
gg<- ggplot(data= tx) +
    geom_violin(draw_quantiles= c(0.25, 0.5, 0.75), aes(x= tpm.M3, y= ifelse(pct_5mC.M8 > 50, 50, pct_5mC.M8), fill= sub('.*_', '', tpm.M3)), size= 0.2) +
    geom_violin(draw_quantiles= c(0.25, 0.5, 0.75), aes(x= tpm.F3, y= ifelse(pct_5mC.T3 > 50, 50, pct_5mC.T3), fill= sub('.*_', '', tpm.F3)), size= 0.2) +
    xlab('Expression level') + ylab('% 5mC') + theme(legend.title= element_blank())
ggsave('boxplot_expr_level_vs_met.pdf', gg, w= 14/2.54, h= 8/2.54, pointsize= 12)
system('rsync --remove-source-files boxplot_expr_level_vs_met.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20160303_rnaseq/')


## Other way: Expression in high/low methylated promoters


```

## Map reads to genome/transcriptome (tophat)

It appears reads have been already trimmed.

```
cd /lustre/sblab/berald01/repository/

bwtidx='/lustre/sblab/berald01/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Sequence/Bowtie2Index/genome'
gtf='/lustre/sblab/berald01/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes.gtf'

for fq1 in fastq/ear04[6-9]*_R1_001.fastq.gz
do
    fq2=${fq1/_R1_/_R2_}
    bname=`basename $fq1 _R1_001.fastq.gz`
    bsub -R "rusage[mem=10000]" -oo tophat_out/${bname}.log -J tophat-${bname} "
        mkdir -p tophat_out/${bname}
        tophat2 --mate-std-dev 50 --no-novel-juncs -o tophat_out/${bname} -p 8 -G ${gtf} ${bwtidx} $fq1 $fq2"
done
```

## Quantitate transcripts with cufflinks

* Merge bam files

```
cd /lustre/sblab/berald01/repository/tophat_out/
for x in ear046_F2 ear047_F3 ear048_M1 ear049_M3
do
    bams=`find ./ -name "accepted_hits.bam" | grep "${x}" | sort`
    bams=`echo $bams`
    bsub -oo ${x}.log "samtools merge -@ 8 ${x}.tophat.bam $bams &&
    samtools index ${x}.tophat.bam &&
    rm $bams"
done
```

* TODO? Run cufflinks

```
gtf='/lustre/sblab/berald01/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes.gtf'
for bam in 
bsub -n 8 -oo cufflinks_test.log -R "rusage[mem=8096]" "cufflinks -o test.output -p 8 --GTF $gff --multi-read-correct $bam"
```

*******

Move fastq to archive

```
ssh hpcgate.cri.camres.org
mvsync /lustre/sblab/berald01/repository/fastq/*.fastq.gz /archive/Groups/SBLab/fs04/berald01/repository/fastq/
```
