<!-- MarkdownTOC -->

- [Software tools and requirements](#software-tools-and-requirements)

<!-- /MarkdownTOC -->

Software tools and requirements
===============================

All data processing was implemented under Linux environment with standard GNU coreutils tools. 
Additional required software:

* [cutadapt](http://cutadapt.readthedocs.org/en/stable/guide.html) version 1.9
* [bwa-meth](https://github.com/brentp/bwa-meth)
* [bwa](https://github.com/lh3/bwa) version 0.7
* [samtools](http://www.htslib.org/) version 1.1
* [picard/MarkDuplicates](https://broadinstitute.github.io/picard/command-line-overview.html) version 1.127
* [bam2methylation.py](https://github.com/dariober/bioinformatics-cafe/tree/master/bam2methylation)
* [BamUtil: clipOverlap](http://genome.sph.umich.edu/wiki/BamUtil:_clipOverlap)
* [bedtools](http://bedtools.readthedocs.org/en/latest/) version 2.25
* [R-3.2.3](https://cran.r-project.org/) 
* Human reference genome version hg19 obtained from [Illumina iGenomes](http://support.illumina.com/sequencing/sequencing_software/igenome.html)
