
Summarise promoter (200bp plus and minus TSS, 200bp upstream only, 1000bp) 5mC and 5hmC levels, RNA expression, SNVs and CNVs for:

- Epigenetic regulators - Gordon list
- Glioblastoma genes - Bai2015, COSMIC

Try using pheatmaps to visualise.



### Define epigenetic regulators

Taken from Gordon:

/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/epigenetic_genes/20151130_genelist_rnaseq_alternative.txt





### Define test genes

Built manually after discussing with Euni:

/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/test_genes/20160505_test_genes.txt





### Define glioblastoma and cancer genes

Obtained from /data/sblab/group_folders/martin03/projects/20150921_BrainMethylomeRoadMap/scripts/20160226_tsg_ocg.md

```bash
cp /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/Census_allFri_Feb_26_17_19_29_2016.tsv /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/COSMICv76/Census_allFri_Feb_26_17_19_29_2016.tsv
```

```python
census=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/COSMICv76/Census_allFri_Feb_26_17_19_29_2016.tsv", "r")
census_lines=census.readlines()
census.close()

len(census_lines)-1 # 572 genes for which mutations have been causally implicated in cancer

cancergenes_all=[]
cancergenes_all_synonyms=[]
cancergenes_glioblastoma=[]
cancergenes_glioblastoma_synonyms=[]

for line in census_lines[1:]:
	fields=line.replace("\n","").split("\t")
	name=fields[0]
	tumour_type_somatic=fields[7]
	tumour_type_germline=fields[8]
	cancergenes_all.append(name)
	list_syn=[]
	if fields[16]!="":
		list_syn=list_syn+fields[16].split('"')[1].split(",")
	list_syn.append(name)
	cancergenes_all_synonyms=cancergenes_all_synonyms+list_syn
	if ("glioblastoma" in tumour_type_somatic) or ("glioblastoma" in tumour_type_germline):
		cancergenes_glioblastoma.append(name)
		cancergenes_glioblastoma_synonyms=cancergenes_glioblastoma_synonyms+list_syn


len(cancergenes_all) # 572
len(set(cancergenes_all)) # 571
len(set(cancergenes_all_synonyms)) # 3997
len(cancergenes_glioblastoma) # 11
len(set(cancergenes_glioblastoma)) # 11
len(set(cancergenes_glioblastoma_synonyms)) # 91


outputfile_glioblastoma = open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/glioblastoma_genes/20160505_glioblastoma_genes.txt", "w")

for gene in set(cancergenes_glioblastoma):
	outputfile_glioblastoma.write("%s\n" % gene)

outputfile_glioblastoma.close()


outputfile_cancer = open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_cancer_genes.txt", "w")

for gene in set(cancergenes_all):
	outputfile_cancer.write("%s\n" % gene)

outputfile_cancer.close()

```





### Write bed files with promoter coordinates (200bp plus and minus TSS) for all, cancer and glioblastoma genes

Obtained from /data/sblab/group_folders/martin03/projects/20150921_BrainMethylomeRoadMap/scripts/20160226_tsg_ocg.md

```python
genes=open("/lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes.gtf", "r")
genes_lines=genes.readlines()
genes.close()

gene_transcript_strand={}
genes_all=set()
gene_chr={}

for line in genes_lines:
	fields=line.split("\t")
	gene_id=fields[8].split(";")[0].split('"')[1]
	transcript_id=fields[8].split(";")[1].split('"')[1]
	leftcoord=fields[3]
	rightcoord=fields[4]
	strand=fields[6]
	chr=fields[0]
	if "_" not in chr:
		genes_all.add(gene_id)
		gene_chr[gene_id]=chr
		if (gene_id, transcript_id, strand) in gene_transcript_strand:
			if strand=="+":
				gene_transcript_strand[(gene_id, transcript_id, strand)].append(str(int(leftcoord)-1))
			if strand=="-":
				gene_transcript_strand[(gene_id, transcript_id, strand)].append(rightcoord)
		else:
			if strand=="+":
				gene_transcript_strand[(gene_id, transcript_id, strand)]=[str(int(leftcoord)-1)]
			if strand=="-":
				gene_transcript_strand[(gene_id, transcript_id, strand)]=[rightcoord]




gene_transcript_strand_merge={}

for tri in gene_transcript_strand:
	gene_id=tri[0]
	transcript_id=tri[1]
	strand=tri[2]
	if strand=="+":
		gene_transcript_strand_merge[(gene_id, transcript_id, strand)]=min(gene_transcript_strand[tri])
	if strand=="-":
		gene_transcript_strand_merge[(gene_id, transcript_id, strand)]=max(gene_transcript_strand[tri])



gene_coords_200={}

count=0
for gene in genes_all:
	count+=1
	print count
	list_coords=[]
	for tri in gene_transcript_strand_merge:
		if gene==tri[0]:
			strand=tri[2]
			list_coords.append(gene_transcript_strand_merge[tri])
	if strand=="+":
		gene_coords_200[gene]=[int(min(list_coords))-200, int(min(list_coords))+200]		
	if strand=="-":
		gene_coords_200[gene]=[int(max(list_coords))-200, int(max(list_coords))+200]		



output_promoters_all_200=open("/lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_200aroundtsseach.bed", "w")

for gene in gene_coords_200:
	output_promoters_all_200.write("%s\t%s\t%s\t%s\n" % (gene_chr[gene], gene_coords_200[gene][0], gene_coords_200[gene][1], gene))


output_promoters_all_200.close()


output_promoters_glioblastoma_200=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/glioblastoma_genes/20160505_genes_promoters_glioblastoma_200aroundtsseach.bed", "w")

for gene in gene_coords_200:
	if gene in genes_all.intersection(set(cancergenes_glioblastoma_synonyms)):
		output_promoters_glioblastoma_200.write("%s\t%s\t%s\t%s\n" % (gene_chr[gene], gene_coords_200[gene][0], gene_coords_200[gene][1], gene))


output_promoters_glioblastoma_200.close()


output_promoters_all_200=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200aroundtsseach.bed", "w")

for gene in gene_coords_200:
	cancer="n"
	glio="n"
	if gene in genes_all.intersection(set(cancergenes_all_synonyms)):
		cancer="y"
	if gene in genes_all.intersection(set(cancergenes_glioblastoma_synonyms)):
		glio="y"
	output_promoters_all_200.write("%s\t%s\t%s\t%s\t%s\t%s\n" % (gene_chr[gene], gene_coords_200[gene][0], gene_coords_200[gene][1], gene, cancer, glio))


output_promoters_all_200.close()

```





### Write bed files with promoter coordinates (200bp upstream only) for all, cancer and glioblastoma genes

Obtained from 200bp plus and minus TSS above

```python
genes=open("/lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes.gtf", "r")
genes_lines=genes.readlines()
genes.close()

gene_transcript_strand={}
genes_all=set()
gene_chr={}

for line in genes_lines:
	fields=line.split("\t")
	gene_id=fields[8].split(";")[0].split('"')[1]
	transcript_id=fields[8].split(";")[1].split('"')[1]
	leftcoord=fields[3]
	rightcoord=fields[4]
	strand=fields[6]
	chr=fields[0]
	if "_" not in chr:
		genes_all.add(gene_id)
		gene_chr[gene_id]=chr
		if (gene_id, transcript_id, strand) in gene_transcript_strand:
			if strand=="+":
				gene_transcript_strand[(gene_id, transcript_id, strand)].append(str(int(leftcoord)-1))
			if strand=="-":
				gene_transcript_strand[(gene_id, transcript_id, strand)].append(rightcoord)
		else:
			if strand=="+":
				gene_transcript_strand[(gene_id, transcript_id, strand)]=[str(int(leftcoord)-1)]
			if strand=="-":
				gene_transcript_strand[(gene_id, transcript_id, strand)]=[rightcoord]




gene_transcript_strand_merge={}

for tri in gene_transcript_strand:
	gene_id=tri[0]
	transcript_id=tri[1]
	strand=tri[2]
	if strand=="+":
		gene_transcript_strand_merge[(gene_id, transcript_id, strand)]=min(gene_transcript_strand[tri])
	if strand=="-":
		gene_transcript_strand_merge[(gene_id, transcript_id, strand)]=max(gene_transcript_strand[tri])



gene_coords_200={}

count=0
for gene in genes_all:
	count+=1
	print count
	list_coords=[]
	for tri in gene_transcript_strand_merge:
		if gene==tri[0]:
			strand=tri[2]
			list_coords.append(gene_transcript_strand_merge[tri])
	if strand=="+":
		gene_coords_200[gene]=[int(min(list_coords))-200, int(min(list_coords))]		
	if strand=="-":
		gene_coords_200[gene]=[int(max(list_coords)), int(max(list_coords))+200]		



output_promoters_all_200=open("/lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_200.bed", "w")

for gene in gene_coords_200:
	output_promoters_all_200.write("%s\t%s\t%s\t%s\n" % (gene_chr[gene], gene_coords_200[gene][0], gene_coords_200[gene][1], gene))


output_promoters_all_200.close()


output_promoters_glioblastoma_200=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/glioblastoma_genes/20160505_genes_promoters_glioblastoma_200.bed", "w")

for gene in gene_coords_200:
	if gene in genes_all.intersection(set(cancergenes_glioblastoma_synonyms)):
		output_promoters_glioblastoma_200.write("%s\t%s\t%s\t%s\n" % (gene_chr[gene], gene_coords_200[gene][0], gene_coords_200[gene][1], gene))


output_promoters_glioblastoma_200.close()


output_promoters_all_200=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200.bed", "w")

for gene in gene_coords_200:
	cancer="n"
	glio="n"
	if gene in genes_all.intersection(set(cancergenes_all_synonyms)):
		cancer="y"
	if gene in genes_all.intersection(set(cancergenes_glioblastoma_synonyms)):
		glio="y"
	output_promoters_all_200.write("%s\t%s\t%s\t%s\t%s\t%s\n" % (gene_chr[gene], gene_coords_200[gene][0], gene_coords_200[gene][1], gene, cancer, glio))


output_promoters_all_200.close()


```






### Sort promoter bed files

Modified from /data/sblab/group_folders/martin03/projects/20150921_BrainMethylomeRoadMap/scripts/20160226_tsg_ocg.md

```bash
sort -k1,1 -k2,2n /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_200aroundtsseach.bed > /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_200aroundtsseach_sorted.bed
rm /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_200aroundtsseach.bed

sort -k1,1 -k2,2n /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/glioblastoma_genes/20160505_genes_promoters_glioblastoma_200aroundtsseach.bed > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/glioblastoma_genes/20160505_genes_promoters_glioblastoma_200aroundtsseach_sorted.bed
rm /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/glioblastoma_genes/20160505_genes_promoters_glioblastoma_200aroundtsseach.bed

sort -k1,1 -k2,2n /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200aroundtsseach.bed > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200aroundtsseach_sorted.bed
rm /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200aroundtsseach.bed

sort -k1,1 -k2,2n /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_200.bed > /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_200_sorted.bed
rm /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_200.bed

sort -k1,1 -k2,2n /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/glioblastoma_genes/20160505_genes_promoters_glioblastoma_200.bed > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/glioblastoma_genes/20160505_genes_promoters_glioblastoma_200_sorted.bed
rm /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/glioblastoma_genes/20160505_genes_promoters_glioblastoma_200.bed

sort -k1,1 -k2,2n /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200.bed > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200_sorted.bed
rm /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200.bed

```





### Intersect promoter bed and methylation files

Modified from /data/sblab/group_folders/martin03/projects/20150921_BrainMethylomeRoadMap/scripts/20160226_tsg_ocg.md

```bash
bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear042_M8BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200aroundtsseach_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10,11,12 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear043_M8oxBS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200aroundtsseach_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10,11,12 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear044_T3BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200aroundtsseach_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10,11,12 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear045_T3oxBS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200aroundtsseach_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10,11,12 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear042_M8BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10,11,12 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear043_M8oxBS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10,11,12 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear044_T3BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10,11,12 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear045_T3oxBS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cancer_genes/20160505_genes_promoters_all_cancer_glioblastoma_200_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10,11,12 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed

```

The columns of the files generated above are:

1st : chr
2nd,3rd : coordinates of nucleotide
4th : cnt_met
5th : cnt_tot
6th : + or - (C or G)
7th : gene names for all genes with methylation data
8th : is it a cancer gene according to COSMIC? (y/n)
9th : is it a glioblastoma gene according to COSMIC? (y/n)


```bash
wc -l ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed # 1233183
wc -l ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed # 1232381
wc -l ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed # 1232879
wc -l ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed # 1232631

cut -f 7 ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed | sort | uniq | wc -l # 22677 genes in total
cut -f 7 ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed | sort | uniq | wc -l # 22672 genes in total
cut -f 7 ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed | sort | uniq | wc -l # 22680 genes in total
cut -f 7 ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed | sort | uniq | wc -l # 22681 genes in total

wc -l ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed # 586402
wc -l ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed # 586212
wc -l ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed # 586369
wc -l ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed # 586325

cut -f 7 ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed | sort | uniq | wc -l # 21651 genes in total
cut -f 7 ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed | sort | uniq | wc -l # 21646 genes in total
cut -f 7 ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed | sort | uniq | wc -l # 21646 genes in total
cut -f 7 ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed | sort | uniq | wc -l # 21648 genes in total

```





### Plot promoter 5mC and 5hmC for a list of test genes using pheatmap (200aroundtsseach)


```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200aroundtsseach.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]

plot(M8.T3.pct.promoter$count, M8.T3.pct.promoter$pct_5mC.M8)
plot(M8.T3.pct.promoter$count, M8.T3.pct.promoter$pct_5hmC.M8)
plot(M8.T3.pct.promoter$count, M8.T3.pct.promoter$pct_5mC.T3)
plot(M8.T3.pct.promoter$count, M8.T3.pct.promoter$pct_5hmC.T3)


library(pheatmap)

testgenes <- read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/test_genes/20160505_test_genes.txt")
testgenes <- as.vector(testgenes$V1)

M8.T3.pct.promoter.data.frame <- data.frame(M8.T3.pct.promoter)
M8.T3.pct.promoter.testgenes <- M8.T3.pct.promoter.data.frame[M8.T3.pct.promoter.data.frame$gene_name %in% testgenes, 2:ncol(M8.T3.pct.promoter.data.frame)]
rownames(M8.T3.pct.promoter.testgenes) <- M8.T3.pct.promoter.data.frame[M8.T3.pct.promoter.data.frame$gene_name %in% testgenes, ]$gene_name
M8.T3.pct.promoter.testgenes <- M8.T3.pct.promoter.testgenes[order(rownames(M8.T3.pct.promoter.testgenes)),]

M8.T3.pct.promoter.testgenes.5mC.5hmC <- M8.T3.pct.promoter.testgenes[c("pct_5mC.M8", "pct_5hmC.M8", "pct_5mC.T3", "pct_5hmC.T3")]
colnames(M8.T3.pct.promoter.testgenes.5mC.5hmC) <- c("5mC M", "5hmC M", "5mC T", "5hmC T")
M8.T3.pct.promoter.testgenes.cancer <- M8.T3.pct.promoter.testgenes["cancer"]
M8.T3.pct.promoter.testgenes.glioblastoma <- M8.T3.pct.promoter.testgenes["glioblastoma"]
M8.T3.pct.promoter.testgenes.count <- M8.T3.pct.promoter.testgenes["count"]

pheatmap(M8.T3.pct.promoter.testgenes.5mC.5hmC, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_M8.T3.pct.promoter.testgenes.5mC.5hmC.200aroundtsseach.pdf")

anno_row <- cbind(M8.T3.pct.promoter.testgenes.glioblastoma, M8.T3.pct.promoter.testgenes.cancer, M8.T3.pct.promoter.testgenes.count)
anno_col <- data.frame(sample = c("Margin", "Margin", "Tumour", "Tumour"))
rownames(anno_col) <- colnames(M8.T3.pct.promoter.testgenes.5mC.5hmC)
anno_colors = list(glioblastoma = c(y = "black", n = "white"), cancer = c(y = "black", n = "white"), sample = c(Margin = "orange", Tumour = "red"))
pheatmap(M8.T3.pct.promoter.testgenes.5mC.5hmC, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_col = anno_col, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, labels_col = c("5mC", "5hmC", "5mC", "5hmC"), filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_M8.T3.pct.promoter.testgenes.5mC.5hmC.200aroundtsseach.annotations.pdf", width = 15/2.54, height = 15/2.54)

# if I want to show the numbers, just add:
# display_numbers = TRUE, number_format = "%.0f", number_color = "black", fontsize_number = 10

```





### Plot promoter 5mC and 5hmC for a list of test genes using pheatmap (1000)

```R
library(data.table)
library(ggplot2)
library(pheatmap)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]

plot(M8.T3.pct.promoter$count, M8.T3.pct.promoter$pct_5mC.M8)
plot(M8.T3.pct.promoter$count, M8.T3.pct.promoter$pct_5hmC.M8)
plot(M8.T3.pct.promoter$count, M8.T3.pct.promoter$pct_5mC.T3)
plot(M8.T3.pct.promoter$count, M8.T3.pct.promoter$pct_5hmC.T3)


testgenes <- read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/test_genes/20160505_test_genes.txt")
testgenes <- as.vector(testgenes$V1)

M8.T3.pct.promoter.data.frame <- data.frame(M8.T3.pct.promoter)
M8.T3.pct.promoter.testgenes <- M8.T3.pct.promoter.data.frame[M8.T3.pct.promoter.data.frame$gene_name %in% testgenes, 2:ncol(M8.T3.pct.promoter.data.frame)]
rownames(M8.T3.pct.promoter.testgenes) <- M8.T3.pct.promoter.data.frame[M8.T3.pct.promoter.data.frame$gene_name %in% testgenes, ]$gene_name
M8.T3.pct.promoter.testgenes <- M8.T3.pct.promoter.testgenes[order(rownames(M8.T3.pct.promoter.testgenes)),]

M8.T3.pct.promoter.testgenes.5mC.5hmC <- M8.T3.pct.promoter.testgenes[c("pct_5mC.M8", "pct_5mC.T3", "pct_5hmC.M8", "pct_5hmC.T3")]
colnames(M8.T3.pct.promoter.testgenes.5mC.5hmC) <- c("5mC M", "5mC T", "5hmC M", "5hmC T")
M8.T3.pct.promoter.testgenes.cancer <- M8.T3.pct.promoter.testgenes["cancer"]
M8.T3.pct.promoter.testgenes.glioblastoma <- M8.T3.pct.promoter.testgenes["glioblastoma"]
M8.T3.pct.promoter.testgenes.count <- M8.T3.pct.promoter.testgenes["count"]

pheatmap(M8.T3.pct.promoter.testgenes.5mC.5hmC, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_M8.T3.pct.promoter.testgenes.5mC.5hmC.1000.pdf")

anno_row <- cbind(M8.T3.pct.promoter.testgenes.glioblastoma, M8.T3.pct.promoter.testgenes.cancer, M8.T3.pct.promoter.testgenes.count)
anno_col <- data.frame(sample = c("Margin", "Tumour", "Margin", "Tumour"))
rownames(anno_col) <- colnames(M8.T3.pct.promoter.testgenes.5mC.5hmC)
anno_colors = list(glioblastoma = c(y = "black", n = "white"), cancer = c(y = "black", n = "white"), sample = c(Margin = "orange", Tumour = "red"))
pheatmap(M8.T3.pct.promoter.testgenes.5mC.5hmC, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_col = anno_col, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, labels_col = c("5mC", "5mC", "5hmC", "5hmC"), filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_M8.T3.pct.promoter.testgenes.5mC.5hmC.1000.annotations.pdf", width = 15/2.54, height = 15/2.54)

```





### Plot promoter 5mC and 5hmC for a list of test genes using pheatmap (200)


```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.200.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]

plot(M8.T3.pct.promoter$count, M8.T3.pct.promoter$pct_5mC.M8)
plot(M8.T3.pct.promoter$count, M8.T3.pct.promoter$pct_5hmC.M8)
plot(M8.T3.pct.promoter$count, M8.T3.pct.promoter$pct_5mC.T3)
plot(M8.T3.pct.promoter$count, M8.T3.pct.promoter$pct_5hmC.T3)


library(pheatmap)

testgenes <- read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/test_genes/20160505_test_genes.txt")
testgenes <- as.vector(testgenes$V1)

M8.T3.pct.promoter.data.frame <- data.frame(M8.T3.pct.promoter)
M8.T3.pct.promoter.testgenes <- M8.T3.pct.promoter.data.frame[M8.T3.pct.promoter.data.frame$gene_name %in% testgenes, 2:ncol(M8.T3.pct.promoter.data.frame)]
rownames(M8.T3.pct.promoter.testgenes) <- M8.T3.pct.promoter.data.frame[M8.T3.pct.promoter.data.frame$gene_name %in% testgenes, ]$gene_name
M8.T3.pct.promoter.testgenes <- M8.T3.pct.promoter.testgenes[order(rownames(M8.T3.pct.promoter.testgenes)),]

M8.T3.pct.promoter.testgenes.5mC.5hmC <- M8.T3.pct.promoter.testgenes[c("pct_5mC.M8", "pct_5hmC.M8", "pct_5mC.T3", "pct_5hmC.T3")]
colnames(M8.T3.pct.promoter.testgenes.5mC.5hmC) <- c("5mC M", "5hmC M", "5mC T", "5hmC T")
M8.T3.pct.promoter.testgenes.cancer <- M8.T3.pct.promoter.testgenes["cancer"]
M8.T3.pct.promoter.testgenes.glioblastoma <- M8.T3.pct.promoter.testgenes["glioblastoma"]
M8.T3.pct.promoter.testgenes.count <- M8.T3.pct.promoter.testgenes["count"]

pheatmap(M8.T3.pct.promoter.testgenes.5mC.5hmC, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_M8.T3.pct.promoter.testgenes.5mC.5hmC.200.pdf")

anno_row <- cbind(M8.T3.pct.promoter.testgenes.glioblastoma, M8.T3.pct.promoter.testgenes.cancer, M8.T3.pct.promoter.testgenes.count)
anno_col <- data.frame(sample = c("Margin", "Margin", "Tumour", "Tumour"))
rownames(anno_col) <- colnames(M8.T3.pct.promoter.testgenes.5mC.5hmC)
anno_colors = list(glioblastoma = c(y = "black", n = "white"), cancer = c(y = "black", n = "white"), sample = c(Margin = "orange", Tumour = "red"))
pheatmap(M8.T3.pct.promoter.testgenes.5mC.5hmC, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_col = anno_col, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, labels_col = c("5mC", "5hmC", "5mC", "5hmC"), filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_M8.T3.pct.promoter.testgenes.5mC.5hmC.200.annotations.pdf", width = 15/2.54, height = 15/2.54)

```





### Plot distributions of promoter 5mC and 5hmC (1000) coloured in the same gradient as the pheatmaps above

This will help to have an idea whether the values in the heatmap are high or low in relative terms

```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]

p <- ggplot(M8.T3.pct.promoter, aes(pct_5mC.M8)) + geom_density() + theme_classic() + xlab("%5mC Margin")
dfp <- ggplot_build(p)$data[[1]]
delta <- diff(dfp$x)[1]
ggplot(dfp) + geom_segment(aes(x = x - delta/2, xend = x + delta/2, y = ymin, yend = ymax, color = x)) + scale_colour_gradientn(colours = colorRampPalette(c("white", "darkblue"))(100)) + theme_classic() + xlab("%5mC Margin") + ylab("density") + theme(legend.position="none")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_M8.promoter.5mC.1000.distribution.pdf", width = 10/2.54, height = 10/2.54)

p <- ggplot(M8.T3.pct.promoter, aes(pct_5hmC.M8)) + geom_density() + theme_classic() + xlab("%5hmC Margin") + coord_cartesian(xlim = c(0, 100))
dfp <- ggplot_build(p)$data[[1]]
delta <- diff(dfp$x)[1]
ggplot(dfp) + geom_segment(aes(x = x - delta/2, xend = x + delta/2, y = ymin, yend = ymax, color = x)) + scale_colour_gradientn(colours = colorRampPalette(c("white", "darkblue"))(100)) + theme_classic() + xlab("%5hmC Margin") + ylab("density") + theme(legend.position="none") + coord_cartesian(xlim = c(0, 100))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_M8.promoter.5hmC.1000.distribution.pdf", width = 10/2.54, height = 10/2.54)

p <- ggplot(M8.T3.pct.promoter, aes(pct_5mC.T3)) + geom_density() + theme_classic() + xlab("%5mC Tumour")
dfp <- ggplot_build(p)$data[[1]]
delta <- diff(dfp$x)[1]
ggplot(dfp) + geom_segment(aes(x = x - delta/2, xend = x + delta/2, y = ymin, yend = ymax, color = x)) + scale_colour_gradientn(colours = colorRampPalette(c("white", "darkblue"))(100)) + theme_classic() + xlab("%5mC Tumour") + ylab("density") + theme(legend.position="none")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_T3.promoter.5mC.1000.distribution.pdf", width = 10/2.54, height = 10/2.54)

p <- ggplot(M8.T3.pct.promoter, aes(pct_5hmC.T3)) + geom_density() + theme_classic() + xlab("%5hmC Tumour") + coord_cartesian(xlim = c(0, 100))
dfp <- ggplot_build(p)$data[[1]]
delta <- diff(dfp$x)[1]
ggplot(dfp) + geom_segment(aes(x = x - delta/2, xend = x + delta/2, y = ymin, yend = ymax, color = x)) + scale_colour_gradientn(colours = colorRampPalette(c("white", "darkblue"))(100)) + theme_classic() + xlab("%5hmC Tumour") + ylab("density") + theme(legend.position="none") + coord_cartesian(xlim = c(0, 100))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_T3.promoter.5hmC.1000.distribution.pdf", width = 10/2.54, height = 10/2.54)

```





### Plot expression of a list of test genes using pheatmap

Also the scatterplot tpmLog2FC vs. tpmLog2Avg to check levels


```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]


tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmLog2Avg<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmLog2Avg)][order(chrom, promStart, promEnd)]

M8.T3.pct.promoter.gx <- merge(M8.T3.pct.promoter, gx, by.x="gene_name", by.y="Associated_Gene_Name")

library(pheatmap)

testgenes <- read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/test_genes/20160505_test_genes.txt")
testgenes <- as.vector(testgenes$V1)

M8.T3.pct.promoter.gx.data.frame <- data.frame(M8.T3.pct.promoter.gx)
M8.T3.pct.promoter.gx.testgenes <- M8.T3.pct.promoter.gx.data.frame[M8.T3.pct.promoter.gx.data.frame$gene_name %in% testgenes, 2:ncol(M8.T3.pct.promoter.gx.data.frame)]
rownames(M8.T3.pct.promoter.gx.testgenes) <- M8.T3.pct.promoter.gx.data.frame[M8.T3.pct.promoter.gx.data.frame$gene_name %in% testgenes, ]$gene_name
M8.T3.pct.promoter.gx.testgenes <- M8.T3.pct.promoter.gx.testgenes[order(rownames(M8.T3.pct.promoter.gx.testgenes)),]

M8.T3.pct.promoter.gx.testgenes.tpmLog2FC <- M8.T3.pct.promoter.gx.testgenes["tpmLog2FC"]
colnames(M8.T3.pct.promoter.gx.testgenes.tpmLog2FC) <- "log2(expression T/M)"
M8.T3.pct.promoter.gx.testgenes.tpmLog2Avg <- M8.T3.pct.promoter.gx.testgenes["tpmLog2Avg"]
colnames(M8.T3.pct.promoter.gx.testgenes.tpmLog2Avg) <- "log2(expression average)"

pheatmap(M8.T3.pct.promoter.gx.testgenes.tpmLog2FC, color = colorRampPalette(c("darkred","white","darkgreen"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_M8.T3.pct.promoter.gx.testgenes.tpmLog2FC.1000.pdf")

pheatmap(M8.T3.pct.promoter.gx.testgenes.tpmLog2Avg, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_M8.T3.pct.promoter.gx.testgenes.tpmLog2Avg.1000.pdf")

ggplot(M8.T3.pct.promoter.gx, aes(x = tpmLog2Avg, y = tpmLog2FC)) + geom_point(alpha = 0.05, size=0.5) + xlab("log2(expression average)") + ylab("log2(expression T/M)") + theme_classic()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160506_M8.T3.promoter.gx.tpmLog2FC.tpmLog2Avg.1000.distribution.pdf", width = 10/2.54, height = 10/2.54)

```







### Plot expression for a list of test genes using pheatmap - adding CNVs as annotations

Modified from /data/sblab/group_folders/martin03/projects/20150921_BrainMethylomeRoadMap/scripts/20160113_cnv.md


```R
library(VariantAnnotation)
library(data.table)
vcf<-readVcf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv/20160211/LP2000729-DNA_C01_LP2000729-DNA_E01_G1_P1.somatic.SV.vcf", "hg19")
vcf

# Examine header(vcf) and vcf
header(vcf)

samples(header(vcf)) # "LP2000729-DNA_C01" "LP2000729-DNA_E01"

as.data.frame(meta(header(vcf))$META) # EstimatedTumorPurity 0.71, OverallPloidy 1.92

fixed(header(vcf))$FILTER # description of features about the different alterations
fixed(header(vcf))$ALT # description of types of alterations
as.data.frame(fixed(vcf)$REF)$x # reference sequences
as.data.frame(fixed(vcf)$ALT)$value # list of types of alterations: CNV and DEL
fixed(vcf)$QUAL # list of quality scores
fixed(vcf)$FILTER # list of features

as.data.frame(info(header(vcf))) # description of more features about the different alterations
info(vcf) # list of more features about the different alterations
rownames(info(vcf)) # list of alterations, e.g. "Canvas:LOSS:chr17:44361921:44565480"
rownames(info(vcf))[grepl("Canvas", rownames(info(vcf)))] # grep "Canvas" to get CNVs
info(vcf)$SVTYPE[grepl("Canvas", rownames(info(vcf)))]
# NA is REF in rownames and refers to diploid, "CNV" is either LOSS or GAIN in rownames, "LOH" is REF in rownames and refers to diploid with loss of heterogeneity (see chr17)

geno(header(vcf)) # description of more features about the different alterations
geno(vcf)$CN[,2] # list of copy number variations
geno(vcf)$MCC[,2] # list of copy number variations with diploid being NA

# Putting all useful info together
cbind(rownames(info(vcf))[grepl("Canvas", rownames(info(vcf)))], info(vcf)$SVTYPE[grepl("Canvas", rownames(info(vcf)))], geno(vcf)$CN[,2][grepl("Canvas", rownames(info(vcf)))])


```


### Produce a bedGraph with gene locations for hg19 (CNV file has also been produced using this assembly):


```python
genes=open("/lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes.gtf", "r")
genes_lines=genes.readlines()
genes.close()

gene_coords={}
gene_chr={}

for line in genes_lines:
	fields=line.split("\t")
	gene_id=fields[8].split(";")[0].split('"')[1]
	leftcoord=fields[3]
	rightcoord=fields[4]
	chr=fields[0]
	if "_" not in chr:
		gene_chr[gene_id] = chr
		if gene_id in gene_coords:
			gene_coords[gene_id] = gene_coords[gene_id] + [int(leftcoord)-1, rightcoord]
		else:
			gene_coords[gene_id] = [int(leftcoord)-1, int(rightcoord)]		





ofile = open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160510_gene_locations.bedGraph", "w")

for gene in gene_coords:
	ofile.write("%s\t%s\t%s\t%s\n" % (gene_chr[gene], int(min(gene_coords[gene])), int(max(gene_coords[gene])), gene))



ofile.close()

```


### Sort gene locations bedGraph:

```bash
sort -k1,1 -k2,2n /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160510_gene_locations.bedGraph > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160510_gene_locations_sorted.bedGraph
rm /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160510_gene_locations.bedGraph
cp /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160510_gene_locations_sorted.bedGraph /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/gene_locations_sorted.bedGraph # copy for future reference

```


### Produce a bedGraph with CNV locations from /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv/20160211/LP2000729-DNA_C01_LP2000729-DNA_E01_G1_P1.somatic.SV.vcf


```python
import vcf # https://pyvcf.readthedocs.org/en/latest/index.html
vcf_reader = vcf.Reader(open('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv/20160211/LP2000729-DNA_C01_LP2000729-DNA_E01_G1_P1.somatic.SV.vcf', 'r'))

ofile = open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160510_CNV.bedGraph", "w")

for record in vcf_reader:
	if "Canvas" in record.ID:
		chr = record.ID.split(":")[2]
		start = record.ID.split(":")[3]
		end = record.ID.split(":")[4]
		ploidy = record.ID.split(":")[1]
		number = record.samples[1]["CN"]
		if 'SVTYPE' in record.INFO.keys():
			if record.INFO['SVTYPE'] == "LOH":
				ploidy = "LOH"
		ofile.write("%s\t%s\t%s\t%s\t%s\n" % (chr, start, end, ploidy, number))


ofile.close()

```

```bash
grep "REF" 20160510_CNV.bedGraph | wc -l # 298 diploid REF regions
grep -v "REF" 20160510_CNV.bedGraph | wc -l # 262 GAIN, LOSS or LOH regions
```

this bedGraph already comes sorted





### Intersect gene locations and CNV bedGraphs

```bash
bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160510_gene_locations_sorted.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160510_CNV.bedGraph -loj > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160511_gene_locations_CNV.bedGraph
wc -l /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160510_gene_locations_sorted.bedGraph # 23346
wc -l /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160511_gene_locations_CNV.bedGraph # 23751

```

```python
ifile = open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160511_gene_locations_CNV.bedGraph", "r")
ilines = ifile.readlines()
ifile.close()

genes = [line.split()[3] for line in ilines]

from collections import Counter

cnt = Counter()

for gene in genes:
	cnt[gene] +=1

cnt.most_common(10) # [('MIR4283-2', 20), ('MIR4283-1', 20), ('MIR1244-3', 20), ('MIR1244-2', 20), ('MIR1244-1', 20), ('POTEB', 11), ('MIR4435-2', 8), ('MIR4435-1', 8), ('DUX4L6', 7), ('DUX4L5', 7)]

```

```bash
grep "MIR4283-2" /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160511_gene_locations_CNV.bedGraph
#chr7	57023491	63081547	MIR4283-2	chr7	61738026	61739086	GAIN	16
#chr7	57023491	63081547	MIR4283-2	chr7	61780081	61783606	GAIN	10
#chr7	57023491	63081547	MIR4283-2	chr7	61785285	61787573	GAIN	14
#chr7	57023491	63081547	MIR4283-2	chr7	61787574	61796164	GAIN	8
#chr7	57023491	63081547	MIR4283-2	chr7	61796534	61799788	GAIN	12
#chr7	57023491	63081547	MIR4283-2	chr7	61818285	61818823	GAIN	25
#chr7	57023491	63081547	MIR4283-2	chr7	61821247	61822012	GAIN	12
#chr7	57023491	63081547	MIR4283-2	chr7	57999566	58054191	GAIN	4
#chr7	57023491	63081547	MIR4283-2	chr7	61661899	61674128	GAIN	4
#chr7	57023491	63081547	MIR4283-2	chr7	61739485	61757691	GAIN	10
#chr7	57023491	63081547	MIR4283-2	chr7	61757692	61779213	GAIN	7
#chr7	57023491	63081547	MIR4283-2	chr7	61800267	61802991	GAIN	7
#chr7	57023491	63081547	MIR4283-2	chr7	61834311	61859914	GAIN	3
#chr7	57023491	63081547	MIR4283-2	chr7	57748707	57996820	GAIN	3
#chr7	57023491	63081547	MIR4283-2	chr7	61054631	61525154	GAIN	4
#chr7	57023491	63081547	MIR4283-2	chr7	61990894	62126082	GAIN	3
#chr7	57023491	63081547	MIR4283-2	chr7	57605306	57748706	GAIN	4
#chr7	57023491	63081547	MIR4283-2	chr7	61859915	61905500	GAIN	7
#chr7	57023491	63081547	MIR4283-2	chr7	17930752	57605305	GAIN	3
#chr7	57023491	63081547	MIR4283-2	chr7	62159209	73851412	GAIN	3

```

There are genes having multiple forms of CNV, e.g. MIR4283-2 is associated to 20 different CNVs



Plotting expression as above and adding CNVs as annotation:

```R
library(data.table)
library(ggplot2)
library(pheatmap)

tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmLog2Avg<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmLog2Avg)][order(chrom, promStart, promEnd)]

testgenes <- read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/test_genes/20160505_test_genes.txt")
testgenes <- as.vector(testgenes$V1)

gx.data.frame <- data.frame(gx)
gx.testgenes <- gx.data.frame[gx.data.frame$Associated_Gene_Name %in% testgenes, ]
rownames(gx.testgenes) <- gx.data.frame[gx.data.frame$Associated_Gene_Name %in% testgenes, ]$Associated_Gene_Name
gx.testgenes <- gx.testgenes[order(rownames(gx.testgenes)),]

gx.testgenes.tpmLog2FC <- gx.testgenes["tpmLog2FC"]
colnames(gx.testgenes.tpmLog2FC) <- "log2(expression T/M)"
gx.testgenes.tpmLog2Avg <- gx.testgenes["tpmLog2Avg"]
colnames(gx.testgenes.tpmLog2Avg) <- "log2(expression average)"


gene_locations_CNV <-read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160511_gene_locations_CNV.bedGraph")

testgenes <- read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/test_genes/20160505_test_genes.txt")
testgenes <- as.vector(testgenes$V1)

gene_locations_CNV.data.frame <- data.frame(gene_locations_CNV)
gene_locations_CNV.testgenes <- gene_locations_CNV.data.frame[gene_locations_CNV.data.frame$V4 %in% testgenes,c(8,9)]
rownames(gene_locations_CNV.testgenes) <- gene_locations_CNV.data.frame[gene_locations_CNV.data.frame$V4 %in% testgenes,]$V4
gene_locations_CNV.testgenes <- gene_locations_CNV.testgenes[order(rownames(gene_locations_CNV.testgenes)),]

gene_locations_CNV.testgenes.cnvtype <- gene_locations_CNV.testgenes["V8"]
colnames(gene_locations_CNV.testgenes.cnvtype) <- "CNV"
gene_locations_CNV.testgenes.cnvnumber <- gene_locations_CNV.testgenes["V9"]
colnames(gene_locations_CNV.testgenes.cnvnumber) <- "CNV"


anno_row <- gene_locations_CNV.testgenes.cnvtype
anno_colors <- list(CNV = c(GAIN = "darkcyan", REF = "white", LOSS = "darkorange3"))

pheatmap(gx.testgenes.tpmLog2FC, color = colorRampPalette(c("darkred","white","darkgreen"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160511_gx.testgenes.tpmLog2FC.CNV.pdf", width = 15/2.54, height = 15/2.54)

pheatmap(gx.testgenes.tpmLog2Avg, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160511_gx.testgenes.tpmLog2Avg.CNV.pdf", width = 15/2.54, height = 15/2.54)

ggplot(gx, aes(x = tpmLog2Avg, y = tpmLog2FC)) + geom_point(alpha = 0.05, size=0.5) + xlab("log2(expression average)") + ylab("log2(expression T/M)") + theme_classic()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160511_gx.tpmLog2FC.tpmLog2Avg.distribution.pdf", width = 10/2.54, height = 10/2.54)



```



### Plot expression for a list of test genes using pheatmap - adding both CNVs and SNVs as annotations

### Intersect gene locations and SNV bedGraphs

```bash
bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160510_gene_locations_sorted.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_C01.somatic.bedGraph -wa -wb > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160516_gene_locations_SNV.bedGraph
wc -l /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160516_gene_locations_SNV.bedGraph # 6531
cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160516_gene_locations_SNV.bedGraph | cut -f 4 | uniq | wc -l # 1892 - which is surprisingly high

```

Plotting expression as above and adding CNVs and SNVs as annotation:

```R
library(data.table)
library(ggplot2)
library(pheatmap)

tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmLog2Avg<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmLog2Avg)][order(chrom, promStart, promEnd)]

testgenes <- read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/test_genes/20160505_test_genes.txt")
testgenes <- as.vector(testgenes$V1)

gx.data.frame <- data.frame(gx)
gx.testgenes <- gx.data.frame[gx.data.frame$Associated_Gene_Name %in% testgenes, ]
rownames(gx.testgenes) <- gx.data.frame[gx.data.frame$Associated_Gene_Name %in% testgenes, ]$Associated_Gene_Name
gx.testgenes <- gx.testgenes[order(rownames(gx.testgenes)),]

gx.testgenes.tpmLog2FC <- gx.testgenes["tpmLog2FC"]
colnames(gx.testgenes.tpmLog2FC) <- "log2(expression T/M)"
gx.testgenes.tpmLog2Avg <- gx.testgenes["tpmLog2Avg"]
colnames(gx.testgenes.tpmLog2Avg) <- "log2(expression average)"


gene_locations_CNV <-read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160511_gene_locations_CNV.bedGraph")

gene_locations_CNV.data.frame <- data.frame(gene_locations_CNV)
gene_locations_CNV.testgenes <- gene_locations_CNV.data.frame[gene_locations_CNV.data.frame$V4 %in% testgenes,c(8,9)]
rownames(gene_locations_CNV.testgenes) <- gene_locations_CNV.data.frame[gene_locations_CNV.data.frame$V4 %in% testgenes,]$V4
gene_locations_CNV.testgenes <- gene_locations_CNV.testgenes[order(rownames(gene_locations_CNV.testgenes)),]

gene_locations_CNV.testgenes.cnvtype <- gene_locations_CNV.testgenes["V8"]
colnames(gene_locations_CNV.testgenes.cnvtype) <- "CNV"
gene_locations_CNV.testgenes.cnvnumber <- gene_locations_CNV.testgenes["V9"]
colnames(gene_locations_CNV.testgenes.cnvnumber) <- "CNV"


gene_locations_SNV <-read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160516_gene_locations_SNV.bedGraph")

gene_locations_SNV.mutatedgenes <- gene_locations_SNV$V4

snv_vector <- c()
for (gene in rownames(gene_locations_CNV.testgenes.cnvtype)){
	if (gene %in% gene_locations_SNV.mutatedgenes){
		snv_vector <- append(snv_vector, "SNV")
	}
	else{
		snv_vector <- append(snv_vector, "nonSNV")
	}
}


anno_row <- data.frame(CNV = as.vector(gene_locations_CNV.testgenes.cnvtype$CNV), SNV = snv_vector, row.names = rownames(gene_locations_CNV.testgenes.cnvtype))
anno_colors <- list(CNV = c(GAIN = "darkcyan", REF = "white", LOSS = "darkorange3"), SNV = c(SNV = "#999999", nonSNV = "white"))

pheatmap(gx.testgenes.tpmLog2FC, color = colorRampPalette(c("darkred","white","darkgreen"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160516_gx.testgenes.tpmLog2FC.CNV.SNV.pdf", width = 15/2.54, height = 15/2.54)

pheatmap(gx.testgenes.tpmLog2Avg, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160516_gx.testgenes.tpmLog2Avg.CNV.SNV.pdf", width = 15/2.54, height = 15/2.54)

```





### Putting it all together for epigenetics genes

FC Expression (sorted)
promoter 5mC/5hmC in Tumour and Margin
CNVs
SNVs
CpG islands
CpG sites count

Where do we get epigenetic genes from?

Manual lists in /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/epigenetic_genes
Original list from Gordon:
20151130_genelist_rnaseq_alternative.txt
Modified and extended lists:
20160516_epigenetic_genes_long.txt # DNA and histone modifiers
20160516_epigenetic_genes_short.txt # DNA modifiers only (changers + binders), e.g. an example of changer is TET and an example of binder is MECP2
20160523_epigenetic_genes_short.txt # DNA modifiers only (changers only)
20160524_epigenetic_genes_TET_DNMT.txt # TET and DNMT only


Another way:
Download from Uniprot a list of gene names associated to GO functions related to the epigenetic activity that we look for, e.g. GO:0070579 methylcytosine dioxygenase activity
http://www.uniprot.org/uniprot/?query=go%3A0070579+AND+organism%3A%22Homo+sapiens+[9606]%22&sort=score
Then put the lists together for the different GO terms



#### 20160523_epigenetic_genes_short.txt

```R
library(data.table)
library(ggplot2)
library(pheatmap)



# Expression
tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]


gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmAvgLog2<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx$tpmLog2Avg<- gx[, list(log2((ear047_F3 + ear049_M3 + 0.01)/2))]
gx$tpmAvg<- rowMeans(gx[, list(ear047_F3, ear049_M3)])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmAvgLog2, tpmLog2Avg, tpmAvg)][order(chrom, promStart, promEnd)]

p <- ggplot(gx, aes(tpmLog2FC)) + geom_density()
dfp <- ggplot_build(p)$data[[1]]
delta <- diff(dfp$x)[1]
ggplot(dfp) + geom_segment(aes(x = x - delta/2, xend = x + delta/2, y = ymin, yend = ymax, color = x)) + scale_colour_gradientn(colours = colorRampPalette(c("darkred","white","darkgreen"))(100)) + theme_classic() + xlab("log2(expression T/M)") + ylab("density") + theme(legend.position="none")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160520_tpmLog2FC_distribution.pdf", width = 10/2.54, height = 10/2.54)

ggplot(gx, aes(x = tpmAvgLog2, y = tpmLog2FC)) + geom_point(alpha = 0.05, size=0.5) + xlab("Average of log2(expression)") + ylab("log2(expression T/M)") + theme_classic()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160520_tpmLog2FC_tpmAvgLog2_distribution.pdf", width = 10/2.54, height = 10/2.54)

ggplot(gx, aes(x = tpmLog2Avg, y = tpmLog2FC)) + geom_point(alpha = 0.05, size=0.5) + xlab("Log2 of average(expression)") + ylab("log2(expression T/M)") + theme_classic()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160520_tpmLog2FC_tpmLog2Avg_distribution.pdf", width = 10/2.54, height = 10/2.54)



# Epigenetic genes
epigenesshort <- read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/epigenetic_genes/20160523_epigenetic_genes_short.txt")
setnames(epigenesshort, c("gene_name", "class", "base_modifications", "function"))

length(intersect(epigenesshort$gene_name, gx$Associated_Gene_Name)) # 26, but however the total is 27
setdiff(epigenesshort$gene_name, gx$Associated_Gene_Name) # "IDH3GL", it is because it is not validated (maybe)

gx.epigenesshort <- gx[Associated_Gene_Name %in% epigenesshort$gene_name]

gx.epigenesshort.sorted <- gx.epigenesshort[order(-tpmLog2FC)]
setnames(gx.epigenesshort.sorted, "Associated_Gene_Name", "gene_name")

gx.epigenesshort.tpmLog2FC <- data.frame(gx.epigenesshort.sorted$tpmLog2FC)
colnames(gx.epigenesshort.tpmLog2FC) <- "log2(expression T/M)"
rownames(gx.epigenesshort.tpmLog2FC) <- gx.epigenesshort.sorted$gene_name

gx.epigenesshort.tpmAvgLog2 <- data.frame(gx.epigenesshort.sorted$tpmAvgLog2)
colnames(gx.epigenesshort.tpmAvgLog2) <- "Average of log2(expression)"
rownames(gx.epigenesshort.tpmAvgLog2) <- gx.epigenesshort.sorted$gene_name

gx.epigenesshort.tpmLog2Avg <- data.frame(gx.epigenesshort.sorted$tpmLog2Avg)
colnames(gx.epigenesshort.tpmLog2Avg) <- "log2 of average(expression)"
rownames(gx.epigenesshort.tpmLog2Avg) <- gx.epigenesshort.sorted$gene_name

gx.epigenesshort.tpmAvg <- data.frame(gx.epigenesshort.sorted$tpmAvg)
colnames(gx.epigenesshort.tpmAvg) <- "average(expression)"
rownames(gx.epigenesshort.tpmAvg) <- gx.epigenesshort.sorted$gene_name

gx.epigenesshort.class <- data.frame(merge(x = gx.epigenesshort.sorted, y = epigenesshort, by = "gene_name")[order(-tpmLog2FC)]$class)
colnames(gx.epigenesshort.class) <- "class"
rownames(gx.epigenesshort.class) <- merge(x = gx.epigenesshort.sorted, y = epigenesshort, by = "gene_name")[order(-tpmLog2FC)]$gene_name



# CNV
gene_locations_CNV <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160511_gene_locations_CNV.bedGraph")
setnames(gene_locations_CNV, c("chr_gene","start_gene", "end_gene", "gene_name", "chr_CNV", "start_CNV", "end_CNV", "type_CNV", "number_CNV"))

length(intersect(gx.epigenesshort$Associated_Gene_Name, gene_locations_CNV$gene_name)) # 26 OK

CNV.epigenesshort <- gene_locations_CNV[gene_name %in% gx.epigenesshort$Associated_Gene_Name]
CNV.epigenesshort # 26
#gene_locations_CNV[gene_name == "MBD2"] # because MBD2 is both in LOSS and REF, however the majority of MBD2 is in LOSS, so we can remove the corresponding REF row
#CNV.epigenesshort <- CNV.epigenesshort[!(gene_name == "MBD2" & type_CNV == "REF")]


gx.CNV.epigenesshort <- merge(x = gx.epigenesshort.sorted, y = CNV.epigenesshort, by = "gene_name")
CNV.epigenesshort.cnvtype <- data.frame(gx.CNV.epigenesshort[order(-tpmLog2FC)]$type_CNV)
colnames(CNV.epigenesshort.cnvtype) <- "CNV"
rownames(CNV.epigenesshort.cnvtype) <- gx.CNV.epigenesshort[order(-tpmLog2FC)]$gene_name



# SNV
gene_locations_SNV <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160516_gene_locations_SNV.bedGraph")
setnames(gene_locations_SNV, c("chr_gene","start_gene", "end_gene", "gene_name", "chr_SNV", "start_SNV", "end_SNV", "base_from", "base_to", "score", "tier"))

length(intersect(gx.epigenesshort$Associated_Gene_Name, gene_locations_SNV$gene_name)) # 1, only "DNMT1" is hit by a SNV

SNV.epigenesshort <- gene_locations_SNV[gene_name %in% gx.epigenesshort$Associated_Gene_Name]

gx.SNV.epigenesshort <- merge(x = gx.epigenesshort.sorted, y = SNV.epigenesshort, all = TRUE, by = "gene_name")
gx.SNV.epigenesshort[, type_SNV := ifelse(is.na(chr_gene), "nonSNV", "SNV")]
SNV.epigenesshort.snvtype <- data.frame(gx.SNV.epigenesshort[order(-tpmLog2FC)]$type_SNV)
colnames(SNV.epigenesshort.snvtype) <- "SNV"
rownames(SNV.epigenesshort.snvtype) <- gx.SNV.epigenesshort[order(-tpmLog2FC)]$gene_name



# promoter 5mC/5hmC
ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]

length(intersect(gx.epigenesshort.sorted$gene_name, M8.T3.pct.promoter$gene_name)) # 26

gx.5mC.5hmC.epigenesshort <- merge(x = gx.epigenesshort.sorted, y = M8.T3.pct.promoter, by = "gene_name")[order(-tpmLog2FC)]

promoter.5mC.5hmC.epigenesshort <- data.frame(gx.5mC.5hmC.epigenesshort$pct_5mC.M8, gx.5mC.5hmC.epigenesshort$pct_5mC.T3, gx.5mC.5hmC.epigenesshort$pct_5hmC.M8, gx.5mC.5hmC.epigenesshort$pct_5hmC.T3)
colnames(promoter.5mC.5hmC.epigenesshort) <- c("% 5mC promoter M", "% 5mC promoter T", "% 5hmC promoter M", "% 5hmC promoter T")
rownames(promoter.5mC.5hmC.epigenesshort) <- gx.5mC.5hmC.epigenesshort$gene_name


# pheatmaps : expression, CNV, SNV
anno_row <- data.frame(CNV = as.vector(CNV.epigenesshort.cnvtype$CNV), SNV = as.vector(SNV.epigenesshort.snvtype$SNV), row.names = rownames(CNV.epigenesshort.cnvtype))
anno_colors <- list(CNV = c(GAIN = "darkcyan", REF = "white", LOSS = "darkorange3"), SNV = c(SNV = "#999999", nonSNV = "white"))

pheatmap(gx.epigenesshort.tpmLog2FC, color = colorRampPalette(c("darkred","white","darkgreen"), bias = 1.05)(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160523_gx.epigenesshort.tpmLog2FC.CNV.SNV.pdf", width = 11/2.54, height = 22/2.54)

pheatmap(gx.epigenesshort.tpmAvgLog2, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160523_gx.epigenesshort.tpmAvgLog2.CNV.SNV.pdf", width = 11/2.54, height = 24/2.54)

pheatmap(gx.epigenesshort.tpmLog2Avg, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160523_gx.epigenesshort.tpmLog2Avg.CNV.SNV.pdf", width = 11/2.54, height = 24/2.54)

pheatmap(gx.epigenesshort.tpmAvg, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160523_gx.epigenesshort.tpmAvg.CNV.SNV.pdf", width = 11/2.54, height = 22/2.54)



# pheatmaps : expression, CNV, SNV + class of enzymes
anno_row <- data.frame(CNV = as.vector(CNV.epigenesshort.cnvtype$CNV), SNV = as.vector(SNV.epigenesshort.snvtype$SNV), class = as.vector(gx.epigenesshort.class$class), row.names = rownames(CNV.epigenesshort.cnvtype))
anno_colors <- list(CNV = c(GAIN = "darkcyan", REF = "white", LOSS = "darkorange3"), SNV = c(SNV = "#999999", nonSNV = "white"), class = c(AID_APOBEC = 1, DNA_glycosylase = 2, DNMT = 3, IDH = 4, MGMT = 5, TET = 6))

pheatmap(gx.epigenesshort.tpmLog2FC, color = colorRampPalette(c("darkred","white","darkgreen"), bias = 1.05)(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, labels_col = expression("log"[2]*"FC"), filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160523_gx.epigenesshort.tpmLog2FC.CNV.SNV.class.pdf", width = 15/2.54, height = 22/2.54)

pheatmap(gx.epigenesshort.tpmAvgLog2, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, labels_col = expression("mean log"[2]*"(TPM+0.01)"), filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160523_gx.epigenesshort.tpmAvgLog2.CNV.SNV.class.pdf", width = 15/2.54, height = 24/2.54)

pheatmap(gx.epigenesshort.tpmLog2Avg, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, labels_col = expression("log"[2]*"mean(TPM+0.01)"), filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160523_gx.epigenesshort.tpmLog2Avg.CNV.SNV.class.pdf", width = 15/2.54, height = 24/2.54)

pheatmap(gx.epigenesshort.tpmAvg, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, labels_col = expression("mean(TPM+0.01)"), filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160523_gx.epigenesshort.tpmAvg.CNV.SNV.class.pdf", width = 15/2.54, height = 22/2.54)



# pheatmaps : 5mC M, 5mC T, 5hmC M, 5hmC T + count of CpGs
anno_row <- data.frame(count_CpG_sites = as.vector(gx.5mC.5hmC.epigenesshort$count), row.names = rownames(promoter.5mC.5hmC.epigenesshort))

pheatmap(promoter.5mC.5hmC.epigenesshort, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160523_promoter.5mC.5hmC.epigenesshort.count_CpG_sites.pdf", width = 16/2.54, height = 22/2.54)


```





#### /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/epigenetic_genes/20160524_epigenetic_genes_TET_DNMT.txt


```R
library(data.table)
library(ggplot2)
library(pheatmap)



# Expression
tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]


gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmAvgLog2<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx$tpmLog2Avg<- gx[, list(log2((ear047_F3 + ear049_M3 + 0.01)/2))]
gx$tpmAvg<- rowMeans(gx[, list(ear047_F3, ear049_M3)])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmAvgLog2, tpmLog2Avg, tpmAvg)][order(chrom, promStart, promEnd)]



# TET and DNMT
epigenesshort <- read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/epigenetic_genes/20160524_epigenetic_genes_TET_DNMT.txt")
setnames(epigenesshort, c("gene_name", "class", "base_modifications", "function"))

length(intersect(epigenesshort$gene_name, gx$Associated_Gene_Name)) # 7

gx.epigenesshort <- gx[Associated_Gene_Name %in% epigenesshort$gene_name]

gx.epigenesshort.sorted <- gx.epigenesshort[order(-tpmLog2FC)]
setnames(gx.epigenesshort.sorted, "Associated_Gene_Name", "gene_name")

gx.epigenesshort.tpmLog2FC <- data.frame(gx.epigenesshort.sorted$tpmLog2FC)
colnames(gx.epigenesshort.tpmLog2FC) <- "log2(expression T/M)"
rownames(gx.epigenesshort.tpmLog2FC) <- gx.epigenesshort.sorted$gene_name

gx.epigenesshort.tpmAvgLog2 <- data.frame(gx.epigenesshort.sorted$tpmAvgLog2)
colnames(gx.epigenesshort.tpmAvgLog2) <- "Average of log2(expression)"
rownames(gx.epigenesshort.tpmAvgLog2) <- gx.epigenesshort.sorted$gene_name

gx.epigenesshort.tpmLog2Avg <- data.frame(gx.epigenesshort.sorted$tpmLog2Avg)
colnames(gx.epigenesshort.tpmLog2Avg) <- "log2 of average(expression)"
rownames(gx.epigenesshort.tpmLog2Avg) <- gx.epigenesshort.sorted$gene_name

gx.epigenesshort.tpmAvg <- data.frame(gx.epigenesshort.sorted$tpmAvg)
colnames(gx.epigenesshort.tpmAvg) <- "average(expression)"
rownames(gx.epigenesshort.tpmAvg) <- gx.epigenesshort.sorted$gene_name

gx.epigenesshort.class <- data.frame(merge(x = gx.epigenesshort.sorted, y = epigenesshort, by = "gene_name")[order(-tpmLog2FC)]$class)
colnames(gx.epigenesshort.class) <- "class"
rownames(gx.epigenesshort.class) <- merge(x = gx.epigenesshort.sorted, y = epigenesshort, by = "gene_name")[order(-tpmLog2FC)]$gene_name



# CNV
gene_locations_CNV <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160511_gene_locations_CNV.bedGraph")
setnames(gene_locations_CNV, c("chr_gene","start_gene", "end_gene", "gene_name", "chr_CNV", "start_CNV", "end_CNV", "type_CNV", "number_CNV"))

length(intersect(gx.epigenesshort$Associated_Gene_Name, gene_locations_CNV$gene_name)) # 7 OK

CNV.epigenesshort <- gene_locations_CNV[gene_name %in% gx.epigenesshort$Associated_Gene_Name]
CNV.epigenesshort # 7

gx.CNV.epigenesshort <- merge(x = gx.epigenesshort.sorted, y = CNV.epigenesshort, by = "gene_name")
CNV.epigenesshort.cnvtype <- data.frame(gx.CNV.epigenesshort[order(-tpmLog2FC)]$type_CNV)
colnames(CNV.epigenesshort.cnvtype) <- "CNV"
rownames(CNV.epigenesshort.cnvtype) <- gx.CNV.epigenesshort[order(-tpmLog2FC)]$gene_name



# SNV
gene_locations_SNV <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160516_gene_locations_SNV.bedGraph")
setnames(gene_locations_SNV, c("chr_gene","start_gene", "end_gene", "gene_name", "chr_SNV", "start_SNV", "end_SNV", "base_from", "base_to", "score", "tier"))

length(intersect(gx.epigenesshort$Associated_Gene_Name, gene_locations_SNV$gene_name)) # 1, only "DNMT1" is hit by a SNV

SNV.epigenesshort <- gene_locations_SNV[gene_name %in% gx.epigenesshort$Associated_Gene_Name]

gx.SNV.epigenesshort <- merge(x = gx.epigenesshort.sorted, y = SNV.epigenesshort, all = TRUE, by = "gene_name")
gx.SNV.epigenesshort[, type_SNV := ifelse(is.na(chr_gene), "nonSNV", "SNV")]
SNV.epigenesshort.snvtype <- data.frame(gx.SNV.epigenesshort[order(-tpmLog2FC)]$type_SNV)
colnames(SNV.epigenesshort.snvtype) <- "SNV"
rownames(SNV.epigenesshort.snvtype) <- gx.SNV.epigenesshort[order(-tpmLog2FC)]$gene_name



# promoter 5mC/5hmC
ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]

length(intersect(gx.epigenesshort.sorted$gene_name, M8.T3.pct.promoter$gene_name)) # 7

gx.5mC.5hmC.epigenesshort <- merge(x = gx.epigenesshort.sorted, y = M8.T3.pct.promoter, by = "gene_name")[order(-tpmLog2FC)]

promoter.5mC.5hmC.epigenesshort <- data.frame(gx.5mC.5hmC.epigenesshort$pct_5mC.M8, gx.5mC.5hmC.epigenesshort$pct_5mC.T3, gx.5mC.5hmC.epigenesshort$pct_5hmC.M8, gx.5mC.5hmC.epigenesshort$pct_5hmC.T3)
colnames(promoter.5mC.5hmC.epigenesshort) <- c("% 5mC M", "% 5mC T", "% 5hmC M", "% 5hmC T")
rownames(promoter.5mC.5hmC.epigenesshort) <- gx.5mC.5hmC.epigenesshort$gene_name



# pheatmaps : expression, CNV, SNV
anno_row <- data.frame(CNV = as.vector(CNV.epigenesshort.cnvtype$CNV), SNV = as.vector(SNV.epigenesshort.snvtype$SNV), row.names = rownames(CNV.epigenesshort.cnvtype))
anno_colors <- list(CNV = c(GAIN = "darkcyan", REF = "white", LOSS = "darkorange3"), SNV = c(SNV = "#999999", nonSNV = "white"))

pheatmap(gx.epigenesshort.tpmLog2FC, color = colorRampPalette(c("darkred","white","darkgreen"), bias = 2.4)(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160524_gx.epigenesshort.TET.DNMT.tpmLog2FC.CNV.SNV.pdf", width = 10/2.54, height = 10/2.54)

pheatmap(gx.epigenesshort.tpmAvgLog2, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160524_gx.epigenesshort.TET.DNMT.tpmAvgLog2.CNV.SNV.pdf", width = 10/2.54, height = 12/2.54)

pheatmap(gx.epigenesshort.tpmLog2Avg, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160524_gx.epigenesshort.TET.DNMT.tpmLog2Avg.CNV.SNV.pdf", width = 10/2.54, height = 12/2.54)

pheatmap(gx.epigenesshort.tpmAvg, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160524_gx.epigenesshort.TET.DNMT.tpmAvg.CNV.SNV.pdf", width = 10/2.54, height = 10/2.54)



# pheatmaps : 5mC M, 5mC T, 5hmC M, 5hmC T + count of CpGs
anno_row <- data.frame(count_CpG_sites = as.vector(gx.5mC.5hmC.epigenesshort$count), row.names = rownames(promoter.5mC.5hmC.epigenesshort))

pheatmap(promoter.5mC.5hmC.epigenesshort, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160524_promoter.5mC.5hmC.epigenesshort.TET.DNMT.count_CpG_sites.pdf", width = 14/2.54, height = 10/2.54)

```










### Putting it all together for glioblastoma genes (Bai2015)

FC Expression (sorted)
promoter 5mC/5hmC in Tumour and Margin
CNVs
SNVs
CpG islands
CpG sites count

Where do we get glioblastoma genes from?

Manual lists in /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/glioblastoma_genes
Built after reading papers like Bai2015
20160601_Bai2015_fig1a.txt

OMIM page of glioblastoma: http://www.omim.org/entry/137800
There might be a way to automatically get the genes

Uniprot search for glioblastoma: http://www.uniprot.org/uniprot/?query=annotation%3A%28type%3Adisease+glioblastoma%29+AND+organism%3A%22Homo+sapiens+[9606]%22&sort=score
Using Uniprot to search OMIM: http://www.uniprot.org/uniprot/?query=database%3A%28type%3Amim+id%3A137800%29&sort=score

GO search for glioblastoma genes perhaps


```R
library(data.table)
library(ggplot2)
library(pheatmap)



# Expression
tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]


gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmAvgLog2<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmAvgLog2)][order(chrom, promStart, promEnd)]



# Bai2015 Fig.1a genes
bai2015 <- read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/glioblastoma_genes/20160601_Bai2015_fig1a.txt", sep = "\t")
setnames(bai2015, c("gene_name", "pathway"))

length(intersect(bai2015$gene_name, gx$Associated_Gene_Name)) # 30

gx.bai2015 <- gx[Associated_Gene_Name %in% bai2015$gene_name]

gx.bai2015.sorted <- gx.bai2015[order(-tpmLog2FC)]
setnames(gx.bai2015.sorted, "Associated_Gene_Name", "gene_name")

gx.bai2015.tpmLog2FC <- data.frame(gx.bai2015.sorted$tpmLog2FC)
colnames(gx.bai2015.tpmLog2FC) <- "log2FC"
rownames(gx.bai2015.tpmLog2FC) <- gx.bai2015.sorted$gene_name

gx.bai2015.tpmAvgLog2 <- data.frame(gx.bai2015.sorted$tpmAvgLog2)
colnames(gx.bai2015.tpmAvgLog2) <- "Mean log2(TPM+0.01)"
rownames(gx.bai2015.tpmAvgLog2) <- gx.bai2015.sorted$gene_name

gx.bai2015.pathway <- data.frame(merge(x = gx.bai2015.sorted, y = bai2015, by = "gene_name")[order(-tpmLog2FC)]$pathway)
colnames(gx.bai2015.pathway) <- "pathway"
rownames(gx.bai2015.pathway) <- merge(x = gx.bai2015.sorted, y = bai2015, by = "gene_name")[order(-tpmLog2FC)]$gene_name



# CNV
gene_locations_CNV <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160511_gene_locations_CNV.bedGraph")
setnames(gene_locations_CNV, c("chr_gene","start_gene", "end_gene", "gene_name", "chr_CNV", "start_CNV", "end_CNV", "type_CNV", "number_CNV"))

length(intersect(gx.bai2015$Associated_Gene_Name, gene_locations_CNV$gene_name)) # 30 OK

CNV.bai2015 <- gene_locations_CNV[gene_name %in% gx.bai2015$Associated_Gene_Name]
CNV.bai2015 # 31, it should be 30
table(CNV.bai2015$gene_name) # ARID2 is present twice
CNV.bai2015 <- CNV.bai2015[!(gene_name == "ARID2" & start_CNV == "45224115")]

gx.CNV.bai2015 <- merge(x = gx.bai2015.sorted, y = CNV.bai2015, by = "gene_name")
CNV.bai2015.cnvtype <- data.frame(gx.CNV.bai2015[order(-tpmLog2FC)]$type_CNV)
colnames(CNV.bai2015.cnvtype) <- "CNV"
rownames(CNV.bai2015.cnvtype) <- gx.CNV.bai2015[order(-tpmLog2FC)]$gene_name



# SNV
gene_locations_SNV <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160516_gene_locations_SNV.bedGraph")
setnames(gene_locations_SNV, c("chr_gene","start_gene", "end_gene", "gene_name", "chr_SNV", "start_SNV", "end_SNV", "base_from", "base_to", "score", "tier"))

length(intersect(gx.bai2015$Associated_Gene_Name, gene_locations_SNV$gene_name)) # 10

SNV.bai2015 <- gene_locations_SNV[gene_name %in% gx.bai2015$Associated_Gene_Name] # 14 mutations, but what type? Also three genes ATRX, MET, NOTCH2 have 2, 3 and 2 SNVs respectively.

gx.SNV.bai2015 <- merge(x = gx.bai2015.sorted, y = SNV.bai2015, all = TRUE, by = "gene_name")
gx.SNV.bai2015[, type_SNV := ifelse(is.na(chr_gene), "nonSNV", "SNV")]
gx.SNV.bai2015 <- gx.SNV.bai2015[!(gene_name == "ATRX" & start_SNV == "77030992")]
gx.SNV.bai2015 <- gx.SNV.bai2015[!(gene_name == "MET" & (start_SNV == "116346998" | start_SNV == "116385520"))]
gx.SNV.bai2015 <- gx.SNV.bai2015[!(gene_name == "NOTCH2" & start_SNV == "120603067")]

SNV.bai2015.snvtype <- data.frame(gx.SNV.bai2015[order(-tpmLog2FC)]$type_SNV)
colnames(SNV.bai2015.snvtype) <- "SNV"
rownames(SNV.bai2015.snvtype) <- gx.SNV.bai2015[order(-tpmLog2FC)]$gene_name



# promoter 5mC/5hmC
ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]

length(intersect(gx.bai2015.sorted$gene_name, M8.T3.pct.promoter$gene_name)) # 30

gx.5mC.5hmC.bai2015 <- merge(x = gx.bai2015.sorted, y = M8.T3.pct.promoter, by = "gene_name")[order(-tpmLog2FC)]

promoter.5mC.5hmC.bai2015 <- data.frame(gx.5mC.5hmC.bai2015$pct_5mC.M8, gx.5mC.5hmC.bai2015$pct_5mC.T3, gx.5mC.5hmC.bai2015$pct_5hmC.M8, gx.5mC.5hmC.bai2015$pct_5hmC.T3)
colnames(promoter.5mC.5hmC.bai2015) <- c("% 5mC promoter M", "% 5mC promoter T", "% 5hmC promoter M", "% 5hmC promoter T")
rownames(promoter.5mC.5hmC.bai2015) <- gx.5mC.5hmC.bai2015$gene_name



# pheatmaps : expression, CNV, SNV, pathway
anno_row <- data.frame(CNV = as.vector(CNV.bai2015.cnvtype$CNV), SNV = as.vector(SNV.bai2015.snvtype$SNV), pathway = as.vector(gx.bai2015.pathway$pathway), row.names = rownames(CNV.bai2015.cnvtype))
pathway_colors <- rainbow(10)
names(pathway_colors) <- c("IDH", "Astrocytic_drivers", "Oligodendroglial_drivers", "NOTCH_signaling", "RTK_RAS_PI3K_signaling", "MYC_RB1_signaling", "FAT_receptors", "DNA_damage_response", "Chromatin_remodeling", "Transcription_modulation")
anno_colors <- list(CNV = c(GAIN = "darkcyan", REF = "white", LOSS = "darkorange3", LOH = "cornflowerblue"), SNV = c(SNV = "#999999", nonSNV = "white"), pathway = pathway_colors)

pheatmap(gx.bai2015.tpmLog2FC, color = colorRampPalette(c("darkred","white","darkgreen"), bias = 0.65)(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, labels_col = expression("log"[2]*"FC"), fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160602_gx.Bai2015.fig1a.tpmLog2FC.CNV.SNV.pathway.pdf", width = 16/2.54, height = 25/2.54)

pheatmap(gx.bai2015.tpmAvgLog2, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, labels_col = expression("mean log"[2]*"(TPM+0.01)"), filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160602_gx.Bai2015.fig1a.tpmAvgLog2.CNV.SNV.pathway.pdf", width = 16/2.54, height = 25/2.54)




# pheatmaps : 5mC M, 5mC T, 5hmC M, 5hmC T + count of CpGs
anno_row <- data.frame(count_CpG_sites = as.vector(gx.5mC.5hmC.bai2015$count), row.names = rownames(promoter.5mC.5hmC.bai2015))

pheatmap(promoter.5mC.5hmC.bai2015, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160602_promoter.5mC.5hmC.Bai2015.fig1a.count_CpG_sites.pdf", width = 16/2.54, height = 25/2.54)


```





### Putting it all together for topoisomerase and histone variant genes


```R
library(data.table)
library(ggplot2)
library(pheatmap)



# Expression
tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]


gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmAvgLog2<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmAvgLog2)][order(chrom, promStart, promEnd)]


# Topoisomerase and histone variant genes
topo_hist <- read.table("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/topoisomerases_histonelinker1variants/20160722_topoisomerases_histonelinker1variants_euni.txt", sep = "\t")
setnames(topo_hist, c("gene_name", "category"))

length(intersect(topo_hist$gene_name, gx$Associated_Gene_Name)) # 13

gx.topo_hist <- gx[Associated_Gene_Name %in% topo_hist$gene_name]

gx.topo_hist.sorted <- gx.topo_hist[order(-tpmLog2FC)]
setnames(gx.topo_hist.sorted, "Associated_Gene_Name", "gene_name")

gx.topo_hist.tpmLog2FC <- data.frame(gx.topo_hist.sorted$tpmLog2FC)
colnames(gx.topo_hist.tpmLog2FC) <- "log2FC"
rownames(gx.topo_hist.tpmLog2FC) <- gx.topo_hist.sorted$gene_name

gx.topo_hist.tpmAvgLog2 <- data.frame(gx.topo_hist.sorted$tpmAvgLog2)
colnames(gx.topo_hist.tpmAvgLog2) <- "Mean log2(TPM+0.01)"
rownames(gx.topo_hist.tpmAvgLog2) <- gx.topo_hist.sorted$gene_name

gx.topo_hist.category <- data.frame(merge(x = gx.topo_hist.sorted, y = topo_hist, by = "gene_name")[order(-tpmLog2FC)]$category)
colnames(gx.topo_hist.category) <- "category"
rownames(gx.topo_hist.category) <- merge(x = gx.topo_hist.sorted, y = topo_hist, by = "gene_name")[order(-tpmLog2FC)]$gene_name


# CNV
gene_locations_CNV <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160511_gene_locations_CNV.bedGraph")
setnames(gene_locations_CNV, c("chr_gene","start_gene", "end_gene", "gene_name", "chr_CNV", "start_CNV", "end_CNV", "type_CNV", "number_CNV"))

length(intersect(gx.topo_hist$Associated_Gene_Name, gene_locations_CNV$gene_name)) # 13 OK

CNV.topo_hist <- gene_locations_CNV[gene_name %in% gx.topo_hist$Associated_Gene_Name]
CNV.topo_hist # 13

gx.CNV.topo_hist <- merge(x = gx.topo_hist.sorted, y = CNV.topo_hist, by = "gene_name")
CNV.topo_hist.cnvtype <- data.frame(gx.CNV.topo_hist[order(-tpmLog2FC)]$type_CNV)
colnames(CNV.topo_hist.cnvtype) <- "CNV"
rownames(CNV.topo_hist.cnvtype) <- gx.CNV.topo_hist[order(-tpmLog2FC)]$gene_name


# SNV
gene_locations_SNV <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/gene_locations/20160516_gene_locations_SNV.bedGraph")
setnames(gene_locations_SNV, c("chr_gene","start_gene", "end_gene", "gene_name", "chr_SNV", "start_SNV", "end_SNV", "base_from", "base_to", "score", "tier"))

length(intersect(gx.topo_hist$Associated_Gene_Name, gene_locations_SNV$gene_name)) # 0

SNV.topo_hist <- gene_locations_SNV[gene_name %in% gx.topo_hist$Associated_Gene_Name] # 14 mutations, but what type? Also three genes ATRX, MET, NOTCH2 have 2, 3 and 2 SNVs respectively.

gx.SNV.topo_hist <- merge(x = gx.topo_hist.sorted, y = SNV.topo_hist, all = TRUE, by = "gene_name")
gx.SNV.topo_hist[, type_SNV := ifelse(is.na(chr_gene), "nonSNV", "SNV")]

SNV.topo_hist.snvtype <- data.frame(gx.SNV.topo_hist[order(-tpmLog2FC)]$type_SNV)
colnames(SNV.topo_hist.snvtype) <- "SNV"
rownames(SNV.topo_hist.snvtype) <- gx.SNV.topo_hist[order(-tpmLog2FC)]$gene_name


# promoter 5mC/5hmC
ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]

length(intersect(gx.topo_hist.sorted$gene_name, M8.T3.pct.promoter$gene_name)) # 13

gx.5mC.5hmC.topo_hist <- merge(x = gx.topo_hist.sorted, y = M8.T3.pct.promoter, by = "gene_name")[order(-tpmLog2FC)]

promoter.5mC.5hmC.topo_hist <- data.frame(gx.5mC.5hmC.topo_hist$pct_5mC.M8, gx.5mC.5hmC.topo_hist$pct_5mC.T3, gx.5mC.5hmC.topo_hist$pct_5hmC.M8, gx.5mC.5hmC.topo_hist$pct_5hmC.T3)
colnames(promoter.5mC.5hmC.topo_hist) <- c("% 5mC promoter M", "% 5mC promoter T", "% 5hmC promoter M", "% 5hmC promoter T")
rownames(promoter.5mC.5hmC.topo_hist) <- gx.5mC.5hmC.topo_hist$gene_name


# pheatmaps : expression, CNV, SNV, category
anno_row <- data.frame(CNV = as.vector(CNV.topo_hist.cnvtype$CNV), SNV = as.vector(SNV.topo_hist.snvtype$SNV), category = as.vector(gx.topo_hist.category$category), row.names = rownames(CNV.topo_hist.cnvtype))
category_colors <- c("khaki", "plum")
names(category_colors) <- c("DNA_topoisomerase", "Histone_linker_1_variant")
anno_colors <- list(CNV = c(GAIN = "darkcyan", REF = "white", LOSS = "darkorange3", LOH = "cornflowerblue"), SNV = c(SNV = "#999999", nonSNV = "white"), category = category_colors)

pheatmap(gx.topo_hist.tpmLog2FC, color = colorRampPalette(c("darkred","white","darkgreen"), bias = 1.2)(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, labels_col = expression("log"[2]*"FC"), fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160723_gx.topo_hist.tpmLog2FC.CNV.SNV.category.pdf", width = 16/2.54, height = 25/2.54)

pheatmap(gx.topo_hist.tpmAvgLog2, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_colors = anno_colors, annotation_legend = TRUE, fontsize=16, labels_col = expression("mean log"[2]*"(TPM+0.01)"), filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160723_gx.topo_hist.tpmAvgLog2.CNV.SNV.category.pdf", width = 16/2.54, height = 25/2.54)


# pheatmaps : 5mC M, 5mC T, 5hmC M, 5hmC T + count of CpGs
anno_row <- data.frame(count_CpG_sites = as.vector(gx.5mC.5hmC.topo_hist$count), row.names = rownames(promoter.5mC.5hmC.topo_hist))

pheatmap(promoter.5mC.5hmC.topo_hist, color = colorRampPalette(c("white", "darkblue"))(100), cellwidth = 16, cellheight = 16, cluster_rows = FALSE, cluster_cols = FALSE, annotation_row = anno_row, annotation_legend = TRUE, fontsize=16, filename="/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160723_promoter.5mC.5hmC.topo_hist.count_CpG_sites.pdf", width = 16/2.54, height = 25/2.54)



```



### ATM

20160804:

Hi Sergio,

could you please tell me the ATM (ataxia telangiectasia mutated) expression levels in tumour versus margin please???

Thaks Euni

Gene name is ATM

http://www.uniprot.org/uniprot/Q13315


```R
library(data.table)

tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]


gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmAvgLog2<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmAvgLog2)][order(chrom, promStart, promEnd)]

write.table(gx, "/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/tables/20160805_gx.txt", quote = FALSE, sep = "\t", row.names=FALSE)

system("rsync /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/tables/20160805_gx.txt martin03@nm149s012925:/Users/martin03/Desktop")

```




### TODO

Finish Bai2015

Check OMIM page above

Check TET papers

Need to split SNVs depending on the type of mutation.
