

This is a continuation of the MGMT promoter methylation section written in 20160226_tsg_ocg.md

### Is the MGMT assay (therascreen MGMT Pyro Kit on a Pyromark Q24 System) an FDA-approved test?
Compliance with 
https://www.qiagen.com/be/shop/detection-solutions/personalized-healthcare/therascreen-mgmt-pyro-kit#orderinginformation
Some components are EU IVD Directive 98/79/EC compliant - see p.9 in QIAGEN2015
Wrote to Marco
Perhaps write to Rita to ask

Need to read more relevant publications: see MGMT and IDH section in cancer_papers_collection.odt to follow up.


### Summary

Moving beyond unreliable thresholds: choosing optimal drug doses for cancer treatment
Developing quantitative clinical decision models for drug treatment

Differences in abundance and activity of the MGMT enzyme between tumour and healthy tissues is key to understand the response of cancer patients to alkylating agents such as temozolomide. 

Temozolomide alkylates guanine in O6:

G --TMZ--> O6mG
O6mG --MGMT--> G

Doctors check different molecular features of the MGMT gene as a proxy for the abundance of MGMT, e.g. promoter methylation using commercial kits, protein expression with immunohistochemistry ... however there is a lack of correlation between the methylation in promoters and the gene expression levels.


### Regulation of MGMT

The idea is to extract the relevant group of genes from Cabrini2015 and prepare summary plots to try to understand why MGMT might be downregulated in our sample. The question is: How is MGMT downregulated in Tumour? A mechanistic explanation (we can do the same with TET downregulation later).

The first hypothesis (from looking at /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160523_gx.epigenesshort.tpmLog2FC.CNV.SNV.class.pdf) would be that MGMT downregulation is due to CNV LOSS.

Second hypotheses would come from looking at the list of genes presented in Cabrini2015.

#### DNA repair genes affecting MGMT:

/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/dna_repair_genes/20160523_dna_repair_genes.txt

Despite MGMT downregulation, if DNA repair genes are upregulated in tumour then O6mG:T pairing might possibly be getting repaired at a rate similar to the MGMT loss.

```R
tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]


gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmAvgLog2<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmAvgLog2)][order(chrom, promStart, promEnd)]

# Cited in Cabrini2015. See also Liu2006.
gx[Associated_Gene_Name == "MLH1"] # 0.09466491, 6.729854
gx[Associated_Gene_Name == "MSH2"] # -0.08625319, 5.313673
gx[Associated_Gene_Name == "MSH6"] # 0.6209074, 6.147635
gx[Associated_Gene_Name == "PMS2"] # -0.2207803, 7.231616

# Obtained directly from Uniprot.
gx[Associated_Gene_Name == "MLH3"] # -1.531214, 5.939912
gx[Associated_Gene_Name == "MSH3"] # -0.3895948, 4.91379
gx[Associated_Gene_Name == "PMS1"] # -1.251714, 5.475057

```

Overall, it is really a mixed bag of up/downregulation. I don't know which gene does preferentially repair O6mG anyway. The genes obtained from Uniprot are all downregulated. This might explain the side effects experienced by the patient.



### Transcription factors (TFs) affecting the expression of MGMT. 

Unfortunately we don't have direct data about TF binding but we can look indirectly at the expression of relevant genes in:

/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/transcriptor_factor_genes/20160523_transcription_factor_genes.txt

Several TFs have been proved to activate MGMT gene regulation. Downregulation of these TFs might explain the downregulation of MGMT.


```R
tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]


gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmAvgLog2<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmAvgLog2)][order(chrom, promStart, promEnd)]

# Cited in Cabrini2015.
gx[Associated_Gene_Name == "SP1"] # 0.9118968 4.40425
gx[Associated_Gene_Name == "JUN"] # 2.539944 5.523182
gx[Associated_Gene_Name == "TFAP2A"] # 0.524163 -0.08440899
gx[Associated_Gene_Name == "TFAP2B"] # -3.075996 -5.105858
gx[Associated_Gene_Name == "TFAP2C"] # 2.176557 -0.5707607
gx[Associated_Gene_Name == "CEBPA"] # 0.2842652  0.8137158
gx[Associated_Gene_Name == "NFKB1"] # 0.8061166   5.257781
gx[Associated_Gene_Name == "NFKB2"] # 1.702643    2.50112
gx[Associated_Gene_Name == "RELA"] # 0.7571688   4.442345
gx[Associated_Gene_Name == "RELB"] # 3.512408   2.093063
gx[Associated_Gene_Name == "REL"] # -0.02714209   2.997184
gx[Associated_Gene_Name == "HIF1A"] # 1.880535   9.107743

```

They are all upregulated therefore this does not explain the downregulation of MGMT.



### Histone deacetylation and methylation

Affects the expression of MGMT in Cabrini2015. Unfortunately we don't have direct data about histone acetylation or methylation marks but we can look indirectly at the expression of relevant genes in:

/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/epigenetic_genes/20160516_epigenetic_genes_long.txt

With regards to the genes cited in Cabrini2015:

```R
tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]


gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmAvgLog2<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmAvgLog2)][order(chrom, promStart, promEnd)]

# Cited in Cabrini2015.
gx[Associated_Gene_Name == "HSP90AA1"] # -0.7511914   9.896535
gx[Associated_Gene_Name == "HSP90AA2P"] # not found
gx[Associated_Gene_Name == "HSP90AA4P"] # 2.97257  -5.157571
gx[Associated_Gene_Name == "HSP90AA5P"] # -1.066137  -6.110787
gx[Associated_Gene_Name == "HSP90AB1"] # 0.002236462   9.662081
gx[Associated_Gene_Name == "HSP90AB2P"] # 0.09649065 -0.5619985
gx[Associated_Gene_Name == "HSP90AB3P"] # 0.3391783  -1.881672
gx[Associated_Gene_Name == "HSP90AB4P"] # not found
gx[Associated_Gene_Name == "HSP90B1"] # 1.320103   8.671708
gx[Associated_Gene_Name == "HSP90B2P"] # -1.570556  -2.135675

gx[Associated_Gene_Name == "EGFR"] # -1.563844   5.569419

gx[Associated_Gene_Name == "AKT1"] # 1.157227   6.154229
gx[Associated_Gene_Name == "AKT2"] # 0.8145778   7.485677
gx[Associated_Gene_Name == "AKT3"] # -3.35748   6.578023
gx[Associated_Gene_Name == "AKT1S1"] # 0.7199667   2.571827
gx[Associated_Gene_Name == "AKTIP"] # -1.658502   5.552997



```

HSP90: the different components seem both up and downregulated so there is not an easy message coming out from this.

EGFR is downregulated.

AKT: two isoforms of the phosphoprotein p-Akt are upregulated, AKT3 is however downregulated



### MGMT promoter methylation

```R
gx[Associated_Gene_Name == "MECP2"] # -1.402496   4.369741

```




### miRNAs affecting the expression of MGMT. 

However I don't know how to have a look at the expression of this list.

/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/miRNAs/20160525_miRNAs.txt

This could be an interesting topic to investigate why MGMT is anyway downregulated in tumour despite the lack of differential promoter methylation in tumour vs. margin.



A good discussion about which method for evaluating MGMT status does best is available in section 6. MGMT expression as predictive biomarker of Cabrini2015


I re-read Cabrini2015 and searched for the expression of some of the genes presented - see above.

An idea for a project would be to search for new chemical inhibitors of MGMT, alternative to O6-benzylguanine. Other drugs that have already been used for different diseases.

Worth searching for a DNA repair gene that is hypermethylated in glioblastoma (this is reflected in the expression) and a drug introducing a chemical change (perhaps alkylation that this gene might be targeting).



Now reading Dullea2016
























