
This script aims to find a relationship between CNVs and the methylation data.

Here are some comments from Jenniffer Becq extracted from our email exchange:

- I think the CNV caller has tripped over a little and is making false positive overcalls (eg. chr12, chr14, chr15, etc are called mainly copy number =3 (triploid) when obviously they are not). I’m sorry about that. I wonder if it’s due to the fact that the “normal” (in this case the margin) has such high coverage and it’s not used to that. The calls using the blood sample as normal are very good and should be used instead of those. 

- I can tell the CNV caller has struggled as it’s been calling chr12, chr14, chr16 etc. as triploid when they are diploid. I think the high depth in both genomes could be the issue and I would use the calls from the other subtractions.

- For CNVs, as I mentioned above, because the caller used is deprecated and not published, you might have the issue of publishing results on it and might prefer to run another caller. As a commercial entity we are usually restricted in the tools we can use because of licensing, and for those without licences we struggle to find time to assess the other available tools. We are currently developing a revamped CNV caller (https://github.com/Illumina/canvas, http://www.biorxiv.org/content/biorxiv/early/2016/01/13/036194.full.pdf?%3Fcollection=), if I have time, I might try it out on the two bams you’ve sent me. Other collaborators have been using BIC-seq (http://compbio.med.harvard.edu/Supplements/PNAS11.html), CONSERTING (http://www.nature.com/nmeth/journal/v12/n6/full/nmeth.3394.html), Sequenza (http://annonc.oxfordjournals.org/content/26/1/64.long) or absolute (https://www.broadinstitute.org/cancer/cga/absolute), but I have no experience in those and couldn’t say which is the best (in fact it’s very difficult to assess them as usually the “truth” is unknown and simulation are never enough to cover all possibilities).

- However you can use our calls for initial investigations, at least to see if you find any correlation between bisulphite data and copy number changes. Eg. what do the bisulphite data look like within the haploid regions: eg. whole of chr10, chr22, long arm of chr6, second half of chr11 and other little bits. How is it for chr17 which is copy neutral LOH? And what about the triploid regions (chr7 and chr20)? I don’t have any hypothesis right now, but just curious to know how it looks/if it looks different than in the diploid regions.

- For SVs, the calls are from Manta, here’s the github and documentation (https://github.com/Illumina/manta), no publication yet (http://biorxiv.org/content/early/2015/08/10/024232) but in the process…  Again, I’d look at the particular SV hotspots and how the bisulphite data look like at those positions: I would focus on translocations between chr6 and chr11 and within chr18. It looks like the oncogene BCL2 is quite rearranged in chr18, I wonder if it’s relevant as it’s a regulator of apoptosis.

We will first look at the calls generated by Strelka then.

### Data

In nas-srv001,

```bash
cd /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap
mkdir cnv
cd cnv
cp /data/sblab_data1/berald01/projects/20150501_methylation_brain/20150914_archive_snp/disk1/CancerLP2000729-DNA_E01_NormalLP200/SomaticVariations/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_A01.somatic.CNV.vcf.gz /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv
cp /data/sblab_data1/berald01/projects/20150501_methylation_brain/20150914_archive_snp/disk1/CancerLP2000729-DNA_C01_NormalLP200/SomaticVariations/CancerLP2000729-DNA_C01_NormalLP2000729-DNA_A01.somatic.CNV.vcf.gz /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv
cp /data/sblab_data1/berald01/projects/20150501_methylation_brain/20150914_archive_snp/20151029_tumour_vs_margin/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_C01/SomaticVariations/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_C01.somatic.CNV.vcf.gz /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv
gzip -d *.vcf.gz
```

### Load CNV data

We will use the VariantAnnotation and data.table packages:
https://bioconductor.org/packages/release/bioc/html/VariantAnnotation.html
https://rawgit.com/wiki/Rdatatable/data.table/vignettes/datatable-intro-vignette.html

```R
library(VariantAnnotation)
library(data.table)
vcf<-readVcf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_A01.somatic.CNV.vcf", "hg19")
vcf

# Examine header for 3 pieces of meta information and 7 pieces of info
meta(header(vcf))$META
meta(header(vcf))$contig
meta(header(vcf))$SAMPLE
fixed(header(vcf))$ALT
info(header(vcf))

chr_l<-as.vector(as.data.frame(rowRanges(vcf))$seqnames)
start_l<-as.data.frame(rowRanges(vcf))$start
end_l<-as.data.frame(info(vcf))$END
cn_l<-as.data.frame(info(vcf))$CN
loh_l<-as.data.frame(info(vcf))$LOH
cnscore_l<-as.data.frame(info(vcf))$CNscore

as.data.frame(info(vcf))$ensembl_gene_id # list of lists of ensemble ids falling into each CNV

data.frame(chr_l, start_l, end_l, cn_l, loh_l, cnscore_l)

cnv<-data.table(chr_l = chr_l, start_l = start_l, end_l = end_l, cn_l = cn_l, loh_l = loh_l, cnscore_l = cnscore_l)

```

### Load BS and oxBS data

Where is the data?
/data/sblab_data1/berald01/projects/20150501_methylation_brain/20151106_genome_methyl

Copy data to martin03 folders, in nas-srv001:

```bash
mkdir /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg
cp /data/sblab_data1/berald01/projects/20150501_methylation_brain/20151106_genome_methyl/*.gz /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg
```

```R
library(data.table)

# Loading Tumour 5mC data (ear045)
meth_cpg_ear045<- fread('zcat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear045_T3oxBS.hiseq201509.cpg.bedGraph.gz')
setnames(meth_cpg_ear045, c("chr", "start", "end", "cnt_met", "cnt_tot"))
meth_cpg_ear045[, pct_met := cnt_met/cnt_tot]

meth_cpg_ear045[, list(avg= mean(pct_met, na.rm= TRUE),  median= median(pct_met, na.rm= TRUE), stdev= sd(pct_met, na.rm= TRUE)), by= chr]

# Loading Margin 5mC data (ear043)

meth_cpg_ear043<- fread('zcat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear043_M8oxBS.hiseq201509.cpg.bedGraph.gz')
setnames(meth_cpg_ear043, c("chr", "start", "end", "cnt_met", "cnt_tot"))
meth_cpg_ear043[, pct_met := cnt_met/cnt_tot]

meth_cpg_ear043[, list(avg= mean(pct_met, na.rm= TRUE),  median= median(pct_met, na.rm= TRUE), stdev= sd(pct_met, na.rm= TRUE)), by= chr]

# Looking at the relationship between CNVs and 5mC (Tumour and Margin) in chr1

x1_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=1 & end < 720000]$pct_met # cn=2
x2_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=720000 & end < 2582000]$pct_met # cn=1
x3_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=2582000 & end < 2697000]$pct_met # cn=2
x4_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=2697000 & end < 8060000]$pct_met # cn=1
x5_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=8060000 & end < 8061000]$pct_met # cn=2
x6_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=8061000 & end < 8919000]$pct_met # cn=0
x7_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=8919000 & end < 8923000]$pct_met # cn=2
x8_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=8923000 & end < 12951000]$pct_met # cn=1
x9_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=12951000 & end < 13779000]$pct_met # cn=2
x10_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=13779000 & end < 24990000]$pct_met # cn=1
x11_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=24990000 & end < 207993000]$pct_met # cn=2
x12_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=207993000 & end < 236775000]$pct_met # cn=1
x13_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=236775000 & end < 241999000]$pct_met # cn=2
x14_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=241999000 & end < 243533000]$pct_met # cn=3
x15_ear045 <- meth_cpg_ear045[chr == "chr1" & start >=243533000 & end < 249250621]$pct_met # cn=2

#install.packages("vioplot")
library(vioplot)
vioplot(x2_ear045, x4_ear045, x6_ear045, x8_ear045, x10_ear045, x12_ear045[!is.na(x12_ear045)], x14_ear045, names=c("haploid", "haploid", "deletion", "haploid", "haploid", "haploid", "triploid"), col=c("gold"))
title("chr1: CNVs and Tumour 5mC")

vioplot(x1_ear045, x2_ear045, x3_ear045, x4_ear045, x5_ear045, x6_ear045, x7_ear045, x8_ear045, x9_ear045, x10_ear045, x11_ear045[!is.na(x11_ear045)], x12_ear045[!is.na(x12_ear045)], x13_ear045, x14_ear045, x15_ear045[!is.na(x15_ear045)], names=c("diploid", "haploid", "diploid", "haploid", "diploid", "deletion", "diploid", "haploid", "diploid", "haploid", "diploid", "haploid", "diploid", "triploid", "diploid"), col="gold")
title("chr1: CNVs and Tumour 5mC")
# There does not seem to be a relationship CNVs chr1 and Tumour 5mC


x1_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=1 & end < 720000]$pct_met # cn=2
x2_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=720000 & end < 2582000]$pct_met # cn=1
x3_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=2582000 & end < 2697000]$pct_met # cn=2
x4_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=2697000 & end < 8060000]$pct_met # cn=1
x5_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=8060000 & end < 8061000]$pct_met # cn=2
x6_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=8061000 & end < 8919000]$pct_met # cn=0
x7_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=8919000 & end < 8923000]$pct_met # cn=2
x8_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=8923000 & end < 12951000]$pct_met # cn=1
x9_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=12951000 & end < 13779000]$pct_met # cn=2
x10_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=13779000 & end < 24990000]$pct_met # cn=1
x11_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=24990000 & end < 207993000]$pct_met # cn=2
x12_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=207993000 & end < 236775000]$pct_met # cn=1
x13_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=236775000 & end < 241999000]$pct_met # cn=2
x14_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=241999000 & end < 243533000]$pct_met # cn=3
x15_ear043 <- meth_cpg_ear043[chr == "chr1" & start >=243533000 & end < 249250621]$pct_met # cn=2

#install.packages("vioplot")
library(vioplot)
vioplot(x2_ear043[!is.na(x2_ear043)], x4_ear043, x6_ear043, x8_ear043, x10_ear043[!is.na(x10_ear043)], x12_ear043, x14_ear043, names=c("haploid", "haploid", "deletion", "haploid", "haploid", "haploid", "triploid"), col=c("gold"))
title("chr1: CNVs and Margin 5mC")

vioplot(x2_ear045, x2_ear043[!is.na(x2_ear043)], x4_ear045, x4_ear043, x6_ear045, x6_ear043, x8_ear045, x8_ear043, x10_ear045, x10_ear043[!is.na(x10_ear043)], x12_ear045[!is.na(x12_ear045)], x12_ear043, x14_ear045, x14_ear043, col=c("gold"))
title("chr1: CNVs and Tumour/Margin 5mC")

vioplot(x1_ear045, x2_ear045, x3_ear045, x4_ear045, x5_ear045, x6_ear045, x7_ear045, x8_ear045, x9_ear045, x10_ear045, x11_ear045[!is.na(x11_ear045)], x12_ear045[!is.na(x12_ear045)], x13_ear045, x14_ear045, x15_ear045[!is.na(x15_ear045)], names=c("diploid", "haploid", "diploid", "haploid", "diploid", "deletion", "diploid", "haploid", "diploid", "haploid", "diploid", "haploid", "diploid", "triploid", "diploid"), col="gold")
title("chr1: CNVs and Tumour 5mC")
# There does not seem to be a relationship CNVs chr1 and Tumour 5mC
```



Now time to investigate the questions posed by Jennifer above

Looking at the relationship between CNVs and 5mC (Tumour and Margin)

### Load new BS and oxBS data

Copy new data to martin03 folders, in nas-srv001:

```bash
cp /data/sblab_data1/berald01/projects/20150501_methylation_brain/20151106_genome_methyl/*.hiseq201512.cpg.bedGraph.gz /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg
```

Check whether hiseq201509 and hiseq201512 are similar:
https://rstudio-pubs-static.s3.amazonaws.com/52230_5ae0d25125b544caab32f75f0360e775.html


```R
# T3oxBS

meth_cpg_T3oxBS_hiseq201509<- fread('zcat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear045_T3oxBS.hiseq201509.cpg.bedGraph.gz')
setnames(meth_cpg_T3oxBS_hiseq201509, c("chr", "start", "end", "cnt_met", "cnt_tot"))
meth_cpg_T3oxBS_hiseq201512<- fread('zcat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear045_T3oxBS.hiseq201512.cpg.bedGraph.gz')
setnames(meth_cpg_T3oxBS_hiseq201512, c("chr", "start", "end", "cnt_met", "cnt_tot"))
# Differences in nrows between the two? CpGs with no reads are not shown

setkey(meth_cpg_T3oxBS_hiseq201509, chr, start, end)
setkey(meth_cpg_T3oxBS_hiseq201512, chr, start, end)
meth_cpg_T3oxBS_intersection <- merge(meth_cpg_T3oxBS_hiseq201509, meth_cpg_T3oxBS_hiseq201512)
# 100*26963349/27107239 = 99.46918 overlap

meth_cpg_T3oxBS_intersection[, pct_met := (cnt_met.x + cnt_met.y)/(cnt_tot.x + cnt_tot.y)]
meth_cpg_T3oxBS_intersection[, list(avg= mean(pct_met, na.rm= TRUE),  median= median(pct_met, na.rm= TRUE), stdev= sd(pct_met, na.rm= TRUE)), by= chr]


# M8oxBS

meth_cpg_M8oxBS_hiseq201509<- fread('zcat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear043_M8oxBS.hiseq201509.cpg.bedGraph.gz')
setnames(meth_cpg_M8oxBS_hiseq201509, c("chr", "start", "end", "cnt_met", "cnt_tot"))
meth_cpg_M8oxBS_hiseq201512<- fread('zcat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear043_M8oxBS.hiseq201512.cpg.bedGraph.gz')
setnames(meth_cpg_M8oxBS_hiseq201512, c("chr", "start", "end", "cnt_met", "cnt_tot"))

setkey(meth_cpg_M8oxBS_hiseq201509, chr, start, end)
setkey(meth_cpg_M8oxBS_hiseq201512, chr, start, end)
meth_cpg_M8oxBS_intersection <- merge(meth_cpg_M8oxBS_hiseq201509, meth_cpg_M8oxBS_hiseq201512)
# 100*26901975/27062028 = 99.40857 % overlap

meth_cpg_M8oxBS_intersection[, pct_met := (cnt_met.x + cnt_met.y)/(cnt_tot.x + cnt_tot.y)]
meth_cpg_M8oxBS_intersection[, list(avg= mean(pct_met, na.rm= TRUE),  median= median(pct_met, na.rm= TRUE), stdev= sd(pct_met, na.rm= TRUE)), by= chr]
```


chr6 - partly haploid


```R
x1_T3oxBS_chr6 <- meth_cpg_T3oxBS_intersection[chr == "chr6" & start >=1 & end < 60000000]$pct_met # cn=2
x2_T3oxBS_chr6 <- meth_cpg_T3oxBS_intersection[chr == "chr6" & start >=63977000 & end < 81029000]$pct_met # cn=1
x3_T3oxBS_chr6 <- meth_cpg_T3oxBS_intersection[chr == "chr6" & start >=81367000 & end < 87009000]$pct_met # cn=1
x4_T3oxBS_chr6 <- meth_cpg_T3oxBS_intersection[chr == "chr6" & start >=87142000 & end < 170910000]$pct_met # cn=1

x1_M8oxBS_chr6 <- meth_cpg_M8oxBS_intersection[chr == "chr6" & start >=1 & end < 60000000]$pct_met # cn=2
x2_M8oxBS_chr6 <- meth_cpg_M8oxBS_intersection[chr == "chr6" & start >=63977000 & end < 81029000]$pct_met # cn=1
x3_M8oxBS_chr6 <- meth_cpg_M8oxBS_intersection[chr == "chr6" & start >=81367000 & end < 87009000]$pct_met # cn=1
x4_M8oxBS_chr6 <- meth_cpg_M8oxBS_intersection[chr == "chr6" & start >=87142000 & end < 170910000]$pct_met # cn=1

vioplot(x1_T3oxBS_chr6[!is.na(x1_T3oxBS_chr6)], x2_T3oxBS_chr6, x3_T3oxBS_chr6, x4_T3oxBS_chr6[!is.na(x4_T3oxBS_chr6)], names=c("diploid", "haploid", "haploid", "haploid"), col=c("gold"))
title("chr6: CNVs and Tumour 5mC")

vioplot(x1_M8oxBS_chr6[!is.na(x1_M8oxBS_chr6)], x2_M8oxBS_chr6, x3_M8oxBS_chr6, x4_M8oxBS_chr6[!is.na(x4_M8oxBS_chr6)], names=c("diploid", "haploid", "haploid", "haploid"), col=c("gold"))
title("chr6: CNVs and Margin 5mC")

vioplot(x1_T3oxBS_chr6[!is.na(x1_T3oxBS_chr6)], x1_M8oxBS_chr6[!is.na(x1_M8oxBS_chr6)], x2_T3oxBS_chr6, x2_M8oxBS_chr6, x3_T3oxBS_chr6, x3_M8oxBS_chr6, x4_T3oxBS_chr6[!is.na(x4_T3oxBS_chr6)], x4_M8oxBS_chr6[!is.na(x4_M8oxBS_chr6)], col=c("gold"))
grid()


setkey(meth_cpg_T3oxBS_intersection, chr, start, end)
setkey(meth_cpg_M8oxBS_intersection, chr, start, end)
meth_cpg_diff <- merge(meth_cpg_M8oxBS_intersection, meth_cpg_T3oxBS_intersection)
# 100*26780993/26963349 = 99.32369 % of CpGs in common between M8oxBS and T3oxBS
meth_cpg_diff[, pct_met_diff := pct_met.y - pct_met.x]

diff1_chr6 <- meth_cpg_diff[chr == "chr6" & start >=1 & end < 60000000]$pct_met_diff # cn=2
diff2_chr6 <- meth_cpg_diff[chr == "chr6" & start >=63977000 & end < 81029000]$pct_met_diff # cn=1
diff3_chr6 <- meth_cpg_diff[chr == "chr6" & start >=81367000 & end < 87009000]$pct_met_diff # cn=1
diff4_chr6 <- meth_cpg_diff[chr == "chr6" & start >=87142000 & end < 170910000]$pct_met_diff # cn=1

vioplot(diff1_chr6[!is.na(diff1_chr6)], diff2_chr6, diff3_chr6, diff4_chr6[!is.na(diff4_chr6)], col=c("gold"))
title("chr6: CNVs and (Tumour - Margin) 5mC")
grid()

source("/home/martin03/sw/R/R-3.2.1/lib64/R/library/vioplot/R/my.vioplot.R")

my.vioplot(diff1_chr6[!is.na(diff1_chr6)], diff2_chr6, diff3_chr6, diff4_chr6[!is.na(diff4_chr6)], col=c("grey30", "green3", "green3", "green3"), names=c("diploid", "haploid", "haploid", "haploid"))
title(main = "chr6: CNVs and (Tumour - Margin) 5mC", ylab="% 5mC (Tumour - Margin)")

my.vioplot(x1_T3oxBS_chr6[!is.na(x1_T3oxBS_chr6)], x1_M8oxBS_chr6[!is.na(x1_M8oxBS_chr6)], x2_T3oxBS_chr6, x2_M8oxBS_chr6, x3_T3oxBS_chr6, x3_M8oxBS_chr6, x4_T3oxBS_chr6[!is.na(x4_T3oxBS_chr6)], x4_M8oxBS_chr6[!is.na(x4_M8oxBS_chr6)], col=c("grey30", "grey30", "green3", "green3", "green3", "green3", "green3", "green3"), names=c("diploid_T", "diploid_M", "haploid_T", "haploid_M", "haploid_T", "haploid_M", "haploid_T", "haploid_M"))
title(main = "chr6: CNVs and (Tumour/Margin) 5mC", ylab="% 5mC (Tumour/Margin)")
grid()
```

There does not seem to be difference between the different regions of chr6





Summary of methylation across all chromosomes

```R
mean(meth_cpg_T3oxBS_intersection$pct_met, na.rm=T) # 0.6718345
median(meth_cpg_T3oxBS_intersection$pct_met, na.rm=T) # 0.8014706
mean(meth_cpg_M8oxBS_intersection$pct_met, na.rm=T) # 0.5524157
median(meth_cpg_M8oxBS_intersection$pct_met, na.rm=T) # 0.6190476

meth_cpg_T3oxBS_intersection_chrsummary <- meth_cpg_T3oxBS_intersection[, list(avg= mean(pct_met, na.rm= TRUE),  median= median(pct_met, na.rm= TRUE), stdev= sd(pct_met, na.rm= TRUE)), by= chr]
meth_cpg_T3oxBS_intersection_chrsummary # no such difference between chromosomes
meth_cpg_M8oxBS_intersection_chrsummary <- meth_cpg_M8oxBS_intersection[, list(avg= mean(pct_met, na.rm= TRUE),  median= median(pct_met, na.rm= TRUE), stdev= sd(pct_met, na.rm= TRUE)), by= chr]
meth_cpg_M8oxBS_intersection_chrsummary # no such difference between chromosomes either

T3oxBS_chr1 <- meth_cpg_T3oxBS_intersection[chr == "chr1"]$pct_met
T3oxBS_chr2 <- meth_cpg_T3oxBS_intersection[chr == "chr2"]$pct_met
T3oxBS_chr3 <- meth_cpg_T3oxBS_intersection[chr == "chr3"]$pct_met
T3oxBS_chr4 <- meth_cpg_T3oxBS_intersection[chr == "chr4"]$pct_met
T3oxBS_chr5 <- meth_cpg_T3oxBS_intersection[chr == "chr5"]$pct_met
T3oxBS_chr6 <- meth_cpg_T3oxBS_intersection[chr == "chr6"]$pct_met
T3oxBS_chr7 <- meth_cpg_T3oxBS_intersection[chr == "chr7"]$pct_met
T3oxBS_chr8 <- meth_cpg_T3oxBS_intersection[chr == "chr8"]$pct_met
T3oxBS_chr9 <- meth_cpg_T3oxBS_intersection[chr == "chr9"]$pct_met
T3oxBS_chr10 <- meth_cpg_T3oxBS_intersection[chr == "chr10"]$pct_met
T3oxBS_chr11 <- meth_cpg_T3oxBS_intersection[chr == "chr11"]$pct_met
T3oxBS_chr12 <- meth_cpg_T3oxBS_intersection[chr == "chr12"]$pct_met
T3oxBS_chr13 <- meth_cpg_T3oxBS_intersection[chr == "chr13"]$pct_met
T3oxBS_chr14 <- meth_cpg_T3oxBS_intersection[chr == "chr14"]$pct_met
T3oxBS_chr15 <- meth_cpg_T3oxBS_intersection[chr == "chr15"]$pct_met
T3oxBS_chr16 <- meth_cpg_T3oxBS_intersection[chr == "chr16"]$pct_met
T3oxBS_chr17 <- meth_cpg_T3oxBS_intersection[chr == "chr17"]$pct_met
T3oxBS_chr18 <- meth_cpg_T3oxBS_intersection[chr == "chr18"]$pct_met
T3oxBS_chr19 <- meth_cpg_T3oxBS_intersection[chr == "chr19"]$pct_met
T3oxBS_chr20 <- meth_cpg_T3oxBS_intersection[chr == "chr20"]$pct_met
T3oxBS_chr21 <- meth_cpg_T3oxBS_intersection[chr == "chr21"]$pct_met
T3oxBS_chr22 <- meth_cpg_T3oxBS_intersection[chr == "chr22"]$pct_met
T3oxBS_chrX <- meth_cpg_T3oxBS_intersection[chr == "chrX"]$pct_met

M8oxBS_chr1 <- meth_cpg_M8oxBS_intersection[chr == "chr1"]$pct_met
M8oxBS_chr2 <- meth_cpg_M8oxBS_intersection[chr == "chr2"]$pct_met
M8oxBS_chr3 <- meth_cpg_M8oxBS_intersection[chr == "chr3"]$pct_met
M8oxBS_chr4 <- meth_cpg_M8oxBS_intersection[chr == "chr4"]$pct_met
M8oxBS_chr5 <- meth_cpg_M8oxBS_intersection[chr == "chr5"]$pct_met
M8oxBS_chr6 <- meth_cpg_M8oxBS_intersection[chr == "chr6"]$pct_met
M8oxBS_chr7 <- meth_cpg_M8oxBS_intersection[chr == "chr7"]$pct_met
M8oxBS_chr8 <- meth_cpg_M8oxBS_intersection[chr == "chr8"]$pct_met
M8oxBS_chr9 <- meth_cpg_M8oxBS_intersection[chr == "chr9"]$pct_met
M8oxBS_chr10 <- meth_cpg_M8oxBS_intersection[chr == "chr10"]$pct_met
M8oxBS_chr11 <- meth_cpg_M8oxBS_intersection[chr == "chr11"]$pct_met
M8oxBS_chr12 <- meth_cpg_M8oxBS_intersection[chr == "chr12"]$pct_met
M8oxBS_chr13 <- meth_cpg_M8oxBS_intersection[chr == "chr13"]$pct_met
M8oxBS_chr14 <- meth_cpg_M8oxBS_intersection[chr == "chr14"]$pct_met
M8oxBS_chr15 <- meth_cpg_M8oxBS_intersection[chr == "chr15"]$pct_met
M8oxBS_chr16 <- meth_cpg_M8oxBS_intersection[chr == "chr16"]$pct_met
M8oxBS_chr17 <- meth_cpg_M8oxBS_intersection[chr == "chr17"]$pct_met
M8oxBS_chr18 <- meth_cpg_M8oxBS_intersection[chr == "chr18"]$pct_met
M8oxBS_chr19 <- meth_cpg_M8oxBS_intersection[chr == "chr19"]$pct_met
M8oxBS_chr20 <- meth_cpg_M8oxBS_intersection[chr == "chr20"]$pct_met
M8oxBS_chr21 <- meth_cpg_M8oxBS_intersection[chr == "chr21"]$pct_met
M8oxBS_chr22 <- meth_cpg_M8oxBS_intersection[chr == "chr22"]$pct_met
M8oxBS_chrX <- meth_cpg_M8oxBS_intersection[chr == "chrX"]$pct_met

diff_chr1 <- meth_cpg_diff[chr == "chr1"]$pct_met_diff
diff_chr2 <- meth_cpg_diff[chr == "chr2"]$pct_met_diff
diff_chr3 <- meth_cpg_diff[chr == "chr3"]$pct_met_diff
diff_chr4 <- meth_cpg_diff[chr == "chr4"]$pct_met_diff
diff_chr5 <- meth_cpg_diff[chr == "chr5"]$pct_met_diff
diff_chr6 <- meth_cpg_diff[chr == "chr6"]$pct_met_diff
diff_chr7 <- meth_cpg_diff[chr == "chr7"]$pct_met_diff
diff_chr8 <- meth_cpg_diff[chr == "chr8"]$pct_met_diff
diff_chr9 <- meth_cpg_diff[chr == "chr9"]$pct_met_diff
diff_chr10 <- meth_cpg_diff[chr == "chr10"]$pct_met_diff
diff_chr11 <- meth_cpg_diff[chr == "chr11"]$pct_met_diff
diff_chr12 <- meth_cpg_diff[chr == "chr12"]$pct_met_diff
diff_chr13 <- meth_cpg_diff[chr == "chr13"]$pct_met_diff
diff_chr14 <- meth_cpg_diff[chr == "chr14"]$pct_met_diff
diff_chr15 <- meth_cpg_diff[chr == "chr15"]$pct_met_diff
diff_chr16 <- meth_cpg_diff[chr == "chr16"]$pct_met_diff
diff_chr17 <- meth_cpg_diff[chr == "chr17"]$pct_met_diff
diff_chr18 <- meth_cpg_diff[chr == "chr18"]$pct_met_diff
diff_chr19 <- meth_cpg_diff[chr == "chr19"]$pct_met_diff
diff_chr20 <- meth_cpg_diff[chr == "chr20"]$pct_met_diff
diff_chr21 <- meth_cpg_diff[chr == "chr21"]$pct_met_diff
diff_chr22 <- meth_cpg_diff[chr == "chr22"]$pct_met_diff
diff_chrX <- meth_cpg_diff[chr == "chrX"]$pct_met_diff

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160121_5mC_cpg_chromosomes_vioplot.pdf", width=40)
my.vioplot(T3oxBS_chr1[!is.na(T3oxBS_chr1)], M8oxBS_chr1[!is.na(M8oxBS_chr1)], T3oxBS_chr2[!is.na(T3oxBS_chr2)], M8oxBS_chr2[!is.na(M8oxBS_chr2)], T3oxBS_chr3[!is.na(T3oxBS_chr3)], M8oxBS_chr3[!is.na(M8oxBS_chr3)], T3oxBS_chr4[!is.na(T3oxBS_chr4)], M8oxBS_chr4[!is.na(M8oxBS_chr4)], T3oxBS_chr5[!is.na(T3oxBS_chr5)], M8oxBS_chr5[!is.na(M8oxBS_chr5)], T3oxBS_chr6[!is.na(T3oxBS_chr6)], M8oxBS_chr6[!is.na(M8oxBS_chr6)], T3oxBS_chr7[!is.na(T3oxBS_chr7)], M8oxBS_chr7[!is.na(M8oxBS_chr7)], T3oxBS_chr8[!is.na(T3oxBS_chr8)], M8oxBS_chr8[!is.na(M8oxBS_chr8)], T3oxBS_chr9[!is.na(T3oxBS_chr9)], M8oxBS_chr9[!is.na(M8oxBS_chr9)], T3oxBS_chr10[!is.na(T3oxBS_chr10)], M8oxBS_chr10[!is.na(M8oxBS_chr10)], T3oxBS_chr11[!is.na(T3oxBS_chr11)], M8oxBS_chr11[!is.na(M8oxBS_chr11)], T3oxBS_chr12[!is.na(T3oxBS_chr12)], M8oxBS_chr12[!is.na(M8oxBS_chr12)], T3oxBS_chr13[!is.na(T3oxBS_chr13)], M8oxBS_chr13[!is.na(M8oxBS_chr13)], T3oxBS_chr14[!is.na(T3oxBS_chr14)], M8oxBS_chr14[!is.na(M8oxBS_chr14)], T3oxBS_chr15[!is.na(T3oxBS_chr15)], M8oxBS_chr15[!is.na(M8oxBS_chr15)], T3oxBS_chr16[!is.na(T3oxBS_chr16)], M8oxBS_chr16[!is.na(M8oxBS_chr16)], T3oxBS_chr17[!is.na(T3oxBS_chr17)], M8oxBS_chr17[!is.na(M8oxBS_chr17)], T3oxBS_chr18[!is.na(T3oxBS_chr18)], M8oxBS_chr18[!is.na(M8oxBS_chr18)], T3oxBS_chr19[!is.na(T3oxBS_chr19)], M8oxBS_chr19[!is.na(M8oxBS_chr19)], T3oxBS_chr20[!is.na(T3oxBS_chr20)], M8oxBS_chr20[!is.na(M8oxBS_chr20)], T3oxBS_chr21[!is.na(T3oxBS_chr21)], M8oxBS_chr21[!is.na(M8oxBS_chr21)], T3oxBS_chr22[!is.na(T3oxBS_chr22)], M8oxBS_chr22[!is.na(M8oxBS_chr22)], T3oxBS_chrX[!is.na(T3oxBS_chrX)], M8oxBS_chrX[!is.na(M8oxBS_chrX)], col=rep("gold", 46) , names=c("chr1_T", "chr1_M", "chr2_T", "chr2_M", "chr3_T", "chr3_M", "chr4_T", "chr4_M", "chr5_T", "chr5_M", "chr6_T", "chr6_M", "chr7_T", "chr7_M", "chr8_T", "chr8_M", "chr9_T", "chr9_M", "chr10_T", "chr10_M", "chr11_T", "chr11_M", "chr12_T", "chr12_M", "chr13_T", "chr13_M", "chr14_T", "chr14_M", "chr15_T", "chr15_M", "chr16_T", "chr16_M", "chr17_T", "chr17_M", "chr18_T", "chr18_M", "chr19_T", "chr19_M", "chr20_T", "chr20_M", "chr21_T", "chr21_M", "chr22_T", "chr22_M", "chrX_T", "chrX_M"))
title(main = "Tumour and Margin 5mC across all chromosomes", ylab="% 5mC (Tumour and Margin)")
grid()
dev.off()

boxplot(T3oxBS_chr1[!is.na(T3oxBS_chr1)], M8oxBS_chr1[!is.na(M8oxBS_chr1)], T3oxBS_chr2[!is.na(T3oxBS_chr2)], M8oxBS_chr2[!is.na(M8oxBS_chr2)], T3oxBS_chr3[!is.na(T3oxBS_chr3)], M8oxBS_chr3[!is.na(M8oxBS_chr3)], T3oxBS_chr4[!is.na(T3oxBS_chr4)], M8oxBS_chr4[!is.na(M8oxBS_chr4)], T3oxBS_chr5[!is.na(T3oxBS_chr5)], M8oxBS_chr5[!is.na(M8oxBS_chr5)], T3oxBS_chr6[!is.na(T3oxBS_chr6)], M8oxBS_chr6[!is.na(M8oxBS_chr6)], T3oxBS_chr7[!is.na(T3oxBS_chr7)], M8oxBS_chr7[!is.na(M8oxBS_chr7)], T3oxBS_chr8[!is.na(T3oxBS_chr8)], M8oxBS_chr8[!is.na(M8oxBS_chr8)], T3oxBS_chr9[!is.na(T3oxBS_chr9)], M8oxBS_chr9[!is.na(M8oxBS_chr9)], T3oxBS_chr10[!is.na(T3oxBS_chr10)], M8oxBS_chr10[!is.na(M8oxBS_chr10)], T3oxBS_chr11[!is.na(T3oxBS_chr11)], M8oxBS_chr11[!is.na(M8oxBS_chr11)], T3oxBS_chr12[!is.na(T3oxBS_chr12)], M8oxBS_chr12[!is.na(M8oxBS_chr12)], T3oxBS_chr13[!is.na(T3oxBS_chr13)], M8oxBS_chr13[!is.na(M8oxBS_chr13)], T3oxBS_chr14[!is.na(T3oxBS_chr14)], M8oxBS_chr14[!is.na(M8oxBS_chr14)], T3oxBS_chr15[!is.na(T3oxBS_chr15)], M8oxBS_chr15[!is.na(M8oxBS_chr15)], T3oxBS_chr16[!is.na(T3oxBS_chr16)], M8oxBS_chr16[!is.na(M8oxBS_chr16)], T3oxBS_chr17[!is.na(T3oxBS_chr17)], M8oxBS_chr17[!is.na(M8oxBS_chr17)], T3oxBS_chr18[!is.na(T3oxBS_chr18)], M8oxBS_chr18[!is.na(M8oxBS_chr18)], T3oxBS_chr19[!is.na(T3oxBS_chr19)], M8oxBS_chr19[!is.na(M8oxBS_chr19)], T3oxBS_chr20[!is.na(T3oxBS_chr20)], M8oxBS_chr20[!is.na(M8oxBS_chr20)], T3oxBS_chr21[!is.na(T3oxBS_chr21)], M8oxBS_chr21[!is.na(M8oxBS_chr21)], T3oxBS_chr22[!is.na(T3oxBS_chr22)], M8oxBS_chr22[!is.na(M8oxBS_chr22)], T3oxBS_chrX[!is.na(T3oxBS_chrX)], M8oxBS_chrX[!is.na(M8oxBS_chrX)])

# both vioplot and boxplot show median rather than mean in the plot


pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160121_5mC_cpg_chromosomes_diff_vioplot.pdf", width=15)
my.vioplot(diff_chr1[!is.na(diff_chr1)], diff_chr2[!is.na(diff_chr2)], diff_chr3[!is.na(diff_chr3)], diff_chr4[!is.na(diff_chr4)], diff_chr5[!is.na(diff_chr5)], diff_chr6[!is.na(diff_chr6)], diff_chr7[!is.na(diff_chr7)], diff_chr8[!is.na(diff_chr8)], diff_chr9[!is.na(diff_chr9)], diff_chr10[!is.na(diff_chr10)], diff_chr11[!is.na(diff_chr11)], diff_chr12[!is.na(diff_chr12)], diff_chr13[!is.na(diff_chr13)], diff_chr14[!is.na(diff_chr14)], diff_chr15[!is.na(diff_chr15)], diff_chr16[!is.na(diff_chr16)], diff_chr17[!is.na(diff_chr17)], diff_chr18[!is.na(diff_chr18)], diff_chr19[!is.na(diff_chr19)], diff_chr20[!is.na(diff_chr20)], diff_chr21[!is.na(diff_chr21)], diff_chr22[!is.na(diff_chr22)], diff_chrX[!is.na(diff_chrX)], col=rep("gold", 23), names=c("chr1", "chr2", "chr3", "chr4", "chr5", "chr6", "chr7", "chr8", "chr9", "chr10", "chr11", "chr12", "chr13", "chr14", "chr15", "chr16", "chr17", "chr18", "chr19", "chr20", "chr21", "chr22", "chrX"))
title(main = "Tumour and Margin differences on 5mC across all chromosomes", ylab="% 5mC (Tumour - Margin)")
grid()
dev.off()

```



chr7 - mostly polyploid
Most of the chromosome is polyploid, the overall comparison with other chromosomes that are also mostly polyploid (e.g. chr20) and mostly diploid (e.g. chr3, chr8) did not reveal any differences.




chr9 - partly haploid

```R
x1_T3oxBS_chr9 <- meth_cpg_T3oxBS_intersection[chr == "chr9" & start >=3471000 & end < 5288000]$pct_met # cn=1
x2_T3oxBS_chr9 <- meth_cpg_T3oxBS_intersection[chr == "chr9" & start >=9592000 & end < 20053000]$pct_met # cn=1
x3_T3oxBS_chr9 <- meth_cpg_T3oxBS_intersection[chr == "chr9" & start >=20054000 & end < 20057000]$pct_met # cn=2, loh=1
x4_T3oxBS_chr9 <- meth_cpg_T3oxBS_intersection[chr == "chr9" & start >=20058000 & end < 28526000]$pct_met # cn=0
x5_T3oxBS_chr9 <- meth_cpg_T3oxBS_intersection[chr == "chr9" & start >=28527000 & end < 34681000]$pct_met # cn=1
x6_T3oxBS_chr9 <- meth_cpg_T3oxBS_intersection[chr == "chr9" & start >=101194000 & end < 102055000]$pct_met # cn=1

x1_M8oxBS_chr9 <- meth_cpg_M8oxBS_intersection[chr == "chr9" & start >=3471000 & end < 5288000]$pct_met # cn=1
x2_M8oxBS_chr9 <- meth_cpg_M8oxBS_intersection[chr == "chr9" & start >=9592000 & end < 20053000]$pct_met # cn=1
x3_M8oxBS_chr9 <- meth_cpg_M8oxBS_intersection[chr == "chr9" & start >=20054000 & end < 20057000]$pct_met # cn=2, loh=1
x4_M8oxBS_chr9 <- meth_cpg_M8oxBS_intersection[chr == "chr9" & start >=20058000 & end < 28526000]$pct_met # cn=0
x5_M8oxBS_chr9 <- meth_cpg_M8oxBS_intersection[chr == "chr9" & start >=28527000 & end < 34681000]$pct_met # cn=1
x6_M8oxBS_chr9 <- meth_cpg_M8oxBS_intersection[chr == "chr9" & start >=101194000 & end < 102055000]$pct_met # cn=1

diff1_chr9 <- meth_cpg_diff[chr == "chr9" & start >=3471000 & end < 5288000]$pct_met_diff # cn=1
diff2_chr9 <- meth_cpg_diff[chr == "chr9" & start >=9592000 & end < 20053000]$pct_met_diff # cn=1
diff3_chr9 <- meth_cpg_diff[chr == "chr9" & start >=20054000 & end < 20057000]$pct_met_diff # cn=2, loh=1
diff4_chr9 <- meth_cpg_diff[chr == "chr9" & start >=20058000 & end < 28526000]$pct_met_diff # cn=0
diff5_chr9 <- meth_cpg_diff[chr == "chr9" & start >=28527000 & end < 34681000]$pct_met_diff # cn=1
diff6_chr9 <- meth_cpg_diff[chr == "chr9" & start >=101194000 & end < 102055000]$pct_met_diff # cn=1

my.vioplot(x1_T3oxBS_chr9[!is.na(x1_T3oxBS_chr9)], x1_M8oxBS_chr9[!is.na(x1_M8oxBS_chr9)], x2_T3oxBS_chr9[!is.na(x2_T3oxBS_chr9)], x2_M8oxBS_chr9[!is.na(x2_M8oxBS_chr9)], x3_T3oxBS_chr9[!is.na(x3_T3oxBS_chr9)], x3_M8oxBS_chr9[!is.na(x3_M8oxBS_chr9)], x4_T3oxBS_chr9[!is.na(x4_T3oxBS_chr9)], x4_M8oxBS_chr9[!is.na(x4_M8oxBS_chr9)], x5_T3oxBS_chr9[!is.na(x5_T3oxBS_chr9)], x5_M8oxBS_chr9[!is.na(x5_M8oxBS_chr9)], x6_T3oxBS_chr9[!is.na(x6_T3oxBS_chr9)], x6_M8oxBS_chr9[!is.na(x6_M8oxBS_chr9)], col=c("green3", "green3", "green3", "green3", "blue", "blue", "white", "white", "green3", "green3", "green3", "green3"), names=c("haploid_T", "haploid_M", "haploid_T", "haploid_M", "diploid_copyneutral_T", "diploid_copyneutral_M", "deletion_T", "deletion_M", "haploid_T", "haploid_M", "haploid_T", "haploid_M"))
title(main = "chr9: CNVs and (Tumour/Margin) 5mC", ylab="% 5mC (Tumour/Margin)")
grid()

my.vioplot(diff1_chr9[!is.na(diff1_chr9)], diff2_chr9[!is.na(diff2_chr9)], diff3_chr9[!is.na(diff3_chr9)], diff4_chr9[!is.na(diff4_chr9)], diff5_chr9[!is.na(diff5_chr9)], diff6_chr9[!is.na(diff6_chr9)], col=c("green3", "green3", "blue", "white", "green3", "green3"), names=c("haploid", "haploid", "diploid_copyneutral", "deletion", "haploid", "haploid"))
title(main = "chr9: CNVs and (Tumour - Margin) 5mC", ylab="% 5mC (Tumour - Margin)")
grid()

```
Perhaps the only obvious observation here is that the levels of 5mC in tumour and margin are more similar to each other in the diploid (copy neutral regions) and the deletion region.




chr10 - mostly haploid
Overall, the chromosome does not look any different to any other chromosome in the genome.

Compare haploid versus diploid - copy neutral.

```R
x1_T3oxBS_chr10 <- meth_cpg_T3oxBS_intersection[chr == "chr10" & start >=1 & end < 90000]$pct_met # cn=2, loh=1
x2_T3oxBS_chr10 <- meth_cpg_T3oxBS_intersection[chr == "chr10" & start >=91000 & end < 38571000]$pct_met # cn=1
x3_T3oxBS_chr10 <- meth_cpg_T3oxBS_intersection[chr == "chr10" & start >=42400000 & end < 46266000]$pct_met # cn=1
x4_T3oxBS_chr10 <- meth_cpg_T3oxBS_intersection[chr == "chr10" & start >=46267000 & end < 46938000]$pct_met # cn=2, loh=1
x5_T3oxBS_chr10 <- meth_cpg_T3oxBS_intersection[chr == "chr10" & start >=46939000 & end < 47701000]$pct_met # cn=1
x6_T3oxBS_chr10 <- meth_cpg_T3oxBS_intersection[chr == "chr10" & start >=47702000 & end < 48290000]$pct_met # cn=2, loh=1
x7_T3oxBS_chr10 <- meth_cpg_T3oxBS_intersection[chr == "chr10" & start >=48291000 & end < 51129000]$pct_met # cn=1
x8_T3oxBS_chr10 <- meth_cpg_T3oxBS_intersection[chr == "chr10" & start >=51130000 & end < 51998000]$pct_met # cn=2, loh=1
x9_T3oxBS_chr10 <- meth_cpg_T3oxBS_intersection[chr == "chr10" & start >=51999000 & end < 135534000]$pct_met # cn=1

x1_M8oxBS_chr10 <- meth_cpg_M8oxBS_intersection[chr == "chr10" & start >=1 & end < 90000]$pct_met # cn=2, loh=1
x2_M8oxBS_chr10 <- meth_cpg_M8oxBS_intersection[chr == "chr10" & start >=91000 & end < 38571000]$pct_met # cn=1
x3_M8oxBS_chr10 <- meth_cpg_M8oxBS_intersection[chr == "chr10" & start >=42400000 & end < 46266000]$pct_met # cn=1
x4_M8oxBS_chr10 <- meth_cpg_M8oxBS_intersection[chr == "chr10" & start >=46267000 & end < 46938000]$pct_met # cn=2, loh=1
x5_M8oxBS_chr10 <- meth_cpg_M8oxBS_intersection[chr == "chr10" & start >=46939000 & end < 47701000]$pct_met # cn=1
x6_M8oxBS_chr10 <- meth_cpg_M8oxBS_intersection[chr == "chr10" & start >=47702000 & end < 48290000]$pct_met # cn=2, loh=1
x7_M8oxBS_chr10 <- meth_cpg_M8oxBS_intersection[chr == "chr10" & start >=48291000 & end < 51129000]$pct_met # cn=1
x8_M8oxBS_chr10 <- meth_cpg_M8oxBS_intersection[chr == "chr10" & start >=51130000 & end < 51998000]$pct_met # cn=2, loh=1
x9_M8oxBS_chr10 <- meth_cpg_M8oxBS_intersection[chr == "chr10" & start >=51999000 & end < 135534000]$pct_met # cn=1

diff1_chr10 <- meth_cpg_diff[chr == "chr10" & start >=1 & end < 90000]$pct_met_diff # cn=2, loh=1
diff2_chr10 <- meth_cpg_diff[chr == "chr10" & start >=91000 & end < 38571000]$pct_met_diff # cn=1
diff3_chr10 <- meth_cpg_diff[chr == "chr10" & start >=42400000 & end < 46266000]$pct_met_diff # cn=1
diff4_chr10 <- meth_cpg_diff[chr == "chr10" & start >=46267000 & end < 46938000]$pct_met_diff # cn=2, loh=1
diff5_chr10 <- meth_cpg_diff[chr == "chr10" & start >=46939000 & end < 47701000]$pct_met_diff # cn=1
diff6_chr10 <- meth_cpg_diff[chr == "chr10" & start >=47702000 & end < 48290000]$pct_met_diff # cn=2, loh=1
diff7_chr10 <- meth_cpg_diff[chr == "chr10" & start >=48291000 & end < 51129000]$pct_met_diff # cn=1
diff8_chr10 <- meth_cpg_diff[chr == "chr10" & start >=51130000 & end < 51998000]$pct_met_diff # cn=2, loh=1
diff9_chr10 <- meth_cpg_diff[chr == "chr10" & start >=51999000 & end < 135534000]$pct_met_diff # cn=1

my.vioplot(x1_T3oxBS_chr10[!is.na(x1_T3oxBS_chr10)], x1_M8oxBS_chr10[!is.na(x1_M8oxBS_chr10)], x2_T3oxBS_chr10[!is.na(x2_T3oxBS_chr10)], x2_M8oxBS_chr10[!is.na(x2_M8oxBS_chr10)], x3_T3oxBS_chr10[!is.na(x3_T3oxBS_chr10)], x3_M8oxBS_chr10[!is.na(x3_M8oxBS_chr10)], x4_T3oxBS_chr10[!is.na(x4_T3oxBS_chr10)], x4_M8oxBS_chr10[!is.na(x4_M8oxBS_chr10)], x5_T3oxBS_chr10[!is.na(x5_T3oxBS_chr10)], x5_M8oxBS_chr10[!is.na(x5_M8oxBS_chr10)], x6_T3oxBS_chr10[!is.na(x6_T3oxBS_chr10)], x6_M8oxBS_chr10[!is.na(x6_M8oxBS_chr10)], x7_T3oxBS_chr10[!is.na(x7_T3oxBS_chr10)], x7_M8oxBS_chr10[!is.na(x7_M8oxBS_chr10)], x8_T3oxBS_chr10[!is.na(x8_T3oxBS_chr10)], x8_M8oxBS_chr10[!is.na(x8_M8oxBS_chr10)], x9_T3oxBS_chr10[!is.na(x9_T3oxBS_chr10)], x9_M8oxBS_chr10[!is.na(x9_M8oxBS_chr10)], col=c("blue", "blue", "green3", "green3", "green3", "green3", "blue", "blue", "green3", "green3", "blue", "blue", "green3", "green3", "blue", "blue", "green3", "green3"), names=c("diploid_copyneutral_T", "diploid_copyneutral_M", "haploid_T", "haploid_M", "haploid_T", "haploid_M", "diploid_copyneutral_T", "diploid_copyneutral_M", "haploid_T", "haploid_M", "diploid_copyneutral_T", "diploid_copyneutral_M", "haploid_T", "haploid_M", "diploid_copyneutral_T", "diploid_copyneutral_M", "haploid_T", "haploid_M"))
title(main = "chr10: CNVs and (Tumour/Margin) 5mC", ylab="% 5mC (Tumour/Margin)")
grid()

my.vioplot(diff1_chr10[!is.na(diff1_chr10)], diff2_chr10[!is.na(diff2_chr10)], diff3_chr10[!is.na(diff3_chr10)], diff4_chr10[!is.na(diff4_chr10)], diff5_chr10[!is.na(diff5_chr10)], diff6_chr10[!is.na(diff6_chr10)], diff7_chr10[!is.na(diff7_chr10)], diff8_chr10[!is.na(diff8_chr10)], diff9_chr10[!is.na(diff9_chr10)], col=c("blue", "green3", "green3", "blue", "green3", "blue", "green3", "blue", "green3"), names=c("diploid_copyneutral", "haploid", "haploid", "diploid_copyneutral", "haploid", "diploid_copyneutral", "haploid", "diploid_copyneutral", "haploid"))
title(main = "chr10: CNVs and (Tumour - Margin) 5mC", ylab="% 5mC (Tumour - Margin)")
grid()

diff1_chr10_diploid_copyneutral<-c(diff1_chr10, diff4_chr10, diff6_chr10, diff8_chr10)
diff2_chr10_haploid<-c(diff2_chr10, diff3_chr10, diff5_chr10, diff7_chr10, diff9_chr10)

my.vioplot(diff1_chr10_diploid_copyneutral[!is.na(diff1_chr10_diploid_copyneutral)], diff2_chr10_haploid[!is.na(diff2_chr10_haploid)], col=c("blue", "green3"), names=c("diploid_copyneutral", "haploid"))
t.test(diff1_chr10_diploid_copyneutral[!is.na(diff1_chr10_diploid_copyneutral)], diff2_chr10_haploid[!is.na(diff2_chr10_haploid)]) # p-value = 1.443e-11
```
Small but perhaps significant difference between the diploid_copyneutral and haploid 5mC levels in chr10.




chr11 - a mixture of many types of CNVs


```R
x1_T3oxBS_chr11 <- meth_cpg_T3oxBS_intersection[chr == "chr11" & start >=187000 & end < 3260000]$pct_met # cn=3
x2_T3oxBS_chr11 <- meth_cpg_T3oxBS_intersection[chr == "chr11" & start >=3260000 & end < 70585000]$pct_met # cn=2
x3_T3oxBS_chr11 <- meth_cpg_T3oxBS_intersection[chr == "chr11" & start >=70585000 & end < 75296000]$pct_met # cn=1
x4_T3oxBS_chr11 <- meth_cpg_T3oxBS_intersection[chr == "chr11" & start >=75297000 & end < 75519000]$pct_met # cn=3
x5_T3oxBS_chr11 <- meth_cpg_T3oxBS_intersection[chr == "chr11" & start >=75520000 & end < 89664000]$pct_met # cn=1
x6_T3oxBS_chr11 <- meth_cpg_T3oxBS_intersection[chr == "chr11" & start >=89665000 & end < 89830000]$pct_met # cn=2, loh=1
x7_T3oxBS_chr11 <- meth_cpg_T3oxBS_intersection[chr == "chr11" & start >=89831000 & end < 135006000]$pct_met # cn=1

x1_M8oxBS_chr11 <- meth_cpg_M8oxBS_intersection[chr == "chr11" & start >=187000 & end < 3260000]$pct_met # cn=3
x2_M8oxBS_chr11 <- meth_cpg_M8oxBS_intersection[chr == "chr11" & start >=3260000 & end < 70585000]$pct_met # cn=2
x3_M8oxBS_chr11 <- meth_cpg_M8oxBS_intersection[chr == "chr11" & start >=70585000 & end < 75296000]$pct_met # cn=1
x4_M8oxBS_chr11 <- meth_cpg_M8oxBS_intersection[chr == "chr11" & start >=75297000 & end < 75519000]$pct_met # cn=3
x5_M8oxBS_chr11 <- meth_cpg_M8oxBS_intersection[chr == "chr11" & start >=75520000 & end < 89664000]$pct_met # cn=1
x6_M8oxBS_chr11 <- meth_cpg_M8oxBS_intersection[chr == "chr11" & start >=89665000 & end < 89830000]$pct_met # cn=2, loh=1
x7_M8oxBS_chr11 <- meth_cpg_M8oxBS_intersection[chr == "chr11" & start >=89831000 & end < 135006000]$pct_met # cn=1

diff1_chr11 <- meth_cpg_diff[chr == "chr11" & start >=187000 & end < 3260000]$pct_met_diff # cn=3
diff2_chr11 <- meth_cpg_diff[chr == "chr11" & start >=3260000 & end < 70585000]$pct_met_diff # cn=2
diff3_chr11 <- meth_cpg_diff[chr == "chr11" & start >=70585000 & end < 75296000]$pct_met_diff # cn=1
diff4_chr11 <- meth_cpg_diff[chr == "chr11" & start >=75297000 & end < 75519000]$pct_met_diff # cn=3
diff5_chr11 <- meth_cpg_diff[chr == "chr11" & start >=75520000 & end < 89664000]$pct_met_diff # cn=1
diff6_chr11 <- meth_cpg_diff[chr == "chr11" & start >=89665000 & end < 89830000]$pct_met_diff # cn=2, loh=1
diff7_chr11 <- meth_cpg_diff[chr == "chr11" & start >=89831000 & end < 135006000]$pct_met_diff # cn=1

my.vioplot(x1_T3oxBS_chr11[!is.na(x1_T3oxBS_chr11)], x1_M8oxBS_chr11[!is.na(x1_M8oxBS_chr11)], x2_T3oxBS_chr11[!is.na(x2_T3oxBS_chr11)], x2_M8oxBS_chr11[!is.na(x2_M8oxBS_chr11)], x3_T3oxBS_chr11[!is.na(x3_T3oxBS_chr11)], x3_M8oxBS_chr11[!is.na(x3_M8oxBS_chr11)], x4_T3oxBS_chr11[!is.na(x4_T3oxBS_chr11)], x4_M8oxBS_chr11[!is.na(x4_M8oxBS_chr11)], x5_T3oxBS_chr11[!is.na(x5_T3oxBS_chr11)], x5_M8oxBS_chr11[!is.na(x5_M8oxBS_chr11)], x6_T3oxBS_chr11[!is.na(x6_T3oxBS_chr11)], x6_M8oxBS_chr11[!is.na(x6_M8oxBS_chr11)], x7_T3oxBS_chr11[!is.na(x7_T3oxBS_chr11)], x7_M8oxBS_chr11[!is.na(x7_M8oxBS_chr11)], col=c("red", "red", "gray30", "gray30", "green3", "green3", "red", "red", "green3", "green3", "blue", "blue", "green3", "green3"), names=c("triploid_T", "triploid_M", "diploid_T", "diploid_M", "haploid_T", "haploid_M", "triploid_T", "triploid_M", "haploid_T", "haploid_M", "diploid_copyneutral_T", "diploid_copyneutral_M", "haploid_T", "haploid_M"))
title(main = "chr11: CNVs and (Tumour/Margin) 5mC", ylab="% 5mC (Tumour/Margin)")
grid()

my.vioplot(diff1_chr11[!is.na(diff1_chr11)], diff2_chr11[!is.na(diff2_chr11)], diff3_chr11[!is.na(diff3_chr11)], diff4_chr11[!is.na(diff4_chr11)], diff5_chr11[!is.na(diff5_chr11)], diff6_chr11[!is.na(diff6_chr11)], diff7_chr11[!is.na(diff7_chr11)], col=c("red", "gray30", "green3", "red", "green3", "blue", "green3"), names=c("triploid", "diploid", "haploid", "triploid", "haploid", "diploid_copyneutral", "haploid"))
title(main = "chr11: CNVs and (Tumour - Margin) 5mC", ylab="% 5mC (Tumour - Margin)")
grid()

```






Intersect the CNV and methylation files (bedtools/python). Use merged methylation files produced by Dario.

Copy merged data to martin03 folders, in nas-srv001:

ear042_M8BS.cpg.bedGraph.gz
ear043_M8oxBS.cpg.bedGraph.gz
ear044_T3BS.cpg.bedGraph.gz
ear045_T3oxBS.cpg.bedGraph.gz


```bash
cp /data/sblab_data1/berald01/projects/20150501_methylation_brain/20151106_genome_methyl/*.cpg.bedGraph.gz /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg
cd /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg
zcat ear043_M8oxBS.cpg.bedGraph.gz | grep -w "10468"
zcat ear043_M8oxBS.hiseq201509.cpg.bedGraph.gz | grep -w "10468"
zcat ear043_M8oxBS.hiseq201512.cpg.bedGraph.gz | grep -w "10468"
```

The sum matches.

```python
import vcf # https://pyvcf.readthedocs.org/en/latest/index.html
vcf_reader = vcf.Reader(open('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_A01.somatic.CNV.vcf', 'r'))

cnv_tvb=[]

for record in vcf_reader:
	chr=record.CHROM
	start=record.POS
	end=record.INFO["END"]
	cn=record.INFO["CN"]
	loh=record.INFO["LOH"]
	cnscore=record.INFO["CNscore"]
	cnv_tvb.append((chr, start, end, cn, loh, cnscore))


len(cnv_tvb) # 88 CNVs in the tumour vs. blood comparison


vcf_reader = vcf.Reader(open('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_C01.somatic.CNV.vcf', 'r'))

cnv_tvm=[]

for record in vcf_reader:
	chr=record.CHROM
	start=record.POS
	end=record.INFO["END"]
	cn=record.INFO["CN"]
	loh=record.INFO["LOH"]
	cnscore=record.INFO["CNscore"]
	cnv_tvm.append((chr, start, end, cn, loh, cnscore))


len(cnv_tvm) # 239 CNVs in the tumour vs. margin comparison


vcf_reader = vcf.Reader(open('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv/CancerLP2000729-DNA_C01_NormalLP2000729-DNA_A01.somatic.CNV.vcf', 'r'))

cnv_mvb=[]

for record in vcf_reader:
	chr=record.CHROM
	start=record.POS
	end=record.INFO["END"]
	cn=record.INFO["CN"]
	loh=record.INFO["LOH"]
	cnscore=record.INFO["CNscore"]
	cnv_mvb.append((chr, start, end, cn, loh, cnscore))


len(cnv_mvb) # 0 CNVs in the margin vs. blood comparison


bed_tvb=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_A01.somatic.CNV.bedGraph", "w")

for cnv in cnv_tvb:
	bed_tvb.write("%s\n" % ("\t".join(["chr"+cnv[0], str(cnv[1]), str(cnv[2]), str(cnv[3]), str(cnv[4]), str(cnv[5])])))


bed_tvb.close()
```


Use bedtools to intersect and find overlaps. Testing bedtools:

```bash
cd /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/
pigz -dk ear042_M8BS.cpg.bedGraph.gz
pigz -dk ear043_M8oxBS.cpg.bedGraph.gz
pigz -dk ear044_T3BS.cpg.bedGraph.gz
pigz -dk ear045_T3oxBS.cpg.bedGraph.gz
wc -l ear042_M8BS.cpg.bedGraph # 27253473
wc -l ear043_M8oxBS.cpg.bedGraph # 27196199
wc -l ear044_T3BS.cpg.bedGraph # 27231244
wc -l ear045_T3oxBS.cpg.bedGraph # 27217723

head -n 100 ear042_M8BS.cpg.bedGraph > ear042_M8BS.cpg.test.bedGraph
head -n 100 ear043_M8oxBS.cpg.bedGraph > ear043_M8oxBS.cpg.test.bedGraph
head -n 100 ear044_T3BS.cpg.bedGraph > ear044_T3BS.cpg.test.bedGraph
head -n 100 ear045_T3oxBS.cpg.bedGraph > ear045_T3oxBS.cpg.test.bedGraph
head -n 10000 ear043_M8oxBS.cpg.bedGraph > ear043_M8oxBS.cpg.test2.bedGraph # so that it covers at least part of the first CNV in chr1

bedtools intersect -a ear043_M8oxBS.cpg.test.bedGraph -b ear045_T3oxBS.cpg.test.bedGraph # respect to ear043_M8oxBS.cpg.test.bedGraph

bedtools intersect -a ear042_M8BS.cpg.test.bedGraph -b ear043_M8oxBS.cpg.test.bedGraph
bedtools intersect -a ear042_M8BS.cpg.test.bedGraph -b ear043_M8oxBS.cpg.test.bedGraph -loj

bedtools intersect -a ear043_M8oxBS.cpg.test2.bedGraph -b ../cnv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_A01.somatic.CNV.bedGraph | head
bedtools intersect -a ear043_M8oxBS.cpg.test2.bedGraph -b ../cnv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_A01.somatic.CNV.bedGraph -loj | head
bedtools intersect -a ear043_M8oxBS.cpg.test2.bedGraph -b ../cnv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_A01.somatic.CNV.bedGraph -v | head
```

```bash
bedtools intersect -a ear042_M8BS.cpg.bedGraph -b ../cnv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_A01.somatic.CNV.bedGraph -loj > ear042_M8BS.cpg.tvb.cnv.bedGraph
wc -l ear042_M8BS.cpg.tvb.cnv.bedGraph # 27253473
bedtools intersect -a ear043_M8oxBS.cpg.bedGraph -b ../cnv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_A01.somatic.CNV.bedGraph -loj > ear043_M8oxBS.cpg.tvb.cnv.bedGraph
wc -l ear043_M8oxBS.cpg.tvb.cnv.bedGraph # 27196199
bedtools intersect -a ear044_T3BS.cpg.bedGraph -b ../cnv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_A01.somatic.CNV.bedGraph -loj > ear044_T3BS.cpg.tvb.cnv.bedGraph
wc -l ear044_T3BS.cpg.tvb.cnv.bedGraph # 27231244
bedtools intersect -a ear045_T3oxBS.cpg.bedGraph -b ../cnv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_A01.somatic.CNV.bedGraph -loj > ear045_T3oxBS.cpg.tvb.cnv.bedGraph
wc -l ear045_T3oxBS.cpg.tvb.cnv.bedGraph # 27217723
```



```R
library(data.table)

ear042_M8BS.cpg.tvb.cnv<- fread('cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear042_M8BS.cpg.tvb.cnv.bedGraph')
ear042_M8BS.cpg.tvb.cnv[,V6:=NULL]
ear042_M8BS.cpg.tvb.cnv[,V7:=NULL]
ear042_M8BS.cpg.tvb.cnv[,V8:=NULL]
setnames(ear042_M8BS.cpg.tvb.cnv, c("chr", "start", "end", "cnt_met", "cnt_tot", "cn", "loh", "cnscore"))
ear042_M8BS.cpg.tvb.cnv[, pct_met := cnt_met/cnt_tot]
ear042_M8BS.cpg.tvb.cnv[cn == ".", cn := 2]
ear042_M8BS.cpg.tvb.cnv[loh == ".", loh := 0]
ear042_M8BS.cpg.tvb.cnv[cnscore== ".", cnscore := 0]

ear043_M8oxBS.cpg.tvb.cnv<- fread('cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear043_M8oxBS.cpg.tvb.cnv.bedGraph')
ear043_M8oxBS.cpg.tvb.cnv[,V6:=NULL]
ear043_M8oxBS.cpg.tvb.cnv[,V7:=NULL]
ear043_M8oxBS.cpg.tvb.cnv[,V8:=NULL]
setnames(ear043_M8oxBS.cpg.tvb.cnv, c("chr", "start", "end", "cnt_met", "cnt_tot", "cn", "loh", "cnscore"))
ear043_M8oxBS.cpg.tvb.cnv[, pct_met := cnt_met/cnt_tot]
ear043_M8oxBS.cpg.tvb.cnv[cn == ".", cn := 2]
ear043_M8oxBS.cpg.tvb.cnv[loh == ".", loh := 0]
ear043_M8oxBS.cpg.tvb.cnv[cnscore== ".", cnscore := 0]

ear044_T3BS.cpg.tvb.cnv <- fread('cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear044_T3BS.cpg.tvb.cnv.bedGraph')
ear044_T3BS.cpg.tvb.cnv[,V6:=NULL]
ear044_T3BS.cpg.tvb.cnv[,V7:=NULL]
ear044_T3BS.cpg.tvb.cnv[,V8:=NULL]
setnames(ear044_T3BS.cpg.tvb.cnv, c("chr", "start", "end", "cnt_met", "cnt_tot", "cn", "loh", "cnscore"))
ear044_T3BS.cpg.tvb.cnv[, pct_met := cnt_met/cnt_tot]
ear044_T3BS.cpg.tvb.cnv[cn == ".", cn := 2]
ear044_T3BS.cpg.tvb.cnv[loh == ".", loh := 0]
ear044_T3BS.cpg.tvb.cnv[cnscore== ".", cnscore := 0]

ear045_T3oxBS.cpg.tvb.cnv <- fread('cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear045_T3oxBS.cpg.tvb.cnv.bedGraph')
ear045_T3oxBS.cpg.tvb.cnv[,V6:=NULL]
ear045_T3oxBS.cpg.tvb.cnv[,V7:=NULL]
ear045_T3oxBS.cpg.tvb.cnv[,V8:=NULL]
setnames(ear045_T3oxBS.cpg.tvb.cnv, c("chr", "start", "end", "cnt_met", "cnt_tot", "cn", "loh", "cnscore"))
ear045_T3oxBS.cpg.tvb.cnv[, pct_met := cnt_met/cnt_tot]
ear045_T3oxBS.cpg.tvb.cnv[cn == ".", cn := 2]
ear045_T3oxBS.cpg.tvb.cnv[loh == ".", loh := 0]
ear045_T3oxBS.cpg.tvb.cnv[cnscore== ".", cnscore := 0]



# How many CpGs fall in each region?

table(ear043_M8oxBS.cpg.tvb.cnv$cn)
#       0        1        2        3        4        5 
#   68061  4302213 19399079  2830993   590059     5794 


# How many CpGs fall in loss of heterozygosity regions within diploid regions?
table(ear043_M8oxBS.cpg.tvb.cnv[cn == 2]$loh)
#       0        1 
#18481892   917187 


# How many CpGs fall in loss of heterozygosity regions within triploid regions?
table(ear043_M8oxBS.cpg.tvb.cnv[cn == 3]$loh)
#      0       1 
#2614694  216299 




# 5mC: oxBS

ear043_M8oxBS.cpg.tvb.cnv[, list(avg= mean(pct_met, na.rm= TRUE),  median= median(pct_met, na.rm= TRUE), stdev= sd(pct_met, na.rm= TRUE)), by= list(cn, loh)]
ear043_M8oxBS.cpg.tvb.cnv[, list(avg= mean(pct_met, na.rm= TRUE),  median= median(pct_met, na.rm= TRUE), stdev= sd(pct_met, na.rm= TRUE)), by= list(chr, cn, loh)]

ear045_T3oxBS.cpg.tvb.cnv[, list(avg= mean(pct_met, na.rm= TRUE),  median= median(pct_met, na.rm= TRUE), stdev= sd(pct_met, na.rm= TRUE)), by= list(cn, loh)]
ear045_T3oxBS.cpg.tvb.cnv[, list(avg= mean(pct_met, na.rm= TRUE),  median= median(pct_met, na.rm= TRUE), stdev= sd(pct_met, na.rm= TRUE)), by= list(chr, cn, loh)]

setkey(ear043_M8oxBS.cpg.tvb.cnv, chr, start, end, cn, loh, cnscore)
setkey(ear045_T3oxBS.cpg.tvb.cnv, chr, start, end, cn, loh, cnscore)
diff.oxBS.cpg.tvb.cnv <- merge(ear043_M8oxBS.cpg.tvb.cnv, ear045_T3oxBS.cpg.tvb.cnv)
# 100*27115544/27217723 = 99.62459% of CpGs in common between M8oxBS and T3oxBS
diff.oxBS.cpg.tvb.cnv[, pct_met_diff := pct_met.y - pct_met.x]



library("ggplot2")

p <- ggplot(ear043_M8oxBS.cpg.tvb.cnv[chr=="chr11"], aes(factor(cn), pct_met))
p + geom_violin(scale = "count", aes(fill = factor(cn)), draw_quantiles = 0.5) + scale_fill_manual(values=c("green3", "gray30", "red")) + xlab("Ploidy") + ylab("%5mC Margin") + labs(title = "chr11") + theme_bw() + theme(legend.position = "none")

p <- ggplot(ear043_M8oxBS.cpg.tvb.cnv[chr=="chr11" & cn == 2], aes(factor(loh), pct_met))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5mC Margin") + labs(title = "chr11 diploid") + theme_bw() + theme(legend.position = "none")

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160129_ear043_M8oxBS.cpg.tvb.cnv.pdf")
p <- ggplot(ear043_M8oxBS.cpg.tvb.cnv, aes(factor(cn), pct_met))
p + geom_violin(scale = "area", aes(fill = factor(cn)), draw_quantiles = 0.5) + scale_fill_manual(values=c("white", "green3", "gray30", "red", "red", "red")) + xlab("Ploidy") + ylab("%5mC Margin") + theme_bw() + theme(legend.position = "none")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160129_ear043_M8oxBS.cpg.tvb.cnv.diploid.pdf")
p <- ggplot(ear043_M8oxBS.cpg.tvb.cnv[cn == 2], aes(factor(loh), pct_met))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5mC Margin") + labs(title = "diploid") + theme_bw() + theme(legend.position = "none")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160129_ear043_M8oxBS.cpg.tvb.cnv.triploid.pdf")
p <- ggplot(ear043_M8oxBS.cpg.tvb.cnv[cn == 3], aes(factor(loh), pct_met))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5mC Margin") + labs(title = "triploid") + theme_bw() + theme(legend.position = "none")
dev.off()


# In 20160129_ear043_M8oxBS.cpg.tvb.cnv.pdf, 5mC decreases in Margin as ploidy increases


p <- ggplot(ear045_T3oxBS.cpg.tvb.cnv[chr=="chr11"], aes(factor(cn), pct_met))
p + geom_violin(scale = "count", aes(fill = factor(cn)), draw_quantiles = 0.5) + scale_fill_manual(values=c("green3", "gray30", "red")) + xlab("Ploidy") + ylab("%5mC Tumour") + labs(title = "chr11") + theme_bw() + theme(legend.position = "none")

p <- ggplot(ear045_T3oxBS.cpg.tvb.cnv[chr=="chr11" & cn == 2], aes(factor(loh), pct_met))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5mC Tumour") + labs(title = "chr11 diploid") + theme_bw() + theme(legend.position = "none")

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160129_ear045_T3oxBS.cpg.tvb.cnv.pdf")
p <- ggplot(ear045_T3oxBS.cpg.tvb.cnv, aes(factor(cn), pct_met))
p + geom_violin(scale = "area", aes(fill = factor(cn)), draw_quantiles = 0.5) + scale_fill_manual(values=c("white", "green3", "gray30", "red", "red", "red")) + xlab("Ploidy") + ylab("%5mC Tumour") + theme_bw() + theme(legend.position = "none")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160129_ear045_T3oxBS.cpg.tvb.cnv.diploid.pdf")
p <- ggplot(ear045_T3oxBS.cpg.tvb.cnv[cn == 2], aes(factor(loh), pct_met))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5mC Tumour") + labs(title = "diploid") + theme_bw() + theme(legend.position = "none")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160129_ear045_T3oxBS.cpg.tvb.cnv.triploid.pdf")
p <- ggplot(ear045_T3oxBS.cpg.tvb.cnv[cn == 3], aes(factor(loh), pct_met))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5mC Tumour") + labs(title = "triploid") + theme_bw() + theme(legend.position = "none")
dev.off()


# Opposite to margin, in 20160129_ear045_T3oxBS.cpg.tvb.cnv.pdf, 5mC increases in Tumour as ploidy increases


p <- ggplot(diff.oxBS.cpg.tvb.cnv[chr=="chr11"], aes(factor(cn), pct_met_diff))
p + geom_violin(scale = "count", aes(fill = factor(cn)), draw_quantiles = 0.5) + scale_fill_manual(values=c("green3", "gray30", "red")) + xlab("Ploidy") + ylab("%5mC (Tumour - Margin)") + labs(title = "chr11") + theme_bw() + theme(legend.position = "none")

p <- ggplot(diff.oxBS.cpg.tvb.cnv[chr=="chr11" & cn == 2], aes(factor(loh), pct_met_diff))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5mC (Tumour - Margin)") + labs(title = "chr11 diploid") + theme_bw() + theme(legend.position = "none")

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160129_diff.oxBS.cpg.tvb.cnv.pdf")
p <- ggplot(diff.oxBS.cpg.tvb.cnv, aes(factor(cn), pct_met_diff))
p + geom_violin(scale = "area", aes(fill = factor(cn)), draw_quantiles = 0.5) + scale_fill_manual(values=c("white", "green3", "gray30", "red", "red", "red")) + xlab("Ploidy") + ylab("%5mC (Tumour - Margin)") + theme_bw() + theme(legend.position = "none")
dev.off()

# We can see the difference here but it isn't as clear as in the individual plots.

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160129_diff.oxBS.cpg.tvb.cnv.diploid.pdf")
p <- ggplot(diff.oxBS.cpg.tvb.cnv[cn == 2], aes(factor(loh), pct_met_diff))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5mC (Tumour - Margin)") + labs(title = "diploid") + theme_bw() + theme(legend.position = "none")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160129_diff.oxBS.cpg.tvb.cnv.triploid.pdf")
p <- ggplot(diff.oxBS.cpg.tvb.cnv[cn == 3], aes(factor(loh), pct_met_diff))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5mC (Tumour - Margin)") + labs(title = "triploid") + theme_bw() + theme(legend.position = "none")
dev.off()


l = list(ear045_T3oxBS.cpg.tvb.cnv, ear043_M8oxBS.cpg.tvb.cnv)
setattr(l, 'names', c("Tumour", "Margin"))
ear045_T3oxBS.cpg.tvb.cnv_ear043_M8oxBS.cpg.tvb.cnv <- rbindlist(l, use.names=TRUE, idcol="ID")

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160129_ear045_T3oxBS.cpg.tvb.cnv_ear043_M8oxBS.cpg.tvb.cnv.pdf", width= 14/2.54, height= 10/2.54)
p <- ggplot(ear045_T3oxBS.cpg.tvb.cnv_ear043_M8oxBS.cpg.tvb.cnv, aes(factor(cn), pct_met))
p + geom_violin(scale = "area", aes(fill = factor(ID,levels(factor(ID))[c(2,1)])), draw_quantiles = 0.5) + scale_fill_manual(values=c("red", "turquoise3")) + xlab("Ploidy") + ylab("%5mC") + theme_bw() + theme(legend.title = element_blank())
dev.off()





# 5hmC: BS - oxBS
# Level of 5hmC is the difference between methylation percentages of BS and oxBS.

setkey(ear042_M8BS.cpg.tvb.cnv, chr, start, end, cn, loh, cnscore)
setkey(ear043_M8oxBS.cpg.tvb.cnv, chr, start, end, cn, loh, cnscore)
setkey(ear044_T3BS.cpg.tvb.cnv, chr, start, end, cn, loh, cnscore)
setkey(ear045_T3oxBS.cpg.tvb.cnv, chr, start, end, cn, loh, cnscore)
M8.cpg.tvb.cnv <- merge(ear042_M8BS.cpg.tvb.cnv, ear043_M8oxBS.cpg.tvb.cnv)
# 100*27133091/27253473 = 99.55829% of CpGs in common between M8BS and M8oxBS
T3.cpg.tvb.cnv <- merge(ear044_T3BS.cpg.tvb.cnv, ear045_T3oxBS.cpg.tvb.cnv)
# 100*27141751/27231244 = 99.67136% of CpGs in common between T3BS and T3oxBS
M8.cpg.tvb.cnv[, pct_met_diff := pct_met.x - pct_met.y]
T3.cpg.tvb.cnv[, pct_met_diff := pct_met.x - pct_met.y]

M8.cpg.tvb.cnv[, list(avg= mean(pct_met_diff, na.rm= TRUE)), by=cn]
table(M8.cpg.tvb.cnv$cn)

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160201_M8BSminusoxBS.cpg.tvb.cnv.pdf")
p <- ggplot(M8.cpg.tvb.cnv, aes(factor(cn), pct_met_diff))
p + geom_violin(scale = "area", aes(fill = factor(cn)), draw_quantiles = 0.5) + scale_fill_manual(values=c("white", "green3", "gray30", "red", "red", "red")) + xlab("Ploidy") + ylab("%5hmC Margin") + theme_bw() + theme(legend.position = "none")
dev.off()

table(M8.cpg.tvb.cnv[cn == 2]$loh)

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160201_M8BSminusoxBS.cpg.tvb.cnv.diploid.pdf")
p <- ggplot(M8.cpg.tvb.cnv[cn == 2], aes(factor(loh), pct_met_diff))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5hmC Margin") + labs(title = "diploid") + theme_bw() + theme(legend.position = "none")
dev.off()

table(M8.cpg.tvb.cnv[cn == 3]$loh)

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160201_M8BSminusoxBS.cpg.tvb.cnv.triploid.pdf")
p <- ggplot(M8.cpg.tvb.cnv[cn == 3], aes(factor(loh), pct_met_diff))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5hmC Margin") + labs(title = "triploid") + theme_bw() + theme(legend.position = "none")
dev.off()

# In 20160201_M8BSminusoxBS.cpg.tvb.cnv.pdf, 5hmC increases in Margin only in pentaploid regions


T3.cpg.tvb.cnv[, list(avg= mean(pct_met_diff, na.rm= TRUE)), by=cn]
table(T3.cpg.tvb.cnv$cn)

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160201_T3BSminusoxBS.cpg.tvb.cnv.pdf")
p <- ggplot(T3.cpg.tvb.cnv, aes(factor(cn), pct_met_diff))
p + geom_violin(scale = "area", aes(fill = factor(cn)), draw_quantiles = 0.5) + scale_fill_manual(values=c("white", "green3", "gray30", "red", "red", "red")) + xlab("Ploidy") + ylab("%5hmC Tumour") + theme_bw() + theme(legend.position = "none")
dev.off()

table(T3.cpg.tvb.cnv[cn == 2]$loh)

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160201_T3BSminusoxBS.cpg.tvb.cnv.diploid.pdf")
p <- ggplot(T3.cpg.tvb.cnv[cn == 2], aes(factor(loh), pct_met_diff))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5hmC Tumour") + labs(title = "diploid") + theme_bw() + theme(legend.position = "none")
dev.off()

table(T3.cpg.tvb.cnv[cn == 3]$loh)

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160201_T3BSminusoxBS.cpg.tvb.cnv.triploid.pdf")
p <- ggplot(T3.cpg.tvb.cnv[cn == 3], aes(factor(loh), pct_met_diff))
p + geom_violin(scale = "area", aes(fill = factor(loh)), draw_quantiles = 0.5) + scale_fill_manual(values=c("gray30", "blue")) + xlab("Loss of heterozygosity") + ylab("%5hmC Tumour") + labs(title = "triploid") + theme_bw() + theme(legend.position = "none")
dev.off()

# In 20160201_T3BSminusoxBS.cpg.tvb.cnv.pdf, 5hmC levels are low overall


l = list(T3.cpg.tvb.cnv, M8.cpg.tvb.cnv)
setattr(l, 'names', c("Tumour", "Margin"))
T3.cpg.tvb.cnv_M8.cpg.tvb.cnv <- rbindlist(l, use.names=TRUE, idcol="ID")

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160201_T3BSminusoxBS.cpg.tvb.cnv_M8BSminusoxBS.cpg.tvb.cnv.pdf", width= 14/2.54, height= 10/2.54)
p <- ggplot(T3.cpg.tvb.cnv_M8.cpg.tvb.cnv, aes(factor(cn), pct_met_diff))
p + geom_violin(scale = "area", aes(fill = factor(ID,levels(factor(ID))[c(2,1)])), draw_quantiles = 0.5) + scale_fill_manual(values=c("red", "turquoise3")) + xlab("Ploidy") + ylab("%5hmC") + theme_bw() + theme(legend.title = element_blank())
dev.off()
```


Copy figures locally:

```bash
rsync --progress 20160129_ear045_T3oxBS.cpg.tvb.cnv_ear043_M8oxBS.cpg.tvb.cnv.pdf martin03@nm149s012925:/home/cri.camres.org/martin03/Desktop
rsync --progress 20160201_T3BSminusoxBS.cpg.tvb.cnv_M8BSminusoxBS.cpg.tvb.cnv.pdf martin03@nm149s012925:/home/cri.camres.org/martin03/Desktop
```








Jenn looked at using CANVAS to get CNV for the tumour (E01) vs. margin (C01) comparison.
The new calls seem much better than the previous one.
The data has been copied to /lustre:

/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv/20160211
Balleles_vs_CN_and_purity.xlsx
LP2000729-DNA_C01_LP2000729-DNA_E01_G1_P1.somatic.SV.vcf.gz
LP2000729-DNA_C01_LP2000729-DNA_E01_G1_P1.somatic.SV.vcf.gz.tbi
LP2000729-DNA_C01_LP2000729-DNA_E01_G1_P1.summary.jpg


Repeat the analysis done above but now using the new datasets:

### Data

```bash
cd /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv/20160211
gzip -d LP2000729-DNA_C01_LP2000729-DNA_E01_G1_P1.somatic.SV.vcf.gz
```

Details of the content of these files is available in: 20160505_summary.md


### Intersect CNV bedGraphs and methylation files

```bash
# Testing:
wc -l /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear042_M8BS.cpg.bedGraph # 54834640
bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear042_M8BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160510_CNV.bedGraph -wa -wb | wc -l # 54283962 - this command excludes what is diploid but has not been called by the CNV caller
bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear042_M8BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160510_CNV.bedGraph -loj | wc -l # 54834640 - this includes everything

# Intersect:
bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear042_M8BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160510_CNV.bedGraph -loj | cut -f 1,2,3,4,5,6,10,11 > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160515_ear042_M8BS.cpg.CNV.bedGraph
wc -l /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160515_ear042_M8BS.cpg.CNV.bedGraph # 54834640

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear043_M8oxBS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160510_CNV.bedGraph -loj | cut -f 1,2,3,4,5,6,10,11 > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160515_ear043_M8oxBS.cpg.CNV.bedGraph
wc -l /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160515_ear043_M8oxBS.cpg.CNV.bedGraph # 54765875

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear044_T3BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160510_CNV.bedGraph -loj | cut -f 1,2,3,4,5,6,10,11 > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160515_ear044_T3BS.cpg.CNV.bedGraph
wc -l /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160515_ear044_T3BS.cpg.CNV.bedGraph # 54801746

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear045_T3oxBS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160510_CNV.bedGraph -loj | cut -f 1,2,3,4,5,6,10,11 > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160515_ear045_T3oxBS.cpg.CNV.bedGraph
wc -l /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160515_ear045_T3oxBS.cpg.CNV.bedGraph # 

```

Boxplot the 5mC/5hmC content in different CNV regions in Margin and Tumour


```R
library(data.table)
library(ggplot2)

ear042_M8BS.cpg.CNV <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160515_ear042_M8BS.cpg.CNV.bedGraph")
setnames(ear042_M8BS.cpg.CNV, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "cnv_type", "cnv_number"))

ear043_M8oxBS.cpg.CNV <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160515_ear043_M8oxBS.cpg.CNV.bedGraph")
setnames(ear043_M8oxBS.cpg.CNV, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "cnv_type", "cnv_number"))

ear044_T3BS.cpg.CNV <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160515_ear044_T3BS.cpg.CNV.bedGraph")
setnames(ear044_T3BS.cpg.CNV, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "cnv_type", "cnv_number"))

ear045_T3oxBS.cpg.CNV <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/data/cnv/20160515_ear045_T3oxBS.cpg.CNV.bedGraph")
setnames(ear045_T3oxBS.cpg.CNV, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "cnv_type", "cnv_number"))

setkey(ear042_M8BS.cpg.CNV, chr, start, end, strand, cnv_type, cnv_number)
setkey(ear043_M8oxBS.cpg.CNV, chr, start, end, strand, cnv_type, cnv_number)
setkey(ear044_T3BS.cpg.CNV, chr, start, end, strand, cnv_type, cnv_number)
setkey(ear045_T3oxBS.cpg.CNV, chr, start, end, strand, cnv_type, cnv_number)

M8.cpg <- merge(ear042_M8BS.cpg.CNV, ear043_M8oxBS.cpg.CNV, suffixes = c(".BS", ".oxBS"))
T3.cpg <- merge(ear044_T3BS.cpg.CNV, ear045_T3oxBS.cpg.CNV, suffixes = c(".BS", ".oxBS"))

rm(ear042_M8BS.cpg.CNV, ear043_M8oxBS.cpg.CNV, ear044_T3BS.cpg.CNV, ear045_T3oxBS.cpg.CNV)

setkey(M8.cpg, chr, start, end, strand, cnv_type, cnv_number)
setkey(T3.cpg, chr, start, end, strand, cnv_type, cnv_number)
M8.T3.cpg <- merge(M8.cpg, T3.cpg, suffixes = c(".M8", ".T3"))
rm(M8.cpg, T3.cpg)

M8.T3.cpg[cnv_type == ".", cnv_type:="REF"]
M8.T3.cpg[, pct_5mC.M8 := 100*cnt_met.oxBS.M8/cnt_tot.oxBS.M8]
M8.T3.cpg[, pct_5hmC.M8 := 100*(cnt_met.BS.M8/cnt_tot.BS.M8 - cnt_met.oxBS.M8/cnt_tot.oxBS.M8)]
M8.T3.cpg[, pct_C.M8 := 100 - 100*cnt_met.BS.M8/cnt_tot.BS.M8]
M8.T3.cpg[, c("cnt_met.BS.M8", "cnt_tot.BS.M8", "cnt_met.oxBS.M8", "cnt_tot.oxBS.M8"):= NULL]
M8.T3.cpg[, pct_5mC.T3 := 100*cnt_met.oxBS.T3/cnt_tot.oxBS.T3]
M8.T3.cpg[, pct_5hmC.T3 := 100*(cnt_met.BS.T3/cnt_tot.BS.T3 - cnt_met.oxBS.T3/cnt_tot.oxBS.T3)]
M8.T3.cpg[, pct_C.T3 := 100 - 100*cnt_met.BS.T3/cnt_tot.BS.T3]
M8.T3.cpg[, c("cnt_met.BS.T3", "cnt_tot.BS.T3", "cnt_met.oxBS.T3", "cnt_tot.oxBS.T3"):= NULL]

M8.T3.cpg.m <- reshape2::melt(M8.T3.cpg, id.vars = c("chr", "start", "end", "strand", "cnv_type", "cnv_number"), variable.name = "type_met", value.name = "pct_met")
rm(M8.T3.cpg)

#M8.T3.cpg.m[type_met %like% "M8"] # select M8 only

#levels(factor(M8.T3.cpg.m$cnv_type))

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160515_M8.cpg.CNV.pdf", width= 14/2.54, height= 10/2.54)
ggplot(M8.T3.cpg.m[type_met %like% "M8"], aes(x = factor(type_met), y = pct_met, fill = factor(cnv_type, levels(factor(cnv_type))[c(1,4,3,2)]))) + geom_boxplot(outlier.shape = NA) + theme_bw() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("darkcyan", "white", "darkorange3", "dodgerblue3")) + xlab("") + ylab("% Modification") + scale_x_discrete(labels = c("5mC","5hmC","C")) + coord_cartesian(ylim = c(0, 100))
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160515_T3.cpg.CNV.pdf", width= 14/2.54, height= 10/2.54)
ggplot(M8.T3.cpg.m[type_met %like% "T3"], aes(x = factor(type_met), y = pct_met, fill = factor(cnv_type, levels(factor(cnv_type))[c(1,4,3,2)]))) + geom_boxplot(outlier.shape = NA) + theme_bw() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("darkcyan", "white", "darkorange3", "dodgerblue3")) + xlab("") + ylab("% Modification") + scale_x_discrete(labels = c("5mC","5hmC","C")) + coord_cartesian(ylim = c(0, 100))
dev.off()


```

I put all the above in a script in lustre and execute it as a job

```bash
fig_dir=/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures
scr_dir=/lustre/sblab/martin03/projects/20150921_BrainMethylomeRoadMap/scripts
bsub -J "cnv" -oo $fig_dir/20160515_M8.T3.cpg.CNV.pdf.log -R "rusage[mem=65536]" "
~/sw/R/R-3.2.1/bin/Rscript $scr_dir/20160515_M8.T3.cpg.CNV.boxplot.R
"

```





20160531 Update from Jennifer Becq

Hi Dario,

Sorry for this very late answer about the purity estimation of the WGS data.

First I got confirmation from the developers of the copy number caller that there is good confidence in the overall purity/ploidy model. In the VCF I sent you earlier in this email chain, you can see two other metrics:

##PurityModelFit=0.0078

##InterModelDistance=0.0355

The lower those values are, the better the fit of the data with the model.  On truth data set, samples with PurityModelFit < 0.02 had >90% accurate CNV calls.

Second, I tried to estimate the purity from the somatic SNVs themselves.

Let me explain my procedure:

-          I’ve selected the high quality SNVs that where in obvious CN=2 regions (in chromosomes 2,3,4,5,8,12,13,14,15,16,21)

-          To check if there were any subclones, I used sciClone (http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1003665, https://github.com/genome/sciclone) to cluster the SNVs based on variant read frequency (VRF) and coverage depth (filtering anything below 15x).

-          See sciClone_image_LP2000729-DNA_E01_selectedChr.png , it detected only one cluster, and removed a bit more variants as noise.

-          Purity can be estimate by simply multiplying by 2 the mean VRF of this cluster of high quality CN=2 SNVs (if the sample was pure, the mean would be 50%): 36.6*2 = 73.2%

So yes I am very confident that the DNA shows a normal:tumour ratio of 30:70 , quite far from the 10:90 ratio you see in the methylation data.

I hope this helps, let me know if you have any questions,

Cheers

Jenn


I copied the attachment of the email here:

/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary/figures/20160531_sciClone_image_LP2000729-DNA_E01_selectedChr.png









