
This is an in-depth analysis of the relationship between SNVs and C, 5mC and 5(h)mC.

Statistics about the nucleotide frequency were already calculated at the bottom of vcf.txt.
The nucleotide frequencies are in /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/Assembly , e.g. LP2000729-DNA_A01.1.txt
Say e.g. chr1:1285142

```bash
grep -w '1285142' LP2000729-DNA_A01.1.txt

# 1	1285142	C	0	19	0	0
```

Check in igv:

```bash
igv &
```


### Generate plots suggested by David Bentley.

As discussed with Dario, there are two ways of calculating %5hmC from the bed files of tumour and margin individually:

* First, pct_5hmC = [sum(cnt_met_BS)/sum(cnt_tot_BS)] - [sum(cnt_met_oxBS)/sum(cnt_tot_oxBS)]. Sum here refers to the sum across all CpGs.
* Second, pct_5hmC = mean([cnt_met_BS/cnt_tot_BS] - [cnt_met_oxBS/cnt_tot_oxBS])


First, the plot of what people have thought that methylation looks like up to now.


```R
library(data.table)

ear042_M8BS.cpg<- fread('cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear042_M8BS.cpg.bedGraph')
setnames(ear042_M8BS.cpg, c("chr", "start", "end", "cnt_met", "cnt_tot"))
ear042_M8BS.cpg[, pct_met := cnt_met/cnt_tot]

ear044_T3BS.cpg<- fread('cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear044_T3BS.cpg.bedGraph')
setnames(ear044_T3BS.cpg, c("chr", "start", "end", "cnt_met", "cnt_tot"))
ear044_T3BS.cpg[, pct_met := cnt_met/cnt_tot]

meth<-c(100*sum(ear044_T3BS.cpg$cnt_met)/sum(ear044_T3BS.cpg$cnt_tot), 100*(1-sum(ear044_T3BS.cpg$cnt_met)/sum(ear044_T3BS.cpg$cnt_tot)), 100*sum(ear042_M8BS.cpg$cnt_met)/sum(ear042_M8BS.cpg$cnt_tot), 100*(1-sum(ear042_M8BS.cpg$cnt_met)/sum(ear042_M8BS.cpg$cnt_tot)))
sample<-c("Tumour", "Tumour", "Margin", "Margin")
label<-c("Methylated", "Non-methylated", "Methylated", "Non-methylated")

summary.tab<-data.frame(meth, sample, label)

library(ggplot2)

ggplot(summary.tab, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "BS") + scale_fill_manual(values=c("red", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160203_ear042_M8BS.cpg_ear044_T3BS.cpg.pdf", width = 11/2.54, height = 10/2.54)

meth2<-c(100*mean(ear044_T3BS.cpg$pct_met), 100*(1-mean(ear044_T3BS.cpg$pct_met)), 100*mean(ear042_M8BS.cpg$pct_met), 100*(1-mean(ear042_M8BS.cpg$pct_met)))
summary.tab2<-data.frame(meth2, sample, label)

ggplot(summary.tab2, aes(x = factor(sample,levels(factor(sample))[c(2,1)]), y = meth2, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "BS") + scale_fill_manual(values=c("red", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160203_ear042_M8BS.cpg_ear044_T3BS.cpg.2.pdf", width = 14/2.54, height = 10/2.54)

# 20160513 update
ggplot(summary.tab, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "BS") + scale_fill_manual(values=c("red", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160513_ear042_M8BS.cpg_ear044_T3BS.cpg.pdf", width = 11/2.54, height = 10/2.54)

# 20160531 update
ggplot(summary.tab, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "BS") + scale_fill_manual(values=c("red", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160531_ear042_M8BS.cpg_ear044_T3BS.cpg.pdf", width = 11.5/2.54, height = 10/2.54)

# 20160823 update
library(data.table)
library(ggplot2)

ear042_M8BS.cpg<- fread('cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear042_M8BS.cpg.bedGraph')
setnames(ear042_M8BS.cpg, c("chr", "start", "end", "cnt_met", "cnt_tot"))
ear042_M8BS.cpg[, pct_met := cnt_met/cnt_tot]

ear044_T3BS.cpg<- fread('cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear044_T3BS.cpg.bedGraph')
setnames(ear044_T3BS.cpg, c("chr", "start", "end", "cnt_met", "cnt_tot"))
ear044_T3BS.cpg[, pct_met := cnt_met/cnt_tot]

meth<-c(100*sum(ear044_T3BS.cpg$cnt_met)/sum(ear044_T3BS.cpg$cnt_tot), 100*(1-sum(ear044_T3BS.cpg$cnt_met)/sum(ear044_T3BS.cpg$cnt_tot)), 100*sum(ear042_M8BS.cpg$cnt_met)/sum(ear042_M8BS.cpg$cnt_tot), 100*(1-sum(ear042_M8BS.cpg$cnt_met)/sum(ear042_M8BS.cpg$cnt_tot)))
sample<-c("Tumour", "Tumour", "Margin", "Margin")
label<-c("Modified", "Unmodified", "Modified", "Unmodified")

summary.tab<-data.frame(meth, sample, label)

gg <- ggplot(summary.tab, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(2,1)]))) +
geom_bar(stat = "identity", width = 0.7) +
xlab("") +
ylab("% Modification") +
theme_classic() +
theme(legend.title = element_blank()) +
labs(title = "BS") +
scale_fill_manual(values=c("lightgray", "#D55E00"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160823_ear042_M8BS.cpg_ear044_T3BS.cpg.pdf", width = 11.5/2.54, height = 10/2.54)


# 20160902 update
gg <- ggplot(summary.tab, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(2,1)]))) +
geom_bar(stat = "identity", width = 0.7) +
xlab("") +
ylab("% Modification") +
theme_classic() +
theme(legend.title = element_blank(), axis.text.x = element_text(size=10), axis.title.y = element_text(size=10)) +
scale_fill_manual(values=c("lightgray", "#D55E00"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160902_ear042_M8BS.cpg_ear044_T3BS.cpg.pdf", width = 7.85/2.54, height = 5.25/2.54)


```

Second, the plot summarising what we observe in our data.

```R
library(data.table)
library(ggplot2)

ear042_M8BS.cpg <- fread('cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear042_M8BS.cpg.bedGraph')
setnames(ear042_M8BS.cpg, c("chr", "start", "end", "cnt_met", "cnt_tot"))
ear042_M8BS.cpg[, pct_met := cnt_met/cnt_tot]

ear043_M8oxBS.cpg <- fread('cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear043_M8oxBS.cpg.bedGraph')
setnames(ear043_M8oxBS.cpg, c("chr", "start", "end", "cnt_met", "cnt_tot"))
ear043_M8oxBS.cpg[, pct_met := cnt_met/cnt_tot]

ear044_T3BS.cpg <- fread('cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear044_T3BS.cpg.bedGraph')
setnames(ear044_T3BS.cpg, c("chr", "start", "end", "cnt_met", "cnt_tot"))
ear044_T3BS.cpg[, pct_met := cnt_met/cnt_tot]

ear045_T3oxBS.cpg <- fread('cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/ear045_T3oxBS.cpg.bedGraph')
setnames(ear045_T3oxBS.cpg, c("chr", "start", "end", "cnt_met", "cnt_tot"))
ear045_T3oxBS.cpg[, pct_met := cnt_met/cnt_tot]

setkey(ear042_M8BS.cpg, chr, start, end)
setkey(ear043_M8oxBS.cpg, chr, start, end)
setkey(ear044_T3BS.cpg, chr, start, end)
setkey(ear045_T3oxBS.cpg, chr, start, end)
M8.cpg <- merge(ear042_M8BS.cpg, ear043_M8oxBS.cpg)
T3.cpg <- merge(ear044_T3BS.cpg, ear045_T3oxBS.cpg)

T3.cpg.5hmC <- 100*(sum(T3.cpg$cnt_met.x)/sum(T3.cpg$cnt_tot.x) - sum(T3.cpg$cnt_met.y)/sum(as.numeric(T3.cpg$cnt_tot.y)))
T3.cpg.5mC <- 100*sum(T3.cpg$cnt_met.y)/sum(as.numeric(T3.cpg$cnt_tot.y))
T3.cpg.Candother <- 100*(1-sum(T3.cpg$cnt_met.x)/sum(T3.cpg$cnt_tot.x))

M8.cpg.5hmC <- 100*(sum(M8.cpg$cnt_met.x)/sum(M8.cpg$cnt_tot.x) - sum(M8.cpg$cnt_met.y)/sum(M8.cpg$cnt_tot.y))
M8.cpg.5mC <- 100*sum(M8.cpg$cnt_met.y)/sum(M8.cpg$cnt_tot.y)
M8.cpg.Candother <- 100*(1-sum(M8.cpg$cnt_met.x)/sum(M8.cpg$cnt_tot.x))

meth<-c(T3.cpg.5mC, T3.cpg.5hmC, T3.cpg.Candother, M8.cpg.5mC, M8.cpg.5hmC, M8.cpg.Candother)
sample<-c("Tumour", "Tumour", "Tumour", "Margin", "Margin", "Margin")
label<-c("5mC", "5hmC", "C", "5mC", "5hmC", "C")

summary.tab<-data.frame(meth, sample, label)

ggplot(summary.tab, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "BS and oxBS") + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160203_ear042_M8BS.cpg_ear043_M8oxBS.cpg_ear044_T3BS.cpg_ear045_T3oxBS.cpg.pdf", width = 10/2.54, height = 10/2.54)

T3.cpg.5hmC.2 <- 100*mean(T3.cpg$pct_met.x - T3.cpg$pct_met.y)
T3.cpg.5mC.2 <- 100*mean(T3.cpg$pct_met.y)
T3.cpg.Candother.2 <- 100*(1-mean(T3.cpg$pct_met.x))

M8.cpg.5hmC.2 <- 100*mean(M8.cpg$pct_met.x - M8.cpg$pct_met.y)
M8.cpg.5mC.2 <- 100*mean(M8.cpg$pct_met.y)
M8.cpg.Candother.2 <- 100*(1-mean(M8.cpg$pct_met.x))

meth2<-c(T3.cpg.5mC.2, T3.cpg.5hmC.2, T3.cpg.Candother.2, M8.cpg.5mC.2, M8.cpg.5hmC.2, M8.cpg.Candother.2)
summary.tab2<-data.frame(meth2, sample, label)

ggplot(summary.tab2, aes(x = factor(sample,levels(factor(sample))[c(2,1)]), y = meth2, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "BS and oxBS") + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160203_ear042_M8BS.cpg_ear043_M8oxBS.cpg_ear044_T3BS.cpg_ear045_T3oxBS.cpg.2.pdf", width = 14/2.54, height = 10/2.54)

# 20160513 update
ggplot(summary.tab, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "BS and oxBS") + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160513_ear042_M8BS.cpg_ear043_M8oxBS.cpg_ear044_T3BS.cpg_ear045_T3oxBS.cpg.pdf", width = 10/2.54, height = 10/2.54)

# 20160823 update
ggplot(summary.tab, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(3,1,2)]))) +
geom_bar(stat = "identity", width = 0.7) +
xlab("") +
ylab("% Modification") +
theme_classic() +
theme(legend.title = element_blank()) +
labs(title = "BS and oxBS") +
scale_fill_manual(values=c("lightgray", "#009E73", "#D55E00"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160823_ear042_M8BS.cpg_ear043_M8oxBS.cpg_ear044_T3BS.cpg_ear045_T3oxBS.cpg.pdf", width = 10/2.54, height = 10/2.54)

# 20160902 update
gg <- ggplot(summary.tab, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(3,1,2)]))) +
geom_bar(stat = "identity", width = 0.7) +
xlab("") +
ylab("% Modification") +
theme_classic() +
theme(legend.title = element_blank(), axis.text.x = element_text(size=10), axis.title.y = element_text(size=10)) +
scale_fill_manual(values=c("lightgray", "#009E73", "#D55E00"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160902_ear042_M8BS.cpg_ear043_M8oxBS.cpg_ear044_T3BS.cpg_ear045_T3oxBS.cpg.pdf", width = 7/2.54, height = 5.25/2.54)


```

Copy figures to Desktop in order to create a slide to share with brain team

```bash
cd /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures
rsync --progress 20160203*.pdf martin03@nm149s012925:/home/cri.camres.org/martin03/Desktop
```






### Link SNV, 5mC and 5hmC

First, explore data manually, we need to find SNVs that hold the following:

- SNV involving C or G in CpG islands
- QSS > 150, play with this a bit and look at the percentage of bases different from the bases involved in the mutation
- (%5hmC_T - %5hmC_M) > 0 conditioned to %5hmC_T > 0 and %5hmC_M > 0
- (%5mC_T - %5mC_M) < 0 conditioned to %5mC_T > 0 and %5mC_M > 0 (I think by definition anyway)


```python
import vcf # https://pyvcf.readthedocs.org/en/latest/index.html
vcf_reader = vcf.Reader(open('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/vcf/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_C01.somatic.vcf', 'r'))

snv_tvm=[]

for record in vcf_reader:
	if record.is_snp:
		chr=record.CHROM
		start=record.start
		end=record.end
		base_m=record.REF
		base_t=str(record.ALT).replace("[","").replace("]","")
		qss=record.INFO["QSS"]
		tqss=record.INFO["TQSS"]
		snv_tvm.append((chr, start, end, base_m, base_t, qss, tqss))



len(snv_tvm) # 8169 SNVs in the tumour vs. margin comparison

bed_tvm=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_C01.somatic.bedGraph", "w")

for snv in snv_tvm:
	bed_tvm.write("%s\n" % ("\t".join(["chr"+snv[0], str(snv[1]), str(snv[2]), snv[3], snv[4], str(snv[5]), str(snv[6])])))


bed_tvm.close()
```


Merge methylation_cpg files and nucleotide frequencies files.
nA	nC	nG	nT

```bash
sed -i "1d" LP2000729-DNA_C01.*.txt
cat LP2000729-DNA_C01.*.txt > LP2000729-DNA_C01.txt
rm LP2000729-DNA_C01.*.txt

sed -i "1d" LP2000729-DNA_E01.*.txt
cat LP2000729-DNA_E01.*.txt > LP2000729-DNA_E01.txt
rm LP2000729-DNA_E01.*.txt
```


```R
library(data.table)

nucleotide.freq.cpg.C01 <- fread("cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/Assembly/LP2000729-DNA_C01.txt")
setnames(nucleotide.freq.cpg.C01, c("chr", "start", "ref", "nA", "nC", "nG", "nT"))

nucleotide.freq.cpg.E01 <- fread("cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/Assembly/LP2000729-DNA_E01.txt")
setnames(nucleotide.freq.cpg.E01, c("chr", "start", "ref", "nA", "nC", "nG", "nT"))

ear042_M8BS.cpg <- fread("zcat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear042_M8BS.cpg.bedGraph.gz")
setnames(ear042_M8BS.cpg, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand"))
ear042_M8BS.cpg[, pct_met := cnt_met/cnt_tot]

ear043_M8oxBS.cpg <- fread("zcat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear043_M8oxBS.cpg.bedGraph.gz")
setnames(ear043_M8oxBS.cpg, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand"))
ear043_M8oxBS.cpg[, pct_met := cnt_met/cnt_tot]

ear044_T3BS.cpg <- fread("zcat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear044_T3BS.cpg.bedGraph.gz")
setnames(ear044_T3BS.cpg, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand"))
ear044_T3BS.cpg[, pct_met := cnt_met/cnt_tot]

ear045_T3oxBS.cpg <- fread("zcat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear045_T3oxBS.cpg.bedGraph.gz")
setnames(ear045_T3oxBS.cpg, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand"))
ear045_T3oxBS.cpg[, pct_met := cnt_met/cnt_tot]

nucleotide.freq.cpg.C01[chr == "1" & start == 2926710]
nucleotide.freq.cpg.C01[chr == "1" & start == 2926711]
nucleotide.freq.cpg.E01[chr == "1" & start == 2926710]
nucleotide.freq.cpg.E01[chr == "1" & start == 2926711]

ear042_M8BS.cpg[chr == "chr1" & start == 2926710]
ear042_M8BS.cpg[chr == "chr1" & start == 2926711]
ear043_M8oxBS.cpg[chr == "chr1" & start == 2926710]
ear043_M8oxBS.cpg[chr == "chr1" & start == 2926711]
ear044_T3BS.cpg[chr == "chr1" & start == 2926710]
ear044_T3BS.cpg[chr == "chr1" & start == 2926711]
ear045_T3oxBS.cpg[chr == "chr1" & start == 2926710]
ear045_T3oxBS.cpg[chr == "chr1" & start == 2926711]

setkey(ear042_M8BS.cpg, chr, start, end, strand)
setkey(ear043_M8oxBS.cpg, chr, start, end, strand)
setkey(ear044_T3BS.cpg, chr, start, end, strand)
setkey(ear045_T3oxBS.cpg, chr, start, end, strand)

M8.cpg <- merge(ear042_M8BS.cpg, ear043_M8oxBS.cpg, suffixes = c(".BS", ".oxBS"))
M8.cpg[, pct_5hmC := pct_met.BS - pct_met.oxBS]
M8.cpg[, pct_5mC := pct_met.oxBS]
M8.cpg[, pct_C_other := 1 - pct_met.BS]

T3.cpg <- merge(ear044_T3BS.cpg, ear045_T3oxBS.cpg, suffixes = c(".BS", ".oxBS"))
T3.cpg[, pct_5hmC := pct_met.BS - pct_met.oxBS]
T3.cpg[, pct_5mC := pct_met.oxBS]
T3.cpg[, pct_C_other := 1 - pct_met.BS]

M8.cpg[chr == "chr1" & start == 2926710]
M8.cpg[chr == "chr1" & start == 2926711]
T3.cpg[chr == "chr1" & start == 2926710]
T3.cpg[chr == "chr1" & start == 2926711]

rm(ear042_M8BS.cpg, ear043_M8oxBS.cpg, ear044_T3BS.cpg, ear045_T3oxBS.cpg)

nucleotide.freq.cpg.C01[, "count_alternative" := ifelse(ref == "C", nA+nG+nT, nA+nC+nT)]
nucleotide.freq.cpg.C01[, "count_total" := nA+nC+nG+nT]
nucleotide.freq.cpg.C01[, "pct_alternative" := ifelse(ref == "C", (nA+nG+nT)/(nA+nC+nG+nT), (nA+nC+nT)/(nA+nC+nG+nT)), with=FALSE]
nucleotide.freq.cpg.E01[, "count_alternative" := ifelse(ref == "C", nA+nG+nT, nA+nC+nT)]
nucleotide.freq.cpg.E01[, "count_total" := nA+nC+nG+nT]
nucleotide.freq.cpg.E01[, "pct_alternative" := ifelse(ref == "C", (nA+nG+nT)/(nA+nC+nG+nT), (nA+nC+nT)/(nA+nC+nG+nT)), with=FALSE]

mean(nucleotide.freq.cpg.C01$pct_alternative, na.rm=T) # 0.03047756
median(nucleotide.freq.cpg.C01$pct_alternative, na.rm=T) # 0.01123596
mean(nucleotide.freq.cpg.C01[ref == "C"]$pct_alternative, na.rm=T) # 0.0305178
median(nucleotide.freq.cpg.C01[ref == "C"]$pct_alternative, na.rm=T) # 0.01123596
mean(nucleotide.freq.cpg.C01[ref == "G"]$pct_alternative, na.rm=T) # 0.03043732
median(nucleotide.freq.cpg.C01[ref == "G"]$pct_alternative, na.rm=T) # 0.01111111

mean(nucleotide.freq.cpg.E01$pct_alternative, na.rm=T) # 0.025457
median(nucleotide.freq.cpg.E01$pct_alternative, na.rm=T) # 0.007722008
mean(nucleotide.freq.cpg.E01[ref == "C"]$pct_alternative, na.rm=T) # 0.02549734
median(nucleotide.freq.cpg.E01[ref == "C"]$pct_alternative, na.rm=T) # 0.007751938
mean(nucleotide.freq.cpg.E01[ref == "G"]$pct_alternative, na.rm=T) # 0.02541667
median(nucleotide.freq.cpg.E01[ref == "G"]$pct_alternative, na.rm=T) # 0.007692308



# Plot alternative allele distributions
library(ggplot2)
ggplot(nucleotide.freq.cpg.C01[ref == "C"], aes(x=pct_alternative)) + geom_histogram(binwidth=.05)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/Assembly/figures/20160206_nucleotide.freq.cpg.C01.pdf")

ggplot(nucleotide.freq.cpg.E01[ref == "C"], aes(x=pct_alternative)) + geom_histogram(binwidth=.05)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/Assembly/figures/20160206_nucleotide.freq.cpg.E01.pdf")



# Plot distributions of %5mC, %5hmC and %C_other for margin and tumour

l = list(T3.cpg, M8.cpg)
setattr(l, 'names', c("Tumour", "Margin"))
T3.cpg_M8.cpg <- rbindlist(l, use.names=TRUE, idcol="ID")
nrow(T3.cpg)+nrow(M8.cpg) == nrow(T3.cpg_M8.cpg) # TRUE

ggplot(T3.cpg_M8.cpg, aes(pct_5mC, colour = ID, fill = ID)) + geom_density(alpha = 0.5)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160207_T3.cpg_M8.cpg.5mC.pdf")
# small bumps due to low number of cnt_tot

ggplot(T3.cpg_M8.cpg, aes(pct_5hmC, colour = ID, fill = ID)) + geom_density(alpha = 0.5)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160207_T3.cpg_M8.cpg.5hmC.pdf")

ggplot(T3.cpg_M8.cpg, aes(pct_C_other, colour = ID, fill = ID)) + geom_density(alpha = 0.5)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160207_T3.cpg_M8.cpg.C_other.pdf")




# link frequency of alternative allele in tumour (E01) to %5hmC, %5mC and %C_other in margin (M08)
# ... and also other combinations, see table below

setkey(nucleotide.freq.cpg.C01, chr, start, ref)
setkey(nucleotide.freq.cpg.E01, chr, start, ref)
nucleotide.freq.cpg.C01.E01 <- merge(nucleotide.freq.cpg.C01, nucleotide.freq.cpg.E01, suffixes = c(".C01", ".E01"))
nucleotide.freq.cpg.C01.E01[, c("nA.C01", "nC.C01", "nG.C01", "nT.C01", "nA.E01", "nC.E01", "nG.E01", "nT.E01"):= NULL]
rm(nucleotide.freq.cpg.C01, nucleotide.freq.cpg.E01)

setkey(M8.cpg, chr, start, end, strand)
setkey(T3.cpg, chr, start, end, strand)
methylation.cpg.M8.T3 <- merge(M8.cpg, T3.cpg, suffixes = c(".M8", ".T3"))
rm(M8.cpg, T3.cpg)

nucleotide.freq.cpg.C01.E01[chr == "1" & start == 2926710]
nucleotide.freq.cpg.C01.E01[chr == "1" & start == 2926711]
methylation.cpg.M8.T3[chr == "chr1" & start == 2926710]
methylation.cpg.M8.T3[chr == "chr1" & start == 2926711]

nucleotide.freq.cpg.C01.E01[,chr:=paste("chr", chr, sep="")]
setkey(nucleotide.freq.cpg.C01.E01, chr, start)
setkey(methylation.cpg.M8.T3, chr, start)
nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3 <- merge(nucleotide.freq.cpg.C01.E01, methylation.cpg.M8.T3)

nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[chr == "chr1" & start == 2926710]
nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[chr == "chr1" & start == 2926711]

dim(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3)
# [1] 54481909       29

rm(nucleotide.freq.cpg.C01.E01, methylation.cpg.M8.T3)


# Save nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3 so it can be used later (repeat the loading of data above but keep the counts of nucleotides as columns):
# write.table(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3, "/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/tables/nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3.csv") # it takes long to write

# library(data.table)
# nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3 <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/tables/nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3.csv")
# It takes ages to load this - use previous strategy of loading files individually and manipulate them


ggplot(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3, aes(x=pct_5hmC.M8, y=pct_alternative.E01)) + geom_point()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160208_pct.alternative.E01_pct.5hmC.M8.pdf")
# The scatterplot contains > 27 million dots, takes ages to open the resulting pdf - try something different

ggplot(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[pct_alternative.E01 > 0.5], aes(x=pct_5hmC.M8, y=pct_alternative.E01)) + geom_point()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160208_pct.alternative.E01_pct.5hmC.M8_withpctalternativelarger0.5.pdf")
# too many dots with negative values of %pct_5hmC in M8

ggplot(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3, aes(x=pct_5hmC.M8, y=pct_alternative.E01)) + stat_binhex()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160208_pct.alternative.E01_pct.5hmC.M8_hexagonalbinning.pdf")

ggplot(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[pct_alternative.E01 > 0.5], aes(x=pct_5hmC.M8, y=pct_alternative.E01)) + stat_binhex()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160208_pct.alternative.E01_pct.5hmC.M8_withpctalternativelarger0.5_hexagonalbinning.pdf")

ggplot(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[pct_5hmC.M8 >= 0], aes(x=pct_5hmC.M8, y=pct_alternative.E01)) + stat_binhex()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160208_pct.alternative.E01_pct.5hmC.M8_withpct5hmCM8larger0_hexagonalbinning.pdf")

ggplot(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[pct_alternative.E01 > 0.5 & pct_5hmC.M8 >= 0], aes(x=pct_5hmC.M8, y=pct_alternative.E01)) + stat_binhex()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160208_pct.alternative.E01_pct.5hmC.M8_withpctalternativelarger0.5_withpct5hmCM8larger0_hexagonalbinning.pdf")

ggplot(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3, aes(x=pct_5hmC.M8, y=pct_alternative.E01)) + geom_bin2d()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160208_pct.alternative.E01_pct.5hmC.M8_rectangularbinning.pdf")

ggplot(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[pct_alternative.E01 > 0.5 & pct_5hmC.M8 >= 0], aes(x=pct_5hmC.M8, y=pct_alternative.E01)) + geom_bin2d()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160208_pct.alternative.E01_pct.5hmC.M8_withpctalternativelarger0.5_withpct5hmCM8larger0_rectangularbinning.pdf")


# summary of smoothScatter plots that follow:
#
#      5hmC.M8   5mC.M8   5hmC.T3   5mC.T3
# E01     i         i        j        j
# C01     j         j        i        i
#
# legend:
# i = of interest, modification in margin leading to high/low alternative allele frequency in tumour or vice versa, high/low alternative allele frequency in margin leading to modification in margin
# j = just in case

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160212_pct.alternative.E01_pct.5hmC.M8_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5hmC.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.E01, nbin = 256, nrpoints = 0, xlab = "%5hmC Margin", ylab = "%alternative allele Tumour")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.alternative.C01_pct.5hmC.M8_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5hmC.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.C01, nbin = 256, nrpoints = 0, xlab = "%5hmC Margin", ylab = "%alternative allele Margin")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.alternative.E01_pct.5mC.M8_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5mC.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.E01, nbin = 256, nrpoints = 0, xlab = "%5mC Margin", ylab = "%alternative allele Tumour")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.alternative.C01_pct.5mC.M8_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5mC.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.C01, nbin = 256, nrpoints = 0, xlab = "%5mC Margin", ylab = "%alternative allele Margin")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.alternative.C01_pct.5hmC.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5hmC.T3, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.C01, nbin = 256, nrpoints = 0, xlab = "%5hmC Tumour", ylab = "%alternative allele Margin")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.alternative.E01_pct.5hmC.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5hmC.T3, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.E01, nbin = 256, nrpoints = 0, xlab = "%5hmC Tumour", ylab = "%alternative allele Tumour")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.alternative.C01_pct.5mC.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5mC.T3, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.C01, nbin = 256, nrpoints = 0, xlab = "%5mC Tumour", ylab = "%alternative allele Margin")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.alternative.E01_pct.5mC.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5mC.T3, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.E01, nbin = 256, nrpoints = 0, xlab = "%5mC Tumour", ylab = "%alternative allele Tumour")
dev.off()


# What is the relationship between the % alternative allele Tumour and % alternative allele Margin

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.alternative.C01_pct.alternative.E01_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.C01, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.E01, nbin = 256, nrpoints = 0, xlab = "% alternative allele Margin", ylab = "% alternative allele Tumour")
dev.off()

# When the % alternative allele is high/low in Margin, it is also correspondingly high/low in Tumour


# What is the relationship between %5mC, %5hmC, %C_other  in Tumour and Margin
#            5hmC.M8   5mC.M8   C_other.M8
# 5hmC.T3
# 5mC.T3
# C_other.T3

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.5hmC.M8_pct.5hmC.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5hmC.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5hmC.T3, nbin = 256, nrpoints = 0, xlab = "%5hmC Margin", ylab = "%5hmC Tumour")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.5hmC.M8_pct.5mC.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5hmC.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5mC.T3, nbin = 256, nrpoints = 0, xlab = "%5hmC Margin", ylab = "%5mC Tumour")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.5hmC.M8_pct.C_other.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5hmC.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_C_other.T3, nbin = 256, nrpoints = 0, xlab = "%5hmC Margin", ylab = "%C/other Tumour")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.5mC.M8_pct.5hmC.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5mC.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5hmC.T3, nbin = 256, nrpoints = 0, xlab = "%5mC Margin", ylab = "%5hmC Tumour")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.5mC.M8_pct.5mC.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5mC.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5mC.T3, nbin = 256, nrpoints = 0, xlab = "%5mC Margin", ylab = "%5mC Tumour")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.5mC.M8_pct.C_other.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5mC.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_C_other.T3, nbin = 256, nrpoints = 0, xlab = "%5mC Margin", ylab = "%C/other Tumour")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.C_other.M8_pct.5hmC.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_C_other.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5hmC.T3, nbin = 256, nrpoints = 0, xlab = "%C/other Margin", ylab = "%5hmC Tumour")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.C_other.M8_pct.5mC.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_C_other.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5mC.T3, nbin = 256, nrpoints = 0, xlab = "%C/other Margin", ylab = "%5mC Tumour")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_pct.C_other.M8_pct.C_other.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_C_other.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_C_other.T3, nbin = 256, nrpoints = 0, xlab = "%C/other Margin", ylab = "%C/other Tumour")
dev.off()


# Now work with the differences/deltas:
#                               Delta%5hmC(M8-T3)   Delta%5mC(T3-M8)
# Delta_altallfreq (E01 - C01)

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_Delta.pct.alternative.E01.C01_Delta.pct.5hmC.M8.T3_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5hmC.M8 - nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5hmC.T3, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.E01 - nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.C01, nbin = 256, nrpoints = 0, xlab = "Delta %5hmC (Margin - Tumour)", ylab = "Delta % alternative allele (Tumour - Margin)")
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_Delta.pct.alternative.E01.C01_Delta.pct.5mC.T3.M8_smoothScatter.pdf")
smoothScatter(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5mC.T3 - nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_5mC.M8, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.E01 - nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3$pct_alternative.C01, nbin = 256, nrpoints = 0, xlab = "Delta %5mC (Tumour - Margin)", ylab = "Delta % alternative allele (Tumour - Margin)")
dev.off()

# The changes in 5mC and 5hmC between tumour and margin take place at low changes of alternative allele frequencies.

```



I now need to look at finding relationships between 5mC, 5hmC in the SNVs reported by Strelka.
It has now become important to me that although cnt_met, cnt_tot is calculated considering the CpG site as a whole, the counts of nucleotides in the genotyping was done considering both C and G separately within the CpG site. Methylation can take place both in the C (positive strand) or in the C complementary to G (negative strand). And there might be important differences between the two.

   m
5' CG 3'
3' GC 5'
    m

C (margin) -> X (tumour), methylation comes from the positive strand
G (margin) -> X (tumour), methylation comes from the negative strand
Both have to be considered

What is the difference in 5mC and 5hmC between strands?


```R
nrow(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[strand=="+"]) # 27240321
nrow(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[ref=="C"]) # 27240321
# There are 27240321 Cs in the positive strand of CpG sites
nrow(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[strand=="-"]) # 27241588
nrow(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[ref=="G"]) # 27241588
# There are 27241588 Cs in the negative strand of CpG sites

data.positive<-nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[strand=="+"]
positive.5hmC.M8<-100*(sum(data.positive$cnt_met.BS.M8)/sum(data.positive$cnt_tot.BS.M8) - sum(data.positive$cnt_met.oxBS.M8)/sum(data.positive$cnt_tot.oxBS.M8))
positive.5mC.M8<-100*sum(data.positive$cnt_met.oxBS.M8)/sum(data.positive$cnt_tot.oxBS.M8)
positive.Candother.M8<-100*(1 - sum(data.positive$cnt_met.BS.M8)/sum(data.positive$cnt_tot.BS.M8))
positive.5hmC.M8+positive.5mC.M8+positive.Candother.M8 # 100, good
positive.5hmC.T3<-100*(sum(data.positive$cnt_met.BS.T3)/sum(data.positive$cnt_tot.BS.T3) - sum(data.positive$cnt_met.oxBS.T3)/sum(data.positive$cnt_tot.oxBS.T3))
positive.5mC.T3<-100*sum(data.positive$cnt_met.oxBS.T3)/sum(data.positive$cnt_tot.oxBS.T3)
positive.Candother.T3<-100*(1 - sum(data.positive$cnt_met.BS.T3)/sum(data.positive$cnt_tot.BS.T3))
positive.5hmC.T3+positive.5mC.T3+positive.Candother.T3 # 100, good

data.negative<-nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[strand=="-"]
negative.5hmC.M8<-100*(sum(data.negative$cnt_met.BS.M8)/sum(data.negative$cnt_tot.BS.M8) - sum(data.negative$cnt_met.oxBS.M8)/sum(data.negative$cnt_tot.oxBS.M8))
negative.5mC.M8<-100*sum(data.negative$cnt_met.oxBS.M8)/sum(data.negative$cnt_tot.oxBS.M8)
negative.Candother.M8<-100*(1 - sum(data.negative$cnt_met.BS.M8)/sum(data.negative$cnt_tot.BS.M8))
negative.5hmC.M8+negative.5mC.M8+negative.Candother.M8 # 100, good
negative.5hmC.T3<-100*(sum(data.negative$cnt_met.BS.T3)/sum(data.negative$cnt_tot.BS.T3) - sum(data.negative$cnt_met.oxBS.T3)/sum(data.negative$cnt_tot.oxBS.T3))
negative.5mC.T3<-100*sum(data.negative$cnt_met.oxBS.T3)/sum(data.negative$cnt_tot.oxBS.T3)
negative.Candother.T3<-100*(1 - sum(data.negative$cnt_met.BS.T3)/sum(data.negative$cnt_tot.BS.T3))
negative.5hmC.T3+negative.5mC.T3+negative.Candother.T3 # 100, good

positive.5hmC.M8 # 17.47474
positive.5mC.M8 # 52.08947
positive.Candother.M8 # 30.43579
negative.5hmC.M8 # 17.51473
negative.5mC.M8 # 52.15355
negative.Candother.M8 # 30.33171
# There does not seem to be much of a difference between + and -

meth.M8<-c(positive.5mC.M8, positive.5hmC.M8, positive.Candother.M8, negative.5mC.M8, negative.5hmC.M8, negative.Candother.M8)
meth.T3<-c(positive.5mC.T3, positive.5hmC.T3, positive.Candother.T3, negative.5mC.T3, negative.5hmC.T3, negative.Candother.T3)
strand<-c("+", "+", "+", "-", "-", "-")
label<-c("5mC", "5hmC", "C/other", "5mC", "5hmC", "C/other")

summary.M8<-data.frame(meth.M8, strand, label)
summary.T3<-data.frame(meth.T3, strand, label)

library(ggplot2)

ggplot(summary.M8, aes(x = factor(strand, levels(factor(strand))[c(2,1)]), y = meth.M8, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("Strand") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "BS and oxBS: Margin") + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_methylation.strand.M8.pdf", width = 10/2.54, height = 10/2.54)

ggplot(summary.T3, aes(x = factor(strand, levels(factor(strand))[c(2,1)]), y = meth.T3, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("Strand") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "BS and oxBS: Tumour") + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/figures/20160215_methylation.strand.T3.pdf", width = 10/2.54, height = 10/2.54)

rm(data.positive, data.negative)

```





Combine variation data obtained from Strelka with 5mC and 5hmC scores

```R

# Load CancerLP2000729-DNA_E01_NormalLP2000729-DNA_C01.somatic and plot distribution of QSS scores

CancerLP2000729_DNA_E01_NormalLP2000729_DNA_C01.somatic <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/CancerLP2000729-DNA_E01_NormalLP2000729-DNA_C01.somatic.bedGraph")
setnames(CancerLP2000729_DNA_E01_NormalLP2000729_DNA_C01.somatic, c("chr", "start", "end", "ref", "alt", "qss", "tqss"))

ggplot(CancerLP2000729_DNA_E01_NormalLP2000729_DNA_C01.somatic, aes(x=qss)) + geom_histogram(binwidth=1)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160209_CancerLP2000729-DNA_E01_NormalLP2000729-DNA_C01.somatic_qssdistribution.pdf")

CancerLP2000729_DNA_E01_NormalLP2000729_DNA_C01.somatic[qss>300]
#     chr    start      end ref alt qss tqss
#1: chr17  7986112  7986113   G   A 325    1
#2: chr17 14839834 14839835   A   C 343    1
#3: chr17 26729058 26729059   G   A 308    1
#4: chr17 32564105 32564106   C   T 310    1
#5: chr17 63848099 63848100   T   C 312    1
#6: chr17 64874300 64874301   C   T 408    1
#7: chr17 70809500 70809501   T   C 315    1
#8: chrMT     3945     3946   G   A 406    1
#9: chrMT     8366     8367   T   C 336    1

# Most top variations happen in chr17

nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[chr=="chr17" & start==7986112] # G in CpG
nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[chr=="chr17" & start==7986111] # C in same CpG
nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[chr=="chr17" & start==32564105] # C in CpG
nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[chr=="chr17" & start==32564106] # G in same CpG

# Examples above worked out in notebook and below

# Merge nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3 and CancerLP2000729_DNA_E01_NormalLP2000729_DNA_C01.somatic

setkey(CancerLP2000729_DNA_E01_NormalLP2000729_DNA_C01.somatic, chr, start, end, ref)
setkey(nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3, chr, start, end, ref)
snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3 <- merge(CancerLP2000729_DNA_E01_NormalLP2000729_DNA_C01.somatic, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3, all=TRUE)
dim(snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3)
#[1] 54487468       32

rm(CancerLP2000729_DNA_E01_NormalLP2000729_DNA_C01.somatic, nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3)

snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[chr=="chr17" & start==32564105]
snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[chr=="chr17" & start==32564106]
snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[chr=="chrMT"]

nrow(snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[alt!="NA"]) # 8169, good, all T v M SNVs
nrow(snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[alt!="NA" & pct_alternative.C01!="NA"]) # 2610
# 2610/8169, almost 1/3 of SNVs fall within CpG sites
nrow(snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[pct_alternative.C01!="NA"]) # 54481909
nrow(snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3) # 54487468
# 54487468 - 54481909 = 5559, more than 2/3 of SNVs do not fall within CpG sites





# Distributions of qss and %alternative allele in C01 and E01

data<-snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[alt!="NA" & pct_alternative.C01!="NA"]

table(data$chr)
table(data$ref)
#   C    G
#1391 1219

table(data$strand)
#   -    +
#1219 1391

table(data$alt)

ggplot(data, aes(x=qss)) + geom_histogram(binwidth=1)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160215_CancerLP2000729-DNA_E01_NormalLP2000729-DNA_C01.somatic_qssdistribution_cpg.pdf")

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160215_pct.alternative.C01_pct.alternative.E01_cpg_smoothScatter.pdf")
smoothScatter(data$pct_alternative.C01, data$pct_alternative.E01, nbin = 256, nrpoints = 0, xlab = "% alternative allele Margin", ylab = "% alternative allele Tumour")
dev.off()
# just top region of ../../methylation_cpg/figures/20160215_pct.alternative.C01_pct.alternative.E01_smoothScatter.pdf





# Barplots showing the change of 5mC, 5hmC and C/other from M to T in the 2610 SNVs falling within CpG sites
# First, without the alternative allele frequency

snv.5hmC.M8<-100*(sum(data$cnt_met.BS.M8)/sum(data$cnt_tot.BS.M8) - sum(data$cnt_met.oxBS.M8)/sum(data$cnt_tot.oxBS.M8))
snv.5mC.M8<-100*sum(data$cnt_met.oxBS.M8)/sum(data$cnt_tot.oxBS.M8)
snv.Candother.M8<-100*(1 - sum(data$cnt_met.BS.M8)/sum(data$cnt_tot.BS.M8))
snv.5hmC.T3<-100*(sum(data$cnt_met.BS.T3)/sum(data$cnt_tot.BS.T3) - sum(data$cnt_met.oxBS.T3)/sum(data$cnt_tot.oxBS.T3))
snv.5mC.T3<-100*sum(data$cnt_met.oxBS.T3)/sum(data$cnt_tot.oxBS.T3)
snv.Candother.T3<-100*(1 - sum(data$cnt_met.BS.T3)/sum(data$cnt_tot.BS.T3))

snv.5hmC.M8 # 14.07141
snv.5mC.M8 # 64.67896
snv.Candother.M8 # 21.24963
snv.5hmC.T3 # 0.4973751
snv.5mC.T3 # 44.33505
snv.Candother.T3 # 55.16757

meth<-c(snv.5mC.T3, snv.5hmC.T3, snv.Candother.T3, snv.5mC.M8, snv.5hmC.M8, snv.Candother.M8)
sample<-c("Tumour", "Tumour", "Tumour", "Margin", "Margin", "Margin")
label<-c("5mC", "5hmC", "C/other", "5mC", "5hmC", "C/other")

summary.noallfreq<-data.frame(meth, sample, label)

ggplot(summary.noallfreq, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "SNVs in CpG sites") + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160215_methylation.snv.cpg.noallfreq.pdf", width = 10/2.54, height = 10/2.54)


# Second, factoring in the alternative allele frequency

snv.alt.M8<-100*sum(data$count_alternative.C01)/sum(data$count_total.C01)
snv.5hmC.M8<-100*(1 - sum(data$count_alternative.C01)/sum(data$count_total.C01))*(sum(data$cnt_met.BS.M8)/sum(data$cnt_tot.BS.M8) - sum(data$cnt_met.oxBS.M8)/sum(data$cnt_tot.oxBS.M8))
snv.5mC.M8<-100*(1 - sum(data$count_alternative.C01)/sum(data$count_total.C01))*sum(data$cnt_met.oxBS.M8)/sum(data$cnt_tot.oxBS.M8)
snv.Candother.M8<-100*(1 - sum(data$count_alternative.C01)/sum(data$count_total.C01))*(1 - sum(data$cnt_met.BS.M8)/sum(data$cnt_tot.BS.M8))
snv.alt.T3<-100*sum(data$count_alternative.E01)/sum(data$count_total.E01)
snv.5hmC.T3<-100*(1 - sum(data$count_alternative.E01)/sum(data$count_total.E01))*(sum(data$cnt_met.BS.T3)/sum(data$cnt_tot.BS.T3) - sum(data$cnt_met.oxBS.T3)/sum(data$cnt_tot.oxBS.T3))
snv.5mC.T3<-100*(1 - sum(data$count_alternative.E01)/sum(data$count_total.E01))*sum(data$cnt_met.oxBS.T3)/sum(data$cnt_tot.oxBS.T3)
snv.Candother.T3<-100*(1 - sum(data$count_alternative.E01)/sum(data$count_total.E01))*(1 - sum(data$cnt_met.BS.T3)/sum(data$cnt_tot.BS.T3))

snv.alt.M8 # 2.707905
snv.5hmC.M8 # 13.69037
snv.5mC.M8 # 62.92752
snv.Candother.M8 # 20.67421
snv.alt.T3 # 33.7973
snv.5hmC.T3 # 0.3292757
snv.5mC.T3 # 29.351
snv.Candother.T3 # 36.52242

meth<-c(snv.5mC.T3, snv.5hmC.T3, snv.Candother.T3, snv.alt.T3, snv.5mC.M8, snv.5hmC.M8, snv.Candother.M8, snv.alt.M8)
sample<-c("Tumour", "Tumour", "Tumour", "Tumour", "Margin", "Margin", "Margin", "Margin")
label<-c("5mC", "5hmC", "C/other", "T", "5mC", "5hmC", "C/other", "T")

summary.withallfreq<-data.frame(meth, sample, label)

ggplot(summary.withallfreq, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(4,3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Methylation/Alternative allele") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "SNVs in CpG sites") + scale_fill_manual(values=c("black", "red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160215_methylation.snv.cpg.withallfreq.pdf", width = 10/2.54, height = 10/2.54)




# First example of a SNV changing the C within a CpG (chr17:32564105) showing the change of 5mC, 5hmC, C/other and % alternative allele from M to T. This is quite an extreme example with a very high QSS score (qss=310). It does not fall though within any gene body region

#    *
# 5' CG 3'
# 3' GC 5'
#
# * = where mutation takes place

# First plot, without the alternative allele frequency

ex<-snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[chr=="chr17" & start==32564105]

ex.snv.5hmC.M8<-100*ex$pct_5hmC.M8
ex.snv.5mC.M8<-100*ex$pct_5mC.M8
ex.snv.Candother.M8<-100*ex$pct_C_other.M8
ex.snv.5hmC.T3<-100*ex$pct_5hmC.T3
ex.snv.5mC.T3<-100*ex$pct_5mC.T3
ex.snv.Candother.T3<-100*ex$pct_C_other.T3

meth<-c(ex.snv.5mC.T3, ex.snv.5hmC.T3, ex.snv.Candother.T3, ex.snv.5mC.M8, ex.snv.5hmC.M8, ex.snv.Candother.M8)
sample<-c("Tumour", "Tumour", "Tumour", "Margin", "Margin", "Margin")
label<-c("5mC", "5hmC", "C/other", "5mC", "5hmC", "C/other")

summary.noallfreq<-data.frame(meth, sample, label)

ggplot(summary.noallfreq, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "chr17:32564105") + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160216_methylation.snv.cpg.noallfreq.example.chr17.32564105.pdf", width = 10/2.54, height = 10/2.54)

# Second plot, factoring in the alternative allele frequency

ex.snv.alt.M8<-100*ex$pct_alternative.C01
ex.snv.5hmC.M8<-100*(1 - ex$pct_alternative.C01)*ex$pct_5hmC.M8
ex.snv.5mC.M8<-100*(1 - ex$pct_alternative.C01)*ex$pct_5mC.M8
ex.snv.Candother.M8<-100*(1 - ex$pct_alternative.C01)*ex$pct_C_other.M8
ex.snv.alt.T3<-100*ex$pct_alternative.E01
ex.snv.5hmC.T3<-100*(1 - ex$pct_alternative.E01)*ex$pct_5hmC.T3
ex.snv.5mC.T3<-100*(1 - ex$pct_alternative.E01)*ex$pct_5mC.T3
ex.snv.Candother.T3<-100*(1 - ex$pct_alternative.E01)*ex$pct_C_other.T3

meth<-c(ex.snv.5mC.T3, ex.snv.5hmC.T3, ex.snv.Candother.T3, ex.snv.alt.T3, ex.snv.5mC.M8, ex.snv.5hmC.M8, ex.snv.Candother.M8, ex.snv.alt.M8)
sample<-c("Tumour", "Tumour", "Tumour", "Tumour", "Margin", "Margin", "Margin", "Margin")
label<-c("5mC", "5hmC", "C/other", "T", "5mC", "5hmC", "C/other", "T")

summary.withallfreq<-data.frame(meth, sample, label)

ggplot(summary.withallfreq, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(4,3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Methylation/Alternative allele") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "chr17:32564105") + scale_fill_manual(values=c("black", "red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160216_methylation.snv.cpg.withallfreq.example.chr17.32564105.pdf", width = 10/2.54, height = 10/2.54)





# Second example of a SNV changing the G within a CpG (chr17:7986112) showing the change of 5mC, 5hmC, C/other and % alternative allele from M to T. This is quite an extreme example with a very high QSS score (qss=325). It does fall within an intron region of the gene ALOX12B (arachidonate 12-lipoxygenase). Methylation changes obviously happen in the corresponding C in the negative strand.

#     *
# 5' CG 3'
# 3' GC 5'
#
# * = where mutation takes place


# First plot, without the alternative allele frequency

ex2<-snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[chr=="chr17" & start==7986112]

ex2.snv.5hmC.M8<-100*ex2$pct_5hmC.M8
ex2.snv.5mC.M8<-100*ex2$pct_5mC.M8
ex2.snv.Candother.M8<-100*ex2$pct_C_other.M8
ex2.snv.5hmC.T3<-100*ex2$pct_5hmC.T3
ex2.snv.5mC.T3<-100*ex2$pct_5mC.T3
ex2.snv.Candother.T3<-100*ex2$pct_C_other.T3

meth<-c(ex2.snv.5mC.T3, ex2.snv.5hmC.T3, ex2.snv.Candother.T3, ex2.snv.5mC.M8, ex2.snv.5hmC.M8, ex2.snv.Candother.M8)
sample<-c("Tumour", "Tumour", "Tumour", "Margin", "Margin", "Margin")
label<-c("5mC", "5hmC", "C/other", "5mC", "5hmC", "C/other")

summary.noallfreq<-data.frame(meth, sample, label)

ggplot(summary.noallfreq, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "chr17:7986112") + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160216_methylation.snv.cpg.noallfreq.example.chr17.7986112.pdf", width = 10/2.54, height = 10/2.54)


# Second plot, factoring in the alternative allele frequency

ex2.snv.alt.M8<-100*ex2$pct_alternative.C01
ex2.snv.5hmC.M8<-100*(1 - ex2$pct_alternative.C01)*ex2$pct_5hmC.M8
ex2.snv.5mC.M8<-100*(1 - ex2$pct_alternative.C01)*ex2$pct_5mC.M8
ex2.snv.Candother.M8<-100*(1 - ex2$pct_alternative.C01)*ex2$pct_C_other.M8
ex2.snv.alt.T3<-100*ex2$pct_alternative.E01
ex2.snv.5hmC.T3<-100*(1 - ex2$pct_alternative.E01)*ex2$pct_5hmC.T3
ex2.snv.5mC.T3<-100*(1 - ex2$pct_alternative.E01)*ex2$pct_5mC.T3
ex2.snv.Candother.T3<-100*(1 - ex2$pct_alternative.E01)*ex2$pct_C_other.T3

meth<-c(ex2.snv.5mC.T3, ex2.snv.5hmC.T3, ex2.snv.Candother.T3, ex2.snv.alt.T3, ex2.snv.5mC.M8, ex2.snv.5hmC.M8, ex2.snv.Candother.M8, ex2.snv.alt.M8)
sample<-c("Tumour", "Tumour", "Tumour", "Tumour", "Margin", "Margin", "Margin", "Margin")
label<-c("5mC", "5hmC", "C/other", "T", "5mC", "5hmC", "C/other", "T")

summary.withallfreq<-data.frame(meth, sample, label)

ggplot(summary.withallfreq, aes(x = factor(sample), y = meth, fill=factor(label, levels(factor(label))[c(4,3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Methylation/Alternative allele") + theme_classic() + theme(legend.title = element_blank()) + labs(title = "chr17:7986112") + scale_fill_manual(values=c("black", "red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160216_methylation.snv.cpg.withallfreq.example.chr17.7986112.pdf", width = 10/2.54, height = 10/2.54)






# Compare 5mC, 5hmC and C_other levels in M and T in SNVs and non-SNVs and bring also the % alternative allele in separate plots

snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3 # 54487468
nonsnv<-snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[is.na(alt),] # 54479299 = 54487468 - 8169
snv<-snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[alt!="NA" & pct_alternative.C01!="NA"] # 2610


# First plots, without the alternative allele frequency,
# Margin

snv.5hmC.M8<-100*(sum(snv$cnt_met.BS.M8)/sum(snv$cnt_tot.BS.M8) - sum(snv$cnt_met.oxBS.M8)/sum(snv$cnt_tot.oxBS.M8))
snv.5mC.M8<-100*sum(snv$cnt_met.oxBS.M8)/sum(snv$cnt_tot.oxBS.M8)
snv.Candother.M8<-100*(1 - sum(snv$cnt_met.BS.M8)/sum(snv$cnt_tot.BS.M8))
nonsnv.5hmC.M8<-100*(sum(nonsnv$cnt_met.BS.M8)/sum(nonsnv$cnt_tot.BS.M8) - sum(nonsnv$cnt_met.oxBS.M8)/sum(nonsnv$cnt_tot.oxBS.M8))
nonsnv.5mC.M8<-100*sum(nonsnv$cnt_met.oxBS.M8)/sum(nonsnv$cnt_tot.oxBS.M8)
nonsnv.Candother.M8<-100*(1 - sum(nonsnv$cnt_met.BS.M8)/sum(nonsnv$cnt_tot.BS.M8))

snv.5hmC.M8 # 14.07141
snv.5mC.M8 # 64.67896
snv.Candother.M8 # 21.24963
nonsnv.5hmC.M8 # 17.49488
nonsnv.5mC.M8 # 52.12085
nonsnv.Candother.M8 # 30.38426

meth<-c(nonsnv.5mC.M8, nonsnv.5hmC.M8, nonsnv.Candother.M8, snv.5mC.M8, snv.5hmC.M8, snv.Candother.M8)
sample<-c("non-SNV", "non-SNV", "non-SNV", "SNV", "SNV", "SNV")
label<-c("5mC", "5hmC", "C/other", "5mC", "5hmC", "C/other")

summary.noallfreq<-data.frame(meth, sample, label)

ggplot(summary.noallfreq, aes(x = factor(sample, levels(factor(sample))[c(2,1)]), y = meth, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Methylation in Margin") + theme_classic() + theme(legend.title = element_blank(), plot.title = element_text(size = rel(.75))) + labs(title = "CpG sites\nSNV (n=2610) and non-SNV (n=54479299)") + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160216_methylation.snv.nonsnv.cpg.M8.noallfreq.pdf", width = 10/2.54, height = 10/2.54)


# Tumour

snv.5hmC.T3<-100*(sum(snv$cnt_met.BS.T3)/sum(snv$cnt_tot.BS.T3) - sum(snv$cnt_met.oxBS.T3)/sum(snv$cnt_tot.oxBS.T3))
snv.5mC.T3<-100*sum(snv$cnt_met.oxBS.T3)/sum(snv$cnt_tot.oxBS.T3)
snv.Candother.T3<-100*(1 - sum(snv$cnt_met.BS.T3)/sum(snv$cnt_tot.BS.T3))
nonsnv.5hmC.T3<-100*(sum(nonsnv$cnt_met.BS.T3)/sum(nonsnv$cnt_tot.BS.T3) - sum(nonsnv$cnt_met.oxBS.T3)/sum(as.numeric(nonsnv$cnt_tot.oxBS.T3)))
nonsnv.5mC.T3<-100*sum(nonsnv$cnt_met.oxBS.T3)/sum(as.numeric(nonsnv$cnt_tot.oxBS.T3))
nonsnv.Candother.T3<-100*(1 - sum(nonsnv$cnt_met.BS.T3)/sum(nonsnv$cnt_tot.BS.T3))

snv.5hmC.T3 # 0.4973751
snv.5mC.T3 # 44.33505
snv.Candother.T3 # 55.16757
nonsnv.5hmC.T3 # 1.635617
nonsnv.5mC.T3 # 63.49754
nonsnv.Candother.T3 # 34.86684

meth<-c(nonsnv.5mC.T3, nonsnv.5hmC.T3, nonsnv.Candother.T3, snv.5mC.T3, snv.5hmC.T3, snv.Candother.T3)
sample<-c("non-SNV", "non-SNV", "non-SNV", "SNV", "SNV", "SNV")
label<-c("5mC", "5hmC", "C/other", "5mC", "5hmC", "C/other")

summary.noallfreq<-data.frame(meth, sample, label)

ggplot(summary.noallfreq, aes(x = factor(sample, levels(factor(sample))[c(2,1)]), y = meth, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Methylation in Tumour") + theme_classic() + theme(legend.title = element_blank(), plot.title = element_text(size = rel(.75))) + labs(title = "CpG sites\nSNV (n=2610) and non-SNV (n=54479299)") + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160216_methylation.snv.nonsnv.cpg.T3.noallfreq.pdf", width = 10/2.54, height = 10/2.54)


# Second plots, with the alternative allele frequency,
# Margin

snv.alt.M8<-100*sum(snv$count_alternative.C01)/sum(snv$count_total.C01)
snv.5hmC.M8<-100*(1 - sum(snv$count_alternative.C01)/sum(snv$count_total.C01))*(sum(snv$cnt_met.BS.M8)/sum(snv$cnt_tot.BS.M8) - sum(snv$cnt_met.oxBS.M8)/sum(snv$cnt_tot.oxBS.M8))
snv.5mC.M8<-100*(1 - sum(snv$count_alternative.C01)/sum(snv$count_total.C01))*sum(snv$cnt_met.oxBS.M8)/sum(snv$cnt_tot.oxBS.M8)
snv.Candother.M8<-100*(1 - sum(snv$count_alternative.C01)/sum(snv$count_total.C01))*(1 - sum(snv$cnt_met.BS.M8)/sum(snv$cnt_tot.BS.M8))
nonsnv.alt.M8<-100*sum(nonsnv$count_alternative.C01)/sum(as.numeric(nonsnv$count_total.C01))
nonsnv.5hmC.M8<-100*(1 - sum(nonsnv$count_alternative.C01)/sum(as.numeric(nonsnv$count_total.C01)))*(sum(nonsnv$cnt_met.BS.M8)/sum(nonsnv$cnt_tot.BS.M8) - sum(nonsnv$cnt_met.oxBS.M8)/sum(nonsnv$cnt_tot.oxBS.M8))
nonsnv.5mC.M8<-100*(1 - sum(nonsnv$count_alternative.C01)/sum(as.numeric(nonsnv$count_total.C01)))*sum(nonsnv$cnt_met.oxBS.M8)/sum(nonsnv$cnt_tot.oxBS.M8)
nonsnv.Candother.M8<-100*(1 - sum(nonsnv$count_alternative.C01)/sum(as.numeric(nonsnv$count_total.C01)))*(1 - sum(nonsnv$cnt_met.BS.M8)/sum(nonsnv$cnt_tot.BS.M8))

snv.alt.M8 # 2.707905
snv.5hmC.M8 # 13.69037
snv.5mC.M8 # 62.92752
snv.Candother.M8 # 20.67421
nonsnv.alt.M8 # 3.467643
nonsnv.5hmC.M8 # 16.88822
nonsnv.5mC.M8 # 50.31349
nonsnv.Candother.M8 # 29.33064

meth<-c(nonsnv.5mC.M8, nonsnv.5hmC.M8, nonsnv.Candother.M8, nonsnv.alt.M8, snv.5mC.M8, snv.5hmC.M8, snv.Candother.M8, snv.alt.M8)
sample<-c("non-SNV", "non-SNV", "non-SNV", "non-SNV", "SNV", "SNV", "SNV", "SNV")
label<-c("5mC", "5hmC", "C/other", "T", "5mC", "5hmC", "C/other", "T")

summary.withallfreq<-data.frame(meth, sample, label)

ggplot(summary.withallfreq, aes(x = factor(sample, levels(factor(sample))[c(2,1)]), y = meth, fill=factor(label, levels(factor(label))[c(4,3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Methylation/Alternative alelle in Margin") + theme_classic() + theme(legend.title = element_blank(), plot.title = element_text(size = rel(.75))) + labs(title = "CpG sites\nSNV (n=2610) and non-SNV (n=54479299)") + scale_fill_manual(values=c("black", "red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160216_methylation.snv.nonsnv.cpg.M8.withallfreq.pdf", width = 10/2.54, height = 10/2.54)


# Tumour

snv.alt.T3<-100*sum(snv$count_alternative.E01)/sum(snv$count_total.E01)
snv.5hmC.T3<-100*(1 - sum(snv$count_alternative.E01)/sum(snv$count_total.E01))*(sum(snv$cnt_met.BS.T3)/sum(snv$cnt_tot.BS.T3) - sum(snv$cnt_met.oxBS.T3)/sum(snv$cnt_tot.oxBS.T3))
snv.5mC.T3<-100*(1 - sum(snv$count_alternative.E01)/sum(snv$count_total.E01))*sum(snv$cnt_met.oxBS.T3)/sum(snv$cnt_tot.oxBS.T3)
snv.Candother.T3<-100*(1 - sum(snv$count_alternative.E01)/sum(snv$count_total.E01))*(1 - sum(snv$cnt_met.BS.T3)/sum(snv$cnt_tot.BS.T3))
nonsnv.alt.T3<-100*sum(nonsnv$count_alternative.E01)/sum(as.numeric(nonsnv$count_total.E01))
nonsnv.5hmC.T3<-100*(1 - sum(nonsnv$count_alternative.E01)/sum(as.numeric(nonsnv$count_total.E01)))*(sum(nonsnv$cnt_met.BS.T3)/sum(nonsnv$cnt_tot.BS.T3) - sum(nonsnv$cnt_met.oxBS.T3)/sum(as.numeric(nonsnv$cnt_tot.oxBS.T3)))
nonsnv.5mC.T3<-100*(1 - sum(nonsnv$count_alternative.E01)/sum(as.numeric(nonsnv$count_total.E01)))*sum(nonsnv$cnt_met.oxBS.T3)/sum(as.numeric(nonsnv$cnt_tot.oxBS.T3))
nonsnv.Candother.T3<-100*(1 - sum(nonsnv$count_alternative.E01)/sum(as.numeric(nonsnv$count_total.E01)))*(1 - sum(nonsnv$cnt_met.BS.T3)/sum(nonsnv$cnt_tot.BS.T3))

snv.alt.T3 # 33.7973
snv.5hmC.T3 # 0.3292757
snv.5mC.T3 # 29.351
snv.Candother.T3 # 36.52242
nonsnv.alt.T3 # 2.965108
nonsnv.5hmC.T3 # 1.58712
nonsnv.5mC.T3 # 61.61477
nonsnv.Candother.T3 # 33.833

meth<-c(nonsnv.5mC.T3, nonsnv.5hmC.T3, nonsnv.Candother.T3, nonsnv.alt.T3, snv.5mC.T3, snv.5hmC.T3, snv.Candother.T3, snv.alt.T3)
sample<-c("non-SNV", "non-SNV", "non-SNV", "non-SNV", "SNV", "SNV", "SNV", "SNV")
label<-c("5mC", "5hmC", "C/other", "T", "5mC", "5hmC", "C/other", "T")

summary.withallfreq<-data.frame(meth, sample, label)

ggplot(summary.withallfreq, aes(x = factor(sample, levels(factor(sample))[c(2,1)]), y = meth, fill=factor(label, levels(factor(label))[c(4,3,1,2)]))) + geom_bar(stat = "identity", width = 0.7) + xlab("") + ylab("% Methylation/Alternative alelle in Tumour") + theme_classic() + theme(legend.title = element_blank(), plot.title = element_text(size = rel(.75))) + labs(title = "CpG sites\nSNV (n=2610) and non-SNV (n=54479299)") + scale_fill_manual(values=c("black", "red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160216_methylation.snv.nonsnv.cpg.T3.withallfreq.pdf", width = 10/2.54, height = 10/2.54)




# How significant are the differences in 20160216_methylation.snv.nonsnv.cpg.M8.noallfreq.pdf and 20160216_methylation.snv.nonsnv.cpg.M8.withallfreq.pdf ?

#5mC
summary(snv$pct_5mC.M8)
summary(nonsnv$pct_5mC.M8)

wilcox.test(snv$pct_5mC.M8, nonsnv$pct_5mC.M8)
# Mann-Whitney test, p-value < 2.2e-16, the mean 5mC levels between SNV and non-SNV are significantly different, Margin 5mC in SNV sites is higher than in non-SNV
prop.test(c(sum(snv$cnt_met.oxBS.M8),sum(nonsnv$cnt_met.oxBS.M8)),c(sum(snv$cnt_tot.oxBS.M8),sum(nonsnv$cnt_tot.oxBS.M8)))
# Binomial test, p-value < 2.2e-16
count<-matrix(c(sum(snv$cnt_met.oxBS.M8), sum(nonsnv$cnt_met.oxBS.M8), sum(snv$cnt_tot.oxBS.M8), sum(as.numeric(nonsnv$cnt_tot.oxBS.M8))), nrow=2)
chisq.test(count)
# Pearson's Chi-squared test with Yates' continuity correction, p-value < 2.2e-16
fisher.test(count)
# Fisher's Exact Test for Count Data, p-value < 2.2e-16
ks.test(snv$pct_5mC.M8, nonsnv$pct_5mC.M8)
# Two-sample Kolmogorov-Smirnov, p-value < 2.2e-16



# 5hmC
summary(snv$pct_5hmC.M8)
summary(nonsnv$pct_5hmC.M8)

wilcox.test(snv$pct_5hmC.M8, nonsnv$pct_5hmC.M8)
# Mann-Whitney test, p-value < 2.2e-16, the mean 5hmC levels between SNV and non-SNV are significantly different, Margin 5hmC in SNV sites is lower than in non-SNV
prop.test(c(sum(snv$cnt_met.BS.M8) - sum(snv$cnt_met.oxBS.M8), sum(nonsnv$cnt_met.BS.M8) - sum(nonsnv$cnt_met.oxBS.M8)), c(sum(as.numeric(snv$cnt_tot.BS.M8))*sum(as.numeric(snv$cnt_tot.oxBS.M8)), sum(as.numeric(nonsnv$cnt_tot.BS.M8))*sum(as.numeric(nonsnv$cnt_tot.oxBS.M8))))
# Binomial test, p-value < 2.2e-16
count<-matrix(c(sum(as.numeric(snv$cnt_met.BS.M8)) - sum(as.numeric(snv$cnt_met.oxBS.M8)), sum(as.numeric(nonsnv$cnt_met.BS.M8)) - sum(as.numeric(nonsnv$cnt_met.oxBS.M8)), sum(as.numeric(snv$cnt_tot.BS.M8))*sum(as.numeric(snv$cnt_tot.oxBS.M8)), sum(as.numeric(nonsnv$cnt_tot.BS.M8))*sum(as.numeric(nonsnv$cnt_tot.oxBS.M8))), nrow=2)
chisq.test(count)
# Pearson's Chi-squared test with Yates' continuity correction, p-value < 2.2e-16
fisher.test(count)
# Error in fisher.test(count) : 'x' has entries too large to be integer
ks.test(snv$pct_5hmC.M8, nonsnv$pct_5hmC.M8)
# Two-sample Kolmogorov-Smirnov test, p-value < 2.2e-16





# Represent the distribution of Margin 5mC/5hmC in SNV and non-SNV using box or violin plots


l = list(snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[alt!="NA" & pct_alternative.C01!="NA"], snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[is.na(alt),])
setattr(l, 'names', c("SNV", "non-SNV"))
snv.nonsnv <- rbindlist(l, use.names=TRUE, idcol="ID")
rm(l)
snv.nonsnv[, c("ref", "alt", "qss", "tqss", "count_alternative.C01", "count_total.C01", "pct_alternative.C01", "count_alternative.E01", "count_total.E01", "pct_alternative.E01", "strand", "cnt_met.BS.M8", "cnt_tot.BS.M8", "pct_met.BS.M8", "cnt_met.oxBS.M8", "cnt_tot.oxBS.M8", "pct_met.oxBS.M8", "cnt_met.BS.T3", "cnt_tot.BS.T3", "pct_met.BS.T3", "cnt_met.oxBS.T3", "cnt_tot.oxBS.T3", "pct_met.oxBS.T3", "pct_5hmC.T3", "pct_5mC.T3", "pct_C_other.T3"):= NULL]
rm(snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3)
gc()


#5mC
pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160218_5mC.snv.nonsnv.cpg.M8.violin.pdf", width= 10/2.54, height= 10/2.54)
ggplot(snv.nonsnv, aes(factor(ID, levels(factor(ID))[c(2,1)]), pct_5mC.M8)) + geom_violin(scale = "area", fill="turquoise3", draw_quantiles = 0.5) + xlab("") + ylab("%5mC") + theme_bw() + theme(legend.title = element_blank())
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160218_5mC.snv.nonsnv.cpg.M8.boxplot.pdf", width= 10/2.54, height= 10/2.54)
ggplot(snv.nonsnv, aes(factor(ID, levels(factor(ID))[c(2,1)]), pct_5mC.M8)) + geom_boxplot(fill="turquoise3", outlier.shape = NA) + xlab("") + ylab("%5mC") + theme_bw() + theme(legend.title = element_blank())
dev.off()


#5hmC
pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160218_5hmC.snv.nonsnv.cpg.M8.violin.pdf", width= 10/2.54, height= 10/2.54)
ggplot(snv.nonsnv, aes(factor(ID, levels(factor(ID))[c(2,1)]), pct_5hmC.M8)) + geom_violin(scale = "area", fill="green", draw_quantiles = 0.5) + xlab("") + ylab("%5hmC") + theme_bw() + theme(legend.title = element_blank())
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160218_5hmC.snv.nonsnv.cpg.M8.boxplot.pdf", width= 10/2.54, height= 10/2.54)
ggplot(snv.nonsnv, aes(factor(ID, levels(factor(ID))[c(2,1)]), pct_5hmC.M8)) + geom_boxplot(fill="green", outlier.shape = NA) + xlab("") + ylab("%5hmC") + theme_bw() + theme(legend.title = element_blank())
dev.off()


#C/other
pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160218_C_other.snv.nonsnv.cpg.M8.violin.pdf", width= 10/2.54, height= 10/2.54)
ggplot(snv.nonsnv, aes(factor(ID, levels(factor(ID))[c(2,1)]), pct_C_other.M8)) + geom_violin(scale = "area", fill="red", draw_quantiles = 0.5) + xlab("") + ylab("%C/other") + theme_bw() + theme(legend.title = element_blank())
dev.off()

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures/20160218_C_other.snv.nonsnv.cpg.M8.boxplot.pdf", width= 10/2.54, height= 10/2.54)
ggplot(snv.nonsnv, aes(factor(ID, levels(factor(ID))[c(2,1)]), pct_C_other.M8)) + geom_boxplot(fill="red", outlier.shape = NA) + xlab("") + ylab("%C/other") + theme_bw() + theme(legend.title = element_blank())
dev.off()

```


# Combined

Option 1:
# Need to submit it as a job with > 32GB, otherwise it crashes
# The script is here:
# /lustre/sblab/martin03/projects/20150921_BrainMethylomeRoadMap/scripts/20160218_combined.snv.nonsnv.cpg.M8.boxplot.R
#
# Option 2:
# Need to sample rows from the non-SNV table snv.nucleotide.freq.cpg.C01.E01_methylation.cpg.M8.T3[is.na(alt),]
# Before binding to snv.non.snv


# Option 1:

```bash
fig_dir=/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures
scr_dir=/lustre/sblab/martin03/projects/20150921_BrainMethylomeRoadMap/scripts
bsub -J "combine" -oo $fig_dir/20160218_combined.snv.nonsnv.cpg.M8.pdf.log -R "rusage[mem=49152]" "
~/sw/R/R-3.2.1/bin/Rscript $scr_dir/20160218_combined.snv.nonsnv.cpg.M8.R
"

```


I branched a small analysis after reading Supek2014 in 20160222_supek2014.md



# 20160514 update

```bash
fig_dir=/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv/figures
scr_dir=/lustre/sblab/martin03/projects/20150921_BrainMethylomeRoadMap/scripts
bsub -J "combine" -oo $fig_dir/20160514_combined.snv.nonsnv.cpg.M8.pdf.log -R "rusage[mem=49152]" "
~/sw/R/R-3.2.1/bin/Rscript $scr_dir/20160514_combined.snv.nonsnv.cpg.M8.R
"

```
