
Very broadly, this script is a set of analyses of the 5mC and 5hmC levels in tumour supressor genes (tsgs) and oncogenes (ocgs) in promoters.

### tsgs : TSGene 2.0 (Zhao2015)

First, a quick data exploration:

```python
tsgene20=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/TSGene2.0/Human_TSGs.txt", "r")
tsgene20_lines=tsgene20.readlines()
tsgene20.close()

len(tsgene20_lines)-1 # 1217 tsgs

genetype=[]

for line in tsgene20_lines[1:]:
	fields=line.replace("\n","").split("\t")
	genetype.append(fields[7])
	if fields[1]=="PTEN":
		print fields


from collections import Counter
genetype_cnt = Counter()
for type in genetype:
	genetype_cnt[type] += 1


genetype_cnt
# Counter({'protein-coding': 1018, 'ncRNA': 194, 'pseudo': 3, 'snoRNA': 1, 'unknown': 1})

```

In TSGene, we will then to find chromosome and coordinates for gene ids using e.g. biopython
http://biopython.org/DIST/docs/tutorial/Tutorial.html#htoc109


### Genes for which mutations have been causally implicated in cancer : COSMIC (Forbes2015)

Downloading data:

```bash
cd /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76
sftp "sergio.martinezcuesta@cruk.cam.ac.uk"@sftp-cancer.sanger.ac.uk
sftp> get /files/grch38/cosmic/v76/cancer_gene_census.csv
sftp> exit
cat cancer_gene_census.csv | wc -l # 523
```

We are missing genes compared to the 572 entries reported here: https://cancer.sanger.ac.uk/census
I manually downloaded the data by clicking on TSV and copying the file over to lustre:

```bash
rsync martin03@nm149s012925:/home/cri.camres.org/martin03/Desktop/Census_allFri_Feb_26_17_19_29_2016.tsv .
cat Census_allFri_Feb_26_17_19_29_2016.tsv | wc -l # 573, right
```



Exploring data:

```python
census=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/Census_allFri_Feb_26_17_19_29_2016.tsv", "r")
census_lines=census.readlines()
census.close()

len(census_lines)-1 # 572 genes for which mutations have been causally implicated in cancer

cancergenes_all=[]
cancergenes_all_synonyms=[]
cancergenes_glioblastoma=[]
cancergenes_glioblastoma_synonyms=[]

for line in census_lines[1:]:
	fields=line.replace("\n","").split("\t")
	name=fields[0]
	tumour_type_somatic=fields[7]
	tumour_type_germline=fields[8]
	cancergenes_all.append(name)
	list_syn=[]
	if fields[16]!="":
		list_syn=list_syn+fields[16].split('"')[1].split(",")
	list_syn.append(name)
	cancergenes_all_synonyms=cancergenes_all_synonyms+list_syn
	if ("glioblastoma" in tumour_type_somatic) or ("glioblastoma" in tumour_type_germline):
		cancergenes_glioblastoma.append(name)
		cancergenes_glioblastoma_synonyms=cancergenes_glioblastoma_synonyms+list_syn


len(cancergenes_all) # 572
len(set(cancergenes_all)) # 571
len(set(cancergenes_all_synonyms)) # 3997
len(cancergenes_glioblastoma) # 11
len(set(cancergenes_glioblastoma)) # 11
len(set(cancergenes_glioblastoma_synonyms)) # 91

```

Create a bed file:
* Genome location of promoters of gene bodies for which mutations have been causally implicated in glioblastoma according to COSMIC (11 genes) (less than 1000 base pairs upstream of transcription start site (tss). Bai2015 used 200bp instead.

How do we know if the coordinates given in the gene census file above match the tss? We just don't know. Giovanni and Dario suggest an alternative approach.

This is actually more complicated than above:
* Each gene is either in + or - strands
* Each gene has multiple transcripts
* Each transcript has a transcription start site (tss), sometimes separated by kilobases

Promoter regions should be defined for each transcript -1000 of the leftmost position if in + strand and +1000 of the rightmost position if in - strand. If the gene is in the + strand:

gene   +   Transcript1   |---(promoter)---|-----(gene body)-----|
gene   +   Transcript2      |---(promoter)---|--------(gene body)--------|

then the promoter of the gene is the union of the promoters of the transcripts:

gene   +                 |-----(promoter)----|


If the gene is in the - strand:

gene   -   Transcript1   |-----(gene body)-----|---(promoter)---|
gene   -   Transcript2      |--------(gene body)--------|---(promoter)---|

union:

gene   -                                       |--------(promoter)-------|

We can get the file with these definitions from the illumina here: https://support.illumina.com/sequencing/sequencing_software/igenome.html
Go to Homo sapiens and hg19. However it is 42 GB size and it will take long to download so we copied it directly from Dario's directory:

The problem with this approach is that sometimes tss of different transcripts are separated by thousands of bp, in some cases falling within regions that look like gene bodies in igv (see e.g. PIK3R1 where promoter has a size of 77812 if we consider this - see below).

So we could instead take the leftmost transcript for + strand and take that as tss (the rightmost transcript if - strand), then substract/add 1000bp to this.


```bash
mkdir /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation
mkdir /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes
cp /lustre/sblab/berald01/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes.gtf /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes
```



```python
genes=open("/lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes.gtf", "r")
genes_lines=genes.readlines()
genes.close()

gene_transcript_strand={}
genes_all=set()
gene_chr={}

for line in genes_lines:
	fields=line.split("\t")
	gene_id=fields[8].split(";")[0].split('"')[1]
	transcript_id=fields[8].split(";")[1].split('"')[1]
	leftcoord=fields[3]
	rightcoord=fields[4]
	strand=fields[6]
	chr=fields[0]
	if "_" not in chr:
		genes_all.add(gene_id)
		gene_chr[gene_id]=chr
		if (gene_id, transcript_id, strand) in gene_transcript_strand:
			if strand=="+":
				gene_transcript_strand[(gene_id, transcript_id, strand)].append(str(int(leftcoord)-1))
			if strand=="-":
				gene_transcript_strand[(gene_id, transcript_id, strand)].append(rightcoord)
		else:
			if strand=="+":
				gene_transcript_strand[(gene_id, transcript_id, strand)]=[str(int(leftcoord)-1)]
			if strand=="-":
				gene_transcript_strand[(gene_id, transcript_id, strand)]=[rightcoord]



for tri in gene_transcript_strand:
	if tri[0]=="TERT":
		print tri, gene_transcript_strand[tri]


for tri in gene_transcript_strand:
	if tri[0]=="TET1":
		print tri, gene_transcript_strand[tri]


for tri in gene_transcript_strand:
	if tri[0]=="TET2":
		print tri, gene_transcript_strand[tri]


for tri in gene_transcript_strand:
	if tri[0]=="DAXX":
		print tri, gene_transcript_strand[tri]


len(genes_all) # 23346
len(genes_all.intersection(set(cancergenes_all))) # 547 out of 571 cancer genes intersect
len(genes_all.intersection(set(cancergenes_glioblastoma))) # 11 out of 11 glioblastoma genes intersect
len(genes_all.intersection(set(cancergenes_all_synonyms))) # 559 out of 571 cancer genes intersect (with synonyms)
len(genes_all.intersection(set(cancergenes_glioblastoma_synonyms))) # 11 out of 11 glioblastoma genes intersect (with synonyms)

len(gene_transcript_strand) # 39680


gene_chr["TERT"] # 'chr5'
gene_chr["TET1"] # 'chr10'
gene_chr["TET2"] # 'chr4'
gene_chr["DAXX"] # 'chr6'

len(gene_chr) # 23346



gene_transcript_strand_merge={}

for tri in gene_transcript_strand:
	gene_id=tri[0]
	transcript_id=tri[1]
	strand=tri[2]
	if strand=="+":
		gene_transcript_strand_merge[(gene_id, transcript_id, strand)]=min(gene_transcript_strand[tri])
	if strand=="-":
		gene_transcript_strand_merge[(gene_id, transcript_id, strand)]=max(gene_transcript_strand[tri])


for tri in gene_transcript_strand_merge:
	if tri[0]=="TERT":
		print tri, gene_transcript_strand_merge[tri]

for tri in gene_transcript_strand_merge:
	if tri[0]=="TET1":
		print tri, gene_transcript_strand_merge[tri]

for tri in gene_transcript_strand_merge:
	if tri[0]=="TET2":
		print tri, gene_transcript_strand_merge[tri]

for tri in gene_transcript_strand_merge:
	if tri[0]=="DAXX":
		print tri, gene_transcript_strand_merge[tri]



gene_coords={} # as drawings above
gene_coords_simple={} # considering only the leftmost/rightmost (+/-) tss

count=0
for gene in genes_all:
	count+=1
	print count
	list_coords=[]
	for tri in gene_transcript_strand_merge:
		if gene==tri[0]:
			strand=tri[2]
			list_coords.append(gene_transcript_strand_merge[tri])
	if strand=="+":
		gene_coords[gene]=[int(min(list_coords))-1000, int(max(list_coords))]
		gene_coords_simple[gene]=[int(min(list_coords))-1000, int(min(list_coords))]		
	if strand=="-":
		gene_coords[gene]=[int(min(list_coords)), int(max(list_coords))+1000]
		gene_coords_simple[gene]=[int(max(list_coords)), int(max(list_coords))+1000]		



gene_coords["TERT"] # [1295162, 1296162]
gene_coords["TET1"] # [70319116, 70320116]
gene_coords["TET2"] # [106066031, 106067841]
gene_coords["DAXX"] # [33290793, 33291793]
gene_coords["PIK3R1"] # [67510583, 67588395]

gene_coords_simple["TERT"] # [1295162, 1296162]
gene_coords_simple["TET1"] # [70319116, 70320116]
gene_coords_simple["TET2"] # [106066031, 106067031]
gene_coords_simple["DAXX"] # [33290793, 33291793]
gene_coords_simple["PIK3R1"] # [67510583, 67511583]

len(gene_coords) # 23346
len(gene_coords_simple) # 23346


for gene in cancergenes_glioblastoma:
	if gene in gene_coords:
		print gene, gene_coords[gene][1]-gene_coords[gene][0]


for gene in cancergenes_glioblastoma:
	if gene in gene_coords_simple:
		print gene, gene_coords_simple[gene][1]-gene_coords_simple[gene][0]



# some genes like APC have huge distance between the tss of their transcripts (31354) in gene_coords, just like PIK3R1
# After discussion with Pierre/Euni, it is anyway ok to consider the whole interval because it has biological relevance, however in some cases it falls in regions that look like gene body in igv

# First, create a bed file with the promoter regions for each gene id and leave it in reference_data for the future
# Second, create a bed file with the promoter regions of glioblastoma genes and use specifically for this project

output_promoters_all=open("/lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters.bed", "w")

for gene in gene_coords:
	output_promoters_all.write("%s\t%s\t%s\t%s\n" % (gene_chr[gene], gene_coords[gene][0], gene_coords[gene][1], gene))


output_promoters_all.close()


output_promoters_all_1000=open("/lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_1000.bed", "w")

for gene in gene_coords_simple:
	output_promoters_all_1000.write("%s\t%s\t%s\t%s\n" % (gene_chr[gene], gene_coords_simple[gene][0], gene_coords_simple[gene][1], gene))


output_promoters_all_1000.close()


output_promoters_glioblastoma=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma.bed", "w")

for gene in gene_coords:
	if gene in genes_all.intersection(set(cancergenes_glioblastoma_synonyms)):
		output_promoters_glioblastoma.write("%s\t%s\t%s\t%s\n" % (gene_chr[gene], gene_coords[gene][0], gene_coords[gene][1], gene))


output_promoters_glioblastoma.close()


output_promoters_glioblastoma_1000=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma_1000.bed", "w")

for gene in gene_coords_simple:
	if gene in genes_all.intersection(set(cancergenes_glioblastoma_synonyms)):
		output_promoters_glioblastoma_1000.write("%s\t%s\t%s\t%s\n" % (gene_chr[gene], gene_coords_simple[gene][0], gene_coords_simple[gene][1], gene))


output_promoters_glioblastoma_1000.close()

quit()
```


Sort all promoter files generated above:

```bash
sort -k1,1 -k2,2n /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters.bed > /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_sorted.bed
rm /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters.bed

sort -k1,1 -k2,2n /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_1000.bed > /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_1000_sorted.bed
rm /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_1000.bed

sort -k1,1 -k2,2n /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma.bed > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma_sorted.bed
rm /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma.bed

sort -k1,1 -k2,2n /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma_1000.bed > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma_1000_sorted.bed
rm /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma_1000.bed

```


### Methylation in promoter regions of genes for which mutations have been causally implicated in glioblastoma especifically and cancer in general by COSMIC


Start interactive job:

```bash
ssh -Y uk-cri-lcst01
cd /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes
bsub -R "rusage[mem=32768]" -q interactive -Is bash
```

Intersect methylation and promoter files:

```bash
cd /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210
pigz -dk ear042_M8BS.cpg.bedGraph.gz
pigz -dk ear043_M8oxBS.cpg.bedGraph.gz
pigz -dk ear044_T3BS.cpg.bedGraph.gz
pigz -dk ear045_T3oxBS.cpg.bedGraph.gz

wc -l ear042_M8BS.cpg.bedGraph # 54834640
wc -l ear043_M8oxBS.cpg.bedGraph # 54765875
wc -l ear044_T3BS.cpg.bedGraph # 54801746
wc -l ear045_T3oxBS.cpg.bedGraph # 54788215

mkdir ../../tsgenes_oncogenes/data/methylation_cpg_promoter

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear042_M8BS.cpg.bedGraph -b /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_1000_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10 | \
bedtools intersect -a - -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma_1000_sorted.bed -loj | \
cut -f 1,2,3,4,5,6,7,11 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.1000.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear043_M8oxBS.cpg.bedGraph -b /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_1000_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10 | \
bedtools intersect -a - -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma_1000_sorted.bed -loj | \
cut -f 1,2,3,4,5,6,7,11 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.1000.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear044_T3BS.cpg.bedGraph -b /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_1000_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10 | \
bedtools intersect -a - -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma_1000_sorted.bed -loj | \
cut -f 1,2,3,4,5,6,7,11 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.1000.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear045_T3oxBS.cpg.bedGraph -b /lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes_promoters_1000_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10 | \
bedtools intersect -a - -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma_1000_sorted.bed -loj | \
cut -f 1,2,3,4,5,6,7,11 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.1000.sorted.bed

```


The columns of the files generated above are:

1st : chr
2nd,3rd : coordinates of nucleotide
4th : cnt_met
5th : cnt_tot
6th : + or - (C or G)
7th : gene names for all genes with methylation data
8th : gene names for glioblastoma genes with methylation data


```bash
wc -l ear042_M8BS.cpg_genes.promoters.1000.sorted.bed # 1758410
wc -l ear043_M8oxBS.cpg_genes.promoters.1000.sorted.bed # 1757250
wc -l ear044_T3BS.cpg_genes.promoters.1000.sorted.bed # 1757885
wc -l ear045_T3oxBS.cpg_genes.promoters.1000.sorted.bed # 1757490

cut -f 7 ear042_M8BS.cpg_genes.promoters.1000.sorted.bed | sort | uniq | wc -l # 23021 genes in total
cut -f 8 ear042_M8BS.cpg_genes.promoters.1000.sorted.bed | sort | uniq | wc -l # 12 - 1 glioblastoma genes (one is blank space)

cut -f 7 ear043_M8oxBS.cpg_genes.promoters.1000.sorted.bed | sort | uniq | wc -l # 23018 genes in total
cut -f 8 ear043_M8oxBS.cpg_genes.promoters.1000.sorted.bed | sort | uniq | wc -l # 12 - 1 glioblastoma genes (one is blank space)

cut -f 7 ear044_T3BS.cpg_genes.promoters.1000.sorted.bed | sort | uniq | wc -l # 23017 genes in total
cut -f 8 ear044_T3BS.cpg_genes.promoters.1000.sorted.bed | sort | uniq | wc -l # 12 - 1 glioblastoma genes (one is blank space)

cut -f 7 ear045_T3oxBS.cpg_genes.promoters.1000.sorted.bed | sort | uniq | wc -l # 23022 genes in total
cut -f 8 ear045_T3oxBS.cpg_genes.promoters.1000.sorted.bed | sort | uniq | wc -l # 12 - 1 glioblastoma genes (one is blank space)
```


Create boxplots of 5mC and 5hmC comparing methylation in M versus T for glioblastoma genes in the context of the rest of promoters

```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "gene_name_glioblastoma"))
ear042_M8BS[, glioblastoma:=ifelse(gene_name_glioblastoma!="", "yes", "no")]
ear042_M8BS[, "gene_name_glioblastoma":= NULL]
ear042_M8BS[, pct_met := cnt_met/cnt_tot]

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "gene_name_glioblastoma"))
ear043_M8oxBS[, glioblastoma:=ifelse(gene_name_glioblastoma!="", "yes", "no")]
ear043_M8oxBS[, "gene_name_glioblastoma":= NULL]
ear043_M8oxBS[, pct_met := cnt_met/cnt_tot]

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "gene_name_glioblastoma"))
ear044_T3BS[, glioblastoma:=ifelse(gene_name_glioblastoma!="", "yes", "no")]
ear044_T3BS[, "gene_name_glioblastoma":= NULL]
ear044_T3BS[, pct_met := cnt_met/cnt_tot]

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "gene_name_glioblastoma"))
ear045_T3oxBS[, glioblastoma:=ifelse(gene_name_glioblastoma!="", "yes", "no")]
ear045_T3oxBS[, "gene_name_glioblastoma":= NULL]
ear045_T3oxBS[, pct_met := cnt_met/cnt_tot]

setkey(ear042_M8BS, chr, start, end, strand, gene_name, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
M8[, pct_5hmC := pct_met.BS - pct_met.oxBS]
M8[, pct_5mC := pct_met.oxBS]
M8[, pct_C_other := 1 - pct_met.BS]

T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))
T3[, pct_5hmC := pct_met.BS - pct_met.oxBS]
T3[, pct_5mC := pct_met.oxBS]
T3[, pct_C_other := 1 - pct_met.BS]

#rm(ear042_M8BS, ear043_M8oxBS, ear044_T3BS, ear045_T3oxBS)



### 5mC

setkey(M8, chr, start, end, strand, gene_name, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

#rm(M8, T3)

M8.T3[, c("cnt_met.BS.M8", "cnt_tot.BS.M8", "pct_met.BS.M8", "cnt_met.oxBS.M8", "cnt_tot.oxBS.M8", "pct_met.oxBS.M8", "pct_5hmC.M8", "pct_C_other.M8", "cnt_met.BS.T3", "cnt_tot.BS.T3", "pct_met.BS.T3", "cnt_met.oxBS.T3", "cnt_tot.oxBS.T3", "pct_met.oxBS.T3", "pct_5hmC.T3", "pct_C_other.T3"):= NULL]
M8.T3.m <- reshape2::melt(M8.T3, id.vars = c("chr", "start", "end", "strand", "gene_name", "glioblastoma"), variable.name = "sample", value.name = "pct_met")

#rm(M8.T3)

# extracting glioblastoma genes
M8.T3.m[glioblastoma=="yes"]

ggplot(M8.T3.m, aes(x = factor(sample), y = pct_met, fill = factor(glioblastoma))) + geom_boxplot(outlier.shape = NA) + theme_bw() + theme(legend.title = element_blank()) + xlab("") + ylab("%5mC") + scale_x_discrete(labels = c("Margin", "Tumour")) + labs(title = "Promoters") + scale_fill_discrete(labels=c("All", "Glioblastoma"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160303_5mC.promoters.M8.T3.glioblastoma.rest.pdf", width = 10/2.54, height = 10/2.54)



### 5hmC

setkey(M8, chr, start, end, strand, gene_name, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

#rm(M8, T3)

M8.T3[, c("cnt_met.BS.M8", "cnt_tot.BS.M8", "pct_met.BS.M8", "cnt_met.oxBS.M8", "cnt_tot.oxBS.M8", "pct_met.oxBS.M8", "pct_5mC.M8", "pct_C_other.M8", "cnt_met.BS.T3", "cnt_tot.BS.T3", "pct_met.BS.T3", "cnt_met.oxBS.T3", "cnt_tot.oxBS.T3", "pct_met.oxBS.T3", "pct_5mC.T3", "pct_C_other.T3"):= NULL]
M8.T3.m <- reshape2::melt(M8.T3, id.vars = c("chr", "start", "end", "strand", "gene_name", "glioblastoma"), variable.name = "sample", value.name = "pct_met")

#rm(M8.T3)

# extracting glioblastoma genes
M8.T3.m[glioblastoma=="yes"]

ggplot(M8.T3.m, aes(x = factor(sample), y = pct_met, fill = factor(glioblastoma))) + geom_boxplot(outlier.shape = NA) + theme_bw() + theme(legend.title = element_blank()) + xlab("") + ylab("%5hmC") + scale_x_discrete(labels = c("Margin", "Tumour")) + labs(title = "Promoters") + scale_fill_discrete(labels=c("All", "Glioblastoma")) + ylim(0, 1)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160303_5hmC.promoters.M8.T3.glioblastoma.rest.pdf", width = 10/2.54, height = 10/2.54)

```



Same boxplots as above but calculating an average % methylation for the promoter by summing cnt_met and cnt_tot across sites within a promoter.


```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "gene_name_glioblastoma"))
ear042_M8BS[, glioblastoma:=ifelse(gene_name_glioblastoma!="", "yes", "no")]
ear042_M8BS[, "gene_name_glioblastoma":= NULL]

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "gene_name_glioblastoma"))
ear043_M8oxBS[, glioblastoma:=ifelse(gene_name_glioblastoma!="", "yes", "no")]
ear043_M8oxBS[, "gene_name_glioblastoma":= NULL]

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "gene_name_glioblastoma"))
ear044_T3BS[, glioblastoma:=ifelse(gene_name_glioblastoma!="", "yes", "no")]
ear044_T3BS[, "gene_name_glioblastoma":= NULL]

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "gene_name_glioblastoma"))
ear045_T3oxBS[, glioblastoma:=ifelse(gene_name_glioblastoma!="", "yes", "no")]
ear045_T3oxBS[, "gene_name_glioblastoma":= NULL]

setkey(ear042_M8BS, chr, start, end, strand, gene_name, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3))), by=list(gene_name, glioblastoma)]

M8.T3.promoter.melt <- reshape2::melt(M8.T3.promoter, id.vars = c("gene_name", "glioblastoma"), variable.name = "sample", value.name = "pct_met")



### 5mC

ggplot(M8.T3.promoter.melt[sample=="pct_5mC.M8" | sample=="pct_5mC.T3"], aes(x = factor(sample), y = pct_met, fill = factor(glioblastoma))) + geom_boxplot(outlier.shape = NA) + theme_bw() + theme(legend.title = element_blank()) + xlab("") + ylab("%5mC") + scale_x_discrete(labels = c("Margin", "Tumour")) + labs(title = "Promoters") + scale_fill_discrete(labels=c("All", "Glioblastoma"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160303_5mC.promoters.M8.T3.glioblastoma.rest.bypromoter.pdf", width = 10/2.54, height = 10/2.54)



### 5hmC

ggplot(M8.T3.promoter.melt[sample=="pct_5hmC.M8" | sample=="pct_5hmC.T3"], aes(x = factor(sample), y = pct_met, fill = factor(glioblastoma))) + geom_boxplot(outlier.shape = NA) + theme_bw() + theme(legend.title = element_blank()) + xlab("") + ylab("%5hmC") + scale_x_discrete(labels = c("Margin", "Tumour")) + labs(title = "Promoters") + scale_fill_discrete(labels=c("All", "Glioblastoma"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160303_5hmC.promoters.M8.T3.glioblastoma.rest.bypromoter.pdf", width = 10/2.54, height = 10/2.54)


```



What is the methylation status of the promoter of genes MGMT?
Interpret in the context of Molenaar2014 and Nakagawachi2003

```R
M8.T3.promoter[gene_name=="MGMT"]
#   gene_name cancer glioblastoma pct_5mC.M8 pct_5hmC.M8 pct_5mC.T3 pct_5hmC.T3
#1:      MGMT      n            n   5.911402     8.39106   10.10435    1.117012

M8.T3[gene_name=="MGMT"] # 76 CpGs (152 nucleotides) in the 1000b promoter.


M8.T3.promoter[gene_name=="IDH1"]
#   gene_name glioblastoma pct_5mC.M8 pct_5hmC.M8 pct_5mC.T3 pct_5hmC.T3
#1:      IDH1          yes   2.685557    6.626928   8.251473     1.08212
```

10% 5mC in T3 (increase of 4% : 6% (M8) -> 10% (T3)), it fits the IDH1wt/MGMTunmet of shortest survival in Molenaar2014.
In parallel, 5hmC drops (decrease of 9% : 10% (M8) -> 1% (T3))

This will be somewhat good David's Bentley focussed plot example.

Another good example is PIK3R6 (here the changes are of 18%). It also has a nonsynonymous mutation and it is related to PIK3R1 and PIK3CA mutated in Bai2015.

```R
M8.T3.promoter[gene_name=="PIK3R6"]
#   gene_name glioblastoma pct_5mC.M8 pct_5hmC.M8 pct_5mC.T3 pct_5hmC.T3
#1:    PIK3R6           no   52.27818    20.00016       70.8    2.710638

M8.T3[gene_name=="PIK3R6"] # 11 CpGs (22 nucleotides) in the 1000b promoter.
```

ATRX, which was found pervasively mutated in Bai2015, might also be a good example to illustrate (15% increase):

```R
M8.T3.promoter[gene_name=="ATRX"]
#   gene_name glioblastoma pct_5mC.M8 pct_5hmC.M8 pct_5mC.T3 pct_5hmC.T3
#1:      ATRX           no    31.2873    11.20834   48.78214   0.1197653

M8.T3[gene_name=="ATRX"] # 19 CpGs (38 nucleotides) in the 1000b
```




Find more extreme paradigms : promoters with high 5hmC in Margin and high 5mC in Tumour overall

```R
M8.T3.promoter[pct_5hmC.M8 > 30 & pct_5mC.T3 > 70][order(-pct_5hmC.M8)][1:100]
M8.T3[gene_name=="FAM21B"]
```

At first sight, there are quite a lot of miRNAs among the top hits.
Need to go through the list in detail and find one of the top relevant ones to combine it with MGMT and PIK3R6.

After reading Molenaar2014, not having a IDH1 mutation - like in our example - can be more deleterious than having it.





Now, I will repeat the plots above but also including all cancer genes from the census as another boxplot

```python

census=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/Census_allFri_Feb_26_17_19_29_2016.tsv", "r")
census_lines=census.readlines()
census.close()

len(census_lines)-1 # 572 genes for which mutations have been causally implicated in cancer

cancergenes_all=[]
cancergenes_all_synonyms=[]
cancergenes_glioblastoma=[]
cancergenes_glioblastoma_synonyms=[]

for line in census_lines[1:]:
	fields=line.replace("\n","").split("\t")
	name=fields[0]
	tumour_type_somatic=fields[7]
	tumour_type_germline=fields[8]
	cancergenes_all.append(name)
	list_syn=[]
	if fields[16]!="":
		list_syn=list_syn+fields[16].split('"')[1].split(",")
	list_syn.append(name)
	cancergenes_all_synonyms=cancergenes_all_synonyms+list_syn
	if ("glioblastoma" in tumour_type_somatic) or ("glioblastoma" in tumour_type_germline):
		cancergenes_glioblastoma.append(name)
		cancergenes_glioblastoma_synonyms=cancergenes_glioblastoma_synonyms+list_syn


len(cancergenes_all) # 572
len(set(cancergenes_all)) # 571
len(set(cancergenes_all_synonyms)) # 3997
len(cancergenes_glioblastoma) # 11
len(set(cancergenes_glioblastoma)) # 11
len(set(cancergenes_glioblastoma_synonyms)) # 91


genes=open("/lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Annotation/Genes/genes.gtf", "r")
genes_lines=genes.readlines()
genes.close()

gene_transcript_strand={}
genes_all=set()
gene_chr={}

for line in genes_lines:
	fields=line.split("\t")
	gene_id=fields[8].split(";")[0].split('"')[1]
	transcript_id=fields[8].split(";")[1].split('"')[1]
	leftcoord=fields[3]
	rightcoord=fields[4]
	strand=fields[6]
	chr=fields[0]
	if "_" not in chr:
		genes_all.add(gene_id)
		gene_chr[gene_id]=chr
		if (gene_id, transcript_id, strand) in gene_transcript_strand:
			if strand=="+":
				gene_transcript_strand[(gene_id, transcript_id, strand)].append(str(int(leftcoord)-1))
			if strand=="-":
				gene_transcript_strand[(gene_id, transcript_id, strand)].append(rightcoord)
		else:
			if strand=="+":
				gene_transcript_strand[(gene_id, transcript_id, strand)]=[str(int(leftcoord)-1)]
			if strand=="-":
				gene_transcript_strand[(gene_id, transcript_id, strand)]=[rightcoord]



len(genes_all) # 23346
len(genes_all.intersection(set(cancergenes_all))) # 547 out of 571 cancer genes intersect
len(genes_all.intersection(set(cancergenes_glioblastoma))) # 11 out of 11 glioblastoma genes intersect
len(genes_all.intersection(set(cancergenes_all_synonyms))) # 559 out of 571 cancer genes intersect (with synonyms)
len(genes_all.intersection(set(cancergenes_glioblastoma_synonyms))) # 11 out of 11 glioblastoma genes intersect (with synonyms)

len(gene_transcript_strand) # 39680

len(gene_chr) # 23346



gene_transcript_strand_merge={}

for tri in gene_transcript_strand:
	gene_id=tri[0]
	transcript_id=tri[1]
	strand=tri[2]
	if strand=="+":
		gene_transcript_strand_merge[(gene_id, transcript_id, strand)]=min(gene_transcript_strand[tri])
	if strand=="-":
		gene_transcript_strand_merge[(gene_id, transcript_id, strand)]=max(gene_transcript_strand[tri])



gene_coords={} # as drawings above
gene_coords_simple={} # considering only the leftmost/rightmost (+/-) tss

count=0
for gene in genes_all:
	count+=1
	print count
	list_coords=[]
	for tri in gene_transcript_strand_merge:
		if gene==tri[0]:
			strand=tri[2]
			list_coords.append(gene_transcript_strand_merge[tri])
	if strand=="+":
		gene_coords[gene]=[int(min(list_coords))-1000, int(max(list_coords))]
		gene_coords_simple[gene]=[int(min(list_coords))-1000, int(min(list_coords))]		
	if strand=="-":
		gene_coords[gene]=[int(min(list_coords)), int(max(list_coords))+1000]
		gene_coords_simple[gene]=[int(max(list_coords)), int(max(list_coords))+1000]		



len(gene_coords) # 23346
len(gene_coords_simple) # 23346



output_promoters_all=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma.bed", "w")

for gene in gene_coords:
	cancer="n"
	glio="n"
	if gene in genes_all.intersection(set(cancergenes_all_synonyms)):
		cancer="y"
	if gene in genes_all.intersection(set(cancergenes_glioblastoma_synonyms)):
		glio="y"
	output_promoters_all.write("%s\t%s\t%s\t%s\t%s\t%s\n" % (gene_chr[gene], gene_coords[gene][0], gene_coords[gene][1], gene, cancer, glio))


output_promoters_all.close()


output_promoters_all_1000=open("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_1000.bed", "w")

for gene in gene_coords_simple:
	cancer="n"
	glio="n"
	if gene in genes_all.intersection(set(cancergenes_all_synonyms)):
		cancer="y"
	if gene in genes_all.intersection(set(cancergenes_glioblastoma_synonyms)):
		glio="y"
	output_promoters_all_1000.write("%s\t%s\t%s\t%s\t%s\t%s\n" % (gene_chr[gene], gene_coords_simple[gene][0], gene_coords_simple[gene][1], gene, cancer, glio))


output_promoters_all_1000.close()

```


Sort:

```bash
sort -k1,1 -k2,2n /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma.bed > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_sorted.bed
rm /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma.bed

sort -k1,1 -k2,2n /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_1000.bed > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_1000_sorted.bed
rm /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_1000.bed

```


Intersect:


```bash
bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear042_M8BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_1000_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10,11,12 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear043_M8oxBS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_1000_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10,11,12 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear044_T3BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_1000_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10,11,12 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear045_T3oxBS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_1000_sorted.bed -wa -wb | \
cut -f 1,2,3,4,5,6,10,11,12 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed

```

The columns of the files generated above are:

1st : chr
2nd,3rd : coordinates of nucleotide
4th : cnt_met
5th : cnt_tot
6th : + or - (C or G)
7th : gene names for all genes with methylation data
8th : is it a cancer gene according to COSMIC? (y/n)
9th : is it a glioblastoma gene according to COSMIC? (y/n)


```bash
wc -l ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed # 1758410
wc -l ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed # 1757250
wc -l ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed # 1757885
wc -l ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed # 1757490

cut -f 7 ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed | sort | uniq | wc -l # 23021 genes in total
cut -f 7 ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed | sort | uniq | wc -l # 23018 genes in total
cut -f 7 ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed | sort | uniq | wc -l # 23017 genes in total
cut -f 7 ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed | sort | uniq | wc -l # 23022 genes in total

```



Same boxplots as above but calculating an average % methylation for the promoter by summing cnt_met and cnt_tot across sites within a promoter.


```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))


# How many total genes in ear042_M8BS ?
ear042_M8BS[, .N, by=gene_name] # 23021

# How many cancer genes in ear042_M8BS ?
ear042_M8BS[cancer=="y", .N, by=gene_name] # 556

# How many glioblastoma genes in ear042_M8BS ?
ear042_M8BS[glioblastoma=="y", .N, by=gene_name] # 11

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3))), by=list(gene_name, cancer, glioblastoma)]

M8.T3.promoter.melt <- reshape2::melt(M8.T3.promoter, id.vars = c("gene_name", "cancer", "glioblastoma"), variable.name = "sample", value.name = "pct_met")

M8.T3.promoter.melt[, type:=ifelse(glioblastoma=="y", "g", ifelse(cancer=="y", "c", "n"))]


### 5mC

ggplot(M8.T3.promoter.melt[sample=="pct_5mC.M8" | sample=="pct_5mC.T3"], aes(x = factor(sample), y = pct_met, fill = factor(type, levels(factor(type))[c(2,1,3)]))) + geom_boxplot(outlier.shape = NA) + theme_bw() + theme(legend.title = element_blank()) + xlab("") + ylab("%5mC") + scale_x_discrete(labels = c("Margin", "Tumour")) + labs(title = "Promoters") + scale_fill_discrete(labels=c("Glioblastoma", "Cancer", "Rest"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160307_5mC.promoters.M8.T3.glioblastoma.cancer.rest.bypromoter.pdf", width = 14/2.54, height = 10/2.54)


### 5hmC

ggplot(M8.T3.promoter.melt[sample=="pct_5hmC.M8" | sample=="pct_5hmC.T3"], aes(x = factor(sample), y = pct_met, fill = factor(type, levels(factor(type))[c(2,1,3)]))) + geom_boxplot(outlier.shape = NA) + theme_bw() + theme(legend.title = element_blank()) + xlab("") + ylab("%5hmC") + scale_x_discrete(labels = c("Margin", "Tumour")) + labs(title = "Promoters") + scale_fill_discrete(labels=c("Glioblastoma", "Cancer", "Rest"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160307_5hmC.promoters.M8.T3.glioblastoma.cancer.rest.bypromoter.pdf", width = 14/2.54, height = 10/2.54)


### 20160513 update

### 5mC

ggplot(M8.T3.promoter.melt[sample=="pct_5mC.M8" | sample=="pct_5mC.T3"], aes(x = factor(sample), y = pct_met, fill = factor(type, levels(factor(type))[c(2,1,3)]))) + geom_boxplot(outlier.shape = NA) + theme_bw() + theme(legend.title = element_blank()) + xlab("") + ylab("% 5mC") + scale_x_discrete(labels = c("Margin", "Tumour")) + scale_fill_discrete(labels=c("Glioblastoma", "Cancer", "Other"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160513_5mC.promoters.M8.T3.glioblastoma.cancer.rest.bypromoter.pdf", width = 14/2.54, height = 10/2.54)

ggplot(M8.T3.promoter.melt[sample=="pct_5mC.M8" | sample=="pct_5mC.T3"], aes(x = factor(type, levels(factor(type))[c(2,1,3)]), y = pct_met, fill = factor(sample))) + geom_boxplot(outlier.shape = NA) + theme_bw() + theme(legend.title = element_blank()) + xlab("") + ylab("% 5mC") + scale_x_discrete(labels = c("Glioblastoma\nn=11", "Cancer\nn=556", "Other\nn=23021")) + scale_fill_manual(labels=c("Margin", "Tumour"), values=c("green4", "red3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160513_5mC.promoters.M8.T3.glioblastoma.cancer.rest.bypromoter.cancerx.pdf", width = 14/2.54, height = 10/2.54)


### 5hmC

ggplot(M8.T3.promoter.melt[sample=="pct_5hmC.M8" | sample=="pct_5hmC.T3"], aes(x = factor(sample), y = pct_met, fill = factor(type, levels(factor(type))[c(2,1,3)]))) + geom_boxplot(outlier.shape = NA) + theme_bw() + theme(legend.title = element_blank()) + xlab("") + ylab("% 5hmC") + scale_x_discrete(labels = c("Margin", "Tumour")) + scale_fill_discrete(labels=c("Glioblastoma", "Cancer", "Other")) + coord_cartesian(ylim = c(0, 100))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160513_5hmC.promoters.M8.T3.glioblastoma.cancer.rest.bypromoter.pdf", width = 14/2.54, height = 10/2.54)

ggplot(M8.T3.promoter.melt[sample=="pct_5hmC.M8" | sample=="pct_5hmC.T3"], aes(x = factor(type, levels(factor(type))[c(2,1,3)]), y = pct_met, fill = factor(sample))) + geom_boxplot(outlier.shape = NA) + theme_bw() + theme(legend.title = element_blank()) + xlab("") + ylab("% 5hmC") + scale_x_discrete(labels = c("Glioblastoma\nn=11", "Cancer\nn=556", "Other\nn=23021")) + scale_fill_manual(labels=c("Margin", "Tumour"), values=c("green4", "red3")) + coord_cartesian(ylim = c(0, 100))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160513_5hmC.promoters.M8.T3.glioblastoma.cancer.rest.bypromoter.cancerx.pdf", width = 14/2.54, height = 10/2.54)


```



Print methylation in promoters to a file to share with Dario in sblab-srv001

```R
write.table(M8.T3.promoter, "/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/tables/M8.T3.promoter.txt", sep="\t", row.names = FALSE, quote=FALSE)
```

In nas-srv001,

```bash
cp /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/tables/M8.T3.promoter.txt /data/sblab_data1/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/tables
```



Example plots of methylation in promoters
* PIK3R6
* ATRX
* MGMT - see below


```R
xdata<-M8.T3[gene_name=="PIK3R6"]

xdata_diff<-diff(xdata$start)
cpg<-c(1)

for (i in xdata_diff){
if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
else{cpg<-append(cpg, tail(cpg, n=1)+1)}
}

xdata[,cpg:=cpg]
xdata # 11 CpG sites

tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",11*3),rep("T",11*3))]
tab.summary[,label:=rep(c(rep("5mC",11), rep("5hmC",11), rep("C/other",11)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160308_PIK3R6.promoter.pdf", width = 14/2.54, height = 10/2.54)



xdata<-M8.T3[gene_name=="ATRX"]

xdata_diff<-diff(xdata$start)
cpg<-c(1)

for (i in xdata_diff){
if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
else{cpg<-append(cpg, tail(cpg, n=1)+1)}
}

xdata[,cpg:=cpg]
xdata # 18 CpG sites

tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",18*3),rep("T",18*3))]
tab.summary[,label:=rep(c(rep("5mC",18), rep("5hmC",18), rep("C/other",18)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160308_ATRX.promoter.pdf", width = 16/2.54, height = 10/2.54)

```



BS and BS+oxBS views of MGMT:

But first of all, what do we consider to be the promoter region of MGMT:

```bash
grep "MGMT" /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_1000_sorted.bed
# chr10	131264453	131265453	MGMT	n	n
```


```R
xdata<-M8.T3[gene_name=="MGMT"]

xdata_diff<-diff(xdata$start)
cpg<-c(1)

for (i in xdata_diff){
if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
else{cpg<-append(cpg, tail(cpg, n=1)+1)}
}

xdata[,cpg:=cpg]
xdata # 68 CpG sites (152 nucleotides), 8 of them appear consecutively

tab<-xdata[, list(pct_meth.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_unmeth.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_meth.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), pct_unmeth.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",68*2),rep("T",68*2))]
tab.summary[,label:=rep(c(rep("Methylated",68), rep("Non-methylated",68)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "turquoise3")) + facet_grid(~ cpg) + labs(title = "BS")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160311_MGMT.promoter.BS.pdf", width=45/2.54, height=10/2.54)




tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",68*3),rep("T",68*3))]
tab.summary[,label:=rep(c(rep("5mC",68), rep("5hmC",68), rep("C/other",68)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg) + labs(title = "BS and oxBS")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160308_MGMT.promoter.BS.oxBS.pdf", width = 45/2.54, height = 10/2.54)


```



Merge all 68 CpG sites in MGMT in a summary plot:

```R
xdata<-M8.T3[gene_name=="MGMT"]
tab<-xdata[, list(pct_meth.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_unmeth.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_meth.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), pct_unmeth.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=gene_name]
tab.summary<-reshape2::melt(tab, id.vars = "gene_name", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",2),rep("T",2))]
tab.summary[,label:=rep(c("Methylated", "Non-methylated"), 2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "turquoise3")) + labs(title = "BS")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160311_MGMT.promoter.BS.summary.pdf", width = 10/2.54, height = 10/2.54)



xdata<-M8.T3[gene_name=="MGMT"]
tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=gene_name]
tab.summary<-reshape2::melt(tab, id.vars = "gene_name", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",3),rep("T",3))]
tab.summary[,label:=rep(c("5mC","5hmC","C/other"),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + labs(title = "BS and oxBS")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160311_MGMT.promoter.BS.oxBS.summary.pdf", width = 8.5/2.54, height = 10/2.54)

```



Extract sequence of MGMT promoter and give it to Euni:

```

from Bio import SeqIO

for seq_record in SeqIO.parse("/lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Sequence/WholeGenomeFasta/genome.fa", "fasta"):
	if seq_record.id == "chr10":
		seq=seq_record.seq


len(seq)

str(seq[131264453:131265453]).upper()
# coordinates from /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_1000_sorted.bed


```


Also factor in the alternative allele if neccessary.

Here is the email exchange between Thomas and Euni to complete the MGMT story.

------------------
Region in chr10 containing what the clinicians consider to be the promoter of the MGMT gene:
131155467-131155565

------------------
Dear Euni,

I have sent the details to Kieren.

Please below see a copy&paste of the MGMT promoter methylation test
result. If MGMT is methylated, the tumour is more likely to respond to
TMZ. Saying that, a case like this would receive radio- and chemotherapy
regardless of the MGMT methylation status. Her TMZ treatment was stopped
as she developed complications of the chemotherapy. At the time of
stopping TMZ and following few months (3-6) she had no visible tumour on
the scan. Then tumour came back and patient died on 22/11/15.

I hope this is helpful.

Regards,

Tom

------------------
SUMMARY OF CLINICAL EVENTS
22/09/2014 - Surgery
29/10/2014 - 09/12/2014 Radiation 60 Gy in 30 fractions + concurrent TMZ
06/01/2015 - 31/03/2015 Chemotherapy Adjuvant TMZ, completed 2 cycles
then stopped.
22/11/15 - date of death

------------------
MGMT PROMOTER METHYLATION TEST RESULT
M6322: A paraffin block (-1) of right frontal lobe sol brain biopsy with glioblastoma WHO-2007 Grade 4.
Date specimen received: 02/10/2014
The H&E slides of the above specimen were reviewed and neoplastic cell-rich tissue was dissected from consecutive unstained sections. DNA was extracted from the dissected tissue using the QIAamp DNA FFPE Tissue Kit (Qiagen) and was bisulphite-converted using the EpiTect Bisulphite Kit (Qiagen). The MGMT promoter methylation was determined by pyrosequencing of four CpG sites (CpGs 76-79) in exon 1 of the MGMT gene using the CE-Marked therascreen MGMT Pyro Kit on a Pyromark Q24 System (Qiagen). The results are as follows:

Neoplastic cell content:                          > 70% of cells dissected
Mean methylation level of four CpGs analysed: 2%
MGMT promoter methylation: Absent

Comment: The above result indicates that MGMT promoter methylation is absent in the neoplastic cells analysed.

Please note MGMT is a DNA repair protein that acts by removing alkyl groups from DNA induced by alkylating agents. Activity of MGMT may lead to resistance to alkylating chemotherapy and transcriptional suppression of the gene by promoter methylation has been shown to be associated with improved survival in patients with high grade glioma and in patients with glioblastoma treated with alkylating agents. However, methylation at other CpGs than those covered by the present assay has not been determined. The results of this test must be interpreted within the clinical context and other relevant data and should not be used as a sole determinant of treatment response to alkylating chemotherapy in standard clinical practice.

Please also note the above assay has been internally validated using the controls supplied in the Kit and DNA extracted from healthy blood samples and formalin-fixed and paraffin-embedded glioma tissues and its performance characteristics verified as described in the Kit. A cut-off of 10% methylation for the four CpG sites is used to assign a glioma as methylated or unmethylated based on the published data (Dunn et al., British Journal of Cancer (2009) 101, 124 - 131; Collins et al,. Acta Neuropathol Commun. (2014) 2:68).

Dr Hongxiang Liu, Consultant Clinical Scientist

Molecular Malignancy Laboratory

-------------------



CE-Marked therascreen MGMT Pyro Kit on a Pyromark Q24 System.
See QIAGEN2015
methylation in four CpG sites in exon 1 (chromosome 10 from 131,265,519 to 131,265,537)

Extract sequence of MGMT promoter used by QIAGEN and Collins2014

```

from Bio import SeqIO

for seq_record in SeqIO.parse("/lustre/sblab/martin03/reference_data/genomes/iGenomes/Homo_sapiens/UCSC/hg19/Sequence/WholeGenomeFasta/genome.fa", "fasta"):
	if seq_record.id == "chr10":
		seq=seq_record.seq


len(seq)

str(seq[131265518:131265537]).upper() # QIAGEN promoter
# coordinates from above. Remember we actually need to indicate in bed (-1)

str(seq[131265491:131265621]).upper() # Collins 2014 promoter


```


Intersect methylation files with a file containing the QIAGEN promoter and the Collins2014

```bash
echo -e "chr10\t131265518\t131265537" > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/MGMT_Pyro_Kit.bed
echo -e "chr10\t131265491\t131265621" > /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/Collins2014.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear042_M8BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/MGMT_Pyro_Kit.bed -wa -wb | \
cut -f 1,2,3,4,5,6 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/ear042_M8BS.cpg_MGMT_Pyro_Kit.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear043_M8oxBS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/MGMT_Pyro_Kit.bed -wa -wb | \
cut -f 1,2,3,4,5,6 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/ear043_M8oxBS.cpg_MGMT_Pyro_Kit.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear044_T3BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/MGMT_Pyro_Kit.bed -wa -wb | \
cut -f 1,2,3,4,5,6 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/ear044_T3BS.cpg_MGMT_Pyro_Kit.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear045_T3oxBS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/MGMT_Pyro_Kit.bed -wa -wb | \
cut -f 1,2,3,4,5,6 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/ear045_T3oxBS.cpg_MGMT_Pyro_Kit.bed


bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear042_M8BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/Collins2014.bed -wa -wb | \
cut -f 1,2,3,4,5,6 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/ear042_M8BS.cpg_Collins2014.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear043_M8oxBS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/Collins2014.bed -wa -wb | \
cut -f 1,2,3,4,5,6 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/ear043_M8oxBS.cpg_Collins2014.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear044_T3BS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/Collins2014.bed -wa -wb | \
cut -f 1,2,3,4,5,6 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/ear044_T3BS.cpg_Collins2014.bed

bedtools intersect -a /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg/20160210/ear045_T3oxBS.cpg.bedGraph -b /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/Collins2014.bed -wa -wb | \
cut -f 1,2,3,4,5,6 > \
/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/ear045_T3oxBS.cpg_Collins2014.bed

```


Boxplot for MGMT Pyro Kit QIAGEN and Collins2014 promoters.


```R
library(data.table)
library(ggplot2)



# MGMT Pyro Kit QIAGEN:

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/ear042_M8BS.cpg_MGMT_Pyro_Kit.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/ear043_M8oxBS.cpg_MGMT_Pyro_Kit.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/ear044_T3BS.cpg_MGMT_Pyro_Kit.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/MGMT_Pyro_Kit_QIAGEN/ear045_T3oxBS.cpg_MGMT_Pyro_Kit.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand"))

setkey(ear042_M8BS, chr, start, end, strand)
setkey(ear043_M8oxBS, chr, start, end, strand)
setkey(ear044_T3BS, chr, start, end, strand)
setkey(ear045_T3oxBS, chr, start, end, strand)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand)
setkey(T3, chr, start, end, strand)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

xdata<-M8.T3

xdata_diff<-diff(xdata$start)
cpg<-c(1)

for (i in xdata_diff){
if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
else{cpg<-append(cpg, tail(cpg, n=1)+1)}
}

xdata[,cpg:=cpg]
xdata # 4 CpG sites

tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",4*3),rep("T",4*3))]
tab.summary[,label:=rep(c(rep("5mC",4), rep("5hmC",4), rep("C/other",4)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg) + labs(title = "BS and oxBS")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160315_MGMT.promoter.BS.oxBS.MGMT_Pyro_Kit.pdf", width = 10/2.54, height = 10/2.54)

tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)))]

tab.summary<-reshape2::melt(tab, variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",3),rep("T",3))]
tab.summary[,label:=rep(c("5mC", "5hmC", "C/other"),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + labs(title = "BS and oxBS")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160315_MGMT.promoter.BS.oxBS.MGMT_Pyro_Kit.summary.pdf", width = 8.5/2.54, height = 10/2.54)



# Collins2014:

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/ear042_M8BS.cpg_Collins2014.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/ear043_M8oxBS.cpg_Collins2014.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/ear044_T3BS.cpg_Collins2014.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/Collins2014/ear045_T3oxBS.cpg_Collins2014.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand"))

setkey(ear042_M8BS, chr, start, end, strand)
setkey(ear043_M8oxBS, chr, start, end, strand)
setkey(ear044_T3BS, chr, start, end, strand)
setkey(ear045_T3oxBS, chr, start, end, strand)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand)
setkey(T3, chr, start, end, strand)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

xdata<-M8.T3

xdata_diff<-diff(xdata$start)
cpg<-c(1)

for (i in xdata_diff){
if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
else{cpg<-append(cpg, tail(cpg, n=1)+1)}
}

xdata[,cpg:=cpg]
xdata # 16 CpG sites because two have two CpG consecutively

tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",16*3),rep("T",16*3))]
tab.summary[,label:=rep(c(rep("5mC",16), rep("5hmC",16), rep("C/other",16)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg) + labs(title = "BS and oxBS")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160315_MGMT.promoter.BS.oxBS.Collins2014.pdf", width = 20/2.54, height = 10/2.54)

tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)))]

tab.summary<-reshape2::melt(tab, variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",3),rep("T",3))]
tab.summary[,label:=rep(c("5mC", "5hmC", "C/other"),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + labs(title = "BS and oxBS")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160315_MGMT.promoter.BS.oxBS.Collins2014.summary.pdf", width = 8.5/2.54, height = 10/2.54)


### 20160513 update
tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",16*3),rep("T",16*3))]
tab.summary[,label:=rep(c(rep("5mC",16), rep("5hmC",16), rep("C",16)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160513_MGMT.promoter.BS.oxBS.Collins2014.pdf", width = 20/2.54, height = 10/2.54)

tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)))]

tab.summary<-reshape2::melt(tab, variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",3),rep("T",3))]
tab.summary[,label:=rep(c("5mC", "5hmC", "C"),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160513_MGMT.promoter.BS.oxBS.Collins2014.summary.pdf", width = 8.5/2.54, height = 10/2.54)


```



Find promoters with:

* high 5hmC
* low 5mC
* low C/other

... in Tumour

```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), cnt_tot.BS.M8=cnt_tot.BS.M8, cnt_tot.oxBS.M8=cnt_tot.oxBS.M8, cnt_tot.BS.T3=cnt_tot.BS.T3, cnt_tot.oxBS.T3=cnt_tot.oxBS.T3), by=list(chr, start, end, strand, gene_name, cancer, glioblastoma)]

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]


M8.T3.pct.promoter[pct_5hmC.T3 > pct_5mC.T3][pct_5hmC.T3 > 10]

sort(table(M8.T3.pct[pct_5hmC.T3 > pct_5mC.T3][pct_5hmC.T3 > 90]$gene_name), decreasing=T)
M8.T3.pct[pct_5hmC.T3 > pct_5mC.T3][pct_5hmC.T3 > 90][cancer=="y"]
M8.T3.pct[pct_5hmC.T3 > pct_5mC.T3][pct_5hmC.T3 > 50][cancer=="y"]
M8.T3.pct[pct_5hmC.T3 > pct_5mC.T3 & cnt_tot.BS.T3 > 20 & cnt_tot.oxBS.T3 > 20 & cancer=="y"] # 5053
sort(table(M8.T3.pct[pct_5hmC.T3 > pct_5mC.T3 & cnt_tot.BS.T3 > 20 & cnt_tot.oxBS.T3 > 20 & cancer=="y"]$gene_name), decreasing=T)
M8.T3.pct[pct_5hmC.T3 > pct_5mC.T3 & pct_5hmC.T3 > pct_C_other.T3 & cnt_tot.BS.T3 > 20 & cnt_tot.oxBS.T3 > 20 & cancer=="y"] # 5
sort(table(M8.T3.pct[pct_5hmC.T3 > pct_5mC.T3 & pct_5hmC.T3 > pct_C_other.T3 & cnt_tot.BS.T3 > 20 & cnt_tot.oxBS.T3 > 20 & cancer=="y"]$gene_name), decreasing=T) # NCOR1, BCR, KAT6A, PSIP1, FANCG

```

KAT6A can be an interesting example to generate BS and BS+oxBS views of the promoter:

But first of all, what do we consider to be the promoter region of KAT6A:

```bash
grep "KAT6A" /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_1000_sorted.bed
# chr8	41909505	41910505	KAT6A	y	n
```


```R
xdata<-M8.T3[gene_name=="KAT6A"]

xdata_diff<-diff(xdata$start)
cpg<-c(1)

for (i in xdata_diff){
if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
else{cpg<-append(cpg, tail(cpg, n=1)+1)}
}

xdata[,cpg:=cpg]
xdata # 60 CpG sites (130 nucleotide), 5 of them appear consecutively

tab<-xdata[, list(pct_meth.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_unmeth.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_meth.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), pct_unmeth.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",60*2),rep("T",60*2))]
tab.summary[,label:=rep(c(rep("Methylated",60), rep("Non-methylated",60)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "turquoise3")) + facet_grid(~ cpg) + labs(title = "BS")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160321_KAT6A.promoter.BS.pdf", width=45/2.54, height=10/2.54)




tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",60*3),rep("T",60*3))]
tab.summary[,label:=rep(c(rep("5mC",60), rep("5hmC",60), rep("C/other",60)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg) + labs(title = "BS and oxBS")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160321_KAT6A.promoter.BS.oxBS.pdf", width = 40/2.54, height = 10/2.54)

M8.T3.pct[pct_5hmC.T3 > pct_5mC.T3 & pct_5hmC.T3 > pct_C_other.T3 & cnt_tot.BS.T3 > 20 & cnt_tot.oxBS.T3 > 20 & gene_name=="KAT6A"]
# chr8 41910423 41910424

xdata[110:130]
# which in xdata corresponds to CpG #58


```



Merge all 60 CpG sites in KAT6A in a summary plot:

```R
xdata<-M8.T3[gene_name=="KAT6A"]
tab<-xdata[, list(pct_meth.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_unmeth.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_meth.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), pct_unmeth.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=gene_name]
tab.summary<-reshape2::melt(tab, id.vars = "gene_name", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",2),rep("T",2))]
tab.summary[,label:=rep(c("Methylated", "Non-methylated"), 2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "turquoise3")) + labs(title = "BS")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160321_KAT6A.promoter.BS.summary.pdf", width = 10/2.54, height = 10/2.54)



xdata<-M8.T3[gene_name=="KAT6A"]
tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=gene_name]
tab.summary<-reshape2::melt(tab, id.vars = "gene_name", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",3),rep("T",3))]
tab.summary[,label:=rep(c("5mC","5hmC","C/other"),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + labs(title = "BS and oxBS")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160321_KAT6A.promoter.BS.oxBS.summary.pdf", width = 8.5/2.54, height = 10/2.54)

```


On the basis of Dario's plot, ROS1 (glioblastoma gene) might be an interesting example to generate BS and BS+oxBS views of the promoter:

What do we think is the promoter region of ROS1?

```bash
grep "ROS1" /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_all_cancer_glioblastoma_1000_sorted.bed
# chr8	chr6	117747018	117748018	ROS1	y	y
```








Find promoters with:

* high 5hmC
* low 5mC
* low C/other

... in Margin and ...

* low 5hmC
* high 5mC
* low C/other

... in Tumour

```R
under construction
```



### Expression versus promoter methylation

Run a model for promoters: expression ~ %5mC + %5hmC + %C/other + cancer(y/n)
lm() function
ARM CRAN package - Bayesian linear modeling
https://cran.r-project.org/web/packages/arm/index.html

Literature:
* Crawley2007
* Altman2015 Points of Significance: Simple linear regression
* Altman2015 Points of Significance: Multiple linear regression
* Kerns2010 chapters 11 and 12
* Everitt2009 chapter 6
* Teetor2011 chapter 11
* James2013 chapter 3

Trying:
* Promoter averages of nucleotides: expression ~ %5mC + %5hmC + %C/other + cancer(y/n)
* Individual nucleotides: expression ~ %5mC + %5hmC + %C/other + cancer(y/n)


First copy RNA data from Dario's folder. In nas-srv001,

```bash
mkdir /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq
mkdir /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data
cp /data/sblab_data1/berald01/projects/20150501_methylation_brain/20160303_rnaseq/tx_quant_lfc.bed /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data
```


In uk-cri-lcst01,

```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]



# Bringing RNA levels
# From Dario's https://github.com/sblab-bioinformatics/projects/blob/master/20150501_methylation_brain/20160303_rnaseq/20160303_rnaseq.md#expression-at-gene-level

tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmLog2Avg<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmLog2Avg)][order(chrom, promStart, promEnd)]


# Merge by gene_name (M8.T3.pct.promoter) and Associated_Gene_Name (gx)
# A different way would be to recalculate the methylation levels in promoters on the basis of the coordinate given in gx - we can try this later

M8.T3.pct.promoter.gx <- merge(M8.T3.pct.promoter, gx, by.x="gene_name", by.y="Associated_Gene_Name")


# Playing with scatterplots a bit:
# Tumour : T3 (methylation) and F3 (expression)
ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.T3, y=log10(ear047_F3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", se = FALSE) + xlab("%5mC promoter") + ylab("log10(gene expression)") + labs(title = "T3 (5mC) vs. F3 (expression)") + theme_classic() # as 5mC increases in promoters, gene expression decreases
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160331_5mC.promoter.T3_expression.F3.pdf", width = 10/2.54, height = 10/2.54)

ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.T3, y=log10(ear047_F3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_density_2d() + stat_density_2d(aes(fill = ..level..), geom = "polygon")

ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.T3, y=log10(ear047_F3+0.01))) + geom_density_2d() + stat_density_2d(geom = "raster", aes(fill = ..density..), contour = FALSE)

ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5hmC.T3, y=log10(ear047_F3+0.01))) + geom_point(alpha = 0.05, size=0.5) + xlab("%5hmC promoter") + ylab("log10(gene expression)") + labs(title = "T3 (5hmC) vs. F3 (expression)") + theme_classic() # as we know most 5hmC in tumour is lost anyway
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160331_5hmC.promoter.T3_expression.F3.pdf", width = 10/2.54, height = 10/2.54)

ggplot(M8.T3.pct.promoter.gx, aes(x=pct_C_other.T3, y=log10(ear047_F3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", se = FALSE) + xlab("%C/other promoter") + ylab("log10(gene expression)") + labs(title = "T3 (C/other) vs. F3 (expression)") + theme_classic() # as C/other increases, expression increases
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160331_C.promoter.T3_expression.F3.pdf", width = 10/2.54, height = 10/2.54)


# Margin : M8 (methylation) and M3 (expression)
ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.M8, y=log10(ear049_M3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", se = FALSE) + xlab("%5mC promoter") + ylab("log10(gene expression)") + labs(title = "M8 (5mC) vs. M3 (expression)") + theme_classic() # similar trend as in Tumour
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160331_5mC.promoter.M8_expression.M3.pdf", width = 10/2.54, height = 10/2.54)

ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5hmC.M8, y=log10(ear049_M3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", se = FALSE) + xlab("%5hmC promoter") + ylab("log10(gene expression)") + labs(title = "M8 (5hmC) vs. M3 (expression)") + theme_classic() + coord_cartesian(xlim = c(0, 50), ylim = c(-2, 4)) # as 5hmC increases in promoters, gene expression decreases but not as drastically as 5mC, the relationship is more complex
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160331_5hmC.promoter.M8_expression.M3.pdf", width = 10/2.54, height = 10/2.54)

ggplot(M8.T3.pct.promoter.gx, aes(x=pct_C_other.M8, y=log10(ear049_M3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", se = FALSE) + xlab("%C/other promoter") + ylab("log10(gene expression)") + labs(title = "M8 (C/other) vs. M3 (expression)") + theme_classic()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160331_C.promoter.M8_expression.M3.pdf", width = 10/2.54, height = 10/2.54)




# Single regression lm()

# Plot the response against each of the explanatory variables separately.

# Tumour
model<-lm(log10(M8.T3.pct.promoter.gx$ear047_F3+0.01) ~ M8.T3.pct.promoter.gx$pct_5mC.T3) # negative slope
summary(model) # Adjusted R-squared:  0.2718
anova(model) # Pr(>F) < 2.2e-16
confint(model)
plot(model)
# using subset=M8.T3.pct.promoter.gx$count>50 does not make a big difference

model<-lm(log10(M8.T3.pct.promoter.gx$ear047_F3+0.01) ~ M8.T3.pct.promoter.gx$pct_5hmC.T3) # positive slope
summary(model) # Adjusted R-squared:  -4.312e-05
anova(model) # Pr(>F) 0.6584

model<-lm(log10(M8.T3.pct.promoter.gx$ear047_F3+0.01) ~ M8.T3.pct.promoter.gx$pct_C_other.T3) # even more positive slope
summary(model) # Adjusted R-squared:  0.2621
anova(model) # Pr(>F) < 2.2e-16

#Margin
model<-lm(log10(M8.T3.pct.promoter.gx$ear049_M3+0.01) ~ M8.T3.pct.promoter.gx$pct_5mC.M8) # negative slope
summary(model) # Adjusted R-squared:  0.327
anova(model) # Pr(>F) < 2.2e-16

model <- lm(log10(M8.T3.pct.promoter.gx$ear049_M3+0.01) ~ M8.T3.pct.promoter.gx$pct_5hmC.M8) # negative slope (different than in tumour)
summary(model) # Adjusted R-squared:  0.06225
anova(model) # Pr(>F) < 2.2e-16

model <- lm(log10(M8.T3.pct.promoter.gx$ear049_M3+0.01) ~ M8.T3.pct.promoter.gx$pct_C_other.M8) # positive slope
summary(model) # Adjusted R-squared:  0.2796
anova(model) # Pr(>F) < 2.2e-16



# Multiple regression

# Plot the explanatory variables against one another
# Margin:
ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.M8, y=pct_5hmC.M8)) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(se = FALSE) + xlab("%5mC promoter") + ylab("%5hmC promoter") + labs(title = "M8 (methylation)") + theme_classic() + coord_cartesian(ylim = c(0, 50)) # interesting relationship
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160401_5mC.5hmC.promoter.M8.pdf", width = 10/2.54, height = 10/2.54)
ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.M8, y=pct_C_other.M8)) + geom_point(alpha = 0.05, size=0.5) + theme_classic() # as expected
ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5hmC.M8, y=pct_C_other.M8)) + geom_point(alpha = 0.05, size=0.5) + theme_classic()
pairs(formula = ~ pct_5mC.M8 + pct_5hmC.M8 + pct_C_other.M8, data = M8.T3.pct.promoter.gx)
# Tumour:
ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.T3, y=pct_5hmC.T3)) + geom_point(alpha = 0.05, size=0.5) + theme_classic() # no bump as in Margin, no 5hmC in tumour
ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.T3, y=pct_C_other.T3)) + geom_point(alpha = 0.05, size=0.5) + theme_classic()
ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5hmC.T3, y=pct_C_other.T3)) + geom_point(alpha = 0.05, size=0.5) + theme_classic()
pairs(formula = ~ pct_5mC.T3 + pct_5hmC.T3 + pct_C_other.T3, data = M8.T3.pct.promoter.gx)


# generalised additive model (gam)
library(mgcv)
# Margin:
par(mfrow=c(2,2))
model<-gam(formula = log10(ear049_M3+0.01) ~ s(pct_5mC.M8) + s(pct_5hmC.M8) + s(pct_C_other.M8), data = M8.T3.pct.promoter.gx)
plot(model)
par(mfrow=c(1,1))
# Tumour:
par(mfrow=c(2,2))
model<-gam(formula = log10(ear047_F3+0.01) ~ s(pct_5mC.T3) + s(pct_5hmC.T3) + s(pct_C_other.T3), data = M8.T3.pct.promoter.gx)
plot(model)
par(mfrow=c(1,1))


# tree model
library(tree)
# Margin:
model<-tree(formula = log10(ear049_M3+0.01) ~ pct_5mC.M8 + pct_5hmC.M8 + pct_C_other.M8, data = M8.T3.pct.promoter.gx)
plot(model)
text(model)
# 5mC is the most important factor affecting gene expression, 5hmC is important at 5mC>32% where interactions might be more important than quadratic terms, no other variables explain a significant amount of the variation in expression for low 5mC levels
# Tumour:
model<-tree(formula = log10(ear047_F3+0.01) ~ pct_5mC.T3 + pct_5hmC.T3 + pct_C_other.T3, data = M8.T3.pct.promoter.gx)
plot(model)
text(model)


# Linear modelling  
# Margin:

model1<-lm(formula = log10(ear049_M3+0.01) ~ pct_5mC.M8 * pct_5hmC.M8 * pct_C_other.M8 + I(pct_5mC.M8^2) + I(pct_5hmC.M8^2) + I(pct_C_other.M8^2), data = M8.T3.pct.promoter.gx) # maximal model, it includes interactions between 5mC, 5hmC and C/other and quadratic terms for 5mC, 5hmC and C/other
summary(model1)

model2<-update(model1, ~. - pct_C_other.M8 - pct_5mC.M8:pct_5hmC.M8 - pct_5mC.M8:pct_C_other.M8 - pct_5hmC.M8:pct_C_other.M8)
summary(model2)

model3<-update(model2, ~. - I(pct_5mC.M8^2))
summary(model3)

model4<-update(model3, ~. - 1)
summary(model4)
plot(model4) # bad heteroscedasticity and curved normality plot
# model4 is close to the minimal adequate model, the relationship of 5hmC with expression seems quadratic

model5<-step(model1)
summary(model5) # it keeps the interaction terms despite the singularities
plot(model5)

model6<-update(model5, ~. - pct_C_other.M8)
summary(model6)

model7<-update(model6, ~. - pct_5hmC.M8)
summary(model7)
plot(model7) # bad heteroscedasticity and curved normality plot
# model7 is also close to a minimal adequate model, different from model4 though

# suggested by Dario:
model8<-lm(formula = log10(ear049_M3+0.01) ~ pct_5mC.M8 * pct_5hmC.M8, data = M8.T3.pct.promoter.gx)
summary(model8) # 5mC is negatively correlated with expression (see Estimate). Also in this model 5hmC is positively correlated with expression.
# The effects of promoter 5mC and 5hmC in gene expression are significant but have different consequences
plot(model8) # bad heteroscedasticity and curved normality plot
ggplot(M8.T3.pct.promoter.gx, aes(x=log10(pct_5hmC.M8/pct_5mC.M8+0.1), y=log10(ear049_M3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", formula = y ~ poly(x, 2), se = FALSE) + xlab("log10(%5hmC/%5mC) promoter") + ylab("log10(gene expression)") + labs(title = "M8 (methylation) vs. M3 (expression)") + theme_classic() + xlim(-1, 1)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160401_5hmC.5mC.promoter.M8_expression.M3.pdf", width = 10/2.54, height = 10/2.54)

# print model8 to text table
library(stargazer)
stargazer(model8, type="text", dep.var.caption="", dep.var.labels=c("log(gene expression)"), covariate.labels=c("%5mC", "%5hmC", "%5mC:%5hmC", "constant"), digits=3, omit.stat=c("adj.rsq","ser","f"), report="vc*", single.row=FALSE, no.space = TRUE)

# weight model8 with the number of nucleotides in CpGs so that promoters with many CpGs will have different effect from promoters with less CpGs
model9 <- lm(formula = log10(ear049_M3+0.01) ~ pct_5mC.M8 * pct_5hmC.M8, data = M8.T3.pct.promoter.gx, weights = count)
summary(model9) # R-squared is lower but coefficients don't change much
# There is no clear difference with the unweighted model8
plot(model9) # bad heteroscedasticity and curved normality plot

# subset model8 on positive and negative strands and check if there are differences
nrow(M8.T3.pct.promoter.gx[Strand == "+"]) # 9387
nrow(M8.T3.pct.promoter.gx[Strand == "-"]) # 9270
model10_pos <- lm(formula = log10(ear049_M3+0.01) ~ pct_5mC.M8 * pct_5hmC.M8, data = M8.T3.pct.promoter.gx, subset = (Strand=="+"))
summary(model10_pos)
# The interaction pct_5mC.M8:pct_5hmC.M8 is not significant in the positive strand
plot(model10_pos)
model10_neg <- lm(formula = log10(ear049_M3+0.01) ~ pct_5mC.M8 * pct_5hmC.M8, data = M8.T3.pct.promoter.gx, subset = (Strand=="-"))
summary(model10_neg)
# The interaction pct_5mC.M8:pct_5hmC.M8 is still significant in the negative strand though
plot(model10_neg)
# Overall, the interaction term seems more important for the negative rather than the positive strand

ggplot(M8.T3.pct.promoter.gx[Strand=="+"], aes(x=pct_5mC.M8, y=log10(ear049_M3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", se = FALSE) + xlab("%5mC promoter") + ylab("log10(gene expression)") + labs(title = "M8 (5mC) vs. M3 (expression) (+ strand)") + theme_classic()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160405_5mC.promoter.M8_expression.M3_posstrand.pdf", width = 10/2.54, height = 10/2.54)

ggplot(M8.T3.pct.promoter.gx[Strand=="+"], aes(x=pct_5hmC.M8, y=log10(ear049_M3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", se = FALSE) + xlab("%5hmC promoter") + ylab("log10(gene expression)") + labs(title = "M8 (5hmC) vs. M3 (expression) (+ strand)") + theme_classic() + coord_cartesian(xlim = c(0, 50), ylim = c(-2, 4))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160405_5hmC.promoter.M8_expression.M3_posstrand.pdf", width = 10/2.54, height = 10/2.54)

ggplot(M8.T3.pct.promoter.gx[Strand=="-"], aes(x=pct_5mC.M8, y=log10(ear049_M3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", se = FALSE) + xlab("%5mC promoter") + ylab("log10(gene expression)") + labs(title = "M8 (5mC) vs. M3 (expression) (- strand)") + theme_classic()
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160405_5mC.promoter.M8_expression.M3_negstrand.pdf", width = 10/2.54, height = 10/2.54)

ggplot(M8.T3.pct.promoter.gx[Strand=="-"], aes(x=pct_5hmC.M8, y=log10(ear049_M3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", se = FALSE) + xlab("%5hmC promoter") + ylab("log10(gene expression)") + labs(title = "M8 (5hmC) vs. M3 (expression) (- strand)") + theme_classic() + coord_cartesian(xlim = c(0, 50), ylim = c(-2, 4))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160405_5hmC.promoter.M8_expression.M3_negstrand.pdf", width = 10/2.54, height = 10/2.54)



# Tumour:

model8_tumour<-lm(formula = log10(ear047_F3+0.01) ~ pct_5mC.T3 * pct_5hmC.T3, data = M8.T3.pct.promoter.gx)
summary(model8_tumour) # Again here, the effects of promoter 5mC and 5hmC in gene expression are both significant but have different consequences. 5mC is negatively correlated with expression however 5hmC is positively correlated with expression.
plot(model8_tumour)
ggplot(M8.T3.pct.promoter.gx, aes(x=log10(pct_5hmC.T3/pct_5mC.T3+0.1), y=log10(ear047_F3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", formula = y ~ poly(x, 2), se = FALSE) + xlab("log10(%5hmC/%5mC) promoter") + ylab("log10(gene expression)") + labs(title = "T3 (methylation) vs. F3 (expression)") + theme_classic() + xlim(-1, 0)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160414_5hmC.5mC.promoter.T3_expression.F3.pdf", width = 10/2.54, height = 10/2.54)

# print model8_tumour to text table
library(stargazer)
stargazer(model8_tumour, type="text", dep.var.caption="", dep.var.labels=c("log(gene expression)"), covariate.labels=c("%5mC", "%5hmC", "%5mC:%5hmC", "constant"), digits=3, omit.stat=c("adj.rsq","ser","f"), report="vc*", single.row=FALSE, no.space = TRUE)




# Delta = Tumour - Margin:
# Delta TPMs ~ Delta %5mC + Delta %5hmC
# How much of the gene expression difference Tumour vs. Margin is due to differences in promoter %5mC and %5hmC levels ?
M8.T3.pct.promoter.gx[,delta_expression := ear047_F3 - ear049_M3]
M8.T3.pct.promoter.gx[,delta_pct_5mC := pct_5mC.T3 - pct_5mC.M8]
M8.T3.pct.promoter.gx[,delta_pct_5hmC := pct_5hmC.T3 - pct_5hmC.M8]
model8_tumour_margin<-lm(formula = delta_expression ~ delta_pct_5mC * delta_pct_5hmC, data = M8.T3.pct.promoter.gx)
summary(model8_tumour_margin)
# None of the tumour vs. margin expression difference is explained by differences in promoter 5mC and 5hmC


# FC = Tumour/Margin:
# FC TPMs ~ FC %5mC + FC %5hmC
# How much of the gene expression fold change Tumour vs. Margin is due to fold changes in promoter %5mC and %5hmC levels ?
M8.T3.pct.promoter.gx[,tpmFC := ear047_F3/ear049_M3]
M8.T3.pct.promoter.gx[,tpmFC := ifelse(tpmFC == "NaN", 0, tpmFC)]
M8.T3.pct.promoter.gx[,pct_5mC_FC := pct_5mC.T3/pct_5mC.M8]
M8.T3.pct.promoter.gx[,pct_5mC_FC := ifelse(pct_5mC_FC == "NaN", 0, pct_5mC_FC)]
M8.T3.pct.promoter.gx[,pct_5hmC_FC := pct_5hmC.T3/pct_5hmC.M8]
M8.T3.pct.promoter.gx[,pct_5hmC_FC := ifelse(pct_5hmC_FC == "NaN", 0, pct_5hmC_FC)]
M8.T3.pct.promoter.gx[tpmFC != "Inf" & pct_5mC_FC != "Inf" & pct_5hmC_FC != "Inf"]
model8_tumour_margin_FC<-lm(formula = tpmFC ~ pct_5mC_FC * pct_5hmC_FC, data = M8.T3.pct.promoter.gx[tpmFC != "Inf" & pct_5mC_FC != "Inf" & pct_5hmC_FC != "Inf"])
summary(model8_tumour_margin_FC)
# I get an error related to NA/NaN/Inf in 'x'. After trying to correct it, I haven't managed to succeed.


```





### Update 20160607:

* Create a scatterplot similar to 20160401_5mC.5hmC.promoter.M8.pdf (%5mC promoter vs. %5hmC promoter) but now showing RNA expression as colour gradient of dots.
* Divide the space of %5mC promoter vs. %5hmC promoter in four regions and plot the RNA expression as four different boxes in a boxplot



```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]


tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx[, tpmLog2FC := log2((ear047_F3 + 0.01) / (ear049_M3 + 0.01))]
gx$tpmLog2Avg<- rowMeans(gx[, list(log2(ear047_F3 + 0.01), log2(ear049_M3 + 0.01))])
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmLog2Avg)][order(chrom, promStart, promEnd)]


M8.T3.pct.promoter.gx <- merge(M8.T3.pct.promoter, gx, by.x="gene_name", by.y="Associated_Gene_Name")


# Scatterplot
gg <- ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.M8, y=pct_5hmC.M8, colour = log2(ear049_M3+0.01))) + geom_point(size=0.05) + geom_smooth(se = FALSE) + xlab("% 5mC") + ylab("% 5hmC") + theme_classic() + coord_cartesian(ylim = c(0, 50)) + scale_colour_gradient2("log2(e+0.01)", midpoint=mean(log2(M8.T3.pct.promoter.gx$ear049_M3+0.01)))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160607_5mC.5hmC.promoter.expression.M8.scatterplot.pdf", width = 14/2.54, height = 10/2.54)


# Boxplot
#mean
M8.T3.pct.promoter.gx[, label.M8 := ifelse(pct_5mC.M8 > mean(pct_5mC.M8), ifelse(pct_5hmC.M8 > mean(pct_5hmC.M8), "h5mC_h5hmC", "h5mC_l5hmC"), ifelse(pct_5hmC.M8 > mean(pct_5hmC.M8), "l5mC_h5hmC", "l5mC_l5hmC"))]
#table(M8.T3.pct.promoter.gx$label.M8)
#h5mC_h5hmC h5mC_l5hmC l5mC_h5hmC l5mC_l5hmC
#      5477        994       2523       9663


#median
#M8.T3.pct.promoter.gx[, label.M8 := ifelse(pct_5mC.M8 > median(pct_5mC.M8), ifelse(pct_5hmC.M8 > median(pct_5hmC.M8), "h5mC_h5hmC", "h5mC_l5hmC"), ifelse(pct_5hmC.M8 > median(pct_5hmC.M8), "l5mC_h5hmC", "l5mC_l5hmC"))]
#table(M8.T3.pct.promoter.gx$label.M8)
#h5mC_h5hmC h5mC_l5hmC l5mC_h5hmC l5mC_l5hmC
#      8137       1191       1191       8138


# quartiles
#M8.T3.pct.promoter.gx[, label.5mC.M8 := ifelse(pct_5mC.M8 > quantile(pct_5mC.M8, c(.75)), "high", ifelse(pct_5mC.M8 < quantile(pct_5mC.M8, c(.25)),"low", "mid"))]
#M8.T3.pct.promoter.gx[, label.5hmC.M8 := ifelse(pct_5hmC.M8 > quantile(pct_5hmC.M8, c(.75)), "high", ifelse(pct_5hmC.M8 < quantile(pct_5hmC.M8, c(.25)),"low", "mid"))]
#M8.T3.pct.promoter.gx[, label.M8 := ifelse(label.5mC.M8 == "high", ifelse(label.5hmC.M8 == "high", "h5mC_h5hmC", ifelse(label.5hmC.M8 == "low", "h5mC_l5hmC", "no")), ifelse(label.5mC.M8=="low", ifelse(label.5hmC.M8 == "high", "l5mC_h5hmC", ifelse(label.5hmC.M8 == "low", "l5mC_l5hmC", "no")), "no"))]
#table(M8.T3.pct.promoter.gx$label.M8)
#h5mC_h5hmC h5mC_l5hmC l5mC_h5hmC l5mC_l5hmC         no
#      2693         62          5       4115      11782


gg2 <- ggplot(M8.T3.pct.promoter.gx, aes(x = factor(label.M8, levels(factor(label.M8))[c(3,4,1,2)]), y = log2(ear049_M3+0.01))) + geom_boxplot(outlier.shape = NA) + theme_classic() + xlab("") + ylab("log2(e+0.01)") + scale_x_discrete(labels = c("low\nhigh", "low\nlow", "high\nhigh", "high\nlow"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160607_5mC.5hmC.promoter.expression.M8.boxplot.pdf", width = 10/2.54, height = 10/2.54)


```


### Update 20160610:

* Represent the plots of %5mC promoter and %5hmC promoter (1kb upstream) vs. log10(e+0.01) as log2(TPM+0.01) instead to see how they look like. Then add this to Figure 2 in the paper.


```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]

tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3)][order(chrom, promStart, promEnd)]

M8.T3.pct.promoter.gx <- merge(M8.T3.pct.promoter, gx, by.x="gene_name", by.y="Associated_Gene_Name")

gg <- ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.M8, y=log2(ear049_M3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", se = FALSE) + xlab("% 5mC promoter") + ylab(expression("log"[2]*"(TPM+0.01)")) + theme_classic() + coord_cartesian(ylim = c(-5, 10))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160610_5mC.promoter.M8_expression.M3.pdf", width = 10/2.54, height = 10/2.54)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160610_5mC.promoter.M8_expression.M3.png", width = 10/2.54, height = 10/2.54)

gg2 <- ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5hmC.M8, y=log2(ear049_M3+0.01))) + geom_point(alpha = 0.05, size=0.5) + geom_smooth(method = "lm", se = FALSE) + xlab("% 5hmC promoter") + ylab(expression("log"[2]*"(TPM+0.01)")) + theme_classic() + coord_cartesian(xlim = c(0, 100), ylim = c(-5, 10))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160610_5hmC.promoter.M8_expression.M3.pdf", width = 10/2.54, height = 10/2.54)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160610_5hmC.promoter.M8_expression.M3.png", width = 10/2.54, height = 10/2.54)

```



* Changing the 20160607 scatterplots and boxplot (20160607_5mC.5hmC.promoter.expression.M8.scatterplot.pdf and 20160607_5mC.5hmC.promoter.expression.M8.boxplot.pdf) with changes on how we refer to the expression axis.


```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]


tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3)][order(chrom, promStart, promEnd)]


M8.T3.pct.promoter.gx <- merge(M8.T3.pct.promoter, gx, by.x="gene_name", by.y="Associated_Gene_Name")



# Scatterplot
gg <- ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.M8, y=pct_5hmC.M8, colour = log2(ear049_M3+0.01))) + geom_point(size=0.05) + xlab("% 5mC promoter") + ylab("% 5hmC promoter") + theme_classic() + coord_cartesian(ylim = c(0, 50)) + scale_colour_gradient2(expression("log"[2]*"(TPM+0.01)"), midpoint=mean(log2(M8.T3.pct.promoter.gx$ear049_M3+0.01))) + geom_vline(xintercept = mean(M8.T3.pct.promoter.gx$pct_5mC.M8), colour= 'blue', linetype= 'dashed') + geom_hline(yintercept = mean(M8.T3.pct.promoter.gx$pct_5hmC.M8), colour= 'blue', linetype= 'dashed')
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160610_5mC.5hmC.promoter.M8_expression.M3.scatterplot.pdf", width = 14/2.54, height = 10/2.54)

mean(M8.T3.pct.promoter.gx$pct_5mC.M8) # 21.30095
mean(M8.T3.pct.promoter.gx$pct_5hmC.M8) # 9.501729



# Boxplot

M8.T3.pct.promoter.gx[, label.M8 := ifelse(pct_5mC.M8 > mean(pct_5mC.M8), ifelse(pct_5hmC.M8 > mean(pct_5hmC.M8), "h5mC_h5hmC", "h5mC_l5hmC"), ifelse(pct_5hmC.M8 > mean(pct_5hmC.M8), "l5mC_h5hmC", "l5mC_l5hmC"))]
table(M8.T3.pct.promoter.gx$label.M8)
#h5mC_h5hmC h5mC_l5hmC l5mC_h5hmC l5mC_l5hmC
#      5477        994       2523       9663

gg2 <- ggplot(M8.T3.pct.promoter.gx, aes(x = factor(label.M8, levels(factor(label.M8))[c(3,4,1,2)]), y = log2(ear049_M3+0.01))) + geom_boxplot(outlier.shape = NA) + theme_classic() + xlab("") + ylab(expression("log"[2]*"(TPM+0.01)")) + scale_x_discrete(labels = c("low\nhigh", "low\nlow", "high\nhigh", "high\nlow"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160610_5mC.5hmC.promoter.M8_expression.M3.boxplot.pdf", width = 10/2.54, height = 10/2.54)



# 20160817 - as requested by Euni, checking average levels of 5mC and 5hmC in promoters

M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)))]

M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3))), by=list(gene_name, cancer, glioblastoma)]

mean(M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3))), by=list(gene_name, cancer, glioblastoma)]$pct_5mC.M8) # 24.23506

mean(M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3))), by=list(gene_name, cancer, glioblastoma)]$pct_5hmC.M8) # 10.39655

mean(M8.T3.pct.promoter.gx$pct_5mC.M8) # 21.30095
mean(M8.T3.pct.promoter.gx$pct_5hmC.M8) # 9.501729



# 20160825 - as requested by Gordon to improve the paper, changing the order in boxplot

M8.T3.pct.promoter.gx[, label.M8 := ifelse(pct_5mC.M8 > mean(pct_5mC.M8), ifelse(pct_5hmC.M8 > mean(pct_5hmC.M8), "h5mC_h5hmC", "h5mC_l5hmC"), ifelse(pct_5hmC.M8 > mean(pct_5hmC.M8), "l5mC_h5hmC", "l5mC_l5hmC"))]

gg2 <- ggplot(M8.T3.pct.promoter.gx, aes(x = factor(label.M8, levels(factor(label.M8))[c(4,3,2,1)]), y = log2(ear049_M3+0.01))) +
geom_boxplot(outlier.shape = NA) +
theme_classic() +
xlab("") +
ylab(expression("log"[2]*"(TPM+0.01)")) +
scale_x_discrete(labels = c("low\nlow", "low\nhigh", "high\nlow", "high\nhigh"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160825_5mC.5hmC.promoter.M8_expression.M3.boxplot.pdf", width = 10/2.54, height = 10/2.54)



# 20160825 - produce the following scatterplots and boxplots to improve Fig. 2c in the paper:

# 1) Scatterplot + boxplot (inset) with averages - similar to above
# 2) Scatterplot + boxplot (inset) with 1/3 and 2/3 quantiles as lines
# 3) Scatterplot + boxplot (inset) with 0.4 and 0.6 quantiles as lines
# 4) Boxplot: expression with 5mC dominant (5mC > 5hmC, 5mC > 66%) and 5hmC dominant (5mC < 5hmC, 5hmC > 66%) with more than 100 nucleotides (50 GpGs)


# 1)

M8.T3.pct.promoter.gx[, log2expression := ifelse(log2(ear049_M3+0.01) > 10, 10, log2(ear049_M3+0.01))]
M8.T3.pct.promoter.gx[, log2expression := ifelse(log2expression < -5, -5, log2expression)]

mean(M8.T3.pct.promoter.gx$pct_5mC.M8) # 21.30095
mean(M8.T3.pct.promoter.gx$pct_5hmC.M8) # 9.501729

gg <- ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.M8, y=pct_5hmC.M8, colour = log2expression)) +
geom_point(size=0.05) +
xlab("% 5mC promoter") +
ylab("% 5hmC promoter") +
theme_classic() +
coord_cartesian(ylim = c(0, 50)) +
scale_colour_gradient2(expression("log"[2]*"(TPM+0.01)"), midpoint = mean(M8.T3.pct.promoter.gx$log2expression)) +
geom_vline(xintercept = mean(M8.T3.pct.promoter.gx$pct_5mC.M8), colour= 'black', linetype= 'dashed') +
geom_hline(yintercept = mean(M8.T3.pct.promoter.gx$pct_5hmC.M8), colour= 'black', linetype= 'dashed')
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160826_5mC.5hmC.promoter.M8_expression.M3.scatterplot.pdf", width = 14/2.54, height = 10/2.54)

M8.T3.pct.promoter.gx[, label.M8 := ifelse(pct_5mC.M8 > mean(pct_5mC.M8), ifelse(pct_5hmC.M8 > mean(pct_5hmC.M8), "h5mC_h5hmC", "h5mC_l5hmC"), ifelse(pct_5hmC.M8 > mean(pct_5hmC.M8), "l5mC_h5hmC", "l5mC_l5hmC"))]

gg2 <- ggplot(M8.T3.pct.promoter.gx, aes(x = factor(label.M8, levels(factor(label.M8))[c(4,3,2,1)]), y = log2(ear049_M3+0.01))) +
geom_boxplot(outlier.shape = NA, lwd = 1) +
theme_classic() +
xlab("") +
ylab(expression("log"[2]*"(TPM+0.01)")) +
scale_x_discrete(labels = c("1", "2", "3", "4")) +
theme(axis.title.y = element_text(size=20), axis.text.y = element_text(size=20), axis.text.x = element_text(size=30), axis.line = element_line(size = 1), axis.ticks = element_line(size = 1))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160826_5mC.5hmC.promoter.M8_expression.M3.boxplot.pdf", width = 10/2.54, height = 10/2.54)


# 2)

M8.T3.pct.promoter.gx[, log2expression := ifelse(log2(ear049_M3+0.01) > 10, 10, log2(ear049_M3+0.01))]
M8.T3.pct.promoter.gx[, log2expression := ifelse(log2expression < -5, -5, log2expression)]

quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.25)[[1]] # 1.74504
quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.75)[[1]] # 39.53811
quantile(M8.T3.pct.promoter.gx$pct_5hmC.M8, probs = 0.25)[[1]] # 1.770875
quantile(M8.T3.pct.promoter.gx$pct_5hmC.M8, probs = 0.75)[[1]] # 15.18476
median(M8.T3.pct.promoter.gx$pct_5hmC.M8) # 7.377025
quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.333)[[1]] # 3.174447
quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.667)[[1]] # 23.15446

gg <- ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.M8, y=pct_5hmC.M8, colour = log2expression)) +
geom_point(size=0.05) +
xlab("% 5mC promoter") +
ylab("% 5hmC promoter") +
theme_classic() +
coord_cartesian(ylim = c(0, 50)) +
scale_colour_gradient2(expression("log"[2]*"(TPM+0.01)"), midpoint = mean(M8.T3.pct.promoter.gx$log2expression)) +
geom_vline(xintercept = quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.333)[[1]], colour= 'black', linetype= 'dashed') +
geom_vline(xintercept = quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.667)[[1]], colour= 'black', linetype= 'dashed') +
geom_hline(yintercept = median(M8.T3.pct.promoter.gx$pct_5hmC.M8), colour= 'black', linetype= 'dashed')
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160826_5mC.5hmC.promoter.M8_expression.M3.scatterplot.quantiles_33_66.pdf", width = 14/2.54, height = 10/2.54)

M8.T3.pct.promoter.gx[, label.M8 := ifelse(pct_5mC.M8 > quantile(pct_5mC.M8, probs = 0.667)[[1]], ifelse(pct_5hmC.M8 > median(pct_5hmC.M8), "h5mC_h5hmC", "h5mC_l5hmC"), ifelse(pct_5mC.M8 < quantile(pct_5mC.M8, probs = 0.333)[[1]], ifelse(pct_5hmC.M8 > median(pct_5hmC.M8), "l5mC_h5hmC", "l5mC_l5hmC"), "no"))]
table(M8.T3.pct.promoter.gx$label.M8)

gg2 <- ggplot(M8.T3.pct.promoter.gx[label.M8 != "no"], aes(x = factor(label.M8, levels(factor(label.M8))[c(4,3,2,1)]), y = log2(ear049_M3+0.01))) +
geom_boxplot(outlier.shape = NA, lwd = 1) +
theme_classic() +
xlab("") +
ylab(expression("log"[2]*"(TPM+0.01)")) +
scale_x_discrete(labels = c("1", "2", "3", "4")) +
theme(axis.title.y = element_text(size=20), axis.text.y = element_text(size=20), axis.text.x = element_text(size=30), axis.line = element_line(size = 1), axis.ticks = element_line(size = 1))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160826_5mC.5hmC.promoter.M8_expression.M3.boxplot.quantiles_33_66.pdf", width = 10/2.54, height = 10/2.54)


# 3)

M8.T3.pct.promoter.gx[, log2expression := ifelse(log2(ear049_M3+0.01) > 10, 10, log2(ear049_M3+0.01))]
M8.T3.pct.promoter.gx[, log2expression := ifelse(log2expression < -5, -5, log2expression)]

quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.4)[[1]] # 4.918909
quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.6)[[1]] # 15.36933

gg <- ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.M8, y=pct_5hmC.M8, colour = log2expression)) +
geom_point(size=0.05) +
xlab("% 5mC promoter") +
ylab("% 5hmC promoter") +
theme_classic() +
coord_cartesian(ylim = c(0, 50)) +
scale_colour_gradient2(expression("log"[2]*"(TPM+0.01)"), midpoint = mean(M8.T3.pct.promoter.gx$log2expression)) +
geom_vline(xintercept = quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.4)[[1]], colour= 'black', linetype= 'dashed') +
geom_vline(xintercept = quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.6)[[1]], colour= 'black', linetype= 'dashed') +
geom_hline(yintercept = median(M8.T3.pct.promoter.gx$pct_5hmC.M8), colour= 'black', linetype= 'dashed')
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160826_5mC.5hmC.promoter.M8_expression.M3.scatterplot.quantiles_40_60.pdf", width = 14/2.54, height = 10/2.54)

M8.T3.pct.promoter.gx[, label.M8 := ifelse(pct_5mC.M8 > quantile(pct_5mC.M8, probs = 0.6)[[1]], ifelse(pct_5hmC.M8 > median(pct_5hmC.M8), "h5mC_h5hmC", "h5mC_l5hmC"), ifelse(pct_5mC.M8 < quantile(pct_5mC.M8, probs = 0.4)[[1]], ifelse(pct_5hmC.M8 > median(pct_5hmC.M8), "l5mC_h5hmC", "l5mC_l5hmC"), "no"))]
table(M8.T3.pct.promoter.gx$label.M8)

gg2 <- ggplot(M8.T3.pct.promoter.gx[label.M8 != "no"], aes(x = factor(label.M8, levels(factor(label.M8))[c(4,3,2,1)]), y = log2(ear049_M3+0.01))) +
geom_boxplot(outlier.shape = NA, lwd = 1) +
theme_classic() +
xlab("") +
ylab(expression("log"[2]*"(TPM+0.01)")) +
scale_x_discrete(labels = c("1", "2", "3", "4")) +
theme(axis.title.y = element_text(size=20), axis.text.y = element_text(size=20), axis.text.x = element_text(size=30), axis.line = element_line(size = 1), axis.ticks = element_line(size = 1))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160826_5mC.5hmC.promoter.M8_expression.M3.boxplot.quantiles_40_60.pdf", width = 10/2.54, height = 10/2.54)


# 4)

summary(M8.T3.pct.promoter.gx$pct_5mC.M8)
summary(M8.T3.pct.promoter.gx$pct_5hmC.M8)

M8.T3.pct.promoter.gx[count>20]

mC_dom <- M8.T3.pct.promoter.gx[pct_5mC.M8 > pct_5hmC.M8][count > 100][pct_5mC.M8 > quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.66)]
hmC_dom <- M8.T3.pct.promoter.gx[pct_5hmC.M8 > pct_5mC.M8][count > 100][pct_5hmC.M8 > quantile(M8.T3.pct.promoter.gx$pct_5hmC.M8, probs = 0.66)]

l <- list(mC_dom, hmC_dom)
setattr(l, 'names', c("5mC_dom", "5hmC_dom"))
dom <- rbindlist(l, use.names=TRUE, idcol="ID")
dom[, ID := factor(ID, levels = c("5mC_dom", "5hmC_dom"))]

gg <- ggplot(dom, aes(x = ID, y = log2(ear049_M3+0.01))) +
geom_boxplot(outlier.shape = NA) +
theme_classic() +
xlab("") +
ylab(expression("log"[2]*"(TPM+0.01)")) +
scale_x_discrete(labels = c("%5mC > %5hmC\n%5mC > 66%", "%5mC < %5hmC\n%5hmC > 66%"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160826_5mC.5hmC.promoter.M8_expression.M3.boxplot.5mC_5hmC_dominant.pdf", width = 9/2.54, height = 10/2.54)

t.test(log2(dom[ID == "5mC_dom"]$ear049_M3+0.01), log2(dom[ID == "5hmC_dom"]$ear049_M3+0.01)) # p-value = 1.703e-13
t.test(dom[ID == "5mC_dom"]$ear049_M3, dom[ID == "5hmC_dom"]$ear049_M3) # p-value = 0.08715



M8.T3.pct.promoter.gx[gene_name == "ATRX"] # ATRX is 5mC dominant with count = 38
M8.T3.pct.promoter.gx[gene_name == "FAT2"] # FAT2 might be an ok example (has higher 5mC in Margin although less count)
M8.T3.pct.promoter.gx[gene_name == "FAT3"] # FAT3 has only count = 2
M8.T3.pct.promoter.gx[gene_name == "PTEN"] # PTEN has really low 5mC in Margin
M8.T3.pct.promoter.gx[gene_name == "PTEN"] # MTOR has more 5hmC than 5mC in margin still low overall though and good number of counts
M8.T3.pct.promoter.gx[gene_name == "MET"] # MET has no 5mC in Margin, but more expression in Tumour than Margin, might be a good example to show in contrast to ATRX, FAT2 or MTOR
M8.T3.pct.promoter.gx[gene_name == "NOTCH2"] # NOTCH2 is not a good example
M8.T3.pct.promoter.gx[gene_name == "NOTCH1"] # NOTCH1 might be a good counterexample, together with MET (see Bai2015)






# 20160906 - produce the following scatterplots and boxplots to improve Fig. 2 in the paper:

# 5) Similar to 2) above : scatterplot + boxplot (inset) with 1/3 and 2/3 quantiles as lines but increasing the size of the text in xy axes ticks and xy axes titles.
# 6) Similar to 4) above : boxplot: expression with 5mC dominant (5mC > 66%), 5hmC dominant (5hmC > 66%) and C dominant (C > 66%) with more than 100 nucleotides (50 GpGs)



# 5)

library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]


tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3)][order(chrom, promStart, promEnd)]


M8.T3.pct.promoter.gx <- merge(M8.T3.pct.promoter, gx, by.x="gene_name", by.y="Associated_Gene_Name")

M8.T3.pct.promoter.gx[, log2expression := ifelse(log2(ear049_M3+0.01) > 10, 10, log2(ear049_M3+0.01))]
M8.T3.pct.promoter.gx[, log2expression := ifelse(log2expression < -5, -5, log2expression)]

gg <- ggplot(M8.T3.pct.promoter.gx, aes(x=pct_5mC.M8, y=pct_5hmC.M8, colour = log2expression)) +
geom_point(size=0.05) +
xlab("% 5mC promoter") +
ylab("% 5hmC promoter") +
theme_classic() +
coord_cartesian(ylim = c(0, 50)) +
scale_colour_gradient2(expression("log"[2]*"(TPM+0.01)"), midpoint = mean(M8.T3.pct.promoter.gx$log2expression)) +
geom_vline(xintercept = quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.333)[[1]], colour= 'black', linetype= 'dashed') +
geom_vline(xintercept = quantile(M8.T3.pct.promoter.gx$pct_5mC.M8, probs = 0.667)[[1]], colour= 'black', linetype= 'dashed') +
geom_hline(yintercept = median(M8.T3.pct.promoter.gx$pct_5hmC.M8), colour= 'black', linetype= 'dashed') +
theme(axis.text = element_text(size=11), axis.title = element_text(size=13), legend.text = element_text(size=12), legend.title = element_text(size=13))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160906_5mC.5hmC.promoter.M8_expression.M3.scatterplot.quantiles_33_66.pdf", width = 12.5/2.54, height = 8.5/2.54)

M8.T3.pct.promoter.gx[, label.M8 := ifelse(pct_5mC.M8 > quantile(pct_5mC.M8, probs = 0.667)[[1]], ifelse(pct_5hmC.M8 > median(pct_5hmC.M8), "h5mC_h5hmC", "h5mC_l5hmC"), ifelse(pct_5mC.M8 < quantile(pct_5mC.M8, probs = 0.333)[[1]], ifelse(pct_5hmC.M8 > median(pct_5hmC.M8), "l5mC_h5hmC", "l5mC_l5hmC"), "no"))]
table(M8.T3.pct.promoter.gx$label.M8)

gg2 <- ggplot(M8.T3.pct.promoter.gx[label.M8 != "no"], aes(x = factor(label.M8, levels(factor(label.M8))[c(4,3,2,1)]), y = log2(ear049_M3+0.01))) +
geom_boxplot(outlier.shape = NA, lwd = 0.75) +
theme_classic() +
xlab("") +
ylab(expression("log"[2]*"(TPM+0.01)")) +
scale_x_discrete(labels = c("1", "2", "3", "4")) +
theme(axis.title.y = element_text(size=13), axis.text.y = element_text(size=12), axis.text.x = element_text(size=18))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160906_5mC.5hmC.promoter.M8_expression.M3.boxplot.quantiles_33_66.pdf", width = 5/2.54, height = 5/2.54)



# 6)

mC_dom <- M8.T3.pct.promoter.gx[pct_5mC.M8 > pct_5hmC.M8 & pct_5mC.M8 > pct_C_other.M8]
hmC_dom <- M8.T3.pct.promoter.gx[pct_5hmC.M8 > pct_5mC.M8 & pct_5hmC.M8 > pct_C_other.M8]
C_dom <- M8.T3.pct.promoter.gx[pct_C_other.M8 > pct_5mC.M8 & pct_C_other.M8 > pct_5hmC.M8]

l <- list(mC_dom, hmC_dom, C_dom)
setattr(l, 'names', c("5mC", "5hmC", "C"))
dom <- rbindlist(l, use.names=TRUE, idcol="dominant")
dom[, dominant := factor(dominant, levels = c("5mC", "5hmC", "C"))]

gg <- ggplot(dom, aes(x = dominant, y = log2(ear049_M3+0.01), fill = dominant)) +
geom_boxplot(outlier.shape = NA, lwd = 0.75) +
theme_classic() +
xlab("") +
ylab(expression("log"[2]*"(TPM+0.01)")) +
scale_x_discrete(labels = c("5mC\n(4647)", "5hmC\n(51)", "C\n(13955)")) +
scale_fill_manual(values=c("#D55E00", "#009E73", "lightgray")) +
theme(legend.position = "none", axis.title.y = element_text(size=13), axis.text.y = element_text(size=12), axis.text.x = element_text(size=10))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160906_5mC.5hmC.promoter.M8_expression.M3.boxplot.5mC_5hmC_C_dominant.pdf", width = 5/2.54, height = 5.5/2.54)

gg <- ggplot(dom, aes(x=pct_5mC.M8, y=pct_5hmC.M8, colour = dominant)) +
geom_point(size=0.05) +
xlab("% 5mC promoter") +
ylab("% 5hmC promoter") +
theme_classic() +
coord_cartesian(ylim = c(0, 50)) +
theme(axis.text = element_text(size=11), axis.title = element_text(size=13), legend.position = "none") +
scale_colour_manual(values = c("#D55E00", "#009E73", "lightgray"))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160906_5mC.5hmC.promoter.M8_expression.M3.scatterplot.5mC_5hmC_C_dominant.pdf", width = 8.5/2.54, height = 8.5/2.54)



```













* Check smFRET experiment in Song2016 to create a slide and discuss in the paper.
* Use IGV to reproduce figure 2B.
* Read Ficz2014 and Bachman2015 and discuss about it in the introduction.
* Iterate through fragments in the oxBS data (R1 and R2) and count how many Cs are 5mC. Use the SNVs to tell whether the methylation comes from a tumour or a normal cell.
* Look at promoters of glioblastoma genes as we did for KAT6A above.
* Look at promoters that have lost the 5hmC in the promoter of the tumour sample, e.g. examples with high 5hmC in margin but low 5hmC in tumour. Then do some form of functional enrichment.
* Look at promoters of epigenetic regulators - take list sent by Dario.
* Discuss tumour microenvironment in the paper, calculations from Dario + Teschendorff2016








### TODO:

summary table: putting together methylation and genetic variation in key glioblastoma
* pheatmap
* gridExtra - CRAN
* tableplot - CRAN


Check if there are any SNVs and CNV in promoters of cancer and glioblastoma genes. TERT?


https://bioconductor.org/packages/3.3/bioc/vignettes/Gviz/inst/doc/Gviz.pdf (page 22)
http://bioconductor.org/packages/release/bioc/html/seqplots.html
suggested by Dario: deeptools
http://www.bioconductor.org/packages/2.11/bioc/html/ggbio.html
https://github.com/shenlab-sinai/ngsplot ***
http://www.annoj.org/



Check overlap of SNVs with SNVs from Bai2015 and TCGA (TCGA2STAT and TGCA Assembler - R packages).
Or perhaps using MutationAligner from the Miller lab.


Run ANNOVAR

Differential methylation:
Which regions differentially increase 5mC in T (high) vs M (low)?
Which regions differentially decrease 5hmC in T (low) vs M (high)?
Check supplementary methods of Bai2015 for hints on how to do this


Do the same for
* genes of interest (notebook)
* Molenaar2014: CDKN2A, EGFR, IDH1, IDH2, PIK3CA, PTEN, and TP53, promoter and gene body SNVs and CNVs


Bring in RNA-seq data
