




The aim of this script is to find examples to discuss in the paper:




## Glioblastoma genes



The glioblastoma genes are:

In uk-cri-lcst01:

```bash
cat /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/COSMICv76/genes_promoters_glioblastoma_1000_sorted.bed
#chr1	204484506	204485506	MDM4
#chr15	90645708	90646708	IDH2
#chr2	209119806	209120806	IDH1
#chr3	178865310	178866310	PIK3CA
#chr4	55094263	55095263	PDGFRA
#chr5	1295162	1296162	TERT
#chr5	67510583	67511583	PIK3R1
#chr6	33290793	33291793	DAXX
#chr6	117747018	117748018	ROS1
#chr6	117923705	117924705	GOPC
#chrX	123093474	123094474	STAG2
```

### Promoter 1kb 5mC and 5hmC

http://www.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000047936;r=6:117288300-117425855


```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), cnt_tot.BS.M8=cnt_tot.BS.M8, cnt_tot.oxBS.M8=cnt_tot.oxBS.M8, cnt_tot.BS.T3=cnt_tot.BS.T3, cnt_tot.oxBS.T3=cnt_tot.oxBS.T3), by=list(chr, start, end, strand, gene_name, cancer, glioblastoma)]

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]


for (gene in levels(factor(ear042_M8BS[glioblastoma == "y"]$gene_name))) {
	xdata<-M8.T3[gene_name==gene]
	xdata_diff<-diff(xdata$start)
	cpg<-c(1)
	for (i in xdata_diff){
		if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
		else{cpg<-append(cpg, tail(cpg, n=1)+1)}
	}
	xdata[,cpg:=cpg]
	ncpgs <- max(xdata$cpg)
	nnucs <- nrow(xdata)
	print(sprintf("%s has %s CpGs distributed across %s nucleotides in the 1kb promoter region", gene, ncpgs, nnucs))

	# promoter BS
	tab<-xdata[, list(pct_meth.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_unmeth.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_meth.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), pct_unmeth.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

	tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
	tab.summary[,sample2:=c(rep("M",ncpgs*2),rep("T",ncpgs*2))]
	tab.summary[,label:=rep(c(rep("Methylated",ncpgs), rep("Non-methylated",ncpgs)),2)]
	tab.summary[,sample:=NULL]

	ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "turquoise3")) + facet_grid(~ cpg) + labs(title = "BS")
	ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160504_%s.promoter.BS.pdf", gene), width=ncpgs/2.54, height=10/2.54)

	# promoter oxBS
	tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

	tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
	tab.summary[,sample2:=c(rep("M",ncpgs*3),rep("T",ncpgs*3))]
	tab.summary[,label:=rep(c(rep("5mC",ncpgs), rep("5hmC",ncpgs), rep("C/other",ncpgs)),2)]
	tab.summary[,sample:=NULL]

	ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg) + labs(title = "BS and oxBS")
	ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160504_%s.promoter.BS.oxBS.pdf", gene), width = ncpgs/2.54, height = 10/2.54)

	# promoter BS summary
	tab<-xdata[, list(pct_meth.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_unmeth.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_meth.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), pct_unmeth.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=gene_name]

	tab.summary<-reshape2::melt(tab, id.vars = "gene_name", variable.name = "sample", value.name = "pct_met")
	tab.summary[,sample2:=c(rep("M",2),rep("T",2))]
	tab.summary[,label:=rep(c("Methylated", "Non-methylated"), 2)]
	tab.summary[,sample:=NULL]

	ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "turquoise3")) + labs(title = "BS")
	ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160504_%s.promoter.BS.summary.pdf", gene), width = 10/2.54, height = 10/2.54)

	# promoter oxBS summary
	tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=gene_name]

	tab.summary<-reshape2::melt(tab, id.vars = "gene_name", variable.name = "sample", value.name = "pct_met")
	tab.summary[,sample2:=c(rep("M",3),rep("T",3))]
	tab.summary[,label:=rep(c("5mC","5hmC","C/other"),2)]
	tab.summary[,sample:=NULL]

	ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + labs(title = "BS and oxBS")
	ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160504_%s.promoter.BS.oxBS.summary.pdf", gene), width = 10/2.54, height = 10/2.54)
}


```





## TET2 and TET3 genes


### Promoter 1kb 5mC and 5hmC

http://www.ensembl.org/Homo_sapiens/Gene/Summary?db=core;g=ENSG00000047936;r=6:117288300-117425855


```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), cnt_tot.BS.M8=cnt_tot.BS.M8, cnt_tot.oxBS.M8=cnt_tot.oxBS.M8, cnt_tot.BS.T3=cnt_tot.BS.T3, cnt_tot.oxBS.T3=cnt_tot.oxBS.T3), by=list(chr, start, end, strand, gene_name, cancer, glioblastoma)]

M8.T3.pct.promoter <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name, cancer, glioblastoma)]


for (gene in c("TET2", "TET3")) {
	xdata<-M8.T3[gene_name==gene]
	xdata_diff<-diff(xdata$start)
	cpg<-c(1)
	for (i in xdata_diff){
		if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
		else{cpg<-append(cpg, tail(cpg, n=1)+1)}
	}
	xdata[,cpg:=cpg]
	ncpgs <- max(xdata$cpg)
	nnucs <- nrow(xdata)
	print(sprintf("%s has %s CpGs distributed across %s nucleotides in the 1kb promoter region", gene, ncpgs, nnucs))

	# promoter BS
	tab<-xdata[, list(pct_meth.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_unmeth.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_meth.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), pct_unmeth.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

	tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
	tab.summary[,sample2:=c(rep("M",ncpgs*2),rep("T",ncpgs*2))]
	tab.summary[,label:=rep(c(rep("Methylated",ncpgs), rep("Non-methylated",ncpgs)),2)]
	tab.summary[,sample:=NULL]

	ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "turquoise3")) + facet_grid(~ cpg) + labs(title = "BS")
	ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160509_%s.promoter.BS.pdf", gene), width=ncpgs/2.54, height=10/2.54)

	# promoter oxBS
	tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

	tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
	tab.summary[,sample2:=c(rep("M",ncpgs*3),rep("T",ncpgs*3))]
	tab.summary[,label:=rep(c(rep("5mC",ncpgs), rep("5hmC",ncpgs), rep("C/other",ncpgs)),2)]
	tab.summary[,sample:=NULL]

	ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg) + labs(title = "BS and oxBS")
	ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160509_%s.promoter.BS.oxBS.pdf", gene), width = ncpgs/2.54, height = 10/2.54)

	# promoter BS summary
	tab<-xdata[, list(pct_meth.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_unmeth.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_meth.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), pct_unmeth.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=gene_name]

	tab.summary<-reshape2::melt(tab, id.vars = "gene_name", variable.name = "sample", value.name = "pct_met")
	tab.summary[,sample2:=c(rep("M",2),rep("T",2))]
	tab.summary[,label:=rep(c("Methylated", "Non-methylated"), 2)]
	tab.summary[,sample:=NULL]

	ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "turquoise3")) + labs(title = "BS")
	ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160509_%s.promoter.BS.summary.pdf", gene), width = 10/2.54, height = 10/2.54)

	# promoter oxBS summary
	tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=gene_name]

	tab.summary<-reshape2::melt(tab, id.vars = "gene_name", variable.name = "sample", value.name = "pct_met")
	tab.summary[,sample2:=c(rep("M",3),rep("T",3))]
	tab.summary[,label:=rep(c("5mC","5hmC","C/other"),2)]
	tab.summary[,sample:=NULL]

	ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + labs(title = "BS and oxBS")
	ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160509_%s.promoter.BS.oxBS.summary.pdf", gene), width = 10/2.54, height = 10/2.54)
}




# repeat overall plots for TET3:

gene <- "TET3"
xdata<-M8.T3[gene_name==gene]
xdata_diff<-diff(xdata$start)
cpg<-c(1)
for (i in xdata_diff){
	if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
	else{cpg<-append(cpg, tail(cpg, n=1)+1)}
}
xdata[,cpg:=cpg]
ncpgs <- max(xdata$cpg)
nnucs <- nrow(xdata)
print(sprintf("%s has %s CpGs distributed across %s nucleotides in the 1kb promoter region", gene, ncpgs, nnucs))


# promoter BS
tab<-xdata[, list(pct_meth.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_unmeth.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_meth.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), pct_unmeth.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",ncpgs*2),rep("T",ncpgs*2))]
tab.summary[,label:=rep(c(rep("Methylated",ncpgs), rep("Non-methylated",ncpgs)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "turquoise3")) + facet_grid(~ cpg) + labs(title = "BS")
ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160509_%s.promoter.BS.pdf", gene), width=10/2.54, height=10/2.54)


# promoter oxBS
tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",ncpgs*3),rep("T",ncpgs*3))]
tab.summary[,label:=rep(c(rep("5mC",ncpgs), rep("5hmC",ncpgs), rep("C/other",ncpgs)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Methylation") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg) + labs(title = "BS and oxBS")
ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160509_%s.promoter.BS.oxBS.pdf", gene), width = 10/2.54, height = 10/2.54)




### 20160513 update

library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))


gene <- "TET2"
xdata<-M8.T3[gene_name==gene]
xdata_diff<-diff(xdata$start)
cpg<-c(1)
for (i in xdata_diff){
	if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
	else{cpg<-append(cpg, tail(cpg, n=1)+1)}
}
xdata[,cpg:=cpg]
ncpgs <- max(xdata$cpg)
nnucs <- nrow(xdata)
print(sprintf("%s has %s CpGs distributed across %s nucleotides in the 1kb promoter region", gene, ncpgs, nnucs))


# promoter BS
tab<-xdata[, list(pct_meth.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_unmeth.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_meth.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), pct_unmeth.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",ncpgs*2),rep("T",ncpgs*2))]
tab.summary[,label:=rep(c(rep("Methylated",ncpgs), rep("Non-methylated",ncpgs)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "turquoise3")) + facet_grid(~ cpg)
ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160514_%s.promoter.BS.pdf", gene), width=ncpgs/2.54, height=10/2.54)


# promoter oxBS
tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",ncpgs*3),rep("T",ncpgs*3))]
tab.summary[,label:=rep(c(rep("5mC",ncpgs), rep("5hmC",ncpgs), rep("C",ncpgs)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg)
ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160514_%s.promoter.BS.oxBS.pdf", gene), width = ncpgs/2.54, height = 10/2.54)


gene <- "TET3"
xdata<-M8.T3[gene_name==gene]
xdata_diff<-diff(xdata$start)
cpg<-c(1)
for (i in xdata_diff){
	if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
	else{cpg<-append(cpg, tail(cpg, n=1)+1)}
}
xdata[,cpg:=cpg]
ncpgs <- max(xdata$cpg)
nnucs <- nrow(xdata)
print(sprintf("%s has %s CpGs distributed across %s nucleotides in the 1kb promoter region", gene, ncpgs, nnucs))


# promoter BS
tab<-xdata[, list(pct_meth.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_unmeth.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_meth.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), pct_unmeth.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",ncpgs*2),rep("T",ncpgs*2))]
tab.summary[,label:=rep(c(rep("Methylated",ncpgs), rep("Non-methylated",ncpgs)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(2,1)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "turquoise3")) + facet_grid(~ cpg)
ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160514_%s.promoter.BS.pdf", gene), width=12/2.54, height=10/2.54)


# promoter oxBS
tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",ncpgs*3),rep("T",ncpgs*3))]
tab.summary[,label:=rep(c(rep("5mC",ncpgs), rep("5hmC",ncpgs), rep("C",ncpgs)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg)
ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160514_%s.promoter.BS.oxBS.pdf", gene), width = 12/2.54, height = 10/2.54)






### 20160702 update

library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))


gene <- "TET2"
xdata<-M8.T3[gene_name==gene]
xdata_diff<-diff(xdata$start)
cpg<-c(1)
for (i in xdata_diff){
	if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
	else{cpg<-append(cpg, tail(cpg, n=1)+1)}
}
xdata[,cpg:=cpg]
ncpgs <- max(xdata$cpg)
nnucs <- nrow(xdata)
print(sprintf("%s has %s CpGs distributed across %s nucleotides in the 1kb promoter region", gene, ncpgs, nnucs))

# promoter oxBS
tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",ncpgs*3),rep("T",ncpgs*3))]
tab.summary[,label:=rep(c(rep("5mC",ncpgs), rep("5hmC",ncpgs), rep("C",ncpgs)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg)
ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160702_%s.promoter.BS.oxBS.pdf", gene), width = ncpgs/2.54, height = 10/2.54)

# promoter oxBS summary
tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=gene_name]

tab.summary<-reshape2::melt(tab, id.vars = "gene_name", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",3),rep("T",3))]
tab.summary[,label:=rep(c("5mC","5hmC","C/other"),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160702_%s.promoter.BS.oxBS.summary.pdf", gene), width = 8/2.54, height = 10/2.54)



gene <- "TET3"
xdata<-M8.T3[gene_name==gene]
xdata_diff<-diff(xdata$start)
cpg<-c(1)
for (i in xdata_diff){
	if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
	else{cpg<-append(cpg, tail(cpg, n=1)+1)}
}
xdata[,cpg:=cpg]
ncpgs <- max(xdata$cpg)
nnucs <- nrow(xdata)
print(sprintf("%s has %s CpGs distributed across %s nucleotides in the 1kb promoter region", gene, ncpgs, nnucs))

# promoter oxBS
tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=cpg]

tab.summary<-reshape2::melt(tab, id.vars = "cpg", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",ncpgs*3),rep("T",ncpgs*3))]
tab.summary[,label:=rep(c(rep("5mC",ncpgs), rep("5hmC",ncpgs), rep("C",ncpgs)),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3")) + facet_grid(~ cpg)
ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160702_%s.promoter.BS.oxBS.pdf", gene), width = 12/2.54, height = 10/2.54)

# promoter oxBS summary
tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=gene_name]

tab.summary<-reshape2::melt(tab, id.vars = "gene_name", variable.name = "sample", value.name = "pct_met")
tab.summary[,sample2:=c(rep("M",3),rep("T",3))]
tab.summary[,label:=rep(c("5mC","5hmC","C/other"),2)]
tab.summary[,sample:=NULL]

ggplot(tab.summary, aes(x = factor(sample2), y = pct_met, fill=factor(label, levels(factor(label))[c(3,1,2)]))) + geom_bar(stat = "identity") + xlab("") + ylab("% Modification") + theme_classic() + theme(legend.title = element_blank()) + scale_fill_manual(values=c("red", "green", "turquoise3"))
ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160702_%s.promoter.BS.oxBS.summary.pdf", gene), width = 8/2.54, height = 10/2.54)




#### Barplot with expression of TET2 and TET3 in Margin and Tumour

tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]


gx<- rbindlist(list(gxPlus, gxMinus))

tab.summary <- data.table(matrix(c(round(log2(gx[Associated_Gene_Name == "TET2"]$ear049_M3 + 0.01), 2), round(log2(gx[Associated_Gene_Name == "TET2"]$ear047_F3 + 0.01), 2), "M", "T"), ncol = 2))
setnames(tab.summary, c("tpm", "sample2") )

ggplot(tab.summary, aes(x = sample2, y = tpm)) + geom_bar(stat = "identity") + xlab("") + ylab(expression("log"[2]*"(TPM+0.01)")) + theme_classic() + theme(legend.title = element_blank())
ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160702_TET2.expression.pdf"), width = 8/2.54, height = 10/2.54)

tab.summary <- data.table(matrix(c(round(log2(gx[Associated_Gene_Name == "TET3"]$ear049_M3 + 0.01), 2), round(log2(gx[Associated_Gene_Name == "TET3"]$ear047_F3 + 0.01), 2), "M", "T"), ncol = 2))
setnames(tab.summary, c("tpm", "sample2") )

ggplot(tab.summary, aes(x = sample2, y = tpm)) + geom_bar(stat = "identity") + xlab("") + ylab(expression("log"[2]*"(TPM+0.01)")) + theme_classic() + theme(legend.title = element_blank())
ggsave(sprintf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160702_TET3.expression.pdf"), width = 8/2.54, height = 10/2.54)

```





### 20160718 update

Take the top 10 genes with highest 5hmC difference between Tumour and Margin and show the difference in expression.

One way of doing this is when by slicing the table containing the 5mC and 5hmC in the promoters.


```R
library(data.table)
library(ggplot2)

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

M8.T3.pct <- M8.T3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3)), count=.N), by=list(gene_name)]

M8.T3.pct[, pct_5hmC.Delta:=pct_5hmC.M8-pct_5hmC.T3]

M8.T3.pct[count > 50][pct_5hmC.M8 > 30][order(-pct_5hmC.Delta)][c(1:10)]

```

Alternatively, we can use the scripts written by Dario:

https://github.com/sblab-bioinformatics/projects/blob/master/20150501_methylation_brain/20160705_top_methyl_proms/20160705_top_methyl_proms.md

See: `20160531_figure_editing.md`






### 20160824 update

Produce two barplots:

* One containing TET2 and TET3 levels of 1kb promoter modification within the same plot
* Second containing the changes in expression for the TET2 and TET3 genes

Perhaps combine the two using:


```R
library(data.table)
library(ggplot2)



# 1kb promoter modification

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

xdata<-M8.T3[gene_name == "TET2" | gene_name == "TET3"]
xdata_diff<-diff(xdata$start)
cpg<-c(1)
for (i in xdata_diff){
	if (i==1){cpg<-append(cpg, tail(cpg, n=1))}
	else{cpg<-append(cpg, tail(cpg, n=1)+1)}
}
xdata[,cpg:=cpg]

tab<-xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=gene_name]

tab.summary <- reshape2::melt(tab, id.vars = "gene_name", variable.name = "sample", value.name = "pct_met")
tab.summary[, sample2 := factor(c(rep("Margin",6),rep("Tumour",6)))]
tab.summary[, label := factor(rep(c("5mC", "5mC", "5hmC", "5hmC", "C", "C"),2), levels = c("C", "5hmC", "5mC"))]
tab.summary[, sample := NULL]

gg <- ggplot(tab.summary, aes(x = sample2, y = pct_met, fill = label)) +
geom_bar(stat = "identity") +
xlab("") +
ylab("% Modification") +
theme_classic() +
theme(legend.title = element_blank(), strip.background = element_blank(), strip.text = element_text(size=12)) +
scale_fill_manual(values=c("lightgray", "#009E73", "#D55E00")) +
facet_grid(. ~ gene_name, switch = "x") # facet_grid(gene_name ~ .) for vertical setting
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160824_TET2.TET3.promoter.BS.oxBS.summary.pdf", width = 10/2.54, height = 10/2.54)




# changes in expression

tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]


gx<- rbindlist(list(gxPlus, gxMinus))

xdata2 <- gx[Associated_Gene_Name == "TET2" | Associated_Gene_Name == "TET3"]

tab2 <- xdata2[, list(Associated_Gene_Name = Associated_Gene_Name, ear047_F3.log = log2(ear047_F3 + 0.01), ear049_M3.log = log2(ear049_M3 + 0.01))]
tab.summary2 <- reshape2::melt(tab2, id.vars = "Associated_Gene_Name", variable.name = "sample", value.name = "expression")
tab.summary2[, sample2 := factor(c(rep("Tumour", 2),rep("Margin", 2)))]
tab.summary2[, sample := NULL]

gg2 <- ggplot(tab.summary2, aes(x = sample2, y = expression)) +
geom_bar(stat = "identity") +
xlab("") +
ylab(expression("log"[2]*"(TPM+0.01)")) +
theme_classic() +
theme(legend.title = element_blank(), strip.background = element_blank(), strip.text = element_text(size=12)) +
facet_grid(. ~ Associated_Gene_Name, switch = "x")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160824_TET2.TET3.expression.pdf", width = 7/2.54, height = 8/2.54)

gg3 <- ggplot(tab.summary2, aes(x = sample2, y = expression)) +
geom_bar(stat = "identity") +
xlab("") +
ylab(expression("log"[2]*"(TPM+0.01)")) +
theme_classic() +
theme(legend.title = element_blank(), strip.background = element_blank(), strip.text = element_text(size=12)) +
facet_grid(. ~ Associated_Gene_Name, switch = "x") +
coord_cartesian(ylim = c(3, 5)) +
scale_y_continuous(breaks = seq(0, 10, 1))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160824_TET2.TET3.expression.ylim3_5.pdf", width = 7.25/2.54, height = 6/2.54)


# combine the two in a single plot

library(gridExtra)

pdf("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160824_TET2.TET3.promoter.BS.oxBS.summary.expression.pdf", width = 10/2.54, height = 17/2.54)
grid.arrange(gg, gg2, nrow = 2, heights = c(1.5, 1))
dev.off()


```



### 20160905 update

Produce three barplots:

* One containing TET2 and TET3 average levels of 1kb promoter modification within the same plot
* Second showing a zoom into single nucleotide resolution of the promoter, say of TET2
* Third containing the changes in expression for the TET2 and TET3 genes


```R
library(data.table)
library(ggplot2)
library(Gviz)
library(BSgenome.Hsapiens.UCSC.hg19)




###############################################
# average levels of 1kb promoter modification #
###############################################

ear042_M8BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear042_M8BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear042_M8BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear043_M8oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear043_M8oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear043_M8oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear044_T3BS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear044_T3BS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear044_T3BS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

ear045_T3oxBS <- fread("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/data/methylation_cpg_promoter/ear045_T3oxBS.cpg_genes.promoters.all.cancer.glioblastoma.1000.sorted.bed")
setnames(ear045_T3oxBS, c("chr", "start", "end", "cnt_met", "cnt_tot", "strand", "gene_name", "cancer", "glioblastoma"))

setkey(ear042_M8BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear043_M8oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear044_T3BS, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(ear045_T3oxBS, chr, start, end, strand, gene_name, cancer, glioblastoma)

M8 <- merge(ear042_M8BS, ear043_M8oxBS, suffixes = c(".BS", ".oxBS"))
T3 <- merge(ear044_T3BS, ear045_T3oxBS, suffixes = c(".BS", ".oxBS"))

setkey(M8, chr, start, end, strand, gene_name, cancer, glioblastoma)
setkey(T3, chr, start, end, strand, gene_name, cancer, glioblastoma)
M8.T3 <- merge(M8, T3, suffixes = c(".M8", ".T3"))

xdata <- M8.T3[gene_name == "TET2" | gene_name == "TET3"]

tab <- xdata[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=gene_name]

tab.summary <- reshape2::melt(tab, id.vars = "gene_name", variable.name = "sample", value.name = "pct_met")
tab.summary[, sample2 := factor(c(rep("Margin",6),rep("Tumour",6)))]
tab.summary[, label := factor(rep(c("5mC", "5mC", "5hmC", "5hmC", "C", "C"),2), levels = c("C", "5hmC", "5mC"))]
tab.summary[, sample := NULL]

gg <- ggplot(tab.summary, aes(x = sample2, y = pct_met, fill = label)) +
geom_bar(stat = "identity") +
xlab("") +
ylab("% Modification") +
theme_classic() +
theme(legend.title = element_blank(), strip.background = element_blank(), strip.text = element_text(size=12), axis.text = element_text(size=12)) +
scale_fill_manual(values=c("lightgray", "#009E73", "#D55E00")) +
facet_grid(. ~ gene_name)
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160905_TET2.TET3.promoter.BS.oxBS.summary.pdf", width = 12/2.54, height = 7/2.54)



##########################################
# zoom single nucleotide resolution TET2 #
##########################################

M8.T3.TET2 <- M8.T3[gene_name == "TET2"]

strack <- SequenceTrack(Hsapiens, chromosome = "chr4")
as.character(subseq(strack, 106066032, 106067031)) # 106066031 + 1

nucleotides <- unlist(strsplit(as.character(subseq(strack, 106066032, 106067031)), split = ""))
chroms <- rep("chr4", length(nucleotides))
starts <- seq(106066031, 106067030)
ends <- seq(106066032, 106067031)

promoter <- data.table(data.frame(chr = chroms, start = starts, end = ends, base = nucleotides))

tab3 <- M8.T3.TET2[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=list(chr, start, end)]

setkey(promoter, chr, start, end)
setkey(tab3, chr, start, end)

promoter_tab3 <- data.frame(merge(promoter, tab3, all = TRUE))

promoter_tab3[is.na(promoter_tab3)] <- 0 # change all NAs to 0s

promoter_tab3 <- data.table(promoter_tab3) # change back to data.table

promoter_tabsummary3 <- reshape2::melt(promoter_tab3, id.vars = c("chr", "start", "end", "base"), variable.name = "sample", value.name = "pct_met")
promoter_tabsummary3[, sample2 := factor(c(rep("M", 1000*3), rep("T", 1000*3)))]
promoter_tabsummary3[, label := factor(rep(c(rep("5mC", 1000), rep("5hmC", 1000), rep("C", 1000)),2), levels = c("C", "5hmC", "5mC"))]
promoter_tabsummary3[, sample:=NULL]
promoter_tabsummary3[, index := rep(1:1000, 6)]

vals <- paste(nucleotides, as.character(complement(DNAStringSet(nucleotides))), sep = "\n")
keys <- as.character(1:1000)
names(vals) <- keys
labels <- vals

gg <- ggplot(promoter_tabsummary3, aes(x = sample2, y = pct_met, fill = label)) +
geom_bar(stat = "identity") +
xlab("") +
ylab("% Modification") +
theme_classic() +
theme(legend.title = element_blank(), strip.background = element_blank(), strip.text = element_text(size=8), panel.margin = unit(0.05, "lines"), axis.text.x = element_text(size=6)) +
scale_fill_manual(values=c("lightgray", "#009E73", "#D55E00")) +
facet_grid(. ~ index, switch = "x", labeller = labeller(index = labels))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160905_TET2.cpg.promoter.1kb.BS.oxBS.all.pdf", width = 1000/2.54, height = 5/2.54, limitsize = FALSE)

gg <- ggplot(promoter_tabsummary3[index > 7 & index < 24], aes(x = sample2, y = pct_met, fill = label)) +
geom_bar(stat = "identity") +
xlab("") +
ylab("% Modification") +
theme_classic() +
theme(legend.position="none", strip.background = element_blank(), plot.title = element_text(size = 12), panel.margin = unit(0.05, "lines"), axis.text.x = element_text(size=7), strip.text = element_text(size=10)) +
scale_fill_manual(values=c("lightgray", "#009E73", "#D55E00")) +
facet_grid(. ~ index, switch = "x", labeller = labeller(index = labels)) +
ggtitle("TET2")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160905_TET2.cpg.promoter.1kb.BS.oxBS.chr4.106066038.106066055.pdf", width = 9/2.54, height = 5/2.54)



##########################################
# zoom single nucleotide resolution TET3 #
##########################################

M8.T3.TET3 <- M8.T3[gene_name == "TET3"]

strack <- SequenceTrack(Hsapiens, chromosome = "chr2")
as.character(subseq(strack, 74272450, 74273449)) # 74272449 + 1

nucleotides <- unlist(strsplit(as.character(subseq(strack, 74272450, 74273449)), split = ""))
chroms <- rep("chr2", length(nucleotides))
starts <- seq(74272449, 74273448)
ends <- seq(74272450, 74273449)

promoter <- data.table(data.frame(chr = chroms, start = starts, end = ends, base = nucleotides))

tab4 <- M8.T3.TET3[, list(pct_5mC.M8=100*sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8), pct_5hmC.M8=100*(sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8) - sum(cnt_met.oxBS.M8)/sum(cnt_tot.oxBS.M8)), pct_C_other.M8=100*(1-sum(cnt_met.BS.M8)/sum(cnt_tot.BS.M8)), pct_5mC.T3=100*sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3), pct_5hmC.T3=100*(sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3) - sum(cnt_met.oxBS.T3)/sum(cnt_tot.oxBS.T3)), pct_C_other.T3=100*(1-sum(cnt_met.BS.T3)/sum(cnt_tot.BS.T3))), by=list(chr, start, end)]

setkey(promoter, chr, start, end)
setkey(tab4, chr, start, end)

promoter_tab4 <- data.frame(merge(promoter, tab4, all = TRUE))

promoter_tab4[is.na(promoter_tab4)] <- 0 # change all NAs to 0s

promoter_tab4 <- data.table(promoter_tab4) # change back to data.table

promoter_tabsummary4 <- reshape2::melt(promoter_tab4, id.vars = c("chr", "start", "end", "base"), variable.name = "sample", value.name = "pct_met")
promoter_tabsummary4[, sample2 := factor(c(rep("M", 1000*3), rep("T", 1000*3)))]
promoter_tabsummary4[, label := factor(rep(c(rep("5mC", 1000), rep("5hmC", 1000), rep("C", 1000)),2), levels = c("C", "5hmC", "5mC"))]
promoter_tabsummary4[, sample:=NULL]
promoter_tabsummary4[, index := rep(1:1000, 6)]

vals <- paste(nucleotides, as.character(complement(DNAStringSet(nucleotides))), sep = "\n")
keys <- as.character(1:1000)
names(vals) <- keys
labels <- vals

gg <- ggplot(promoter_tabsummary4, aes(x = sample2, y = pct_met, fill = label)) +
geom_bar(stat = "identity") +
xlab("") +
ylab("% Modification") +
theme_classic() +
theme(legend.title = element_blank(), strip.background = element_blank(), strip.text = element_text(size=8), panel.margin = unit(0.05, "lines"), axis.text.x = element_text(size=6)) +
scale_fill_manual(values=c("lightgray", "#009E73", "#D55E00")) +
facet_grid(. ~ index, switch = "x")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160905_TET3.cpg.promoter.1kb.BS.oxBS.all.pdf", width = 1000/2.54, height = 5/2.54, limitsize = FALSE)

gg <- ggplot(promoter_tabsummary4[index > 301 & index < 308], aes(x = sample2, y = pct_met, fill = label)) +
geom_bar(stat = "identity") +
xlab("") +
ylab("% Modification") +
theme_classic() +
theme(legend.position="none", strip.background = element_blank(), plot.title = element_text(size = 12), panel.margin = unit(0.05, "lines"), axis.text.x = element_text(size=7), strip.text = element_text(size=10)) +
scale_fill_manual(values=c("lightgray", "#009E73", "#D55E00")) +
facet_grid(. ~ index, switch = "x", labeller = labeller(index = labels)) +
ggtitle("TET3")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160905_TET3.cpg.promoter.1kb.BS.oxBS.chr2.74272750.74272756.pdf", width = 4.5/2.54, height = 5/2.54)

gg <- ggplot(promoter_tabsummary4[index > 321 & index < 331], aes(x = sample2, y = pct_met, fill = label)) +
geom_bar(stat = "identity") +
xlab("") +
ylab("% Modification") +
theme_classic() +
theme(legend.position="none", strip.background = element_blank(), plot.title = element_text(size = 12), panel.margin = unit(0.05, "lines"), axis.text.x = element_text(size=7), strip.text = element_text(size=10)) +
scale_fill_manual(values=c("lightgray", "#009E73", "#D55E00")) +
facet_grid(. ~ index, switch = "x", labeller = labeller(index = labels)) +
ggtitle("TET3")
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160905_TET3.cpg.promoter.1kb.BS.oxBS.chr2.74272770.74272779.pdf", width = 6/2.54, height = 5/2.54)




##############
# Expression #
##############

tx<- fread('/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq/data/tx_quant_lfc.bed')
gxPlus<- tx[Strand == '+', list(
    chrom= min(chrom),
    promStart= min(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]

gxMinus<- tx[Strand == '-', list(
    chrom= min(chrom),
    promStart= max(promStart),
    Strand= min(Strand),
    ear047_F3= sum(ear047_F3),
    ear049_M3= sum(ear049_M3)), by= list(Associated_Gene_Name)]


gx<- rbindlist(list(gxPlus, gxMinus))
gx[, promEnd := promStart + 1000]
gx<- gx[, list(chrom, promStart, promEnd, Associated_Gene_Name, Strand, ear047_F3, ear049_M3, tpmLog2FC, tpmLog2Avg)][order(chrom, promStart, promEnd)]

xdata2 <- gx[Associated_Gene_Name == "TET2" | Associated_Gene_Name == "TET3"]

tab2 <- xdata2[, list(Associated_Gene_Name = Associated_Gene_Name, ear047_F3.log = log2(ear047_F3 + 0.01), ear049_M3.log = log2(ear049_M3 + 0.01))]
tab.summary2 <- reshape2::melt(tab2, id.vars = "Associated_Gene_Name", variable.name = "sample", value.name = "expression")
tab.summary2[, sample2 := factor(c(rep("Tumour", 2),rep("Margin", 2)))]
tab.summary2[, sample := NULL]

gg <- ggplot(tab.summary2, aes(x = sample2, y = expression)) +
geom_bar(stat = "identity") +
xlab("") +
ylab(expression("log"[2]*"(TPM+0.01)")) +
theme_classic() +
theme(legend.title = element_blank(), strip.background = element_blank(), strip.text = element_text(size=12), axis.text = element_text(size=12), axis.title = element_text(size=12)) +
facet_grid(. ~ Associated_Gene_Name) +
coord_cartesian(ylim = c(3, 5)) +
scale_y_continuous(breaks = seq(0, 10, 1))
ggsave("/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes/figures/20160905_TET2.TET3.expression.ylim3_5.pdf", width = 9.25/2.54, height = 7/2.54)


```
