### Estimating 5hmC within tumor and margin

This is the same approach as in [20151105_bayesBetaBimodal](20151105_bayesBetaBimodal.md)
Instead of comparing tumor vs margin within oxBS, we compare BS - oxBS within tissue.

#### 5hmC in margin

* Estimate prior difference

```R
R
setwd('/nas/sblab_data1/berald01/projects/20150501_methylation_brain/20151106_genome_methyl/')

library(fitdistrplus)
library(scales)
library(data.table)
library(gridExtra)
library(parallel)
library(ggplot2)

Mode <- function(x) {
  x<- round(x, 3)
  ux <- unique(round(x, 3))
  ux[which.max(tabulate(match(x, ux)))]
}

inv.logit<- function(x){
    exp(x)/(1+exp(x))
}

NSIM<- 100000

## Data preparation
first<- TRUE
for(x in c('ear042_M8BS.cpg.bedGraph.gz', 'ear043_M8oxBS.cpg.bedGraph.gz', 'ear044_T3BS.cpg.bedGraph.gz', 'ear045_T3oxBS.cpg.bedGraph.gz')){
    dat<- fread(sprintf('tabix %s chr18', x))
    dat[, library_id := sub('\\..*', '', x)]
    if(first){
        bdg<- dat
        setnames(bdg, names(bdg), c('chrom', 'start', 'end', 'cnt_met', 'cnt_tot', 'library_id'))
        first<- FALSE
    } else {
        bdg<- rbindlist(list(bdg, dat))
    }
}

bdg[, pct_met := cnt_met/cnt_tot]

smryAll<- bdg[, list(avg= mean(pct_met, na.rm= TRUE),  median= median(pct_met, na.rm= TRUE), stdev= sd(pct_met, na.rm= TRUE)), by= library_id]
#
#      library_id       avg    median     stdev
#1:   ear042_M8BS 0.7500425 0.8500000 0.2550484
#2: ear043_M8oxBS 0.5765066 0.6406250 0.2382102
#3:   ear044_T3BS 0.6982640 0.8160920 0.2694428
#4: ear045_T3oxBS 0.6851064 0.8023256 0.2690240

gg<- ggplot(data= bdg, aes(x= pct_met)) +
    geom_histogram(fill= 'grey60', colour= 'grey20') +
    facet_wrap(~library_id)
ggsave('hist_pct_met_global.pdf', w= 20, h= 12, units= 'cm')
system('rsync -v --remove-source-files hist_pct_met_global.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20151030_betaBinEstimate/')
# Only >x
smry<- bdg[pct_met > 0.5, list(avg= mean(pct_met, na.rm= TRUE), stdev= sd(pct_met, na.rm= TRUE)), by= library_id]

# Cross tabulate the comparison of interest
cnt_met<- dcast.data.table(data= bdg[library_id %in% c('ear042_M8BS', 'ear043_M8oxBS')], chrom+start+end ~ library_id, value.var= 'cnt_met')
cnt_tot<- dcast.data.table(data= bdg[library_id %in% c('ear042_M8BS', 'ear043_M8oxBS')], chrom+start+end ~ library_id, value.var= 'cnt_tot')
cmp<- merge(cnt_met, cnt_tot, by= c('chrom', 'start', 'end'), suffixes= c('.M', '.Tot'))

cmp<- cmp[ear042_M8BS.Tot >= 30 & ear043_M8oxBS.Tot >= 30,]

## PRIOR: BETA BIMODAL
## ===================
# Estmate prior beta params for pct_met > x and pct_met < x by matching quantiles
# Leave same suffixes _T and _M where _T: BS _M: oxBS
cutoff<- 0.2
priorBetaParam_T_High<- fitdist(bdg[library_id == 'ear042_M8BS' & pct_met > cutoff & !is.na(pct_met), pct_met], 'beta', method= 'qme', probs= c(0.1, 0.9))
priorBetaParam_T_Low<- fitdist(bdg[library_id == 'ear042_M8BS' & pct_met < cutoff & !is.na(pct_met), pct_met], 'beta', method= 'qme', probs= c(0.1, 0.9))
priorBetaParam_M_High<- fitdist(bdg[library_id == 'ear043_M8oxBS' & pct_met > cutoff & !is.na(pct_met), pct_met], 'beta', method= 'qme', probs= c(0.1, 0.9))
priorBetaParam_M_Low<- fitdist(bdg[library_id == 'ear043_M8oxBS' & pct_met < cutoff & !is.na(pct_met), pct_met], 'beta', method= 'qme', probs= c(0.1, 0.9))

# Plot priors
prop_T_high<- sum(cmp[, (ear042_M8BS.M / ear042_M8BS.Tot) > cutoff]) / nrow(cmp)
prop_M_high<- sum(cmp[, (ear043_M8oxBS.M / ear043_M8oxBS.Tot) > cutoff]) / nrow(cmp)

rndPrior_T<- c(
    rbeta(10 * NSIM * prop_T_high, priorBetaParam_T_High$estimate[1], priorBetaParam_T_High$estimate[2]),
    rbeta(10 * NSIM * (1-prop_T_high), priorBetaParam_T_Low$estimate[1], priorBetaParam_T_Low$estimate[2]))
rndPrior_M<- c(
    rbeta(10 * NSIM * prop_M_high, priorBetaParam_M_High$estimate[1], priorBetaParam_M_High$estimate[2]),
    rbeta(10 * NSIM * (1-prop_M_high), priorBetaParam_M_Low$estimate[1], priorBetaParam_M_Low$estimate[2]))

pdf('bimodalBetaPrior.BS-oxBS.Margin.pdf', w= 12/2.54, h= 12/2.54, pointsize= 10)
par(mgp= c(1.75, 0.5, 0), las= 1)
plot(density(rndPrior_T), xlim= c(0, 1), col= 'red', lwd= 2, xlab= '% 5(h)mC', main= 'Prior distribution')
points(density(rndPrior_M), col= 'blue', type= 'l', lwd= 2)
grid(col= 'grey30')
legend('topleft', lwd= 2, col= c('red', 'blue'), legend= c('BS - 5(h)mC', 'oxBS 5mC'), bg= 'white')
dev.off()
system('rsync --remove-source-files bimodalBetaPrior.BS-oxBS.Margin.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20151030_betaBinEstimate/')

## DATA: Uniform prior
## ===================
x<- cmp[1,]
yT = x$ear042_M8BS.M   # BS C
nT = x$ear042_M8BS.Tot # BS C+T

yM = x$ear043_M8oxBS.M   # oxBS C
nM = x$ear043_M8oxBS.Tot # oxBS C+T

data_T<- c(
    rbeta(NSIM * prop_T_high, yT + 1, (nT - yT) + 1),
    rbeta(NSIM * (1-prop_T_high), yT + 1, (nT - yT) + 1))
data_M<- c(
    rbeta(NSIM * prop_M_high, yM + 1, (nM - yM) + 1),
    rbeta(NSIM * (1-prop_M_high), yM + 1, (nM - yM) + 1))

pdf('example2.tmp.pdf', w= 16/2.54, h= 12/2.54, pointsize= 10)
par(mgp= c(1.75, 0.5, 0), las= 0)
hist(data_M, breaks= 30, border= 'white', col= '#0000FF90', xlim= c(0, 1), freq= FALSE)
hist(data_T, breaks= 30, col= '#FF000090', border= 'white', add= TRUE, freq= FALSE)
abline(v= yT/nT, col= 'red', lty= 'dashed', lwd= 2)
abline(v= yM/nM, col= 'blue', lty= 'dashed', lwd= 2)
legend('topleft', lwd= 2, col= c('red', 'blue'), legend= c('Tumor', 'Margin'))
grid(col= 'grey30')
dev.off()
system(sprintf('rsync --remove-source-files example2.tmp.pdf $mac_office:/media/groups/Research/sblab/public_folders/berald01/projects/20150501_methylation_brain/20151030_betaBinEstimate/exampleDataObs.%s-%s.%s-%s.pdf', yT, nT, yM, nM))

## POSTERIOR: Update high and low priors
## =====================================
post_T<- c(
    rbeta(NSIM * prop_T_high, yT + priorBetaParam_T_High$estimate[1], (nT - yT) + priorBetaParam_T_High$estimate[2]),
    rbeta(NSIM * (1-prop_T_high), yT + priorBetaParam_T_Low$estimate[1], (nT - yT) + priorBetaParam_T_Low$estimate[2]))
post_M<- c(
    rbeta(NSIM * prop_M_high, yM + priorBetaParam_M_High$estimate[1], (nM - yM) + priorBetaParam_M_High$estimate[2]),
    rbeta(NSIM * (1-prop_M_high), yM + priorBetaParam_M_Low$estimate[1], (nM - yM) + priorBetaParam_M_Low$estimate[2]))

pdf('example.tmp.pdf', w= 16/2.54, h= 12/2.54, pointsize= 10)
par(mgp= c(1.75, 0.5, 0), las= 0)
hist(post_M, breaks= 30, border= 'white', col= '#0000FF90', xlim= c(0, 1), freq= FALSE)
hist(post_T, breaks= 30, col= '#FF000090', border= 'white', add= TRUE, freq= FALSE)
abline(v= yT/nT, col= 'red', lty= 'dashed', lwd= 2)
abline(v= yM/nM, col= 'blue', lty= 'dashed', lwd= 2)
grid(col= 'grey30')
legend('topleft', lwd= 2, col= c('red', 'blue'), legend= c('Tumor', 'Margin'))
dev.off()
system(sprintf('scp example.tmp.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20151030_betaBinEstimate/examplePostObs.%s-%s.%s-%s.pdf', yT, nT, yM, nM))

## DIFFERENCE
pdf('example.tmp.pdf', w= 16/2.54, h= 12/2.54, pointsize= 10)
par(mgp= c(1.75, 0.5, 0), las= 0)
hist(post_T - post_M, breaks= 30, border= 'white', col= 'grey60', xlim= c(-1, 1), freq= FALSE)
abline(v= 0, col= 'black', lty= 'dashed', lwd= 1)
grid(col= 'grey30')
legend('topleft', lwd= 2, col= c('red', 'blue'), legend= c('Tumor', 'Margin'))
dev.off()
system(sprintf('scp example.tmp.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20151030_betaBinEstimate/exampleDiffObs.%s-%s.%s-%s.pdf', yT, nT, yM, nM))

## Posterior for all sites
## =======================
p<- c(0.005, 0.025, 0.25, 0.5, 0.75, 0.975, 0.995)
simPostDiff<- function(x, priorBetaParam_T_High, priorBetaParam_T_Low, priorBetaParam_M_High, priorBetaParam_M_Low,
    prop_T_high, prop_M_high){
    NSIM<- 100000
    x<- unlist(x)

    yM = x[1]   # Met in oxBS
    yT = x[2]   # Met in BS
    nM = x[3]   # Counts
    nT = x[4]

    post_T<- c(
        rbeta(NSIM * prop_T_high, yT + priorBetaParam_T_High$estimate[1], (nT - yT) + priorBetaParam_T_High$estimate[2]),
        rbeta(NSIM * (1-prop_T_high), yT + priorBetaParam_T_Low$estimate[1], (nT - yT) + priorBetaParam_T_Low$estimate[2]))
    post_M<- c(
        rbeta(NSIM * prop_M_high, yM + priorBetaParam_M_High$estimate[1], (nM - yM) + priorBetaParam_M_High$estimate[2]),
        rbeta(NSIM * (1-prop_M_high), yM + priorBetaParam_M_Low$estimate[1], (nM - yM) + priorBetaParam_M_Low$estimate[2]))

    nn<- sprintf('p%s', p)
    Tq<- quantile(post_T, p);
    names(Tq)<- nn
    Tq<- c(Tq, mode= Mode(post_T))
    Mq<- quantile(post_M, p)
    names(Mq)<- nn
    Mq<- c(Mq, mode= Mode(post_M))
    Dq<- quantile(post_T - post_M, p)
    names(Dq)<- nn
    Dq<- c(Dq, mode= Mode(post_T - post_M))
    
    qq<- list(post_T= Tq, post_M= Mq, post_D= Dq)
    return(qq)
}

datOxBS<- cmp[, list(ear043_M8oxBS.M, ear042_M8BS.M, ear043_M8oxBS.Tot, ear042_M8BS.Tot)] ## Order of columns matters!
clus<- makeCluster(24)
clusterExport(clus, list('simPostDiff', 'Mode', 'priorBetaParam_T_High', 'priorBetaParam_T_Low', 'priorBetaParam_M_High', 'priorBetaParam_M_Low',
    'prop_T_high', 'prop_M_high', 'p'))
outqq<- parRapply(clus, datOxBS, function(x, y, z) simPostDiff(x, priorBetaParam_T_High, priorBetaParam_T_Low, priorBetaParam_M_High, priorBetaParam_M_Low,
    prop_T_high, prop_M_high))
stopCluster(clus)
save(outqq, file= 'outqq.tmp.Rdata')

# NB: It's not very smart to go thorugh the long list outqq 3 times!
postDT<- data.table(do.call('rbind', lapply(outqq, function(x) c(x[['post_T']], library_id= 'ear042_M8BS')))) # 5(h)mC in BS
postDT<- rbindlist(list(data.table(do.call('rbind', lapply(outqq, function(x) c(x[['post_M']], library_id= 'ear043_M8oxBS')))), postDT)) # 5mC in oxBS
postDT<- rbindlist(list(data.table(do.call('rbind', lapply(outqq, function(x) c(x[['post_D']], library_id= '5hmC_margin')))), postDT)) # 5hmC: difference BS - oxBS
for(x in p){
    pp<- paste('p', x, sep= '')
    postDT[[pp]]<- as.numeric(postDT[[pp]])
}
postDT[['mode']]<- as.numeric(postDT[['mode']])
stopifnot(nrow(postDT) == nrow(cmp) * 3)
postDT[, chrom := rep(cmp$chrom, 3)]
postDT[, start := rep(cmp$start, 3)]
postDT[, end := rep(cmp$end, 3)]

BSoxBS_post<- postDT[library_id %in% c('ear043_M8oxBS', 'ear042_M8BS'),]
write.table(file= 'margin_posterior.chr18.txt', x= BSoxBS_post, quote= FALSE, row.names= FALSE, sep= '\t')
system('pigz -f margin_posterior.chr18.txt')

hmC<- postDT[library_id == '5hmC_margin',]
hmC[, library_id := NULL]

## Infer the quantile in the posterior distr where the 0 point is crossed
## This part might be included in the parallel loop although it's pretty quick
## ======================================================================
getQAtZero<- function(x, p){
    xlm<- glm(p ~ x, family= quasibinomial)
    qz<- inv.logit(xlm$coefficients[1])
    return(qz)
}
dat<- as.matrix((hmC[, paste('p', p, sep= ''), with= FALSE]))
clus<- makeCluster(24)
clusterExport(clus, list('getQAtZero', 'inv.logit', 'p'))
q0<- parRapply(clus, dat, function(x) getQAtZero(x, p))
stopCluster(clus)

## Convert quantile to a p-value stats. The smallest the more far in the tail]
## the zero point is.
hmC[, pqAt0 := ifelse(q0 > 0.5, 1-q0, q0)]

write.table(file= 'hmC_margin_posterior.chr18.txt', x= hmC, quote= FALSE, row.names= FALSE, sep= '\t')
system('pigz -f hmC_margin_posterior.chr18.txt')
```

## Margin summary plots

```
cd /mnt/nfs/nas/sblab_data1/berald01/projects/20150501_methylation_brain/20151106_genome_methyl/
R

library(scales)
library(data.table)
library(gridExtra)
library(ggplot2)

hmC<- fread('zcat hmC_margin_posterior.chr18.txt.gz')
BSoxBS_post<- fread('zcat margin_posterior.chr18.txt.gz') 

## Distributiuon of 5hmC
## =====================
gg<- ggplot(data= hmC, aes(x= mode)) +
    geom_histogram(colour= 'white', fill= 'blue', alpha= 0.3) +
    geom_histogram(data= hmC[pqAt0 < 0.01], aes(x= p0.5), colour= 'white', fill= 'red', alpha= 0.3) +
    xlab('5hmC - Posterior median') +
    ggtitle('5hmC in margin') +
    scale_x_continuous(breaks=pretty_breaks(n=10)) +
    geom_vline(xintercept= 0, colour= 'grey30', size= 1, linetype= 'dashed')
ggsave('5hmC_margin_hist.chr18.pdf', gg, width= 16, height= 12, units= 'cm')
system('rsync --remove-source-files 5hmC_margin_hist.chr18.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20151030_betaBinEstimate/')

## Plot width of CI vs mean depth
## ==============================
hmC<- merge(hmC, cmp, by= c('chrom', 'start', 'end'))
xat<- seq(1, nrow(hmC), length.out= 100000)
gg<- ggplot(data= hmC[xat,],aes(x= rowMeans(hmC[xat, list(ear043_M8oxBS.Tot, ear042_M8BS.Tot)]),
        y= hmC[xat, p0.995 - p0.005], colour= p0.5)) +
    geom_point(alpha= 0.2, size= 1) +
    geom_smooth(se= FALSE) +
    xlim(NA, 200) +
    xlab('Depth in margin [mean(BS, oxBS)]') +
    ylab('Width CI') +
    scale_colour_gradient2(limits=c(0, 0.4), midpoint= 0.2, low=  muted("blue"), high= muted("red"), mid= "yellow", space= 'Lab') +
    ggtitle('5hmC in Margin\nWidth of credible interval in relation to sequencing depth')
ggsave('5hmC_margin_width_CI.chr18.png', width= 14, height= 12, units= 'cm')
system('rsync --remove-source-files 5hmC_margin_width_CI.chr18.png $mac_office:$cri_public_projects/20150501_methylation_brain/20151030_betaBinEstimate/')

## 5mC vs 5hmC
## ===========

dat<- merge(hmC[, list(chrom, start, end, mode)],
    BSoxBS_post[library_id == 'ear043_M8oxBS', list(chrom, start, end, mode)],
    by= c('chrom', 'start', 'end'), suffixes= c('.hmC', '.mC'))
xat<- seq(1, nrow(dat), length.out= 20000)
gg<- ggplot(data= dat[xat], aes(x= 100 * mode.mC, y= 100 * mode.hmC)) +
    geom_point(alpha= 0.2, size= 1, colour= 'grey30') +
    geom_hline(yintercept= 0, colour= 'blue', size= 1, linetype= 'dotted') +
    geom_abline(intercept = 100, slope = -1, colour= 'grey10', size= 0.5, linetype= 'dashed') +
    xlab('% 5mC') + ylab('% 5hmC') + ggtitle('Relationship between 5mC and 5hmC in margin')
ggsave('hmC_vs_mC_margin.pdf', width= 14, height= 12, units= 'cm')
system('rsync --remove-source-files hmC_vs_mC_margin.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20151030_betaBinEstimate/')
```

#### 5hmC in tumor

TO BE DONE WITH NEW DATA

"ear044_T3BS"   "ear045_T3oxBS"

* Estimate prior difference

```R
R
setwd('/nas/sblab_data1/berald01/projects/20150501_methylation_brain/20150921_BrainMethylomeRoadMap/20151013_methylation_chr18')

library(fitdistrplus)
library(scales)
library(data.table)
library(gridExtra)
library(parallel)
library(ggplot2)

Mode <- function(x) {
  x<- round(x, 3)
  ux <- unique(round(x, 3))
  ux[which.max(tabulate(match(x, ux)))]
}

inv.logit<- function(x){
    exp(x)/(1+exp(x))
}

NSIM<- 100000

## Data preparation
bdg<- fread('zcat hiseq201509.chr18.cpg.bdg.gz')
bdg[, pct_met := cnt_met/cnt_tot]

# Cross tabulate the comparison of interest
cnt_met<- dcast.data.table(data= bdg[library_id %in% c('ear044_T3BS', 'ear045_T3oxBS')], chrom+start+end ~ library_id, value.var= 'cnt_met')
cnt_tot<- dcast.data.table(data= bdg[library_id %in% c('ear044_T3BS', 'ear045_T3oxBS')], chrom+start+end ~ library_id, value.var= 'cnt_tot')
cmp<- merge(cnt_met, cnt_tot, by= c('chrom', 'start', 'end'), suffixes= c('.M', '.Tot'))

keep<- rowSums(cmp[,list(ear044_T3BS.Tot, ear045_T3oxBS.Tot)]) > 40
cmp<- cmp[keep, ]

## PRIOR: BETA BIMODAL
## ===================
# Estmate prior beta params for pct_met > x and pct_met < x by matching quantiles
# Leave same suffixes _T and _M where _T: BS _M: oxBS
cutoff<- 0.2
priorBetaParam_T_High<- fitdist(bdg[library_id == 'ear044_T3BS' & pct_met > cutoff & !is.na(pct_met), pct_met], 'beta', method= 'qme', probs= c(0.1, 0.9))
priorBetaParam_T_Low<- fitdist(bdg[library_id == 'ear044_T3BS' & pct_met < cutoff & !is.na(pct_met), pct_met], 'beta', method= 'qme', probs= c(0.1, 0.9))
priorBetaParam_M_High<- fitdist(bdg[library_id == 'ear045_T3oxBS' & pct_met > cutoff & !is.na(pct_met), pct_met], 'beta', method= 'qme', probs= c(0.1, 0.9))
priorBetaParam_M_Low<- fitdist(bdg[library_id == 'ear045_T3oxBS' & pct_met < cutoff & !is.na(pct_met), pct_met], 'beta', method= 'qme', probs= c(0.1, 0.9))

# Plot priors
prop_T_high<- sum(cmp[, (ear044_T3BS.M / ear044_T3BS.Tot) > cutoff]) / nrow(cmp)
prop_M_high<- sum(cmp[, (ear045_T3oxBS.M / ear045_T3oxBS.Tot) > cutoff]) / nrow(cmp)

rndPrior_T<- c(
    rbeta(10 * NSIM * prop_T_high, priorBetaParam_T_High$estimate[1], priorBetaParam_T_High$estimate[2]),
    rbeta(10 * NSIM * (1-prop_T_high), priorBetaParam_T_Low$estimate[1], priorBetaParam_T_Low$estimate[2]))
rndPrior_M<- c(
    rbeta(10 * NSIM * prop_M_high, priorBetaParam_M_High$estimate[1], priorBetaParam_M_High$estimate[2]),
    rbeta(10 * NSIM * (1-prop_M_high), priorBetaParam_M_Low$estimate[1], priorBetaParam_M_Low$estimate[2]))

pdf('bimodalBetaPrior.BS-oxBS.Tumor.pdf', w= 12/2.54, h= 12/2.54, pointsize= 10)
par(mgp= c(1.75, 0.5, 0), las= 1)
plot(density(rndPrior_T), xlim= c(0, 1), col= 'red', lwd= 2, xlab= '% 5(h)mC', main= 'Prior distribution')
points(density(rndPrior_M), col= 'blue', type= 'l', lwd= 2)
grid(col= 'grey30')
legend('topleft', lwd= 2, col= c('red', 'blue'), legend= c('BS - 5(h)mC', 'oxBS 5mC'), bg= 'white')
dev.off()
system('rsync --remove-source-files bimodalBetaPrior.BS-oxBS.Tumor.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20151030_betaBinEstimate/')

## DATA: Uniform prior
## ===================
x<- cmp[1,] # x<- cmp[start == 77773050]
yT = x$ear044_T3BS.M   # BS C
nT = x$ear044_T3BS.Tot # BS C+T

yM = x$ear045_T3oxBS.M   # oxBS C
nM = x$ear045_T3oxBS.Tot # oxBS C+T

data_T<- c(
    rbeta(NSIM * prop_T_high, yT + 1, (nT - yT) + 1),
    rbeta(NSIM * (1-prop_T_high), yT + 1, (nT - yT) + 1))
data_M<- c(
    rbeta(NSIM * prop_M_high, yM + 1, (nM - yM) + 1),
    rbeta(NSIM * (1-prop_M_high), yM + 1, (nM - yM) + 1))

pdf('example2.tmp.pdf', w= 16/2.54, h= 12/2.54, pointsize= 10)
    par(mgp= c(1.75, 0.5, 0), las= 0)
    hist(data_M, breaks= 30, border= 'white', col= '#0000FF90', xlim= c(0, 1), freq= FALSE)
    hist(data_T, breaks= 30, col= '#FF000090', border= 'white', add= TRUE, freq= FALSE)
    abline(v= yT/nT, col= 'red', lty= 'dashed', lwd= 2)
    abline(v= yM/nM, col= 'blue', lty= 'dashed', lwd= 2)
    legend('topleft', lwd= 2, col= c('red', 'blue'), legend= c('Tumor', 'Margin'))
    grid(col= 'grey30')
dev.off()
system(sprintf('rsync --remove-source-files example2.tmp.pdf $mac_office:/media/groups/Research/sblab/public_folders/berald01/projects/20150501_methylation_brain/20151030_betaBinEstimate/exampleDataObs.%s-%s.%s-%s.pdf', yT, nT, yM, nM))

## POSTERIOR: Update high and low priors
## =====================================
post_T<- c(
    rbeta(NSIM * prop_T_high, yT + priorBetaParam_T_High$estimate[1], (nT - yT) + priorBetaParam_T_High$estimate[2]),
    rbeta(NSIM * (1-prop_T_high), yT + priorBetaParam_T_Low$estimate[1], (nT - yT) + priorBetaParam_T_Low$estimate[2]))
post_M<- c(
    rbeta(NSIM * prop_M_high, yM + priorBetaParam_M_High$estimate[1], (nM - yM) + priorBetaParam_M_High$estimate[2]),
    rbeta(NSIM * (1-prop_M_high), yM + priorBetaParam_M_Low$estimate[1], (nM - yM) + priorBetaParam_M_Low$estimate[2]))

pdf('example.tmp.pdf', w= 16/2.54, h= 12/2.54, pointsize= 10)
par(mgp= c(1.75, 0.5, 0), las= 0)
hist(post_M, breaks= 30, border= 'white', col= '#0000FF90', xlim= c(0, 1), freq= FALSE)
hist(post_T, breaks= 30, col= '#FF000090', border= 'white', add= TRUE, freq= FALSE)
abline(v= yT/nT, col= 'red', lty= 'dashed', lwd= 2)
abline(v= yM/nM, col= 'blue', lty= 'dashed', lwd= 2)
grid(col= 'grey30')
legend('topleft', lwd= 2, col= c('red', 'blue'), legend= c('Tumor', 'Margin'))
dev.off()
system(sprintf('scp example.tmp.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20151030_betaBinEstimate/examplePostObs.%s-%s.%s-%s.pdf', yT, nT, yM, nM))

## DIFFERENCE
pdf('example.tmp.pdf', w= 16/2.54, h= 12/2.54, pointsize= 10)
par(mgp= c(1.75, 0.5, 0), las= 0)
hist(post_T - post_M, breaks= 30, border= 'white', col= 'grey60', xlim= c(-1, 1), freq= FALSE)
abline(v= 0, col= 'black', lty= 'dashed', lwd= 1)
grid(col= 'grey30')
legend('topleft', lwd= 2, col= c('red', 'blue'), legend= c('Tumor', 'Margin'))
dev.off()
system(sprintf('scp example.tmp.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20151030_betaBinEstimate/exampleDiffObs.%s-%s.%s-%s.pdf', yT, nT, yM, nM))

## Posterior for all sites
## =======================
p<- c(0.005, 0.025, 0.25, 0.5, 0.75, 0.975, 0.995)
simPostDiff<- function(x, priorBetaParam_T_High, priorBetaParam_T_Low, priorBetaParam_M_High, priorBetaParam_M_Low,
    prop_T_high, prop_M_high){
    NSIM<- 100000
    x<- unlist(x)

    yM = x[1]   # Met in oxBS
    yT = x[2]   # Met in BS
    nM = x[3]   # Counts
    nT = x[4]

    post_T<- c(
        rbeta(NSIM * prop_T_high, yT + priorBetaParam_T_High$estimate[1], (nT - yT) + priorBetaParam_T_High$estimate[2]),
        rbeta(NSIM * (1-prop_T_high), yT + priorBetaParam_T_Low$estimate[1], (nT - yT) + priorBetaParam_T_Low$estimate[2]))
    post_M<- c(
        rbeta(NSIM * prop_M_high, yM + priorBetaParam_M_High$estimate[1], (nM - yM) + priorBetaParam_M_High$estimate[2]),
        rbeta(NSIM * (1-prop_M_high), yM + priorBetaParam_M_Low$estimate[1], (nM - yM) + priorBetaParam_M_Low$estimate[2]))

    nn<- sprintf('p%s', p)
    Tq<- quantile(post_T, p);
    names(Tq)<- nn
    Tq<- c(Tq, mode= Mode(post_T))
    Mq<- quantile(post_M, p)
    names(Mq)<- nn
    Mq<- c(Mq, mode= Mode(post_M))
    Dq<- quantile(post_T - post_M, p)
    names(Dq)<- nn
    Dq<- c(Dq, mode= Mode(post_T - post_M))
    
    qq<- list(post_T= Tq, post_M= Mq, post_D= Dq)
    return(qq)
}

datOxBS<- cmp[, list(ear045_T3oxBS.M, ear044_T3BS.M, ear045_T3oxBS.Tot, ear044_T3BS.Tot)] ## Order of columns matters!
clus<- makeCluster(24)
clusterExport(clus, list('simPostDiff', 'Mode', 'priorBetaParam_T_High', 'priorBetaParam_T_Low', 'priorBetaParam_M_High', 'priorBetaParam_M_Low',
    'prop_T_high', 'prop_M_high', 'p'))
outqq<- parRapply(clus, datOxBS, function(x, y, z) simPostDiff(x, priorBetaParam_T_High, priorBetaParam_T_Low, priorBetaParam_M_High, priorBetaParam_M_Low,
    prop_T_high, prop_M_high))
stopCluster(clus)
save(outqq, file= 'outqq.tmp.Rdata')

# NB: It's not very smart to go thorugh the long list outqq 3 times!
postDT<- data.table(do.call('rbind', lapply(outqq, function(x) c(x[['post_T']], library_id= 'ear044_T3BS')))) # 5(h)mC in BS
postDT<- rbindlist(list(data.table(do.call('rbind', lapply(outqq, function(x) c(x[['post_M']], library_id= 'ear045_T3oxBS')))), postDT)) # 5mC in oxBS
postDT<- rbindlist(list(data.table(do.call('rbind', lapply(outqq, function(x) c(x[['post_D']], library_id= '5hmC_tumor')))), postDT)) # 5hmC: difference BS - oxBS
for(x in p){
    pp<- paste('p', x, sep= '')
    postDT[[pp]]<- as.numeric(postDT[[pp]])
}
postDT[['mode']]<- as.numeric(postDT[['mode']])
stopifnot(nrow(postDT) == nrow(cmp) * 3)
postDT[, chrom := rep(cmp$chrom, 3)]
postDT[, start := rep(cmp$start, 3)]
postDT[, end := rep(cmp$end, 3)]

BSoxBS_post<- postDT[library_id %in% c('ear045_T3oxBS', 'ear044_T3BS'),]
write.table(file= 'tumor_posterior.txt', x= BSoxBS_post, quote= FALSE, row.names= FALSE, sep= '\t')
system('pigz -f tumor_posterior.txt')

hmC<- postDT[library_id == '5hmC_tumor',]
hmC[, library_id := NULL]

## Infer the quantile in the posterior distr where the 0 point is crossed
## This part might be included in the parallel loop although it's pretty quick
## ======================================================================
getQAtZero<- function(x, p){
    xlm<- glm(p ~ x, family= quasibinomial)
    qz<- inv.logit(xlm$coefficients[1])
    return(qz)
}
dat<- as.matrix((hmC[, paste('p', p, sep= ''), with= FALSE]))
clus<- makeCluster(24)
clusterExport(clus, list('getQAtZero', 'inv.logit', 'p'))
q0<- parRapply(clus, dat, function(x) getQAtZero(x, p))
stopCluster(clus)

## Convert quantile to a p-value stats. The smallest the more far in the tail]
## the zero point is.
hmC[, pqAt0 := ifelse(q0 > 0.5, 1-q0, q0)]

write.table(file= 'hmC_tumor_posterior.txt', x= hmC, quote= FALSE, row.names= FALSE, sep= '\t')
system('pigz -f hmC_tumor_posterior.txt')
```

## Tumor summary plots

```
cd /nas/sblab_data1/berald01/projects/20150501_methylation_brain/20150921_BrainMethylomeRoadMap/20151013_methylation_chr18
R

library(scales)
library(data.table)
library(gridExtra)
library(ggplot2)

hmC<- fread('zcat hmC_tumor_posterior.txt.gz')
BSoxBS_post<- fread('zcat tumor_posterior.txt.gz') 

## Distributiuon of 5hmC
## =====================
gg<- ggplot(data= hmC, aes(x= mode)) +
    geom_histogram(colour= 'white', fill= 'blue', alpha= 0.3) +
    geom_histogram(data= hmC[pqAt0 < 0.01], aes(x= p0.5), colour= 'white', fill= 'red', alpha= 0.3) +
    xlab('5hmC - Posterior median') +
    ggtitle('5hmC in tumor') +
    scale_x_continuous(breaks=pretty_breaks(n=10)) +
    geom_vline(xintercept= 0, colour= 'grey30', linewidth= 2, linetype= 'dotted')
ggsave('5hmC_tumor_hist.pdf', gg, width= 16, height= 12, units= 'cm')
system('rsync --remove-source-files 5hmC_tumor_hist.pdf $mac_office:$cri_public_projects/20150921_BrainMethylomeRoadMap/20151013_methylation_chr18/20151102_Tumor_vs_Margin_by_bayesian_approach/')

## Plot width of CI vs mean depth
## ==============================
hmC<- merge(hmC, cmp, by= c('chrom', 'start', 'end'))
xat<- seq(1, nrow(hmC), length.out= 100000)
gg<- ggplot(data= hmC[xat,],aes(x= rowMeans(hmC[xat, list(ear045_T3oxBS.Tot, ear044_T3BS.Tot)]),
        y= hmC[xat, p0.995 - p0.005], colour= ifelse(abs(p0.5) < 0.5, abs(p0.5), 1-abs(p0.5)))) +
    geom_point(alpha= 0.2, size= 1) +
    xlim(NA, 100) +
    xlab('Depth in tumor [mean(BS, oxBS)]') +
    ylab('Width CI') +
    scale_colour_gradient2(limits=c(0, 0.1), midpoint= 0.05, low=  muted("blue"), high= muted("red"), mid= "yellow", space= 'Lab') +
    ggtitle('5hmC in tumor\nWidth of credible interval in relation to sequencing depth')
ggsave('5hmC_tumor_width_CI.png', width= 22, height= 12, units= 'cm')
system('rsync --remove-source-files 5hmC_tumor_width_CI.png
    $mac_office:$cri_public_projects/20150921_BrainMethylomeRoadMap/20151013_methylation_chr18/20151102_Tumor_vs_Margin_by_bayesian_approach/')

## CI width for BS (posterior 5(h)mC) and oxBS (posterior 5mC)
## ===========================================================
dat<- merge(BSoxBS_post, bdg, by= c('chrom', 'start', 'end', 'library_id'))
xat<- seq(1, nrow(dat), length.out= 100000)
gg<- ggplot(data= dat[xat], aes(x= cnt_tot, y= p0.995-p0.005, colour= ifelse(p0.5 < 0.5, p0.5, 1-p0.5))) +
    geom_point(alpha= 0.2, size= 1) +
    xlim(NA, 100) +
    xlab('Depth in tumor') +
    ylab('Width CI') +
    facet_wrap(~library_id) +
    scale_colour_gradient2(limits=c(0, 0.5), midpoint= 0.25, low=  muted("blue"), high= muted("red"), mid= "yellow", space= 'Lab') +
    ggtitle('5(h)mC in tumor\nWidth of credible interval in relation to sequencing depth')
ggsave('tumor_libs_width_CI.png', width= 24, height= 10, units= 'cm')
system('rsync --remove-source-files tumor_libs_width_CI.png $mac_office:$cri_public_projects/20150921_BrainMethylomeRoadMap/20151013_methylation_chr18/20151102_Tumor_vs_Margin_by_bayesian_approach/')


## 5mC vs 5hmC
## ===========

dat<- merge(hmC[, list(chrom, start, end, mode)],
    BSoxBS_post[library_id == 'ear045_T3oxBS', list(chrom, start, end, mode)],
    by= c('chrom', 'start', 'end'), suffixes= c('.hmC', '.mC'))
xat<- seq(1, nrow(dat), length.out= 20000)
gg<- ggplot(data= dat[xat], aes(x= 100 * mode.mC, y= 100 * mode.hmC)) +
    geom_point(alpha= 0.2, size= 1, colour= 'grey30') +
    geom_hline(yintercept= 0, colour= 'blue', size= 1, linetype= 'dotted') +
    geom_abline(intercept = 100, slope = -1, colour= 'grey10', size= 0.5, linetype= 'dashed') +
    xlab('% 5mC') + ylab('% 5hmC') + ggtitle('Relationship between 5mC and 5hmC in tumor')
    ggsave('hmC_vs_mC_tumor.pdf', width= 14, height= 12, units= 'cm')
    system('rsync --remove-source-files hmC_vs_mC_tumor.pdf $mac_office:$cri_public_projects/20150921_BrainMethylomeRoadMap/20151013_methylation_chr18/20151102_Tumor_vs_Margin_by_bayesian_approach/')
```