
Cleaning up `/lustre`

```bash
cd /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap

du -h --max-depth=1 /lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap
#120K	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/fastq
#28K	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/fastq_trimmed
#26M	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/bsub
#38G	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/bwameth
#43M	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/vcf
#488G	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/Assembly
#12K	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/scripts
#60K	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/samples
#3.3M	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/cnv
#30G	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/methylation_cpg
#844K	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/snv
#628K	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/supek2014
#574M	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/tsgenes_oncogenes
#23G	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/rnaseq
#8.2G	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/summary
#124M	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/figure_editing
#332K	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/examples
#56K	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/ma2016
#4.3G	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap/genebody_5hmC
#591G	/lustre/sblab/martin03/repository/20150921_BrainMethylomeRoadMap

# alexandrov2013/
rm -rf alexandrov2013

# Assembly/
rm -rf Assembly

# bsub/
rm -rf bsub

# bwameth/
# the bam files are in Sergio's external disk here: /media/martin03/Seagate\ Backup\ Plus\ Drive/martin03/bwameth/
# the bedGraph files were copied to sblab-srv001
for bedgraph in `find bwameth/ -name "*cpg.bedGraph*"`
do
  rsync -arvuP $bedgraph martin03@sblab-srv001:/nas/sblab_data1/group_folders/martin03/repository/bedGraph/
done

rm -rf bwameth/ # the remaining files are owned by Dario

# cnv/
for vcf in `find cnv/ -name "*.vcf*"`
do
  rsync -arvuP $vcf martin03@nas-srv001:/archive/Groups/SBLab/fs05/martin03/repository/vcf/
done

rm -rf cnv

# examples/
rm -rf examples

# fastq/
rm -rf fastq

# fastq_trimmed/
rm -rf fastq_trimmed

# figure_editing/
rm -rf figure_editing

# genebody_5hmC/ , I am still working in some of things so i just left it as it is

# ma2016/
rm -rf ma2016

# methylation_cpg/ , the individual hiseq201509 and hiseq201512 bedGraph files are already saved in sblab-srv001 so no need to save them again
rm -rf methylation_cpg

# rnaseq/
# the raw fastq files are already backed-up: nas-srv001:/archive/Groups/SBLab/fs04/berald01/repository/fastq/ear04{6..9}* , so no need to keep them
cd rnaseq/data/bam
rsync -arvuP rnaseq/data/bam/*.bam martin03@sblab-srv001:/nas/sblab_data1/group_folders/martin03/repository/bam/

rm -rf rnaseq

# samples/ contain sample information so I will just leave as it is for now

# scripts/
rsync -arvuP scripts/basecount_cgsites.py martin03@sblab-srv001:/nas/sblab_data1/group_folders/martin03/projects/20150921_BrainMethylomeRoadMap/scripts/

rm -rf scripts

# snv/
rm -rf snv

# summary/
rm -rf summary

# supek2014/
rm -rf supek2014

# tsgenes_oncogenes/
rm -rf tsgenes_oncogenes

# vcf/
rm -rf vcf

```
