## Distribution of 5mC in margin by genomic feature

We want to see where we find 5mC in margin and tumor. To keep things simple just use the oxBS files.

```
cd /nas/sblab_data1/berald01/projects/20150501_methylation_brain/20160126_methyl_distr/

for bed in hg19.cpgIslandExt.bed \
           hg19.cpgIslandExtShelves.bed \
           hg19.cpgIslandExtShores.bed \
           hg19.intergenic.bed \
           hg19.refgene.3utr.bed \
           hg19.refgene.5utr.bed \
           hg19.refgene.exons.bed \
           hg19.refgene.introns.bed \
           hg19.refgene.promoters_1000upstream_0downstream.bed
do
echo $bed
bname=`basename $bed .bed`
nohup intersectBed -sorted -a <(sortBed -i ../reference_data/$bed | mergeBed) -b <(bigWigToBedGraph ../20151106_genome_methyl/margin.pct_5mC.bw /dev/stdout) -wa -wb \
| groupBy -g 1,2,3 -c 1,7 -o count,mean > $bname.margin.tmp.bed &

nohup intersectBed -sorted -a <(sortBed -i ../reference_data/$bed | mergeBed) -b <(bigWigToBedGraph ../20151106_genome_methyl/tumor.pct_5mC.bw /dev/stdout) -wa -wb \
| groupBy -g 1,2,3 -c 1,7 -o count,mean > $bname.tumor.tmp.bed &
done

echo "chrom start end n_cpg pct_met feature tissue" | tr ' ' '\t' > 5mC_MarginTumor.bed
tableCat.py -i hg19.*.tmp.bed -r '.tmp.bed' | sed 's/hg19\.//' | sed 's/refgene.//' | awk -v OFS="\t" '{gsub(/\./, "\t", $6); print $0}' >> 5mC_MarginTumor.bed
gzip 5mC_MarginTumor.bed
rm hg19.*.tmp.bed

R
library(ggplot2)
library(data.table)
library(reshape2)
bed<- fread('zcat 5mC_MarginTumor.bed.gz')
bed[, pct_met := pct_met]
bed[, feature := sub('cpgIslandExtShelves', 'CpGi Shelves', feature)]
bed[, feature := sub('cpgIslandExtShores', 'CpGi Shores', feature)]
bed[, feature := sub('cpgIslandExt', 'CpG islands', feature)]
bed[, feature := sub('promoters_1000upstream_0downstream', 'Promoters', feature)]

dat<- bed
# dat<- bed[feature %in% c("CpG islands", "5utr", "3utr", "exons", "introns", "intergenic", "Promoters")]

## Ordering of boxes by oxBS median
## --------------------------------
mdn<- dat[tissue == 'margin', list(median= median(pct_met)), by= list(feature)][order(median)]
dat[, feature := factor(dat$feature, levels= mdn$feature)]

dat[, tissue := factor(tissue, levels= c('tumor', 'margin'))]

gg<- ggplot(data= dat, aes(x= feature, y= pct_met, by= tissue, colour= tissue)) +
   geom_violin(draw_quantiles= 0.5, position=position_dodge(0.75)) +
   ggtitle("5mC by genomic feature") + 
   ylab('% 5mC') +
   theme(axis.text.x= element_text(angle= 45, hjust= 1))
ggsave('boxplot_pct_5mC_cpg_by_region_TumorMargin.pdf', w= 18, h= 10, units= 'cm')
system("rsync --remove-source-files boxplot_pct_5mC_cpg_by_region_TumorMargin.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20160126_methyl_distr/")

## Same as boxplot. Not used
#gg<- ggplot(data= dat, aes(x= feature, y= pct_met, by= tissue, colour= tissue)) +
#   geom_boxplot(position=position_dodge(0.75), outlier.size= 0.75, outlier.colour= 'grey30') +
#   geom_hline(yintercept= 0, colour= 'blue', alpha= 0.2, size= 1) + 
#   ggtitle("5mC by genomic feature") + 
#   ylab('% 5mC') +
#    ylim(0, 60) + 
#   theme(axis.text.x= element_text(angle= 45, hjust= 1))
#ggsave('boxplot_pct_5hmC_chr18_cpg_by_region_TumorMargin.pdf', w= 18, h= 12, units= 'cm')
#system(
#"rsync --remove-source-files boxplot_pct_5hmC_chr18_cpg_by_region_TumorMargin.pdf $mac_office:$cri_public_projects/20150501_methylation_brain/20150921_BrainMethylomeRoadMap/20151013_methylation_chr18/")

quit(save= 'no')
rm /tmp/5hmC_MarginTumor.bed
```

